{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "0cb72484-0e2e-4d13-bda9-58fe13e8ae34",
   "metadata": {},
   "source": [
    "# Prep SLURM commands"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "16eddb68-67bf-48d8-9343-a7e48c69b312",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "'/expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/scripts'"
      ],
      "text/latex": [
       "'/expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/scripts'"
      ],
      "text/markdown": [
       "'/expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/scripts'"
      ],
      "text/plain": [
       "[1] \"/expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/scripts\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "getwd()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "82f55bee-2e04-4858-a040-64869a761882",
   "metadata": {},
   "outputs": [],
   "source": [
    "setwd(\"../\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "750b3ac4-4232-4549-b317-faa48852db41",
   "metadata": {},
   "outputs": [],
   "source": [
    "#' Generate Chunk Ranges with Fixed Number of Chunks\n",
    "#'\n",
    "#' This function divides a range into a specified number of chunks, with each chunk\n",
    "#' having approximately the same size.\n",
    "#'\n",
    "#' @param start The starting point of the range to be divided.\n",
    "#' @param end The ending point of the range to be divided.\n",
    "#' @param num_chunks The fixed number of chunks to divide the range into.\n",
    "#'\n",
    "#' @return A matrix where each row represents a chunk, with the first column being\n",
    "#'         the start of the chunk and the second column being the end of the chunk.\n",
    "#' @examples\n",
    "#' chunk_fixed_n(1, 100, 5)\n",
    "#' @export\n",
    "chunk_fixed_n <- function(start, end, num_chunks) {\n",
    "  chunk_size <- ceiling((end - start + 1) / num_chunks)\n",
    "  chunk_ranges <- sapply(1:num_chunks, function(i) {\n",
    "    chunk_start <- start + (i - 1) * chunk_size\n",
    "    chunk_end <- min(chunk_start + chunk_size - 1, end)\n",
    "    c(chunk_start, chunk_end)\n",
    "  })\n",
    "  return(chunk_ranges)\n",
    "}\n",
    "\n",
    "#' Generate Chunk Ranges with Fixed Chunk Size\n",
    "#'\n",
    "#' This function divides a range into chunks up to a given maximum size, dynamically\n",
    "#' determining the number of chunks based on the range and maximum chunk size.\n",
    "#'\n",
    "#' @param start The starting point of the range to be divided.\n",
    "#' @param end The ending point of the range to be divided.\n",
    "#' @param max_chunk_size The maximum size that each chunk can have.\n",
    "#'\n",
    "#' @return A matrix where each row represents a chunk, with the first column being\n",
    "#'         the start of the chunk and the second column being the end of the chunk.\n",
    "#'         The last chunk may be smaller than `max_chunk_size` to fit the range.\n",
    "#' @examples\n",
    "#' chunk_fixed_size(1, 100, 20)\n",
    "#' @export\n",
    "chunk_fixed_size <- function(start, end, max_chunk_size) {\n",
    "  if (length(start) > 1 || length(end) > 1) {\n",
    "    stop(\"start and end must be single values\")\n",
    "  }\n",
    "  if (start > end || max_chunk_size <= 0) {\n",
    "    stop(\"Invalid arguments: ensure start <= end and max_chunk_size > 0\")\n",
    "  }\n",
    "  \n",
    "  num_chunks <- ceiling((end - start + 1) / max_chunk_size)\n",
    "  chunk_ranges <- matrix(nrow = num_chunks, ncol = 2)\n",
    "  \n",
    "  for (i in 1:num_chunks) {\n",
    "    chunk_start <- start + (i - 1) * max_chunk_size\n",
    "    chunk_end <- min(chunk_start + max_chunk_size - 1, end)\n",
    "    chunk_ranges[i, ] <- c(chunk_start, chunk_end)\n",
    "  }\n",
    "  \n",
    "  return(t(chunk_ranges))\n",
    "}\n",
    "\n",
    "generate_slurm_script <- function(args, tag_pt1, tag_pt2, mem_per_cpu = \"16G\") {\n",
    "  cpus_per_task <- if (is.null(args$num_cores) || args$num_cores == \"all\") {\n",
    "    \"#SBATCH --exclusive\\n\"\n",
    "  } else {\n",
    "    paste0(\"#SBATCH --cpus-per-task=\", args$num_cores, \"\\n\")\n",
    "  }\n",
    "\n",
    "  # Always set mem_per_cpu flag, even in exclusive mode\n",
    "  mem_allocation <- paste0(\"#SBATCH --mem-per-cpu=\", mem_per_cpu, \"\\n\")\n",
    "\n",
    "  args_string <- paste(\"--\", names(args), \"=\", args, sep = \"\", collapse = \" \")\n",
    "  args_string <- paste(args_string, \" --tag=\", tag_pt1, \"-\", tag_pt2, sep = \"\")\n",
    "\n",
    "  slurm_script <- paste0(\n",
    "    \"#!/bin/bash\\n\",\n",
    "    if (cluster_specific_parameters) paste0(\n",
    "      \"#SBATCH --partition=\", partition, \"\\n\",\n",
    "      \"#SBATCH -A \", acct, \"\\n\",\n",
    "      \"#SBATCH --time=\", time, \"\\n\"\n",
    "    ),\n",
    "    if (!is.null(nodes) && !is.null(ntasks_per_node)) paste0(\n",
    "      \"#SBATCH --nodes=\", nodes, \"\\n\",\n",
    "      \"#SBATCH --ntasks-per-node=\", ntasks_per_node, \"\\n\"\n",
    "    ),\n",
    "    cpus_per_task,\n",
    "    mem_allocation,\n",
    "    \"#SBATCH --output=slurm_output_\", tag_pt1, \"-\", tag_pt2, \".out\\n\",\n",
    "    \"#SBATCH --job-name=\", tag_pt1, \"-\", tag_pt2, \"\\n\",\n",
    "    if (module_load_conda) \"module load conda\\n\",\n",
    "    \"conda activate mwas\\n\",\n",
    "    \"echo 'Executing Rscript with arguments: Rscript scripts/CLI.R \", args_string, \"'\\n\",\n",
    "    \"Rscript scripts/CLI.R \", args_string, \"\\n\"\n",
    "  )\n",
    "\n",
    "  return(list(script = slurm_script, path = paste0(\"slurm_scripts/\", tag_pt1, \"-\", tag_pt2, \".sh\")))\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "e618d140-ad01-469b-9db1-93176c568260",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Cluster-specific parameters\n",
    "cluster_specific_parameters <- TRUE # Example condition\n",
    "acct <- \"jhu152\"\n",
    "time <- \"24:00:00\"\n",
    "partition <- \"shared\"\n",
    "\n",
    "module_load_conda <- FALSE\n",
    "\n",
    "nodes <- 1\n",
    "ntasks_per_node <- 1\n",
    "\n",
    "# Overwrite flag (set by user)\n",
    "overwrite <- FALSE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "8e415d48-449a-40da-8547-832968839644",
   "metadata": {},
   "outputs": [],
   "source": [
    "# chunk_ranges <- chunk_fixed_size(files$first_meth_index_with_SNP_coverage[i],\n",
    "#                                  files$last_meth_index_with_SNP_coverage[i],\n",
    "#                                  1000)\n",
    "                         \n",
    "\n",
    "# chunk_ranges <- chunk_ranges[1:2,1:2]\n",
    "\n",
    "chunk_ranges <- chunk_fixed_n(1, 10000, 5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "9c7b2054-67b9-42e5-9f90-1bd0bdac846e",
   "metadata": {},
   "outputs": [],
   "source": [
    "library(data.table)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "815ee1a2-cbf5-4d10-ab9e-74457df5b83b",
   "metadata": {},
   "outputs": [],
   "source": [
    "matched_df <- fread(\"../CpGWAS/scripts/09-OUT_matched_SNP_meth_cov.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "5fa7b0c6-222d-4383-a44f-8f28aabcf55a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 198 × 10</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Chr</th><th scope=col>SNP_data</th><th scope=col>methylation_data</th><th scope=col>last_meth_value_with_SNP_coverage</th><th scope=col>first_meth_value_with_SNP_coverage</th><th scope=col>last_meth_index_with_SNP_coverage</th><th scope=col>first_meth_index_with_SNP_coverage</th><th scope=col>population</th><th scope=col>region</th><th scope=col>cov_file</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td> 1</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr1.pgen </td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr1_AA.rda </td><td>248918358</td><td> 1069461</td><td>2202702</td><td> 8982</td><td>AA </td><td>caud</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_caud.csv </td></tr>\n",
       "\t<tr><td> 2</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr2.pgen </td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr2_AA.rda </td><td>241863783</td><td>   10001</td><td>2019984</td><td>    1</td><td>AA </td><td>caud</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_caud.csv </td></tr>\n",
       "\t<tr><td> 3</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr3.pgen </td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr3_AA.rda </td><td>198099789</td><td>   11602</td><td>1538467</td><td>    1</td><td>AA </td><td>caud</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_caud.csv </td></tr>\n",
       "\t<tr><td> 4</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr4.pgen </td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr4_AA.rda </td><td>189877411</td><td>   69399</td><td>1387731</td><td>    1</td><td>AA </td><td>caud</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_caud.csv </td></tr>\n",
       "\t<tr><td> 5</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr5.pgen </td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr5_AA.rda </td><td>181172584</td><td>   44104</td><td>1409038</td><td>    1</td><td>AA </td><td>caud</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_caud.csv </td></tr>\n",
       "\t<tr><td> 6</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr6.pgen </td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr6_AA.rda </td><td>170619093</td><td>  192453</td><td>1412543</td><td> 1138</td><td>AA </td><td>caud</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_caud.csv </td></tr>\n",
       "\t<tr><td> 7</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr7.pgen </td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr7_AA.rda </td><td>159334659</td><td>   49742</td><td>1490198</td><td>    1</td><td>AA </td><td>caud</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_caud.csv </td></tr>\n",
       "\t<tr><td> 8</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr8.pgen </td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr8_AA.rda </td><td>145078546</td><td>  196751</td><td>1225856</td><td> 1483</td><td>AA </td><td>caud</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_caud.csv </td></tr>\n",
       "\t<tr><td> 9</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr9.pgen </td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr9_AA.rda </td><td>136933101</td><td>  164314</td><td>1064218</td><td> 2159</td><td>AA </td><td>caud</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_caud.csv </td></tr>\n",
       "\t<tr><td>10</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr10.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr10_AA.rda</td><td>133625493</td><td>   45719</td><td>1288155</td><td>    1</td><td>AA </td><td>caud</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_caud.csv </td></tr>\n",
       "\t<tr><td>11</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr11.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr11_AA.rda</td><td>135075586</td><td>  194509</td><td>1236846</td><td>    1</td><td>AA </td><td>caud</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_caud.csv </td></tr>\n",
       "\t<tr><td>12</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr12.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr12_AA.rda</td><td>133239922</td><td>   77807</td><td>1222018</td><td>    1</td><td>AA </td><td>caud</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_caud.csv </td></tr>\n",
       "\t<tr><td>13</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr13.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr13_AA.rda</td><td>114353123</td><td>18702898</td><td> 773488</td><td> 6198</td><td>AA </td><td>caud</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_caud.csv </td></tr>\n",
       "\t<tr><td>14</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr14.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr14_AA.rda</td><td>106883713</td><td>19991195</td><td> 809631</td><td> 1646</td><td>AA </td><td>caud</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_caud.csv </td></tr>\n",
       "\t<tr><td>15</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr15.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr15_AA.rda</td><td>101892692</td><td>23433917</td><td> 786237</td><td>11250</td><td>AA </td><td>caud</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_caud.csv </td></tr>\n",
       "\t<tr><td>16</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr16.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr16_AA.rda</td><td> 90100455</td><td>   29324</td><td>1057285</td><td>  465</td><td>AA </td><td>caud</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_caud.csv </td></tr>\n",
       "\t<tr><td>17</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr17.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr17_AA.rda</td><td> 83098547</td><td>  193254</td><td>1119214</td><td> 1513</td><td>AA </td><td>caud</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_caud.csv </td></tr>\n",
       "\t<tr><td>18</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr18.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr18_AA.rda</td><td> 80257169</td><td>   83755</td><td> 644498</td><td>  386</td><td>AA </td><td>caud</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_caud.csv </td></tr>\n",
       "\t<tr><td>19</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr19.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr19_AA.rda</td><td> 58538684</td><td>  271202</td><td>1007230</td><td>    1</td><td>AA </td><td>caud</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_caud.csv </td></tr>\n",
       "\t<tr><td>20</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr20.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr20_AA.rda</td><td> 64296371</td><td>   71317</td><td> 705461</td><td>   12</td><td>AA </td><td>caud</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_caud.csv </td></tr>\n",
       "\t<tr><td>21</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr21.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr21_AA.rda</td><td> 46682679</td><td>13391623</td><td> 358983</td><td>11194</td><td>AA </td><td>caud</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_caud.csv </td></tr>\n",
       "\t<tr><td>22</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr22.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr22_AA.rda</td><td> 50767586</td><td>16603949</td><td> 543839</td><td>14844</td><td>AA </td><td>caud</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_caud.csv </td></tr>\n",
       "\t<tr><td> 1</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr1.pgen </td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr1_all.rda</td><td>248918358</td><td> 1069461</td><td>2202702</td><td> 8982</td><td>all</td><td>caud</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_caud.csv</td></tr>\n",
       "\t<tr><td> 2</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr2.pgen </td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr2_all.rda</td><td>241863783</td><td>   10001</td><td>2019984</td><td>    1</td><td>all</td><td>caud</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_caud.csv</td></tr>\n",
       "\t<tr><td> 3</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr3.pgen </td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr3_all.rda</td><td>198099789</td><td>   11602</td><td>1538467</td><td>    1</td><td>all</td><td>caud</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_caud.csv</td></tr>\n",
       "\t<tr><td> 4</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr4.pgen </td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr4_all.rda</td><td>189877411</td><td>   69399</td><td>1387731</td><td>    1</td><td>all</td><td>caud</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_caud.csv</td></tr>\n",
       "\t<tr><td> 5</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr5.pgen </td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr5_all.rda</td><td>181172584</td><td>   44104</td><td>1409038</td><td>    1</td><td>all</td><td>caud</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_caud.csv</td></tr>\n",
       "\t<tr><td> 6</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr6.pgen </td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr6_all.rda</td><td>170619093</td><td>  192453</td><td>1412543</td><td> 1138</td><td>all</td><td>caud</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_caud.csv</td></tr>\n",
       "\t<tr><td> 7</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr7.pgen </td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr7_all.rda</td><td>159334659</td><td>   49742</td><td>1490198</td><td>    1</td><td>all</td><td>caud</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_caud.csv</td></tr>\n",
       "\t<tr><td> 8</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr8.pgen </td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr8_all.rda</td><td>145078546</td><td>  196751</td><td>1225856</td><td> 1483</td><td>all</td><td>caud</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_caud.csv</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>15</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr15.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr15_all.rda</td><td>101892692</td><td>23433917</td><td> 786237</td><td>11250</td><td>all</td><td>hippo</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_hippo.csv</td></tr>\n",
       "\t<tr><td>16</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr16.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr16_all.rda</td><td> 90100455</td><td>   29324</td><td>1057285</td><td>  465</td><td>all</td><td>hippo</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_hippo.csv</td></tr>\n",
       "\t<tr><td>17</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr17.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr17_all.rda</td><td> 83098547</td><td>  193254</td><td>1119214</td><td> 1513</td><td>all</td><td>hippo</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_hippo.csv</td></tr>\n",
       "\t<tr><td>18</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr18.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr18_all.rda</td><td> 80257169</td><td>   83755</td><td> 644498</td><td>  386</td><td>all</td><td>hippo</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_hippo.csv</td></tr>\n",
       "\t<tr><td>19</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr19.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr19_all.rda</td><td> 58538684</td><td>  271202</td><td>1007230</td><td>    1</td><td>all</td><td>hippo</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_hippo.csv</td></tr>\n",
       "\t<tr><td>20</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr20.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr20_all.rda</td><td> 64296371</td><td>   71317</td><td> 705461</td><td>   12</td><td>all</td><td>hippo</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_hippo.csv</td></tr>\n",
       "\t<tr><td>21</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr21.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr21_all.rda</td><td> 46682679</td><td>13391623</td><td> 358983</td><td>11194</td><td>all</td><td>hippo</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_hippo.csv</td></tr>\n",
       "\t<tr><td>22</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr22.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr22_all.rda</td><td> 50767586</td><td>16603949</td><td> 543839</td><td>14844</td><td>all</td><td>hippo</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_hippo.csv</td></tr>\n",
       "\t<tr><td> 1</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr1.pgen </td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr1_EA.rda  </td><td>248918358</td><td> 1069461</td><td>2202702</td><td> 8982</td><td>EA </td><td>hippo</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_hippo.csv </td></tr>\n",
       "\t<tr><td> 2</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr2.pgen </td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr2_EA.rda  </td><td>241863783</td><td>   10001</td><td>2019984</td><td>    1</td><td>EA </td><td>hippo</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_hippo.csv </td></tr>\n",
       "\t<tr><td> 3</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr3.pgen </td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr3_EA.rda  </td><td>198099789</td><td>   11602</td><td>1538467</td><td>    1</td><td>EA </td><td>hippo</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_hippo.csv </td></tr>\n",
       "\t<tr><td> 4</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr4.pgen </td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr4_EA.rda  </td><td>189877411</td><td>   69399</td><td>1387731</td><td>    1</td><td>EA </td><td>hippo</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_hippo.csv </td></tr>\n",
       "\t<tr><td> 5</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr5.pgen </td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr5_EA.rda  </td><td>181172584</td><td>   44104</td><td>1409038</td><td>    1</td><td>EA </td><td>hippo</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_hippo.csv </td></tr>\n",
       "\t<tr><td> 6</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr6.pgen </td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr6_EA.rda  </td><td>170619093</td><td>  192453</td><td>1412543</td><td> 1138</td><td>EA </td><td>hippo</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_hippo.csv </td></tr>\n",
       "\t<tr><td> 7</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr7.pgen </td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr7_EA.rda  </td><td>159334659</td><td>   49742</td><td>1490198</td><td>    1</td><td>EA </td><td>hippo</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_hippo.csv </td></tr>\n",
       "\t<tr><td> 8</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr8.pgen </td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr8_EA.rda  </td><td>145078546</td><td>  196751</td><td>1225856</td><td> 1483</td><td>EA </td><td>hippo</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_hippo.csv </td></tr>\n",
       "\t<tr><td> 9</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr9.pgen </td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr9_EA.rda  </td><td>136933101</td><td>  164314</td><td>1064218</td><td> 2159</td><td>EA </td><td>hippo</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_hippo.csv </td></tr>\n",
       "\t<tr><td>10</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr10.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr10_EA.rda </td><td>133625493</td><td>   45719</td><td>1288155</td><td>    1</td><td>EA </td><td>hippo</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_hippo.csv </td></tr>\n",
       "\t<tr><td>11</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr11.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr11_EA.rda </td><td>135075586</td><td>  194509</td><td>1236846</td><td>    1</td><td>EA </td><td>hippo</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_hippo.csv </td></tr>\n",
       "\t<tr><td>12</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr12.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr12_EA.rda </td><td>133239922</td><td>   77807</td><td>1222018</td><td>    1</td><td>EA </td><td>hippo</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_hippo.csv </td></tr>\n",
       "\t<tr><td>13</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr13.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr13_EA.rda </td><td>114353123</td><td>18702898</td><td> 773488</td><td> 6198</td><td>EA </td><td>hippo</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_hippo.csv </td></tr>\n",
       "\t<tr><td>14</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr14.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr14_EA.rda </td><td>106883713</td><td>19991195</td><td> 809631</td><td> 1646</td><td>EA </td><td>hippo</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_hippo.csv </td></tr>\n",
       "\t<tr><td>15</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr15.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr15_EA.rda </td><td>101892692</td><td>23433917</td><td> 786237</td><td>11250</td><td>EA </td><td>hippo</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_hippo.csv </td></tr>\n",
       "\t<tr><td>16</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr16.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr16_EA.rda </td><td> 90100455</td><td>   29324</td><td>1057285</td><td>  465</td><td>EA </td><td>hippo</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_hippo.csv </td></tr>\n",
       "\t<tr><td>17</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr17.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr17_EA.rda </td><td> 83098547</td><td>  193254</td><td>1119214</td><td> 1513</td><td>EA </td><td>hippo</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_hippo.csv </td></tr>\n",
       "\t<tr><td>18</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr18.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr18_EA.rda </td><td> 80257169</td><td>   83755</td><td> 644498</td><td>  386</td><td>EA </td><td>hippo</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_hippo.csv </td></tr>\n",
       "\t<tr><td>19</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr19.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr19_EA.rda </td><td> 58538684</td><td>  271202</td><td>1007230</td><td>    1</td><td>EA </td><td>hippo</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_hippo.csv </td></tr>\n",
       "\t<tr><td>20</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr20.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr20_EA.rda </td><td> 64296371</td><td>   71317</td><td> 705461</td><td>   12</td><td>EA </td><td>hippo</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_hippo.csv </td></tr>\n",
       "\t<tr><td>21</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr21.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr21_EA.rda </td><td> 46682679</td><td>13391623</td><td> 358983</td><td>11194</td><td>EA </td><td>hippo</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_hippo.csv </td></tr>\n",
       "\t<tr><td>22</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr22.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr22_EA.rda </td><td> 50767586</td><td>16603949</td><td> 543839</td><td>14844</td><td>EA </td><td>hippo</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_hippo.csv </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 198 × 10\n",
       "\\begin{tabular}{llllllllll}\n",
       " Chr & SNP\\_data & methylation\\_data & last\\_meth\\_value\\_with\\_SNP\\_coverage & first\\_meth\\_value\\_with\\_SNP\\_coverage & last\\_meth\\_index\\_with\\_SNP\\_coverage & first\\_meth\\_index\\_with\\_SNP\\_coverage & population & region & cov\\_file\\\\\n",
       " <int> & <chr> & <chr> & <int> & <int> & <int> & <int> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t  1 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr1.pgen  & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr1\\_AA.rda  & 248918358 &  1069461 & 2202702 &  8982 & AA  & caud & /expanse/lustre/projects/jhu152/naglemi/mwas/full\\_covariates/AA\\_caud.csv \\\\\n",
       "\t  2 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr2.pgen  & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr2\\_AA.rda  & 241863783 &    10001 & 2019984 &     1 & AA  & caud & /expanse/lustre/projects/jhu152/naglemi/mwas/full\\_covariates/AA\\_caud.csv \\\\\n",
       "\t  3 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr3.pgen  & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr3\\_AA.rda  & 198099789 &    11602 & 1538467 &     1 & AA  & caud & /expanse/lustre/projects/jhu152/naglemi/mwas/full\\_covariates/AA\\_caud.csv \\\\\n",
       "\t  4 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr4.pgen  & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr4\\_AA.rda  & 189877411 &    69399 & 1387731 &     1 & AA  & caud & /expanse/lustre/projects/jhu152/naglemi/mwas/full\\_covariates/AA\\_caud.csv \\\\\n",
       "\t  5 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr5.pgen  & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr5\\_AA.rda  & 181172584 &    44104 & 1409038 &     1 & AA  & caud & /expanse/lustre/projects/jhu152/naglemi/mwas/full\\_covariates/AA\\_caud.csv \\\\\n",
       "\t  6 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr6.pgen  & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr6\\_AA.rda  & 170619093 &   192453 & 1412543 &  1138 & AA  & caud & /expanse/lustre/projects/jhu152/naglemi/mwas/full\\_covariates/AA\\_caud.csv \\\\\n",
       "\t  7 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr7.pgen  & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr7\\_AA.rda  & 159334659 &    49742 & 1490198 &     1 & AA  & caud & /expanse/lustre/projects/jhu152/naglemi/mwas/full\\_covariates/AA\\_caud.csv \\\\\n",
       "\t  8 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr8.pgen  & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr8\\_AA.rda  & 145078546 &   196751 & 1225856 &  1483 & AA  & caud & /expanse/lustre/projects/jhu152/naglemi/mwas/full\\_covariates/AA\\_caud.csv \\\\\n",
       "\t  9 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr9.pgen  & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr9\\_AA.rda  & 136933101 &   164314 & 1064218 &  2159 & AA  & caud & /expanse/lustre/projects/jhu152/naglemi/mwas/full\\_covariates/AA\\_caud.csv \\\\\n",
       "\t 10 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr10.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr10\\_AA.rda & 133625493 &    45719 & 1288155 &     1 & AA  & caud & /expanse/lustre/projects/jhu152/naglemi/mwas/full\\_covariates/AA\\_caud.csv \\\\\n",
       "\t 11 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr11.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr11\\_AA.rda & 135075586 &   194509 & 1236846 &     1 & AA  & caud & /expanse/lustre/projects/jhu152/naglemi/mwas/full\\_covariates/AA\\_caud.csv \\\\\n",
       "\t 12 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr12.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr12\\_AA.rda & 133239922 &    77807 & 1222018 &     1 & AA  & caud & /expanse/lustre/projects/jhu152/naglemi/mwas/full\\_covariates/AA\\_caud.csv \\\\\n",
       "\t 13 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr13.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr13\\_AA.rda & 114353123 & 18702898 &  773488 &  6198 & AA  & caud & /expanse/lustre/projects/jhu152/naglemi/mwas/full\\_covariates/AA\\_caud.csv \\\\\n",
       "\t 14 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr14.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr14\\_AA.rda & 106883713 & 19991195 &  809631 &  1646 & AA  & caud & /expanse/lustre/projects/jhu152/naglemi/mwas/full\\_covariates/AA\\_caud.csv \\\\\n",
       "\t 15 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr15.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr15\\_AA.rda & 101892692 & 23433917 &  786237 & 11250 & AA  & caud & /expanse/lustre/projects/jhu152/naglemi/mwas/full\\_covariates/AA\\_caud.csv \\\\\n",
       "\t 16 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr16.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr16\\_AA.rda &  90100455 &    29324 & 1057285 &   465 & AA  & caud & /expanse/lustre/projects/jhu152/naglemi/mwas/full\\_covariates/AA\\_caud.csv \\\\\n",
       "\t 17 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr17.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr17\\_AA.rda &  83098547 &   193254 & 1119214 &  1513 & AA  & caud & /expanse/lustre/projects/jhu152/naglemi/mwas/full\\_covariates/AA\\_caud.csv \\\\\n",
       "\t 18 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr18.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr18\\_AA.rda &  80257169 &    83755 &  644498 &   386 & AA  & caud & /expanse/lustre/projects/jhu152/naglemi/mwas/full\\_covariates/AA\\_caud.csv \\\\\n",
       "\t 19 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr19.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr19\\_AA.rda &  58538684 &   271202 & 1007230 &     1 & AA  & caud & /expanse/lustre/projects/jhu152/naglemi/mwas/full\\_covariates/AA\\_caud.csv \\\\\n",
       "\t 20 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr20.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr20\\_AA.rda &  64296371 &    71317 &  705461 &    12 & AA  & caud & /expanse/lustre/projects/jhu152/naglemi/mwas/full\\_covariates/AA\\_caud.csv \\\\\n",
       "\t 21 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr21.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr21\\_AA.rda &  46682679 & 13391623 &  358983 & 11194 & AA  & caud & /expanse/lustre/projects/jhu152/naglemi/mwas/full\\_covariates/AA\\_caud.csv \\\\\n",
       "\t 22 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr22.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr22\\_AA.rda &  50767586 & 16603949 &  543839 & 14844 & AA  & caud & /expanse/lustre/projects/jhu152/naglemi/mwas/full\\_covariates/AA\\_caud.csv \\\\\n",
       "\t  1 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr1.pgen  & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr1\\_all.rda & 248918358 &  1069461 & 2202702 &  8982 & all & caud & /expanse/lustre/projects/jhu152/naglemi/mwas/full\\_covariates/all\\_caud.csv\\\\\n",
       "\t  2 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr2.pgen  & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr2\\_all.rda & 241863783 &    10001 & 2019984 &     1 & all & caud & /expanse/lustre/projects/jhu152/naglemi/mwas/full\\_covariates/all\\_caud.csv\\\\\n",
       "\t  3 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr3.pgen  & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr3\\_all.rda & 198099789 &    11602 & 1538467 &     1 & all & caud & /expanse/lustre/projects/jhu152/naglemi/mwas/full\\_covariates/all\\_caud.csv\\\\\n",
       "\t  4 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr4.pgen  & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr4\\_all.rda & 189877411 &    69399 & 1387731 &     1 & all & caud & /expanse/lustre/projects/jhu152/naglemi/mwas/full\\_covariates/all\\_caud.csv\\\\\n",
       "\t  5 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr5.pgen  & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr5\\_all.rda & 181172584 &    44104 & 1409038 &     1 & all & caud & /expanse/lustre/projects/jhu152/naglemi/mwas/full\\_covariates/all\\_caud.csv\\\\\n",
       "\t  6 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr6.pgen  & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr6\\_all.rda & 170619093 &   192453 & 1412543 &  1138 & all & caud & /expanse/lustre/projects/jhu152/naglemi/mwas/full\\_covariates/all\\_caud.csv\\\\\n",
       "\t  7 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr7.pgen  & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr7\\_all.rda & 159334659 &    49742 & 1490198 &     1 & all & caud & /expanse/lustre/projects/jhu152/naglemi/mwas/full\\_covariates/all\\_caud.csv\\\\\n",
       "\t  8 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr8.pgen  & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr8\\_all.rda & 145078546 &   196751 & 1225856 &  1483 & all & caud & /expanse/lustre/projects/jhu152/naglemi/mwas/full\\_covariates/all\\_caud.csv\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t 15 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr15.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr15\\_all.rda & 101892692 & 23433917 &  786237 & 11250 & all & hippo & /expanse/lustre/projects/jhu152/naglemi/mwas/full\\_covariates/all\\_hippo.csv\\\\\n",
       "\t 16 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr16.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr16\\_all.rda &  90100455 &    29324 & 1057285 &   465 & all & hippo & /expanse/lustre/projects/jhu152/naglemi/mwas/full\\_covariates/all\\_hippo.csv\\\\\n",
       "\t 17 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr17.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr17\\_all.rda &  83098547 &   193254 & 1119214 &  1513 & all & hippo & /expanse/lustre/projects/jhu152/naglemi/mwas/full\\_covariates/all\\_hippo.csv\\\\\n",
       "\t 18 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr18.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr18\\_all.rda &  80257169 &    83755 &  644498 &   386 & all & hippo & /expanse/lustre/projects/jhu152/naglemi/mwas/full\\_covariates/all\\_hippo.csv\\\\\n",
       "\t 19 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr19.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr19\\_all.rda &  58538684 &   271202 & 1007230 &     1 & all & hippo & /expanse/lustre/projects/jhu152/naglemi/mwas/full\\_covariates/all\\_hippo.csv\\\\\n",
       "\t 20 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr20.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr20\\_all.rda &  64296371 &    71317 &  705461 &    12 & all & hippo & /expanse/lustre/projects/jhu152/naglemi/mwas/full\\_covariates/all\\_hippo.csv\\\\\n",
       "\t 21 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr21.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr21\\_all.rda &  46682679 & 13391623 &  358983 & 11194 & all & hippo & /expanse/lustre/projects/jhu152/naglemi/mwas/full\\_covariates/all\\_hippo.csv\\\\\n",
       "\t 22 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr22.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr22\\_all.rda &  50767586 & 16603949 &  543839 & 14844 & all & hippo & /expanse/lustre/projects/jhu152/naglemi/mwas/full\\_covariates/all\\_hippo.csv\\\\\n",
       "\t  1 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr1.pgen  & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr1\\_EA.rda   & 248918358 &  1069461 & 2202702 &  8982 & EA  & hippo & /expanse/lustre/projects/jhu152/naglemi/mwas/full\\_covariates/EA\\_hippo.csv \\\\\n",
       "\t  2 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr2.pgen  & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr2\\_EA.rda   & 241863783 &    10001 & 2019984 &     1 & EA  & hippo & /expanse/lustre/projects/jhu152/naglemi/mwas/full\\_covariates/EA\\_hippo.csv \\\\\n",
       "\t  3 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr3.pgen  & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr3\\_EA.rda   & 198099789 &    11602 & 1538467 &     1 & EA  & hippo & /expanse/lustre/projects/jhu152/naglemi/mwas/full\\_covariates/EA\\_hippo.csv \\\\\n",
       "\t  4 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr4.pgen  & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr4\\_EA.rda   & 189877411 &    69399 & 1387731 &     1 & EA  & hippo & /expanse/lustre/projects/jhu152/naglemi/mwas/full\\_covariates/EA\\_hippo.csv \\\\\n",
       "\t  5 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr5.pgen  & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr5\\_EA.rda   & 181172584 &    44104 & 1409038 &     1 & EA  & hippo & /expanse/lustre/projects/jhu152/naglemi/mwas/full\\_covariates/EA\\_hippo.csv \\\\\n",
       "\t  6 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr6.pgen  & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr6\\_EA.rda   & 170619093 &   192453 & 1412543 &  1138 & EA  & hippo & /expanse/lustre/projects/jhu152/naglemi/mwas/full\\_covariates/EA\\_hippo.csv \\\\\n",
       "\t  7 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr7.pgen  & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr7\\_EA.rda   & 159334659 &    49742 & 1490198 &     1 & EA  & hippo & /expanse/lustre/projects/jhu152/naglemi/mwas/full\\_covariates/EA\\_hippo.csv \\\\\n",
       "\t  8 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr8.pgen  & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr8\\_EA.rda   & 145078546 &   196751 & 1225856 &  1483 & EA  & hippo & /expanse/lustre/projects/jhu152/naglemi/mwas/full\\_covariates/EA\\_hippo.csv \\\\\n",
       "\t  9 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr9.pgen  & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr9\\_EA.rda   & 136933101 &   164314 & 1064218 &  2159 & EA  & hippo & /expanse/lustre/projects/jhu152/naglemi/mwas/full\\_covariates/EA\\_hippo.csv \\\\\n",
       "\t 10 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr10.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr10\\_EA.rda  & 133625493 &    45719 & 1288155 &     1 & EA  & hippo & /expanse/lustre/projects/jhu152/naglemi/mwas/full\\_covariates/EA\\_hippo.csv \\\\\n",
       "\t 11 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr11.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr11\\_EA.rda  & 135075586 &   194509 & 1236846 &     1 & EA  & hippo & /expanse/lustre/projects/jhu152/naglemi/mwas/full\\_covariates/EA\\_hippo.csv \\\\\n",
       "\t 12 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr12.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr12\\_EA.rda  & 133239922 &    77807 & 1222018 &     1 & EA  & hippo & /expanse/lustre/projects/jhu152/naglemi/mwas/full\\_covariates/EA\\_hippo.csv \\\\\n",
       "\t 13 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr13.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr13\\_EA.rda  & 114353123 & 18702898 &  773488 &  6198 & EA  & hippo & /expanse/lustre/projects/jhu152/naglemi/mwas/full\\_covariates/EA\\_hippo.csv \\\\\n",
       "\t 14 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr14.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr14\\_EA.rda  & 106883713 & 19991195 &  809631 &  1646 & EA  & hippo & /expanse/lustre/projects/jhu152/naglemi/mwas/full\\_covariates/EA\\_hippo.csv \\\\\n",
       "\t 15 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr15.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr15\\_EA.rda  & 101892692 & 23433917 &  786237 & 11250 & EA  & hippo & /expanse/lustre/projects/jhu152/naglemi/mwas/full\\_covariates/EA\\_hippo.csv \\\\\n",
       "\t 16 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr16.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr16\\_EA.rda  &  90100455 &    29324 & 1057285 &   465 & EA  & hippo & /expanse/lustre/projects/jhu152/naglemi/mwas/full\\_covariates/EA\\_hippo.csv \\\\\n",
       "\t 17 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr17.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr17\\_EA.rda  &  83098547 &   193254 & 1119214 &  1513 & EA  & hippo & /expanse/lustre/projects/jhu152/naglemi/mwas/full\\_covariates/EA\\_hippo.csv \\\\\n",
       "\t 18 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr18.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr18\\_EA.rda  &  80257169 &    83755 &  644498 &   386 & EA  & hippo & /expanse/lustre/projects/jhu152/naglemi/mwas/full\\_covariates/EA\\_hippo.csv \\\\\n",
       "\t 19 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr19.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr19\\_EA.rda  &  58538684 &   271202 & 1007230 &     1 & EA  & hippo & /expanse/lustre/projects/jhu152/naglemi/mwas/full\\_covariates/EA\\_hippo.csv \\\\\n",
       "\t 20 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr20.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr20\\_EA.rda  &  64296371 &    71317 &  705461 &    12 & EA  & hippo & /expanse/lustre/projects/jhu152/naglemi/mwas/full\\_covariates/EA\\_hippo.csv \\\\\n",
       "\t 21 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr21.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr21\\_EA.rda  &  46682679 & 13391623 &  358983 & 11194 & EA  & hippo & /expanse/lustre/projects/jhu152/naglemi/mwas/full\\_covariates/EA\\_hippo.csv \\\\\n",
       "\t 22 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr22.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr22\\_EA.rda  &  50767586 & 16603949 &  543839 & 14844 & EA  & hippo & /expanse/lustre/projects/jhu152/naglemi/mwas/full\\_covariates/EA\\_hippo.csv \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 198 × 10\n",
       "\n",
       "| Chr &lt;int&gt; | SNP_data &lt;chr&gt; | methylation_data &lt;chr&gt; | last_meth_value_with_SNP_coverage &lt;int&gt; | first_meth_value_with_SNP_coverage &lt;int&gt; | last_meth_index_with_SNP_coverage &lt;int&gt; | first_meth_index_with_SNP_coverage &lt;int&gt; | population &lt;chr&gt; | region &lt;chr&gt; | cov_file &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|\n",
       "|  1 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr1.pgen  | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr1_AA.rda  | 248918358 |  1069461 | 2202702 |  8982 | AA  | caud | /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_caud.csv  |\n",
       "|  2 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr2.pgen  | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr2_AA.rda  | 241863783 |    10001 | 2019984 |     1 | AA  | caud | /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_caud.csv  |\n",
       "|  3 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr3.pgen  | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr3_AA.rda  | 198099789 |    11602 | 1538467 |     1 | AA  | caud | /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_caud.csv  |\n",
       "|  4 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr4.pgen  | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr4_AA.rda  | 189877411 |    69399 | 1387731 |     1 | AA  | caud | /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_caud.csv  |\n",
       "|  5 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr5.pgen  | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr5_AA.rda  | 181172584 |    44104 | 1409038 |     1 | AA  | caud | /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_caud.csv  |\n",
       "|  6 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr6.pgen  | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr6_AA.rda  | 170619093 |   192453 | 1412543 |  1138 | AA  | caud | /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_caud.csv  |\n",
       "|  7 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr7.pgen  | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr7_AA.rda  | 159334659 |    49742 | 1490198 |     1 | AA  | caud | /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_caud.csv  |\n",
       "|  8 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr8.pgen  | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr8_AA.rda  | 145078546 |   196751 | 1225856 |  1483 | AA  | caud | /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_caud.csv  |\n",
       "|  9 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr9.pgen  | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr9_AA.rda  | 136933101 |   164314 | 1064218 |  2159 | AA  | caud | /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_caud.csv  |\n",
       "| 10 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr10.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr10_AA.rda | 133625493 |    45719 | 1288155 |     1 | AA  | caud | /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_caud.csv  |\n",
       "| 11 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr11.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr11_AA.rda | 135075586 |   194509 | 1236846 |     1 | AA  | caud | /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_caud.csv  |\n",
       "| 12 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr12.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr12_AA.rda | 133239922 |    77807 | 1222018 |     1 | AA  | caud | /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_caud.csv  |\n",
       "| 13 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr13.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr13_AA.rda | 114353123 | 18702898 |  773488 |  6198 | AA  | caud | /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_caud.csv  |\n",
       "| 14 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr14.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr14_AA.rda | 106883713 | 19991195 |  809631 |  1646 | AA  | caud | /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_caud.csv  |\n",
       "| 15 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr15.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr15_AA.rda | 101892692 | 23433917 |  786237 | 11250 | AA  | caud | /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_caud.csv  |\n",
       "| 16 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr16.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr16_AA.rda |  90100455 |    29324 | 1057285 |   465 | AA  | caud | /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_caud.csv  |\n",
       "| 17 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr17.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr17_AA.rda |  83098547 |   193254 | 1119214 |  1513 | AA  | caud | /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_caud.csv  |\n",
       "| 18 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr18.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr18_AA.rda |  80257169 |    83755 |  644498 |   386 | AA  | caud | /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_caud.csv  |\n",
       "| 19 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr19.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr19_AA.rda |  58538684 |   271202 | 1007230 |     1 | AA  | caud | /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_caud.csv  |\n",
       "| 20 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr20.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr20_AA.rda |  64296371 |    71317 |  705461 |    12 | AA  | caud | /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_caud.csv  |\n",
       "| 21 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr21.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr21_AA.rda |  46682679 | 13391623 |  358983 | 11194 | AA  | caud | /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_caud.csv  |\n",
       "| 22 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr22.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr22_AA.rda |  50767586 | 16603949 |  543839 | 14844 | AA  | caud | /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_caud.csv  |\n",
       "|  1 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr1.pgen  | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr1_all.rda | 248918358 |  1069461 | 2202702 |  8982 | all | caud | /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_caud.csv |\n",
       "|  2 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr2.pgen  | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr2_all.rda | 241863783 |    10001 | 2019984 |     1 | all | caud | /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_caud.csv |\n",
       "|  3 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr3.pgen  | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr3_all.rda | 198099789 |    11602 | 1538467 |     1 | all | caud | /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_caud.csv |\n",
       "|  4 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr4.pgen  | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr4_all.rda | 189877411 |    69399 | 1387731 |     1 | all | caud | /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_caud.csv |\n",
       "|  5 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr5.pgen  | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr5_all.rda | 181172584 |    44104 | 1409038 |     1 | all | caud | /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_caud.csv |\n",
       "|  6 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr6.pgen  | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr6_all.rda | 170619093 |   192453 | 1412543 |  1138 | all | caud | /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_caud.csv |\n",
       "|  7 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr7.pgen  | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr7_all.rda | 159334659 |    49742 | 1490198 |     1 | all | caud | /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_caud.csv |\n",
       "|  8 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr8.pgen  | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr8_all.rda | 145078546 |   196751 | 1225856 |  1483 | all | caud | /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_caud.csv |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| 15 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr15.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr15_all.rda | 101892692 | 23433917 |  786237 | 11250 | all | hippo | /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_hippo.csv |\n",
       "| 16 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr16.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr16_all.rda |  90100455 |    29324 | 1057285 |   465 | all | hippo | /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_hippo.csv |\n",
       "| 17 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr17.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr17_all.rda |  83098547 |   193254 | 1119214 |  1513 | all | hippo | /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_hippo.csv |\n",
       "| 18 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr18.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr18_all.rda |  80257169 |    83755 |  644498 |   386 | all | hippo | /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_hippo.csv |\n",
       "| 19 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr19.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr19_all.rda |  58538684 |   271202 | 1007230 |     1 | all | hippo | /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_hippo.csv |\n",
       "| 20 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr20.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr20_all.rda |  64296371 |    71317 |  705461 |    12 | all | hippo | /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_hippo.csv |\n",
       "| 21 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr21.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr21_all.rda |  46682679 | 13391623 |  358983 | 11194 | all | hippo | /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_hippo.csv |\n",
       "| 22 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr22.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr22_all.rda |  50767586 | 16603949 |  543839 | 14844 | all | hippo | /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_hippo.csv |\n",
       "|  1 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr1.pgen  | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr1_EA.rda   | 248918358 |  1069461 | 2202702 |  8982 | EA  | hippo | /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_hippo.csv  |\n",
       "|  2 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr2.pgen  | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr2_EA.rda   | 241863783 |    10001 | 2019984 |     1 | EA  | hippo | /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_hippo.csv  |\n",
       "|  3 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr3.pgen  | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr3_EA.rda   | 198099789 |    11602 | 1538467 |     1 | EA  | hippo | /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_hippo.csv  |\n",
       "|  4 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr4.pgen  | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr4_EA.rda   | 189877411 |    69399 | 1387731 |     1 | EA  | hippo | /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_hippo.csv  |\n",
       "|  5 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr5.pgen  | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr5_EA.rda   | 181172584 |    44104 | 1409038 |     1 | EA  | hippo | /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_hippo.csv  |\n",
       "|  6 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr6.pgen  | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr6_EA.rda   | 170619093 |   192453 | 1412543 |  1138 | EA  | hippo | /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_hippo.csv  |\n",
       "|  7 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr7.pgen  | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr7_EA.rda   | 159334659 |    49742 | 1490198 |     1 | EA  | hippo | /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_hippo.csv  |\n",
       "|  8 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr8.pgen  | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr8_EA.rda   | 145078546 |   196751 | 1225856 |  1483 | EA  | hippo | /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_hippo.csv  |\n",
       "|  9 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr9.pgen  | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr9_EA.rda   | 136933101 |   164314 | 1064218 |  2159 | EA  | hippo | /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_hippo.csv  |\n",
       "| 10 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr10.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr10_EA.rda  | 133625493 |    45719 | 1288155 |     1 | EA  | hippo | /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_hippo.csv  |\n",
       "| 11 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr11.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr11_EA.rda  | 135075586 |   194509 | 1236846 |     1 | EA  | hippo | /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_hippo.csv  |\n",
       "| 12 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr12.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr12_EA.rda  | 133239922 |    77807 | 1222018 |     1 | EA  | hippo | /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_hippo.csv  |\n",
       "| 13 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr13.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr13_EA.rda  | 114353123 | 18702898 |  773488 |  6198 | EA  | hippo | /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_hippo.csv  |\n",
       "| 14 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr14.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr14_EA.rda  | 106883713 | 19991195 |  809631 |  1646 | EA  | hippo | /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_hippo.csv  |\n",
       "| 15 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr15.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr15_EA.rda  | 101892692 | 23433917 |  786237 | 11250 | EA  | hippo | /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_hippo.csv  |\n",
       "| 16 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr16.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr16_EA.rda  |  90100455 |    29324 | 1057285 |   465 | EA  | hippo | /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_hippo.csv  |\n",
       "| 17 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr17.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr17_EA.rda  |  83098547 |   193254 | 1119214 |  1513 | EA  | hippo | /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_hippo.csv  |\n",
       "| 18 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr18.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr18_EA.rda  |  80257169 |    83755 |  644498 |   386 | EA  | hippo | /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_hippo.csv  |\n",
       "| 19 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr19.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr19_EA.rda  |  58538684 |   271202 | 1007230 |     1 | EA  | hippo | /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_hippo.csv  |\n",
       "| 20 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr20.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr20_EA.rda  |  64296371 |    71317 |  705461 |    12 | EA  | hippo | /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_hippo.csv  |\n",
       "| 21 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr21.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr21_EA.rda  |  46682679 | 13391623 |  358983 | 11194 | EA  | hippo | /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_hippo.csv  |\n",
       "| 22 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr22.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr22_EA.rda  |  50767586 | 16603949 |  543839 | 14844 | EA  | hippo | /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_hippo.csv  |\n",
       "\n"
      ],
      "text/plain": [
       "    Chr SNP_data                                                          \n",
       "1    1  /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr1.pgen \n",
       "2    2  /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr2.pgen \n",
       "3    3  /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr3.pgen \n",
       "4    4  /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr4.pgen \n",
       "5    5  /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr5.pgen \n",
       "6    6  /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr6.pgen \n",
       "7    7  /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr7.pgen \n",
       "8    8  /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr8.pgen \n",
       "9    9  /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr9.pgen \n",
       "10  10  /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr10.pgen\n",
       "11  11  /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr11.pgen\n",
       "12  12  /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr12.pgen\n",
       "13  13  /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr13.pgen\n",
       "14  14  /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr14.pgen\n",
       "15  15  /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr15.pgen\n",
       "16  16  /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr16.pgen\n",
       "17  17  /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr17.pgen\n",
       "18  18  /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr18.pgen\n",
       "19  19  /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr19.pgen\n",
       "20  20  /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr20.pgen\n",
       "21  21  /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr21.pgen\n",
       "22  22  /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr22.pgen\n",
       "23   1  /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr1.pgen \n",
       "24   2  /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr2.pgen \n",
       "25   3  /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr3.pgen \n",
       "26   4  /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr4.pgen \n",
       "27   5  /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr5.pgen \n",
       "28   6  /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr6.pgen \n",
       "29   7  /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr7.pgen \n",
       "30   8  /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr8.pgen \n",
       "⋮   ⋮   ⋮                                                                 \n",
       "169 15  /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr15.pgen\n",
       "170 16  /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr16.pgen\n",
       "171 17  /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr17.pgen\n",
       "172 18  /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr18.pgen\n",
       "173 19  /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr19.pgen\n",
       "174 20  /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr20.pgen\n",
       "175 21  /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr21.pgen\n",
       "176 22  /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr22.pgen\n",
       "177  1  /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr1.pgen \n",
       "178  2  /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr2.pgen \n",
       "179  3  /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr3.pgen \n",
       "180  4  /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr4.pgen \n",
       "181  5  /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr5.pgen \n",
       "182  6  /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr6.pgen \n",
       "183  7  /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr7.pgen \n",
       "184  8  /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr8.pgen \n",
       "185  9  /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr9.pgen \n",
       "186 10  /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr10.pgen\n",
       "187 11  /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr11.pgen\n",
       "188 12  /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr12.pgen\n",
       "189 13  /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr13.pgen\n",
       "190 14  /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr14.pgen\n",
       "191 15  /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr15.pgen\n",
       "192 16  /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr16.pgen\n",
       "193 17  /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr17.pgen\n",
       "194 18  /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr18.pgen\n",
       "195 19  /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr19.pgen\n",
       "196 20  /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr20.pgen\n",
       "197 21  /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr21.pgen\n",
       "198 22  /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr22.pgen\n",
       "    methylation_data                                                          \n",
       "1   /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr1_AA.rda   \n",
       "2   /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr2_AA.rda   \n",
       "3   /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr3_AA.rda   \n",
       "4   /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr4_AA.rda   \n",
       "5   /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr5_AA.rda   \n",
       "6   /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr6_AA.rda   \n",
       "7   /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr7_AA.rda   \n",
       "8   /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr8_AA.rda   \n",
       "9   /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr9_AA.rda   \n",
       "10  /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr10_AA.rda  \n",
       "11  /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr11_AA.rda  \n",
       "12  /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr12_AA.rda  \n",
       "13  /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr13_AA.rda  \n",
       "14  /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr14_AA.rda  \n",
       "15  /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr15_AA.rda  \n",
       "16  /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr16_AA.rda  \n",
       "17  /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr17_AA.rda  \n",
       "18  /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr18_AA.rda  \n",
       "19  /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr19_AA.rda  \n",
       "20  /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr20_AA.rda  \n",
       "21  /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr21_AA.rda  \n",
       "22  /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr22_AA.rda  \n",
       "23  /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr1_all.rda  \n",
       "24  /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr2_all.rda  \n",
       "25  /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr3_all.rda  \n",
       "26  /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr4_all.rda  \n",
       "27  /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr5_all.rda  \n",
       "28  /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr6_all.rda  \n",
       "29  /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr7_all.rda  \n",
       "30  /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr8_all.rda  \n",
       "⋮   ⋮                                                                         \n",
       "169 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr15_all.rda\n",
       "170 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr16_all.rda\n",
       "171 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr17_all.rda\n",
       "172 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr18_all.rda\n",
       "173 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr19_all.rda\n",
       "174 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr20_all.rda\n",
       "175 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr21_all.rda\n",
       "176 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr22_all.rda\n",
       "177 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr1_EA.rda  \n",
       "178 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr2_EA.rda  \n",
       "179 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr3_EA.rda  \n",
       "180 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr4_EA.rda  \n",
       "181 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr5_EA.rda  \n",
       "182 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr6_EA.rda  \n",
       "183 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr7_EA.rda  \n",
       "184 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr8_EA.rda  \n",
       "185 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr9_EA.rda  \n",
       "186 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr10_EA.rda \n",
       "187 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr11_EA.rda \n",
       "188 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr12_EA.rda \n",
       "189 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr13_EA.rda \n",
       "190 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr14_EA.rda \n",
       "191 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr15_EA.rda \n",
       "192 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr16_EA.rda \n",
       "193 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr17_EA.rda \n",
       "194 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr18_EA.rda \n",
       "195 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr19_EA.rda \n",
       "196 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr20_EA.rda \n",
       "197 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr21_EA.rda \n",
       "198 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr22_EA.rda \n",
       "    last_meth_value_with_SNP_coverage first_meth_value_with_SNP_coverage\n",
       "1   248918358                          1069461                          \n",
       "2   241863783                            10001                          \n",
       "3   198099789                            11602                          \n",
       "4   189877411                            69399                          \n",
       "5   181172584                            44104                          \n",
       "6   170619093                           192453                          \n",
       "7   159334659                            49742                          \n",
       "8   145078546                           196751                          \n",
       "9   136933101                           164314                          \n",
       "10  133625493                            45719                          \n",
       "11  135075586                           194509                          \n",
       "12  133239922                            77807                          \n",
       "13  114353123                         18702898                          \n",
       "14  106883713                         19991195                          \n",
       "15  101892692                         23433917                          \n",
       "16   90100455                            29324                          \n",
       "17   83098547                           193254                          \n",
       "18   80257169                            83755                          \n",
       "19   58538684                           271202                          \n",
       "20   64296371                            71317                          \n",
       "21   46682679                         13391623                          \n",
       "22   50767586                         16603949                          \n",
       "23  248918358                          1069461                          \n",
       "24  241863783                            10001                          \n",
       "25  198099789                            11602                          \n",
       "26  189877411                            69399                          \n",
       "27  181172584                            44104                          \n",
       "28  170619093                           192453                          \n",
       "29  159334659                            49742                          \n",
       "30  145078546                           196751                          \n",
       "⋮   ⋮                                 ⋮                                 \n",
       "169 101892692                         23433917                          \n",
       "170  90100455                            29324                          \n",
       "171  83098547                           193254                          \n",
       "172  80257169                            83755                          \n",
       "173  58538684                           271202                          \n",
       "174  64296371                            71317                          \n",
       "175  46682679                         13391623                          \n",
       "176  50767586                         16603949                          \n",
       "177 248918358                          1069461                          \n",
       "178 241863783                            10001                          \n",
       "179 198099789                            11602                          \n",
       "180 189877411                            69399                          \n",
       "181 181172584                            44104                          \n",
       "182 170619093                           192453                          \n",
       "183 159334659                            49742                          \n",
       "184 145078546                           196751                          \n",
       "185 136933101                           164314                          \n",
       "186 133625493                            45719                          \n",
       "187 135075586                           194509                          \n",
       "188 133239922                            77807                          \n",
       "189 114353123                         18702898                          \n",
       "190 106883713                         19991195                          \n",
       "191 101892692                         23433917                          \n",
       "192  90100455                            29324                          \n",
       "193  83098547                           193254                          \n",
       "194  80257169                            83755                          \n",
       "195  58538684                           271202                          \n",
       "196  64296371                            71317                          \n",
       "197  46682679                         13391623                          \n",
       "198  50767586                         16603949                          \n",
       "    last_meth_index_with_SNP_coverage first_meth_index_with_SNP_coverage\n",
       "1   2202702                            8982                             \n",
       "2   2019984                               1                             \n",
       "3   1538467                               1                             \n",
       "4   1387731                               1                             \n",
       "5   1409038                               1                             \n",
       "6   1412543                            1138                             \n",
       "7   1490198                               1                             \n",
       "8   1225856                            1483                             \n",
       "9   1064218                            2159                             \n",
       "10  1288155                               1                             \n",
       "11  1236846                               1                             \n",
       "12  1222018                               1                             \n",
       "13   773488                            6198                             \n",
       "14   809631                            1646                             \n",
       "15   786237                           11250                             \n",
       "16  1057285                             465                             \n",
       "17  1119214                            1513                             \n",
       "18   644498                             386                             \n",
       "19  1007230                               1                             \n",
       "20   705461                              12                             \n",
       "21   358983                           11194                             \n",
       "22   543839                           14844                             \n",
       "23  2202702                            8982                             \n",
       "24  2019984                               1                             \n",
       "25  1538467                               1                             \n",
       "26  1387731                               1                             \n",
       "27  1409038                               1                             \n",
       "28  1412543                            1138                             \n",
       "29  1490198                               1                             \n",
       "30  1225856                            1483                             \n",
       "⋮   ⋮                                 ⋮                                 \n",
       "169  786237                           11250                             \n",
       "170 1057285                             465                             \n",
       "171 1119214                            1513                             \n",
       "172  644498                             386                             \n",
       "173 1007230                               1                             \n",
       "174  705461                              12                             \n",
       "175  358983                           11194                             \n",
       "176  543839                           14844                             \n",
       "177 2202702                            8982                             \n",
       "178 2019984                               1                             \n",
       "179 1538467                               1                             \n",
       "180 1387731                               1                             \n",
       "181 1409038                               1                             \n",
       "182 1412543                            1138                             \n",
       "183 1490198                               1                             \n",
       "184 1225856                            1483                             \n",
       "185 1064218                            2159                             \n",
       "186 1288155                               1                             \n",
       "187 1236846                               1                             \n",
       "188 1222018                               1                             \n",
       "189  773488                            6198                             \n",
       "190  809631                            1646                             \n",
       "191  786237                           11250                             \n",
       "192 1057285                             465                             \n",
       "193 1119214                            1513                             \n",
       "194  644498                             386                             \n",
       "195 1007230                               1                             \n",
       "196  705461                              12                             \n",
       "197  358983                           11194                             \n",
       "198  543839                           14844                             \n",
       "    population region\n",
       "1   AA         caud  \n",
       "2   AA         caud  \n",
       "3   AA         caud  \n",
       "4   AA         caud  \n",
       "5   AA         caud  \n",
       "6   AA         caud  \n",
       "7   AA         caud  \n",
       "8   AA         caud  \n",
       "9   AA         caud  \n",
       "10  AA         caud  \n",
       "11  AA         caud  \n",
       "12  AA         caud  \n",
       "13  AA         caud  \n",
       "14  AA         caud  \n",
       "15  AA         caud  \n",
       "16  AA         caud  \n",
       "17  AA         caud  \n",
       "18  AA         caud  \n",
       "19  AA         caud  \n",
       "20  AA         caud  \n",
       "21  AA         caud  \n",
       "22  AA         caud  \n",
       "23  all        caud  \n",
       "24  all        caud  \n",
       "25  all        caud  \n",
       "26  all        caud  \n",
       "27  all        caud  \n",
       "28  all        caud  \n",
       "29  all        caud  \n",
       "30  all        caud  \n",
       "⋮   ⋮          ⋮     \n",
       "169 all        hippo \n",
       "170 all        hippo \n",
       "171 all        hippo \n",
       "172 all        hippo \n",
       "173 all        hippo \n",
       "174 all        hippo \n",
       "175 all        hippo \n",
       "176 all        hippo \n",
       "177 EA         hippo \n",
       "178 EA         hippo \n",
       "179 EA         hippo \n",
       "180 EA         hippo \n",
       "181 EA         hippo \n",
       "182 EA         hippo \n",
       "183 EA         hippo \n",
       "184 EA         hippo \n",
       "185 EA         hippo \n",
       "186 EA         hippo \n",
       "187 EA         hippo \n",
       "188 EA         hippo \n",
       "189 EA         hippo \n",
       "190 EA         hippo \n",
       "191 EA         hippo \n",
       "192 EA         hippo \n",
       "193 EA         hippo \n",
       "194 EA         hippo \n",
       "195 EA         hippo \n",
       "196 EA         hippo \n",
       "197 EA         hippo \n",
       "198 EA         hippo \n",
       "    cov_file                                                                  \n",
       "1   /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_caud.csv  \n",
       "2   /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_caud.csv  \n",
       "3   /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_caud.csv  \n",
       "4   /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_caud.csv  \n",
       "5   /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_caud.csv  \n",
       "6   /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_caud.csv  \n",
       "7   /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_caud.csv  \n",
       "8   /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_caud.csv  \n",
       "9   /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_caud.csv  \n",
       "10  /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_caud.csv  \n",
       "11  /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_caud.csv  \n",
       "12  /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_caud.csv  \n",
       "13  /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_caud.csv  \n",
       "14  /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_caud.csv  \n",
       "15  /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_caud.csv  \n",
       "16  /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_caud.csv  \n",
       "17  /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_caud.csv  \n",
       "18  /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_caud.csv  \n",
       "19  /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_caud.csv  \n",
       "20  /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_caud.csv  \n",
       "21  /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_caud.csv  \n",
       "22  /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_caud.csv  \n",
       "23  /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_caud.csv \n",
       "24  /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_caud.csv \n",
       "25  /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_caud.csv \n",
       "26  /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_caud.csv \n",
       "27  /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_caud.csv \n",
       "28  /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_caud.csv \n",
       "29  /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_caud.csv \n",
       "30  /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_caud.csv \n",
       "⋮   ⋮                                                                         \n",
       "169 /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_hippo.csv\n",
       "170 /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_hippo.csv\n",
       "171 /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_hippo.csv\n",
       "172 /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_hippo.csv\n",
       "173 /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_hippo.csv\n",
       "174 /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_hippo.csv\n",
       "175 /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_hippo.csv\n",
       "176 /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_hippo.csv\n",
       "177 /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_hippo.csv \n",
       "178 /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_hippo.csv \n",
       "179 /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_hippo.csv \n",
       "180 /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_hippo.csv \n",
       "181 /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_hippo.csv \n",
       "182 /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_hippo.csv \n",
       "183 /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_hippo.csv \n",
       "184 /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_hippo.csv \n",
       "185 /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_hippo.csv \n",
       "186 /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_hippo.csv \n",
       "187 /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_hippo.csv \n",
       "188 /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_hippo.csv \n",
       "189 /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_hippo.csv \n",
       "190 /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_hippo.csv \n",
       "191 /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_hippo.csv \n",
       "192 /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_hippo.csv \n",
       "193 /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_hippo.csv \n",
       "194 /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_hippo.csv \n",
       "195 /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_hippo.csv \n",
       "196 /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_hippo.csv \n",
       "197 /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_hippo.csv \n",
       "198 /expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_hippo.csv "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "matched_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "bb1308f1-f068-4f2c-8097-cca4fdc14c2b",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#!/bin/bash\n",
      "#SBATCH --partition=shared\n",
      "#SBATCH -A jhu152\n",
      "#SBATCH --time=24:00:00\n",
      "#SBATCH --nodes=1\n",
      "#SBATCH --ntasks-per-node=1\n",
      "#SBATCH --cpus-per-task=1\n",
      "#SBATCH --mem-per-cpu=16G\n",
      "#SBATCH --output=slurm_output_libd_chr1-chr1_all-1-2000-dynamic-1corestotal-allcorepera-20240404-110552.out\n",
      "#SBATCH --job-name=libd_chr1-chr1_all-1-2000-dynamic-1corestotal-allcorepera-20240404-110552\n",
      "conda activate mwas\n",
      "echo 'Executing Rscript with arguments: Rscript scripts/CLI.R --outdir=./output/500k_window/ --snp_data_path=/expanse/lustre/projects/jhu152/naglemi/mwas/gwas/libd_chr1.pgen --methylation_data_path=/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr1_all.rda --cov=/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_dlpfc.csv --verbose=FALSE --lambda_choice=1se --alphas=0.5 --allow_inefficient_parallelization=FALSE --n_fold=5 --window_sizes=500000 --save_evaluation_results_each_fold=FALSE --save_glmnet_object=FALSE --omit_folds_with_na_r=TRUE --methInput_rds_path=/expanse/lustre/projects/jhu152/naglemi/mwas/chr1_dfplc_all_methylation_10k_samples_a3.rds --chunk1=1 --chunk2=2000 --cv_eval_mode=dynamic --num_cores=1 --cores_per_alpha=all --tag=libd_chr1-chr1_all-1-2000-dynamic-1corestotal-allcorepera-20240404-110552'\n",
      "Rscript scripts/CLI.R --outdir=./output/500k_window/ --snp_data_path=/expanse/lustre/projects/jhu152/naglemi/mwas/gwas/libd_chr1.pgen --methylation_data_path=/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr1_all.rda --cov=/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_dlpfc.csv --verbose=FALSE --lambda_choice=1se --alphas=0.5 --allow_inefficient_parallelization=FALSE --n_fold=5 --window_sizes=500000 --save_evaluation_results_each_fold=FALSE --save_glmnet_object=FALSE --omit_folds_with_na_r=TRUE --methInput_rds_path=/expanse/lustre/projects/jhu152/naglemi/mwas/chr1_dfplc_all_methylation_10k_samples_a3.rds --chunk1=1 --chunk2=2000 --cv_eval_mode=dynamic --num_cores=1 --cores_per_alpha=all --tag=libd_chr1-chr1_all-1-2000-dynamic-1corestotal-allcorepera-20240404-110552\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Sleeping for 120 seconds\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#!/bin/bash\n",
      "#SBATCH --partition=shared\n",
      "#SBATCH -A jhu152\n",
      "#SBATCH --time=24:00:00\n",
      "#SBATCH --nodes=1\n",
      "#SBATCH --ntasks-per-node=1\n",
      "#SBATCH --cpus-per-task=12\n",
      "#SBATCH --mem-per-cpu=16G\n",
      "#SBATCH --output=slurm_output_libd_chr1-chr1_all-1-2000-dynamic-12corestotal-allcorepera-20240404-110552.out\n",
      "#SBATCH --job-name=libd_chr1-chr1_all-1-2000-dynamic-12corestotal-allcorepera-20240404-110552\n",
      "conda activate mwas\n",
      "echo 'Executing Rscript with arguments: Rscript scripts/CLI.R --outdir=./output/500k_window/ --snp_data_path=/expanse/lustre/projects/jhu152/naglemi/mwas/gwas/libd_chr1.pgen --methylation_data_path=/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr1_all.rda --cov=/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_dlpfc.csv --verbose=FALSE --lambda_choice=1se --alphas=0.5 --allow_inefficient_parallelization=FALSE --n_fold=5 --window_sizes=500000 --save_evaluation_results_each_fold=FALSE --save_glmnet_object=FALSE --omit_folds_with_na_r=TRUE --methInput_rds_path=/expanse/lustre/projects/jhu152/naglemi/mwas/chr1_dfplc_all_methylation_10k_samples_a3.rds --chunk1=1 --chunk2=2000 --cv_eval_mode=dynamic --num_cores=12 --cores_per_alpha=all --tag=libd_chr1-chr1_all-1-2000-dynamic-12corestotal-allcorepera-20240404-110552'\n",
      "Rscript scripts/CLI.R --outdir=./output/500k_window/ --snp_data_path=/expanse/lustre/projects/jhu152/naglemi/mwas/gwas/libd_chr1.pgen --methylation_data_path=/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr1_all.rda --cov=/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_dlpfc.csv --verbose=FALSE --lambda_choice=1se --alphas=0.5 --allow_inefficient_parallelization=FALSE --n_fold=5 --window_sizes=500000 --save_evaluation_results_each_fold=FALSE --save_glmnet_object=FALSE --omit_folds_with_na_r=TRUE --methInput_rds_path=/expanse/lustre/projects/jhu152/naglemi/mwas/chr1_dfplc_all_methylation_10k_samples_a3.rds --chunk1=1 --chunk2=2000 --cv_eval_mode=dynamic --num_cores=12 --cores_per_alpha=all --tag=libd_chr1-chr1_all-1-2000-dynamic-12corestotal-allcorepera-20240404-110552\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Sleeping for 120 seconds\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#!/bin/bash\n",
      "#SBATCH --partition=shared\n",
      "#SBATCH -A jhu152\n",
      "#SBATCH --time=24:00:00\n",
      "#SBATCH --nodes=1\n",
      "#SBATCH --ntasks-per-node=1\n",
      "#SBATCH --cpus-per-task=1\n",
      "#SBATCH --mem-per-cpu=16G\n",
      "#SBATCH --output=slurm_output_libd_chr1-chr1_all-2001-4000-dynamic-1corestotal-allcorepera-20240404-110552.out\n",
      "#SBATCH --job-name=libd_chr1-chr1_all-2001-4000-dynamic-1corestotal-allcorepera-20240404-110552\n",
      "conda activate mwas\n",
      "echo 'Executing Rscript with arguments: Rscript scripts/CLI.R --outdir=./output/500k_window/ --snp_data_path=/expanse/lustre/projects/jhu152/naglemi/mwas/gwas/libd_chr1.pgen --methylation_data_path=/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr1_all.rda --cov=/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_dlpfc.csv --verbose=FALSE --lambda_choice=1se --alphas=0.5 --allow_inefficient_parallelization=FALSE --n_fold=5 --window_sizes=500000 --save_evaluation_results_each_fold=FALSE --save_glmnet_object=FALSE --omit_folds_with_na_r=TRUE --methInput_rds_path=/expanse/lustre/projects/jhu152/naglemi/mwas/chr1_dfplc_all_methylation_10k_samples_a3.rds --chunk1=2001 --chunk2=4000 --cv_eval_mode=dynamic --num_cores=1 --cores_per_alpha=all --tag=libd_chr1-chr1_all-2001-4000-dynamic-1corestotal-allcorepera-20240404-110552'\n",
      "Rscript scripts/CLI.R --outdir=./output/500k_window/ --snp_data_path=/expanse/lustre/projects/jhu152/naglemi/mwas/gwas/libd_chr1.pgen --methylation_data_path=/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr1_all.rda --cov=/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_dlpfc.csv --verbose=FALSE --lambda_choice=1se --alphas=0.5 --allow_inefficient_parallelization=FALSE --n_fold=5 --window_sizes=500000 --save_evaluation_results_each_fold=FALSE --save_glmnet_object=FALSE --omit_folds_with_na_r=TRUE --methInput_rds_path=/expanse/lustre/projects/jhu152/naglemi/mwas/chr1_dfplc_all_methylation_10k_samples_a3.rds --chunk1=2001 --chunk2=4000 --cv_eval_mode=dynamic --num_cores=1 --cores_per_alpha=all --tag=libd_chr1-chr1_all-2001-4000-dynamic-1corestotal-allcorepera-20240404-110552\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Sleeping for 120 seconds\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#!/bin/bash\n",
      "#SBATCH --partition=shared\n",
      "#SBATCH -A jhu152\n",
      "#SBATCH --time=24:00:00\n",
      "#SBATCH --nodes=1\n",
      "#SBATCH --ntasks-per-node=1\n",
      "#SBATCH --cpus-per-task=12\n",
      "#SBATCH --mem-per-cpu=16G\n",
      "#SBATCH --output=slurm_output_libd_chr1-chr1_all-2001-4000-dynamic-12corestotal-allcorepera-20240404-110553.out\n",
      "#SBATCH --job-name=libd_chr1-chr1_all-2001-4000-dynamic-12corestotal-allcorepera-20240404-110553\n",
      "conda activate mwas\n",
      "echo 'Executing Rscript with arguments: Rscript scripts/CLI.R --outdir=./output/500k_window/ --snp_data_path=/expanse/lustre/projects/jhu152/naglemi/mwas/gwas/libd_chr1.pgen --methylation_data_path=/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr1_all.rda --cov=/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_dlpfc.csv --verbose=FALSE --lambda_choice=1se --alphas=0.5 --allow_inefficient_parallelization=FALSE --n_fold=5 --window_sizes=500000 --save_evaluation_results_each_fold=FALSE --save_glmnet_object=FALSE --omit_folds_with_na_r=TRUE --methInput_rds_path=/expanse/lustre/projects/jhu152/naglemi/mwas/chr1_dfplc_all_methylation_10k_samples_a3.rds --chunk1=2001 --chunk2=4000 --cv_eval_mode=dynamic --num_cores=12 --cores_per_alpha=all --tag=libd_chr1-chr1_all-2001-4000-dynamic-12corestotal-allcorepera-20240404-110553'\n",
      "Rscript scripts/CLI.R --outdir=./output/500k_window/ --snp_data_path=/expanse/lustre/projects/jhu152/naglemi/mwas/gwas/libd_chr1.pgen --methylation_data_path=/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr1_all.rda --cov=/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_dlpfc.csv --verbose=FALSE --lambda_choice=1se --alphas=0.5 --allow_inefficient_parallelization=FALSE --n_fold=5 --window_sizes=500000 --save_evaluation_results_each_fold=FALSE --save_glmnet_object=FALSE --omit_folds_with_na_r=TRUE --methInput_rds_path=/expanse/lustre/projects/jhu152/naglemi/mwas/chr1_dfplc_all_methylation_10k_samples_a3.rds --chunk1=2001 --chunk2=4000 --cv_eval_mode=dynamic --num_cores=12 --cores_per_alpha=all --tag=libd_chr1-chr1_all-2001-4000-dynamic-12corestotal-allcorepera-20240404-110553\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Sleeping for 120 seconds\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#!/bin/bash\n",
      "#SBATCH --partition=shared\n",
      "#SBATCH -A jhu152\n",
      "#SBATCH --time=24:00:00\n",
      "#SBATCH --nodes=1\n",
      "#SBATCH --ntasks-per-node=1\n",
      "#SBATCH --cpus-per-task=1\n",
      "#SBATCH --mem-per-cpu=16G\n",
      "#SBATCH --output=slurm_output_libd_chr1-chr1_all-4001-6000-dynamic-1corestotal-allcorepera-20240404-110553.out\n",
      "#SBATCH --job-name=libd_chr1-chr1_all-4001-6000-dynamic-1corestotal-allcorepera-20240404-110553\n",
      "conda activate mwas\n",
      "echo 'Executing Rscript with arguments: Rscript scripts/CLI.R --outdir=./output/500k_window/ --snp_data_path=/expanse/lustre/projects/jhu152/naglemi/mwas/gwas/libd_chr1.pgen --methylation_data_path=/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr1_all.rda --cov=/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_dlpfc.csv --verbose=FALSE --lambda_choice=1se --alphas=0.5 --allow_inefficient_parallelization=FALSE --n_fold=5 --window_sizes=500000 --save_evaluation_results_each_fold=FALSE --save_glmnet_object=FALSE --omit_folds_with_na_r=TRUE --methInput_rds_path=/expanse/lustre/projects/jhu152/naglemi/mwas/chr1_dfplc_all_methylation_10k_samples_a3.rds --chunk1=4001 --chunk2=6000 --cv_eval_mode=dynamic --num_cores=1 --cores_per_alpha=all --tag=libd_chr1-chr1_all-4001-6000-dynamic-1corestotal-allcorepera-20240404-110553'\n",
      "Rscript scripts/CLI.R --outdir=./output/500k_window/ --snp_data_path=/expanse/lustre/projects/jhu152/naglemi/mwas/gwas/libd_chr1.pgen --methylation_data_path=/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr1_all.rda --cov=/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_dlpfc.csv --verbose=FALSE --lambda_choice=1se --alphas=0.5 --allow_inefficient_parallelization=FALSE --n_fold=5 --window_sizes=500000 --save_evaluation_results_each_fold=FALSE --save_glmnet_object=FALSE --omit_folds_with_na_r=TRUE --methInput_rds_path=/expanse/lustre/projects/jhu152/naglemi/mwas/chr1_dfplc_all_methylation_10k_samples_a3.rds --chunk1=4001 --chunk2=6000 --cv_eval_mode=dynamic --num_cores=1 --cores_per_alpha=all --tag=libd_chr1-chr1_all-4001-6000-dynamic-1corestotal-allcorepera-20240404-110553\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Sleeping for 120 seconds\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#!/bin/bash\n",
      "#SBATCH --partition=shared\n",
      "#SBATCH -A jhu152\n",
      "#SBATCH --time=24:00:00\n",
      "#SBATCH --nodes=1\n",
      "#SBATCH --ntasks-per-node=1\n",
      "#SBATCH --cpus-per-task=12\n",
      "#SBATCH --mem-per-cpu=16G\n",
      "#SBATCH --output=slurm_output_libd_chr1-chr1_all-4001-6000-dynamic-12corestotal-allcorepera-20240404-110553.out\n",
      "#SBATCH --job-name=libd_chr1-chr1_all-4001-6000-dynamic-12corestotal-allcorepera-20240404-110553\n",
      "conda activate mwas\n",
      "echo 'Executing Rscript with arguments: Rscript scripts/CLI.R --outdir=./output/500k_window/ --snp_data_path=/expanse/lustre/projects/jhu152/naglemi/mwas/gwas/libd_chr1.pgen --methylation_data_path=/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr1_all.rda --cov=/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_dlpfc.csv --verbose=FALSE --lambda_choice=1se --alphas=0.5 --allow_inefficient_parallelization=FALSE --n_fold=5 --window_sizes=500000 --save_evaluation_results_each_fold=FALSE --save_glmnet_object=FALSE --omit_folds_with_na_r=TRUE --methInput_rds_path=/expanse/lustre/projects/jhu152/naglemi/mwas/chr1_dfplc_all_methylation_10k_samples_a3.rds --chunk1=4001 --chunk2=6000 --cv_eval_mode=dynamic --num_cores=12 --cores_per_alpha=all --tag=libd_chr1-chr1_all-4001-6000-dynamic-12corestotal-allcorepera-20240404-110553'\n",
      "Rscript scripts/CLI.R --outdir=./output/500k_window/ --snp_data_path=/expanse/lustre/projects/jhu152/naglemi/mwas/gwas/libd_chr1.pgen --methylation_data_path=/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr1_all.rda --cov=/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_dlpfc.csv --verbose=FALSE --lambda_choice=1se --alphas=0.5 --allow_inefficient_parallelization=FALSE --n_fold=5 --window_sizes=500000 --save_evaluation_results_each_fold=FALSE --save_glmnet_object=FALSE --omit_folds_with_na_r=TRUE --methInput_rds_path=/expanse/lustre/projects/jhu152/naglemi/mwas/chr1_dfplc_all_methylation_10k_samples_a3.rds --chunk1=4001 --chunk2=6000 --cv_eval_mode=dynamic --num_cores=12 --cores_per_alpha=all --tag=libd_chr1-chr1_all-4001-6000-dynamic-12corestotal-allcorepera-20240404-110553\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Sleeping for 120 seconds\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#!/bin/bash\n",
      "#SBATCH --partition=shared\n",
      "#SBATCH -A jhu152\n",
      "#SBATCH --time=24:00:00\n",
      "#SBATCH --nodes=1\n",
      "#SBATCH --ntasks-per-node=1\n",
      "#SBATCH --cpus-per-task=1\n",
      "#SBATCH --mem-per-cpu=16G\n",
      "#SBATCH --output=slurm_output_libd_chr1-chr1_all-6001-8000-dynamic-1corestotal-allcorepera-20240404-110553.out\n",
      "#SBATCH --job-name=libd_chr1-chr1_all-6001-8000-dynamic-1corestotal-allcorepera-20240404-110553\n",
      "conda activate mwas\n",
      "echo 'Executing Rscript with arguments: Rscript scripts/CLI.R --outdir=./output/500k_window/ --snp_data_path=/expanse/lustre/projects/jhu152/naglemi/mwas/gwas/libd_chr1.pgen --methylation_data_path=/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr1_all.rda --cov=/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_dlpfc.csv --verbose=FALSE --lambda_choice=1se --alphas=0.5 --allow_inefficient_parallelization=FALSE --n_fold=5 --window_sizes=500000 --save_evaluation_results_each_fold=FALSE --save_glmnet_object=FALSE --omit_folds_with_na_r=TRUE --methInput_rds_path=/expanse/lustre/projects/jhu152/naglemi/mwas/chr1_dfplc_all_methylation_10k_samples_a3.rds --chunk1=6001 --chunk2=8000 --cv_eval_mode=dynamic --num_cores=1 --cores_per_alpha=all --tag=libd_chr1-chr1_all-6001-8000-dynamic-1corestotal-allcorepera-20240404-110553'\n",
      "Rscript scripts/CLI.R --outdir=./output/500k_window/ --snp_data_path=/expanse/lustre/projects/jhu152/naglemi/mwas/gwas/libd_chr1.pgen --methylation_data_path=/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr1_all.rda --cov=/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_dlpfc.csv --verbose=FALSE --lambda_choice=1se --alphas=0.5 --allow_inefficient_parallelization=FALSE --n_fold=5 --window_sizes=500000 --save_evaluation_results_each_fold=FALSE --save_glmnet_object=FALSE --omit_folds_with_na_r=TRUE --methInput_rds_path=/expanse/lustre/projects/jhu152/naglemi/mwas/chr1_dfplc_all_methylation_10k_samples_a3.rds --chunk1=6001 --chunk2=8000 --cv_eval_mode=dynamic --num_cores=1 --cores_per_alpha=all --tag=libd_chr1-chr1_all-6001-8000-dynamic-1corestotal-allcorepera-20240404-110553\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Sleeping for 120 seconds\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#!/bin/bash\n",
      "#SBATCH --partition=shared\n",
      "#SBATCH -A jhu152\n",
      "#SBATCH --time=24:00:00\n",
      "#SBATCH --nodes=1\n",
      "#SBATCH --ntasks-per-node=1\n",
      "#SBATCH --cpus-per-task=12\n",
      "#SBATCH --mem-per-cpu=16G\n",
      "#SBATCH --output=slurm_output_libd_chr1-chr1_all-6001-8000-dynamic-12corestotal-allcorepera-20240404-110553.out\n",
      "#SBATCH --job-name=libd_chr1-chr1_all-6001-8000-dynamic-12corestotal-allcorepera-20240404-110553\n",
      "conda activate mwas\n",
      "echo 'Executing Rscript with arguments: Rscript scripts/CLI.R --outdir=./output/500k_window/ --snp_data_path=/expanse/lustre/projects/jhu152/naglemi/mwas/gwas/libd_chr1.pgen --methylation_data_path=/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr1_all.rda --cov=/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_dlpfc.csv --verbose=FALSE --lambda_choice=1se --alphas=0.5 --allow_inefficient_parallelization=FALSE --n_fold=5 --window_sizes=500000 --save_evaluation_results_each_fold=FALSE --save_glmnet_object=FALSE --omit_folds_with_na_r=TRUE --methInput_rds_path=/expanse/lustre/projects/jhu152/naglemi/mwas/chr1_dfplc_all_methylation_10k_samples_a3.rds --chunk1=6001 --chunk2=8000 --cv_eval_mode=dynamic --num_cores=12 --cores_per_alpha=all --tag=libd_chr1-chr1_all-6001-8000-dynamic-12corestotal-allcorepera-20240404-110553'\n",
      "Rscript scripts/CLI.R --outdir=./output/500k_window/ --snp_data_path=/expanse/lustre/projects/jhu152/naglemi/mwas/gwas/libd_chr1.pgen --methylation_data_path=/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr1_all.rda --cov=/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_dlpfc.csv --verbose=FALSE --lambda_choice=1se --alphas=0.5 --allow_inefficient_parallelization=FALSE --n_fold=5 --window_sizes=500000 --save_evaluation_results_each_fold=FALSE --save_glmnet_object=FALSE --omit_folds_with_na_r=TRUE --methInput_rds_path=/expanse/lustre/projects/jhu152/naglemi/mwas/chr1_dfplc_all_methylation_10k_samples_a3.rds --chunk1=6001 --chunk2=8000 --cv_eval_mode=dynamic --num_cores=12 --cores_per_alpha=all --tag=libd_chr1-chr1_all-6001-8000-dynamic-12corestotal-allcorepera-20240404-110553\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Sleeping for 120 seconds\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#!/bin/bash\n",
      "#SBATCH --partition=shared\n",
      "#SBATCH -A jhu152\n",
      "#SBATCH --time=24:00:00\n",
      "#SBATCH --nodes=1\n",
      "#SBATCH --ntasks-per-node=1\n",
      "#SBATCH --cpus-per-task=1\n",
      "#SBATCH --mem-per-cpu=16G\n",
      "#SBATCH --output=slurm_output_libd_chr1-chr1_all-8001-10000-dynamic-1corestotal-allcorepera-20240404-110553.out\n",
      "#SBATCH --job-name=libd_chr1-chr1_all-8001-10000-dynamic-1corestotal-allcorepera-20240404-110553\n",
      "conda activate mwas\n",
      "echo 'Executing Rscript with arguments: Rscript scripts/CLI.R --outdir=./output/500k_window/ --snp_data_path=/expanse/lustre/projects/jhu152/naglemi/mwas/gwas/libd_chr1.pgen --methylation_data_path=/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr1_all.rda --cov=/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_dlpfc.csv --verbose=FALSE --lambda_choice=1se --alphas=0.5 --allow_inefficient_parallelization=FALSE --n_fold=5 --window_sizes=500000 --save_evaluation_results_each_fold=FALSE --save_glmnet_object=FALSE --omit_folds_with_na_r=TRUE --methInput_rds_path=/expanse/lustre/projects/jhu152/naglemi/mwas/chr1_dfplc_all_methylation_10k_samples_a3.rds --chunk1=8001 --chunk2=10000 --cv_eval_mode=dynamic --num_cores=1 --cores_per_alpha=all --tag=libd_chr1-chr1_all-8001-10000-dynamic-1corestotal-allcorepera-20240404-110553'\n",
      "Rscript scripts/CLI.R --outdir=./output/500k_window/ --snp_data_path=/expanse/lustre/projects/jhu152/naglemi/mwas/gwas/libd_chr1.pgen --methylation_data_path=/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr1_all.rda --cov=/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_dlpfc.csv --verbose=FALSE --lambda_choice=1se --alphas=0.5 --allow_inefficient_parallelization=FALSE --n_fold=5 --window_sizes=500000 --save_evaluation_results_each_fold=FALSE --save_glmnet_object=FALSE --omit_folds_with_na_r=TRUE --methInput_rds_path=/expanse/lustre/projects/jhu152/naglemi/mwas/chr1_dfplc_all_methylation_10k_samples_a3.rds --chunk1=8001 --chunk2=10000 --cv_eval_mode=dynamic --num_cores=1 --cores_per_alpha=all --tag=libd_chr1-chr1_all-8001-10000-dynamic-1corestotal-allcorepera-20240404-110553\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Sleeping for 120 seconds\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#!/bin/bash\n",
      "#SBATCH --partition=shared\n",
      "#SBATCH -A jhu152\n",
      "#SBATCH --time=24:00:00\n",
      "#SBATCH --nodes=1\n",
      "#SBATCH --ntasks-per-node=1\n",
      "#SBATCH --cpus-per-task=12\n",
      "#SBATCH --mem-per-cpu=16G\n",
      "#SBATCH --output=slurm_output_libd_chr1-chr1_all-8001-10000-dynamic-12corestotal-allcorepera-20240404-110553.out\n",
      "#SBATCH --job-name=libd_chr1-chr1_all-8001-10000-dynamic-12corestotal-allcorepera-20240404-110553\n",
      "conda activate mwas\n",
      "echo 'Executing Rscript with arguments: Rscript scripts/CLI.R --outdir=./output/500k_window/ --snp_data_path=/expanse/lustre/projects/jhu152/naglemi/mwas/gwas/libd_chr1.pgen --methylation_data_path=/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr1_all.rda --cov=/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_dlpfc.csv --verbose=FALSE --lambda_choice=1se --alphas=0.5 --allow_inefficient_parallelization=FALSE --n_fold=5 --window_sizes=500000 --save_evaluation_results_each_fold=FALSE --save_glmnet_object=FALSE --omit_folds_with_na_r=TRUE --methInput_rds_path=/expanse/lustre/projects/jhu152/naglemi/mwas/chr1_dfplc_all_methylation_10k_samples_a3.rds --chunk1=8001 --chunk2=10000 --cv_eval_mode=dynamic --num_cores=12 --cores_per_alpha=all --tag=libd_chr1-chr1_all-8001-10000-dynamic-12corestotal-allcorepera-20240404-110553'\n",
      "Rscript scripts/CLI.R --outdir=./output/500k_window/ --snp_data_path=/expanse/lustre/projects/jhu152/naglemi/mwas/gwas/libd_chr1.pgen --methylation_data_path=/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr1_all.rda --cov=/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_dlpfc.csv --verbose=FALSE --lambda_choice=1se --alphas=0.5 --allow_inefficient_parallelization=FALSE --n_fold=5 --window_sizes=500000 --save_evaluation_results_each_fold=FALSE --save_glmnet_object=FALSE --omit_folds_with_na_r=TRUE --methInput_rds_path=/expanse/lustre/projects/jhu152/naglemi/mwas/chr1_dfplc_all_methylation_10k_samples_a3.rds --chunk1=8001 --chunk2=10000 --cv_eval_mode=dynamic --num_cores=12 --cores_per_alpha=all --tag=libd_chr1-chr1_all-8001-10000-dynamic-12corestotal-allcorepera-20240404-110553\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Sleeping for 120 seconds\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Constant Arguments Setup\n",
    "constant_args_df <- data.frame(\n",
    "  outdir = \"./output/500k_window/\",\n",
    "  snp_data_path = \"/expanse/lustre/projects/jhu152/naglemi/mwas/gwas/libd_chr1.pgen\",\n",
    "  methylation_data_path = \"/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr1_all.rda\",\n",
    "  cov = \"/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_dlpfc.csv\",\n",
    "  verbose = FALSE,\n",
    "  lambda_choice = \"1se\",\n",
    "  alphas = 0.5,\n",
    "  allow_inefficient_parallelization = FALSE,\n",
    "  n_fold = 5,\n",
    "  window_sizes = \"500000\",\n",
    "  #window_sizes = \"500000\",\n",
    "  save_evaluation_results_each_fold = FALSE,\n",
    "  save_glmnet_object = FALSE,\n",
    "  omit_folds_with_na_r = TRUE,\n",
    "  methInput_rds_path = \"/expanse/lustre/projects/jhu152/naglemi/mwas/chr1_dfplc_all_methylation_10k_samples_a3.rds\"\n",
    ")\n",
    "\n",
    "# Varying parameters\n",
    "cv_eval_modes <- c(\"dynamic\")\n",
    "cores_per_alphas <- c(\"all\") #NA)  # Include NA to signify the default value should be used\n",
    "num_cores_options <- c(1)\n",
    "\n",
    "for (chunk_range in 1:ncol(chunk_ranges)) {\n",
    "  constant_args_df$chunk1 <- chunk_ranges[1, chunk_range]\n",
    "  constant_args_df$chunk2 <- chunk_ranges[2, chunk_range]\n",
    "    # Loop through each combination\n",
    "    for (cv_eval_mode in cv_eval_modes) {\n",
    "      for (cores_per_alpha in cores_per_alphas) {\n",
    "        for (num_cores in num_cores_options) {\n",
    "          # Update constant_args_df for the current combination\n",
    "          constant_args_df$cv_eval_mode <- cv_eval_mode\n",
    "          constant_args_df$num_cores <- num_cores\n",
    "          if (!is.na(cores_per_alpha)) {\n",
    "            constant_args_df$cores_per_alpha <- cores_per_alpha\n",
    "          } else {\n",
    "            constant_args_df$cores_per_alpha <- NULL\n",
    "          }\n",
    "\n",
    "          # Generate tags\n",
    "          snp_base <- tools::file_path_sans_ext(basename(constant_args_df$snp_data_path))\n",
    "          meth_base <- tools::file_path_sans_ext(basename(constant_args_df$methylation_data_path))\n",
    "          datetime_str <- format(Sys.time(), \"%Y%m%d-%H%M%S\")\n",
    "          tag_pt1 <- paste(snp_base, meth_base, format(constant_args_df$chunk1, scientific = FALSE), format(constant_args_df$chunk2, scientific = FALSE), cv_eval_mode, paste0(num_cores, \"corestotal\"), ifelse(is.na(cores_per_alpha), \"defaultcore\", paste0(cores_per_alpha, \"corepera\")), sep = \"-\")\n",
    "          tag_pt2 <- datetime_str\n",
    "\n",
    "          # Check for existing file\n",
    "          existing_files <- list.files(path = constant_args_df$outdir, pattern = paste0(\"^\", tag_pt1, \".*\\\\.rds$\"), full.names = TRUE)\n",
    "          if (!overwrite && length(existing_files) > 0) {\n",
    "            message(\"File with tag \", tag_pt1, \" already exists. Skipping...\")\n",
    "            next\n",
    "          }\n",
    "\n",
    "          # Generate and print SLURM script\n",
    "          script_info <- generate_slurm_script(constant_args_df, tag_pt1, tag_pt2,\n",
    "                                               mem_per_cpu = \"16G\")\n",
    "\n",
    "          cat(script_info$script)\n",
    "\n",
    "          writeLines(script_info$script, script_info$path)\n",
    "\n",
    "          # Submit the SLURM job using the sbatch command\n",
    "          system(paste(\"sbatch\", script_info$path))\n",
    "\n",
    "          # Implement job submission limits and intervals if necessary\n",
    "          sleeptime <- 120\n",
    "          #Sys.sleep(sleeptime)\n",
    "          message(paste0(\"Sleeping for \", sleeptime, \" seconds\"))\n",
    "        }\n",
    "      }\n",
    "    }\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "6e12ac75-bd0f-40c9-ab75-a24ba56637c2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "'libd_chr1-chr1_all-8001-10000-dynamic-12corestotal-allcorepera'"
      ],
      "text/latex": [
       "'libd\\_chr1-chr1\\_all-8001-10000-dynamic-12corestotal-allcorepera'"
      ],
      "text/markdown": [
       "'libd_chr1-chr1_all-8001-10000-dynamic-12corestotal-allcorepera'"
      ],
      "text/plain": [
       "[1] \"libd_chr1-chr1_all-8001-10000-dynamic-12corestotal-allcorepera\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "tag_pt1"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c9a9aa25-1415-456b-b5b0-3ab6703941c7",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Define parameters"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "d1924a3d-0a35-4f4d-8dc6-3a18ecd1f543",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# # Constant Arguments Setup\n",
    "# constant_args_df <- data.frame(\n",
    "#   outdir = \"./output/\",\n",
    "#   chunk1 = 1000000,\n",
    "#   chunk2 = 1001000,\n",
    "#   snp_data_path = \"/dcs04/lieber/statsgen/mnagle/mwas/gwas/libd_chr1.pgen\",\n",
    "#   methylation_data_path = \"/dcs04/lieber/statsgen/mnagle/mwas/pheno/dlpfc/out/chr1_AA.rda\",\n",
    "#   #snp_data_path = \"/Users/michaelnagle/code/mwas/gwas/libd_chr1.pgen\",\n",
    "#   #methylation_data_path = \"/Users/michaelnagle/code/mwas/pheno/dlpfc/out/chr1_AA.rda\",  \n",
    "#   verbose = FALSE,\n",
    "#   lambda_choice = \"1se\",\n",
    "#   alphas = \"0.25,0.5,0.75,1\",\n",
    "#   #num_cores = \"all\",\n",
    "#   allow_inefficient_parallelization = FALSE,\n",
    "#   n_fold = 5,\n",
    "#   window_sizes = \"1000,2000,5000,10000,20000,50000,100000,500000\",\n",
    "#   #tag = format(Sys.time(), \"%Y%m%d-%H%M%S\"),\n",
    "#   save_evaluation_results_each_fold = FALSE,\n",
    "#   save_glmnet_object = FALSE,\n",
    "#   cv_eval_mode = \"dynamic\"\n",
    "# )\n",
    "\n",
    "# # Convert alphas to a comma-separated string\n",
    "# #constant_args_df$alphas <- sapply(constant_args_df$alphas, function(x) paste(x, collapse = \",\"))\n",
    "\n",
    "\n",
    "# # Varying parameters\n",
    "# cv_eval_modes <- c(\"static\", \"dynamic\")\n",
    "# cores_per_alphas <- c(\"all\", \"1\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e58c74ca-a71b-43b0-967b-185802ec8cb1",
   "metadata": {},
   "source": [
    "## Generate and deploy SLURM calls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "27def26a-2583-4b17-947b-b40cacd9e562",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# generate_slurm_script <- function(args, tag) {\n",
    "#   # Check if cores_per_alpha is set and not NULL\n",
    "#   if (!is.null(args$cores_per_alpha) && args$cores_per_alpha == \"all\") {\n",
    "#     mem_allocation <- \"0\"\n",
    "#     cpus_per_task <- \"#SBATCH --exclusive\\n\"\n",
    "#   } else {\n",
    "#     mem_allocation <- \"16G\"\n",
    "#     cpus_per_task <- \"#SBATCH --cpus-per-task=1\\n\"\n",
    "#   }\n",
    "\n",
    "#   args_string <- paste(\"--\", names(args), \"=\", args, sep = \"\", collapse = \" \")\n",
    "#   args_string <- paste(args_string, \" --tag=\", tag, sep = \"\")\n",
    "\n",
    "#   slurm_script <- paste(\n",
    "#     \"#!/bin/bash\\n\",\n",
    "#     cpus_per_task,\n",
    "#     \"#SBATCH --mem=\", mem_allocation, \"\\n\",\n",
    "#     \"#SBATCH --output=slurm_output_\", tag, \".out\\n\",\n",
    "#     \"#SBATCH --job-name=\", tag, \"\\n\",\n",
    "#     \"Rscript scripts/CLI.R \", args_string, \"\\n\",\n",
    "#     sep = \"\"\n",
    "#   )\n",
    "\n",
    "#   return(slurm_script)\n",
    "# }\n",
    "\n",
    "# # Varying parameters\n",
    "# cv_eval_modes <- c(\"static\", \"dynamic\")\n",
    "# cores_per_alphas <- c(\"1\", NA)  # Include NA to signify the default value should be used\n",
    "\n",
    "# # Loop through each combination\n",
    "# for (cv_eval_mode in cv_eval_modes) {\n",
    "#   for (cores_per_alpha in cores_per_alphas) {\n",
    "#     # Update constant_args_df for the current combination\n",
    "#     constant_args_df$cv_eval_mode <- cv_eval_mode\n",
    "#     # Use an if-statement to decide whether to assign the value or leave it to default\n",
    "#     if (!is.na(cores_per_alpha)) {\n",
    "#       constant_args_df$cores_per_alpha <- cores_per_alpha\n",
    "#     } else {\n",
    "#       constant_args_df$cores_per_alpha <- NULL  # Setting it to NULL to use the function's default\n",
    "#     }\n",
    "\n",
    "#     # Generate tag with consideration for NA\n",
    "#     snp_base <- tools::file_path_sans_ext(basename(constant_args_df$snp_data_path))\n",
    "#     meth_base <- tools::file_path_sans_ext(basename(constant_args_df$methylation_data_path))\n",
    "#     datetime_str <- format(Sys.time(), \"%Y%m%d-%H%M%S\")\n",
    "#     cores_tag <- ifelse(is.na(cores_per_alpha), \"defaultcore\", paste0(cores_per_alpha, \"core\"))\n",
    "#     tag <- paste(snp_base, meth_base, cv_eval_mode, cores_tag, datetime_str, sep = \"-\")\n",
    "\n",
    "#     # Generate and print SLURM script\n",
    "#     slurm_script <- generate_slurm_script(constant_args_df, tag)\n",
    "#     cat(slurm_script, \"\\n\\n\")\n",
    "\n",
    "#     slurm_script_path <- paste0(\"slurm_scripts/\", tag, \".sh\")  # Define path to save SLURM script\n",
    "\n",
    "      \n",
    "#     # Save SLURM script to a file\n",
    "#     writeLines(slurm_script, slurm_script_path)\n",
    "\n",
    "#     # Submit the SLURM job using the sbatch command\n",
    "#     system(paste(\"sbatch\", slurm_script_path))\n",
    "\n",
    "#     # Implement job submission limits and intervals if necessary\n",
    "#     # For example, to wait for 1 second between submissions:\n",
    "#     Sys.sleep(42)\n",
    "#     message(\"ZZZZzzzz....\")\n",
    "#   }\n",
    "# }\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5371a450-78e9-4625-bc65-67fc22693584",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d42889af-e0fb-43b7-9ea0-f84c90aacbbb",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.3.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
