{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "943739c8-3422-4b8b-8c10-7e9db3d42912",
   "metadata": {},
   "source": [
    "# 37. Plot and summarize MWAS stage 2 outputs"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "672ca198-ffaa-4892-b0ed-31f31e02eb9c",
   "metadata": {},
   "source": [
    "This actualyl comes right before notebook 40"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e57c50fc-a199-4819-95da-10af56625dea",
   "metadata": {},
   "source": [
    "## Test for one"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "74d7112e-15ba-43b4-81aa-5511ac6ac1b4",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "For example usage please run: vignette('qqman')\n",
      "\n",
      "\n",
      "\n",
      "Citation appreciated but not required:\n",
      "\n",
      "Turner, (2018). qqman: an R package for visualizing GWAS results using Q-Q and manhattan plots. Journal of Open Source Software, 3(25), 731, https://doi.org/10.21105/joss.00731.\n",
      "\n",
      "\n",
      "\n",
      "Loading required package: iterators\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(data.table)\n",
    "library(parallel)\n",
    "library(qqman)\n",
    "library(ggplot2)\n",
    "library(stringr)\n",
    "library(foreach)\n",
    "library(doParallel)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "71bac16e-2dd9-4515-abc7-957781c577e6",
   "metadata": {},
   "outputs": [],
   "source": [
    "library(data.table)\n",
    "library(parallel)\n",
    "library(stringr)\n",
    "\n",
    "df <- fread(\"12-OUT_matched_SNP_meth_cov_outputs.csv\")\n",
    "\n",
    "df <- df[!grepl(\"empty\", df$path), ]\n",
    "\n",
    "df$tag <- paste0(df$region, \"_\", df$population, \"_\", df$Chr, \"_\", df$chunk_start, \"_\", df$chunk_end)\n",
    "\n",
    "library(stringr)\n",
    "\n",
    "df$path_unmodified <- df$path\n",
    "\n",
    "df$path <- gsub(\"-caud\", \"\", df$path)\n",
    "df$path <- gsub(\"-hippo\", \"\", df$path)\n",
    "df$path <- gsub(\"-dlpfc\", \"\", df$path)\n",
    "\n",
    "datetime <- str_split_fixed(df$path, \"allcorepera-\", 2)[,2]\n",
    "datetime <- gsub(\"\\\\.rds\", \"\", datetime)\n",
    "datetime <- str_split_fixed(datetime, \"-\", 2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "c61e35cb-06b0-4429-858d-30eaa34d48e0",
   "metadata": {},
   "outputs": [],
   "source": [
    "df$date <- as.numeric(datetime[, 1])\n",
    "df$time <- as.numeric(datetime[, 2])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "5a92e272-8b2d-4cd1-8143-d87d9ef79291",
   "metadata": {},
   "outputs": [],
   "source": [
    "df[, datetime := as.POSIXct(paste0(date, \" \", time), format = \"%Y%m%d %H%M%S\")]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "a23e16ea-d712-4af7-98fd-4d77e64b0dbb",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Order the data by datetime in descending order\n",
    "setorder(df, region, Chr, population, chunk_start, chunk_end, -datetime)\n",
    "\n",
    "# Remove duplicate groups, keeping the first (most recent) entry\n",
    "df <- df[, .SD[1], by = .(region, Chr, population, chunk_start, chunk_end)]\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "8f096346-92b8-4837-805e-f063e0b219c9",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 6 × 19</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>region</th><th scope=col>Chr</th><th scope=col>population</th><th scope=col>chunk_start</th><th scope=col>chunk_end</th><th scope=col>SNP_data</th><th scope=col>methylation_data</th><th scope=col>last_meth_value_with_SNP_coverage</th><th scope=col>first_meth_value_with_SNP_coverage</th><th scope=col>last_meth_index_with_SNP_coverage</th><th scope=col>first_meth_index_with_SNP_coverage</th><th scope=col>cov_file</th><th scope=col>modified_methylation_data</th><th scope=col>path</th><th scope=col>tag</th><th scope=col>path_unmodified</th><th scope=col>date</th><th scope=col>time</th><th scope=col>datetime</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dttm&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>caud</td><td>1</td><td>AA</td><td>  8982</td><td> 28981</td><td>/dcs04/lieber/statsgen/shizhong/michael/mwas/gwas//libd_chr1.pgen</td><td>/dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/caud/out/chr1_AA.rda</td><td>248918358</td><td>1069461</td><td>2202702</td><td>8982</td><td>/dcs04/lieber/statsgen/mnagle/mwas/full_covariates/AA_caud.csv</td><td>/dcs04/lieber/statsgen/mnagle/mwas/pheno/caud/out/chr1_AA_8982-28981.rds   </td><td>..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-8982-28981-dynamic-1corestotal-allcorepera-20240510-121942.rds   </td><td>caud_AA_1_8982_28981   </td><td>..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-8982-28981-dynamic-1corestotal-allcorepera-caud-20240510-121942.rds   </td><td>20240510</td><td>121942</td><td>2024-05-10 12:19:42</td></tr>\n",
       "\t<tr><td>caud</td><td>1</td><td>AA</td><td> 28982</td><td> 48981</td><td>/dcs04/lieber/statsgen/shizhong/michael/mwas/gwas//libd_chr1.pgen</td><td>/dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/caud/out/chr1_AA.rda</td><td>248918358</td><td>1069461</td><td>2202702</td><td>8982</td><td>/dcs04/lieber/statsgen/mnagle/mwas/full_covariates/AA_caud.csv</td><td>/dcs04/lieber/statsgen/mnagle/mwas/pheno/caud/out/chr1_AA_28982-48981.rds  </td><td>..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-28982-48981-dynamic-1corestotal-allcorepera-20240510-122045.rds  </td><td>caud_AA_1_28982_48981  </td><td>..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-28982-48981-dynamic-1corestotal-allcorepera-caud-20240510-122045.rds  </td><td>20240510</td><td>122045</td><td>2024-05-10 12:20:45</td></tr>\n",
       "\t<tr><td>caud</td><td>1</td><td>AA</td><td> 48982</td><td> 68981</td><td>/dcs04/lieber/statsgen/shizhong/michael/mwas/gwas//libd_chr1.pgen</td><td>/dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/caud/out/chr1_AA.rda</td><td>248918358</td><td>1069461</td><td>2202702</td><td>8982</td><td>/dcs04/lieber/statsgen/mnagle/mwas/full_covariates/AA_caud.csv</td><td>/dcs04/lieber/statsgen/mnagle/mwas/pheno/caud/out/chr1_AA_48982-68981.rds  </td><td>..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-48982-68981-dynamic-1corestotal-allcorepera-20240510-122147.rds  </td><td>caud_AA_1_48982_68981  </td><td>..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-48982-68981-dynamic-1corestotal-allcorepera-caud-20240510-122147.rds  </td><td>20240510</td><td>122147</td><td>2024-05-10 12:21:47</td></tr>\n",
       "\t<tr><td>caud</td><td>1</td><td>AA</td><td> 68982</td><td> 88981</td><td>/dcs04/lieber/statsgen/shizhong/michael/mwas/gwas//libd_chr1.pgen</td><td>/dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/caud/out/chr1_AA.rda</td><td>248918358</td><td>1069461</td><td>2202702</td><td>8982</td><td>/dcs04/lieber/statsgen/mnagle/mwas/full_covariates/AA_caud.csv</td><td>/dcs04/lieber/statsgen/mnagle/mwas/pheno/caud/out/chr1_AA_68982-88981.rds  </td><td>..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-68982-88981-dynamic-1corestotal-allcorepera-20240510-122249.rds  </td><td>caud_AA_1_68982_88981  </td><td>..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-68982-88981-dynamic-1corestotal-allcorepera-caud-20240510-122249.rds  </td><td>20240510</td><td>122249</td><td>2024-05-10 12:22:49</td></tr>\n",
       "\t<tr><td>caud</td><td>1</td><td>AA</td><td> 88982</td><td>108981</td><td>/dcs04/lieber/statsgen/shizhong/michael/mwas/gwas//libd_chr1.pgen</td><td>/dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/caud/out/chr1_AA.rda</td><td>248918358</td><td>1069461</td><td>2202702</td><td>8982</td><td>/dcs04/lieber/statsgen/mnagle/mwas/full_covariates/AA_caud.csv</td><td>/dcs04/lieber/statsgen/mnagle/mwas/pheno/caud/out/chr1_AA_88982-108981.rds </td><td>..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-88982-108981-dynamic-1corestotal-allcorepera-20240510-122350.rds </td><td>caud_AA_1_88982_108981 </td><td>..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-88982-108981-dynamic-1corestotal-allcorepera-caud-20240510-122350.rds </td><td>20240510</td><td>122350</td><td>2024-05-10 12:23:50</td></tr>\n",
       "\t<tr><td>caud</td><td>1</td><td>AA</td><td>108982</td><td>128981</td><td>/dcs04/lieber/statsgen/shizhong/michael/mwas/gwas//libd_chr1.pgen</td><td>/dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/caud/out/chr1_AA.rda</td><td>248918358</td><td>1069461</td><td>2202702</td><td>8982</td><td>/dcs04/lieber/statsgen/mnagle/mwas/full_covariates/AA_caud.csv</td><td>/dcs04/lieber/statsgen/mnagle/mwas/pheno/caud/out/chr1_AA_108982-128981.rds</td><td>..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-108982-128981-dynamic-1corestotal-allcorepera-20240510-122451.rds</td><td>caud_AA_1_108982_128981</td><td>..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-108982-128981-dynamic-1corestotal-allcorepera-caud-20240510-122451.rds</td><td>20240510</td><td>122451</td><td>2024-05-10 12:24:51</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 6 × 19\n",
       "\\begin{tabular}{lllllllllllllllllll}\n",
       " region & Chr & population & chunk\\_start & chunk\\_end & SNP\\_data & methylation\\_data & last\\_meth\\_value\\_with\\_SNP\\_coverage & first\\_meth\\_value\\_with\\_SNP\\_coverage & last\\_meth\\_index\\_with\\_SNP\\_coverage & first\\_meth\\_index\\_with\\_SNP\\_coverage & cov\\_file & modified\\_methylation\\_data & path & tag & path\\_unmodified & date & time & datetime\\\\\n",
       " <chr> & <int> & <chr> & <int> & <dbl> & <chr> & <chr> & <int> & <int> & <int> & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <dbl> & <dbl> & <dttm>\\\\\n",
       "\\hline\n",
       "\t caud & 1 & AA &   8982 &  28981 & /dcs04/lieber/statsgen/shizhong/michael/mwas/gwas//libd\\_chr1.pgen & /dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/caud/out/chr1\\_AA.rda & 248918358 & 1069461 & 2202702 & 8982 & /dcs04/lieber/statsgen/mnagle/mwas/full\\_covariates/AA\\_caud.csv & /dcs04/lieber/statsgen/mnagle/mwas/pheno/caud/out/chr1\\_AA\\_8982-28981.rds    & ..//output\\_EXPANSE\\_a2\\_caud/libd\\_chr1-chr1\\_AA-libd\\_chr1-chr1\\_AA-8982-28981-dynamic-1corestotal-allcorepera-20240510-121942.rds    & caud\\_AA\\_1\\_8982\\_28981    & ..//output\\_EXPANSE\\_a2\\_caud/libd\\_chr1-chr1\\_AA-libd\\_chr1-chr1\\_AA-8982-28981-dynamic-1corestotal-allcorepera-caud-20240510-121942.rds    & 20240510 & 121942 & 2024-05-10 12:19:42\\\\\n",
       "\t caud & 1 & AA &  28982 &  48981 & /dcs04/lieber/statsgen/shizhong/michael/mwas/gwas//libd\\_chr1.pgen & /dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/caud/out/chr1\\_AA.rda & 248918358 & 1069461 & 2202702 & 8982 & /dcs04/lieber/statsgen/mnagle/mwas/full\\_covariates/AA\\_caud.csv & /dcs04/lieber/statsgen/mnagle/mwas/pheno/caud/out/chr1\\_AA\\_28982-48981.rds   & ..//output\\_EXPANSE\\_a2\\_caud/libd\\_chr1-chr1\\_AA-libd\\_chr1-chr1\\_AA-28982-48981-dynamic-1corestotal-allcorepera-20240510-122045.rds   & caud\\_AA\\_1\\_28982\\_48981   & ..//output\\_EXPANSE\\_a2\\_caud/libd\\_chr1-chr1\\_AA-libd\\_chr1-chr1\\_AA-28982-48981-dynamic-1corestotal-allcorepera-caud-20240510-122045.rds   & 20240510 & 122045 & 2024-05-10 12:20:45\\\\\n",
       "\t caud & 1 & AA &  48982 &  68981 & /dcs04/lieber/statsgen/shizhong/michael/mwas/gwas//libd\\_chr1.pgen & /dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/caud/out/chr1\\_AA.rda & 248918358 & 1069461 & 2202702 & 8982 & /dcs04/lieber/statsgen/mnagle/mwas/full\\_covariates/AA\\_caud.csv & /dcs04/lieber/statsgen/mnagle/mwas/pheno/caud/out/chr1\\_AA\\_48982-68981.rds   & ..//output\\_EXPANSE\\_a2\\_caud/libd\\_chr1-chr1\\_AA-libd\\_chr1-chr1\\_AA-48982-68981-dynamic-1corestotal-allcorepera-20240510-122147.rds   & caud\\_AA\\_1\\_48982\\_68981   & ..//output\\_EXPANSE\\_a2\\_caud/libd\\_chr1-chr1\\_AA-libd\\_chr1-chr1\\_AA-48982-68981-dynamic-1corestotal-allcorepera-caud-20240510-122147.rds   & 20240510 & 122147 & 2024-05-10 12:21:47\\\\\n",
       "\t caud & 1 & AA &  68982 &  88981 & /dcs04/lieber/statsgen/shizhong/michael/mwas/gwas//libd\\_chr1.pgen & /dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/caud/out/chr1\\_AA.rda & 248918358 & 1069461 & 2202702 & 8982 & /dcs04/lieber/statsgen/mnagle/mwas/full\\_covariates/AA\\_caud.csv & /dcs04/lieber/statsgen/mnagle/mwas/pheno/caud/out/chr1\\_AA\\_68982-88981.rds   & ..//output\\_EXPANSE\\_a2\\_caud/libd\\_chr1-chr1\\_AA-libd\\_chr1-chr1\\_AA-68982-88981-dynamic-1corestotal-allcorepera-20240510-122249.rds   & caud\\_AA\\_1\\_68982\\_88981   & ..//output\\_EXPANSE\\_a2\\_caud/libd\\_chr1-chr1\\_AA-libd\\_chr1-chr1\\_AA-68982-88981-dynamic-1corestotal-allcorepera-caud-20240510-122249.rds   & 20240510 & 122249 & 2024-05-10 12:22:49\\\\\n",
       "\t caud & 1 & AA &  88982 & 108981 & /dcs04/lieber/statsgen/shizhong/michael/mwas/gwas//libd\\_chr1.pgen & /dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/caud/out/chr1\\_AA.rda & 248918358 & 1069461 & 2202702 & 8982 & /dcs04/lieber/statsgen/mnagle/mwas/full\\_covariates/AA\\_caud.csv & /dcs04/lieber/statsgen/mnagle/mwas/pheno/caud/out/chr1\\_AA\\_88982-108981.rds  & ..//output\\_EXPANSE\\_a2\\_caud/libd\\_chr1-chr1\\_AA-libd\\_chr1-chr1\\_AA-88982-108981-dynamic-1corestotal-allcorepera-20240510-122350.rds  & caud\\_AA\\_1\\_88982\\_108981  & ..//output\\_EXPANSE\\_a2\\_caud/libd\\_chr1-chr1\\_AA-libd\\_chr1-chr1\\_AA-88982-108981-dynamic-1corestotal-allcorepera-caud-20240510-122350.rds  & 20240510 & 122350 & 2024-05-10 12:23:50\\\\\n",
       "\t caud & 1 & AA & 108982 & 128981 & /dcs04/lieber/statsgen/shizhong/michael/mwas/gwas//libd\\_chr1.pgen & /dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/caud/out/chr1\\_AA.rda & 248918358 & 1069461 & 2202702 & 8982 & /dcs04/lieber/statsgen/mnagle/mwas/full\\_covariates/AA\\_caud.csv & /dcs04/lieber/statsgen/mnagle/mwas/pheno/caud/out/chr1\\_AA\\_108982-128981.rds & ..//output\\_EXPANSE\\_a2\\_caud/libd\\_chr1-chr1\\_AA-libd\\_chr1-chr1\\_AA-108982-128981-dynamic-1corestotal-allcorepera-20240510-122451.rds & caud\\_AA\\_1\\_108982\\_128981 & ..//output\\_EXPANSE\\_a2\\_caud/libd\\_chr1-chr1\\_AA-libd\\_chr1-chr1\\_AA-108982-128981-dynamic-1corestotal-allcorepera-caud-20240510-122451.rds & 20240510 & 122451 & 2024-05-10 12:24:51\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 6 × 19\n",
       "\n",
       "| region &lt;chr&gt; | Chr &lt;int&gt; | population &lt;chr&gt; | chunk_start &lt;int&gt; | chunk_end &lt;dbl&gt; | SNP_data &lt;chr&gt; | methylation_data &lt;chr&gt; | last_meth_value_with_SNP_coverage &lt;int&gt; | first_meth_value_with_SNP_coverage &lt;int&gt; | last_meth_index_with_SNP_coverage &lt;int&gt; | first_meth_index_with_SNP_coverage &lt;int&gt; | cov_file &lt;chr&gt; | modified_methylation_data &lt;chr&gt; | path &lt;chr&gt; | tag &lt;chr&gt; | path_unmodified &lt;chr&gt; | date &lt;dbl&gt; | time &lt;dbl&gt; | datetime &lt;dttm&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| caud | 1 | AA |   8982 |  28981 | /dcs04/lieber/statsgen/shizhong/michael/mwas/gwas//libd_chr1.pgen | /dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/caud/out/chr1_AA.rda | 248918358 | 1069461 | 2202702 | 8982 | /dcs04/lieber/statsgen/mnagle/mwas/full_covariates/AA_caud.csv | /dcs04/lieber/statsgen/mnagle/mwas/pheno/caud/out/chr1_AA_8982-28981.rds    | ..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-8982-28981-dynamic-1corestotal-allcorepera-20240510-121942.rds    | caud_AA_1_8982_28981    | ..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-8982-28981-dynamic-1corestotal-allcorepera-caud-20240510-121942.rds    | 20240510 | 121942 | 2024-05-10 12:19:42 |\n",
       "| caud | 1 | AA |  28982 |  48981 | /dcs04/lieber/statsgen/shizhong/michael/mwas/gwas//libd_chr1.pgen | /dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/caud/out/chr1_AA.rda | 248918358 | 1069461 | 2202702 | 8982 | /dcs04/lieber/statsgen/mnagle/mwas/full_covariates/AA_caud.csv | /dcs04/lieber/statsgen/mnagle/mwas/pheno/caud/out/chr1_AA_28982-48981.rds   | ..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-28982-48981-dynamic-1corestotal-allcorepera-20240510-122045.rds   | caud_AA_1_28982_48981   | ..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-28982-48981-dynamic-1corestotal-allcorepera-caud-20240510-122045.rds   | 20240510 | 122045 | 2024-05-10 12:20:45 |\n",
       "| caud | 1 | AA |  48982 |  68981 | /dcs04/lieber/statsgen/shizhong/michael/mwas/gwas//libd_chr1.pgen | /dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/caud/out/chr1_AA.rda | 248918358 | 1069461 | 2202702 | 8982 | /dcs04/lieber/statsgen/mnagle/mwas/full_covariates/AA_caud.csv | /dcs04/lieber/statsgen/mnagle/mwas/pheno/caud/out/chr1_AA_48982-68981.rds   | ..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-48982-68981-dynamic-1corestotal-allcorepera-20240510-122147.rds   | caud_AA_1_48982_68981   | ..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-48982-68981-dynamic-1corestotal-allcorepera-caud-20240510-122147.rds   | 20240510 | 122147 | 2024-05-10 12:21:47 |\n",
       "| caud | 1 | AA |  68982 |  88981 | /dcs04/lieber/statsgen/shizhong/michael/mwas/gwas//libd_chr1.pgen | /dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/caud/out/chr1_AA.rda | 248918358 | 1069461 | 2202702 | 8982 | /dcs04/lieber/statsgen/mnagle/mwas/full_covariates/AA_caud.csv | /dcs04/lieber/statsgen/mnagle/mwas/pheno/caud/out/chr1_AA_68982-88981.rds   | ..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-68982-88981-dynamic-1corestotal-allcorepera-20240510-122249.rds   | caud_AA_1_68982_88981   | ..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-68982-88981-dynamic-1corestotal-allcorepera-caud-20240510-122249.rds   | 20240510 | 122249 | 2024-05-10 12:22:49 |\n",
       "| caud | 1 | AA |  88982 | 108981 | /dcs04/lieber/statsgen/shizhong/michael/mwas/gwas//libd_chr1.pgen | /dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/caud/out/chr1_AA.rda | 248918358 | 1069461 | 2202702 | 8982 | /dcs04/lieber/statsgen/mnagle/mwas/full_covariates/AA_caud.csv | /dcs04/lieber/statsgen/mnagle/mwas/pheno/caud/out/chr1_AA_88982-108981.rds  | ..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-88982-108981-dynamic-1corestotal-allcorepera-20240510-122350.rds  | caud_AA_1_88982_108981  | ..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-88982-108981-dynamic-1corestotal-allcorepera-caud-20240510-122350.rds  | 20240510 | 122350 | 2024-05-10 12:23:50 |\n",
       "| caud | 1 | AA | 108982 | 128981 | /dcs04/lieber/statsgen/shizhong/michael/mwas/gwas//libd_chr1.pgen | /dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/caud/out/chr1_AA.rda | 248918358 | 1069461 | 2202702 | 8982 | /dcs04/lieber/statsgen/mnagle/mwas/full_covariates/AA_caud.csv | /dcs04/lieber/statsgen/mnagle/mwas/pheno/caud/out/chr1_AA_108982-128981.rds | ..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-108982-128981-dynamic-1corestotal-allcorepera-20240510-122451.rds | caud_AA_1_108982_128981 | ..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-108982-128981-dynamic-1corestotal-allcorepera-caud-20240510-122451.rds | 20240510 | 122451 | 2024-05-10 12:24:51 |\n",
       "\n"
      ],
      "text/plain": [
       "  region Chr population chunk_start chunk_end\n",
       "1 caud   1   AA           8982       28981   \n",
       "2 caud   1   AA          28982       48981   \n",
       "3 caud   1   AA          48982       68981   \n",
       "4 caud   1   AA          68982       88981   \n",
       "5 caud   1   AA          88982      108981   \n",
       "6 caud   1   AA         108982      128981   \n",
       "  SNP_data                                                         \n",
       "1 /dcs04/lieber/statsgen/shizhong/michael/mwas/gwas//libd_chr1.pgen\n",
       "2 /dcs04/lieber/statsgen/shizhong/michael/mwas/gwas//libd_chr1.pgen\n",
       "3 /dcs04/lieber/statsgen/shizhong/michael/mwas/gwas//libd_chr1.pgen\n",
       "4 /dcs04/lieber/statsgen/shizhong/michael/mwas/gwas//libd_chr1.pgen\n",
       "5 /dcs04/lieber/statsgen/shizhong/michael/mwas/gwas//libd_chr1.pgen\n",
       "6 /dcs04/lieber/statsgen/shizhong/michael/mwas/gwas//libd_chr1.pgen\n",
       "  methylation_data                                                       \n",
       "1 /dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/caud/out/chr1_AA.rda\n",
       "2 /dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/caud/out/chr1_AA.rda\n",
       "3 /dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/caud/out/chr1_AA.rda\n",
       "4 /dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/caud/out/chr1_AA.rda\n",
       "5 /dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/caud/out/chr1_AA.rda\n",
       "6 /dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/caud/out/chr1_AA.rda\n",
       "  last_meth_value_with_SNP_coverage first_meth_value_with_SNP_coverage\n",
       "1 248918358                         1069461                           \n",
       "2 248918358                         1069461                           \n",
       "3 248918358                         1069461                           \n",
       "4 248918358                         1069461                           \n",
       "5 248918358                         1069461                           \n",
       "6 248918358                         1069461                           \n",
       "  last_meth_index_with_SNP_coverage first_meth_index_with_SNP_coverage\n",
       "1 2202702                           8982                              \n",
       "2 2202702                           8982                              \n",
       "3 2202702                           8982                              \n",
       "4 2202702                           8982                              \n",
       "5 2202702                           8982                              \n",
       "6 2202702                           8982                              \n",
       "  cov_file                                                      \n",
       "1 /dcs04/lieber/statsgen/mnagle/mwas/full_covariates/AA_caud.csv\n",
       "2 /dcs04/lieber/statsgen/mnagle/mwas/full_covariates/AA_caud.csv\n",
       "3 /dcs04/lieber/statsgen/mnagle/mwas/full_covariates/AA_caud.csv\n",
       "4 /dcs04/lieber/statsgen/mnagle/mwas/full_covariates/AA_caud.csv\n",
       "5 /dcs04/lieber/statsgen/mnagle/mwas/full_covariates/AA_caud.csv\n",
       "6 /dcs04/lieber/statsgen/mnagle/mwas/full_covariates/AA_caud.csv\n",
       "  modified_methylation_data                                                  \n",
       "1 /dcs04/lieber/statsgen/mnagle/mwas/pheno/caud/out/chr1_AA_8982-28981.rds   \n",
       "2 /dcs04/lieber/statsgen/mnagle/mwas/pheno/caud/out/chr1_AA_28982-48981.rds  \n",
       "3 /dcs04/lieber/statsgen/mnagle/mwas/pheno/caud/out/chr1_AA_48982-68981.rds  \n",
       "4 /dcs04/lieber/statsgen/mnagle/mwas/pheno/caud/out/chr1_AA_68982-88981.rds  \n",
       "5 /dcs04/lieber/statsgen/mnagle/mwas/pheno/caud/out/chr1_AA_88982-108981.rds \n",
       "6 /dcs04/lieber/statsgen/mnagle/mwas/pheno/caud/out/chr1_AA_108982-128981.rds\n",
       "  path                                                                                                                            \n",
       "1 ..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-8982-28981-dynamic-1corestotal-allcorepera-20240510-121942.rds   \n",
       "2 ..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-28982-48981-dynamic-1corestotal-allcorepera-20240510-122045.rds  \n",
       "3 ..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-48982-68981-dynamic-1corestotal-allcorepera-20240510-122147.rds  \n",
       "4 ..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-68982-88981-dynamic-1corestotal-allcorepera-20240510-122249.rds  \n",
       "5 ..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-88982-108981-dynamic-1corestotal-allcorepera-20240510-122350.rds \n",
       "6 ..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-108982-128981-dynamic-1corestotal-allcorepera-20240510-122451.rds\n",
       "  tag                    \n",
       "1 caud_AA_1_8982_28981   \n",
       "2 caud_AA_1_28982_48981  \n",
       "3 caud_AA_1_48982_68981  \n",
       "4 caud_AA_1_68982_88981  \n",
       "5 caud_AA_1_88982_108981 \n",
       "6 caud_AA_1_108982_128981\n",
       "  path_unmodified                                                                                                                      \n",
       "1 ..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-8982-28981-dynamic-1corestotal-allcorepera-caud-20240510-121942.rds   \n",
       "2 ..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-28982-48981-dynamic-1corestotal-allcorepera-caud-20240510-122045.rds  \n",
       "3 ..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-48982-68981-dynamic-1corestotal-allcorepera-caud-20240510-122147.rds  \n",
       "4 ..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-68982-88981-dynamic-1corestotal-allcorepera-caud-20240510-122249.rds  \n",
       "5 ..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-88982-108981-dynamic-1corestotal-allcorepera-caud-20240510-122350.rds \n",
       "6 ..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-108982-128981-dynamic-1corestotal-allcorepera-caud-20240510-122451.rds\n",
       "  date     time   datetime           \n",
       "1 20240510 121942 2024-05-10 12:19:42\n",
       "2 20240510 122045 2024-05-10 12:20:45\n",
       "3 20240510 122147 2024-05-10 12:21:47\n",
       "4 20240510 122249 2024-05-10 12:22:49\n",
       "5 20240510 122350 2024-05-10 12:23:50\n",
       "6 20240510 122451 2024-05-10 12:24:51"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "d164235f-ae39-44dd-8edf-792eb9d4e600",
   "metadata": {},
   "outputs": [],
   "source": [
    "fwrite(df, \"37-OUT/37-OUT_result_table_no_dups.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "82d6a352-52db-4ea6-b2b3-280d3760cb05",
   "metadata": {},
   "outputs": [],
   "source": [
    "traits <- c(\"bp\", \"scz\", \"mdd\")\n",
    "for(i in 1:length(traits)){\n",
    "    this_trait <- traits[i]\n",
    "\n",
    "    dir1 <- \"/expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud/\"\n",
    "    dir2 <- \"/expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_dlpfc/\"\n",
    "    dir3 <- \"/expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_hippo/\"\n",
    "    # List of files\n",
    "    data_files.1 <- list.files(dir1, full.names = TRUE, pattern = paste0(\"gwas_stat_\",this_trait,\"_alleleprocessed_a3\"))\n",
    "    data_files.2 <- list.files(dir2, full.names = TRUE, pattern = paste0(\"gwas_stat_\",this_trait,\"_alleleprocessed_a3\"))\n",
    "    data_files.3 <- list.files(dir3, full.names = TRUE, pattern = paste0(\"gwas_stat_\",this_trait,\"_alleleprocessed_a3\"))\n",
    "    data_files <- c(data_files.1, data_files.2, data_files.3)\n",
    "    data_files <- data_files[grepl(\"101324\", data_files)]\n",
    "    data_files <- data_files[grepl(\"a3\", data_files)]\n",
    "    data_files <- data_files[grepl(\"csv\", data_files)]\n",
    "    data_files_alt_name <- gsub(\"-caud\", \"\", data_files)\n",
    "    data_files_alt_name <- gsub(\"-dlpfc\", \"\", data_files_alt_name)\n",
    "    data_files_alt_name <- gsub(\"-hippo\", \"\", data_files_alt_name)\n",
    "    #data_files_alt_name <- data_files_alt_name[grepl(\"101324\", data_files_alt_name)]\n",
    "    # Split the file paths based on '-' and '_', keeping all parts\n",
    "    result_table <- data.table(str_split_fixed(data_files_alt_name, \"[-_]\", n = Inf))\n",
    "    \n",
    "    result_table[, c(1:3, 5, 8:11, 14:16, 19:20, 22:25)] <- NULL\n",
    "    \n",
    "    \n",
    "    colnames(result_table) <- c(\"region\", \"Chr\", \"population\", \"chunk_start\", \"chunk_end\", \"date\", \"time\", \"trait\")\n",
    "    result_table$chunk_start <- as.numeric(as.character(result_table$chunk_start))\n",
    "    result_table$chunk_end <- as.numeric(as.character(result_table$chunk_end))\n",
    "    these_paths <- as.data.frame(data_files)\n",
    "    colnames(these_paths) <- paste0(\"result_\", this_trait)\n",
    "result_table <- cbind(result_table, these_paths)\n",
    "\n",
    "    result_table$Chr <- as.numeric(gsub(\"chr\", \"\", result_table$Chr))\n",
    "    result_table$region <- gsub(\"//libd\", \"\", result_table$region)\n",
    "    result_table$tag <- paste0(result_table$region, \"_\", result_table$population, \"_\", result_table$Chr, \"_\", result_table$chunk_start, \"_\", result_table$chunk_end, \"_\", result_table$trait)\n",
    "    head(result_table)\n",
    "    # region\tChr\tpopulation\tchunk_start\tchunk_end\tdate\ttime\ttrait\tresult_bp\ttag\n",
    "    # <chr>\t<dbl>\t<chr>\t<dbl>\t<dbl>\t<chr>\t<chr>\t<chr>\t<chr>\t<chr>\n",
    "    # caud\t1\tAA\t1008982\t1028981\t20240416\t154932\tbp\t/expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-1008982-1028981-dynamic-1corestotal-allcorepera-20240416-154932_gwas_stat_bp_alleleprocessed_a3_results-101324.csv\tcaud_AA__1008982_1028981_bp\n",
    "    # caud\t1\tAA\t1008982\t1028981\t20240510\t131051\tbp\t/expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-1008982-1028981-dynamic-1corestotal-allcorepera-caud-20240510-131051_gwas_stat_bp_alleleprocessed_a3_results-101324.csv\tcaud_AA__1008982_1028981_bp\n",
    "    # caud\t1\tAA\t1028982\t1048981\t20240415\t140831\tbp\t/expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-1028982-1048981-dynamic-1corestotal-allcorepera-20240415-140831_gwas_stat_bp_alleleprocessed_a3_results-101324.csv\tcaud_AA__1028982_1048981_bp\n",
    "    # caud\t1\tAA\t1028982\t1048981\t20240510\t131153\tbp\t/expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-1028982-1048981-dynamic-1corestotal-allcorepera-caud-20240510-131153_gwas_stat_bp_alleleprocessed_a3_results-101324.csv\tcaud_AA__1028982_1048981_bp\n",
    "    # caud\t1\tAA\t1048982\t1068981\t20240415\t141231\tbp\t/expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-1048982-1068981-dynamic-1corestotal-allcorepera-20240415-141231_gwas_stat_bp_alleleprocessed_a3_results-101324.csv\tcaud_AA__1048982_1068981_bp\n",
    "    # caud\t1\tAA\t1048982\t1068981\t20240510\t131254\tbp\t/expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-1048982-1068981-dynamic-1corestotal-allcorepera-caud-20240510-131254_gwas_stat_bp_alleleprocessed_a3_results-101324.csv\tcaud_AA__1048982_1068981_bp\n",
    "    \n",
    "    \n",
    "    # for each combo of region x Chr x population x chunk_start x chunk_end, I only want the row with the most recent date, time\n",
    "    \n",
    "    # prove to me that it works by demonstrating an example to the console (and don't flood my console)\n",
    "    \n",
    "    # Convert 'date' and 'time' to a single POSIXct datetime column for accurate comparison\n",
    "    result_table[, datetime := as.POSIXct(paste0(date, \" \", time), format = \"%Y%m%d %H%M%S\")]\n",
    "    \n",
    "    # Order the data by datetime in descending order\n",
    "    setorder(result_table, region, Chr, population, chunk_start, chunk_end, -datetime)\n",
    "    \n",
    "    # Remove duplicate groups, keeping the first (most recent) entry\n",
    "    \n",
    "    result_table <- result_table[, .SD[1], by = .(region, Chr, population, chunk_start, chunk_end)]\n",
    "    \n",
    "\n",
    "    df <- df[!grepl(\"empty\", df$path), ]\n",
    "    result_table$trait <- result_table$date <- result_table$time <- result_table$tag <- result_table$datetime <- NULL\n",
    "    df <- merge(df, result_table, by = c(\"region\", \"Chr\", \"population\", \"chunk_start\", \"chunk_end\"))\n",
    "\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "de3cb497-ddff-4caf-9bb9-619f56f0c965",
   "metadata": {},
   "outputs": [],
   "source": [
    "result_table <- df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "13196e2f-4c72-454f-9b68-ba89cdf8f757",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 6 × 22</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>region</th><th scope=col>Chr</th><th scope=col>population</th><th scope=col>chunk_start</th><th scope=col>chunk_end</th><th scope=col>SNP_data</th><th scope=col>methylation_data</th><th scope=col>last_meth_value_with_SNP_coverage</th><th scope=col>first_meth_value_with_SNP_coverage</th><th scope=col>last_meth_index_with_SNP_coverage</th><th scope=col>⋯</th><th scope=col>modified_methylation_data</th><th scope=col>path</th><th scope=col>tag</th><th scope=col>path_unmodified</th><th scope=col>date</th><th scope=col>time</th><th scope=col>datetime</th><th scope=col>result_bp</th><th scope=col>result_scz</th><th scope=col>result_mdd</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>⋯</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dttm&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>caud</td><td>1</td><td>AA</td><td>  8982</td><td> 28981</td><td>/dcs04/lieber/statsgen/shizhong/michael/mwas/gwas//libd_chr1.pgen</td><td>/dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/caud/out/chr1_AA.rda</td><td>248918358</td><td>1069461</td><td>2202702</td><td>⋯</td><td>/dcs04/lieber/statsgen/mnagle/mwas/pheno/caud/out/chr1_AA_8982-28981.rds   </td><td>..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-8982-28981-dynamic-1corestotal-allcorepera-20240510-121942.rds   </td><td>caud_AA_1_8982_28981   </td><td>..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-8982-28981-dynamic-1corestotal-allcorepera-caud-20240510-121942.rds   </td><td>20240510</td><td>121942</td><td>2024-05-10 12:19:42</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-8982-28981-dynamic-1corestotal-allcorepera-caud-20240510-121942_gwas_stat_bp_alleleprocessed_a3_results-101324.csv   </td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-8982-28981-dynamic-1corestotal-allcorepera-caud-20240510-121942_gwas_stat_scz_alleleprocessed_a3_results-101324.csv   </td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-8982-28981-dynamic-1corestotal-allcorepera-caud-20240510-121942_gwas_stat_mdd_alleleprocessed_a3_results-101324.csv   </td></tr>\n",
       "\t<tr><td>caud</td><td>1</td><td>AA</td><td> 28982</td><td> 48981</td><td>/dcs04/lieber/statsgen/shizhong/michael/mwas/gwas//libd_chr1.pgen</td><td>/dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/caud/out/chr1_AA.rda</td><td>248918358</td><td>1069461</td><td>2202702</td><td>⋯</td><td>/dcs04/lieber/statsgen/mnagle/mwas/pheno/caud/out/chr1_AA_28982-48981.rds  </td><td>..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-28982-48981-dynamic-1corestotal-allcorepera-20240510-122045.rds  </td><td>caud_AA_1_28982_48981  </td><td>..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-28982-48981-dynamic-1corestotal-allcorepera-caud-20240510-122045.rds  </td><td>20240510</td><td>122045</td><td>2024-05-10 12:20:45</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-28982-48981-dynamic-1corestotal-allcorepera-caud-20240510-122045_gwas_stat_bp_alleleprocessed_a3_results-101324.csv  </td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-28982-48981-dynamic-1corestotal-allcorepera-caud-20240510-122045_gwas_stat_scz_alleleprocessed_a3_results-101324.csv  </td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-28982-48981-dynamic-1corestotal-allcorepera-caud-20240510-122045_gwas_stat_mdd_alleleprocessed_a3_results-101324.csv  </td></tr>\n",
       "\t<tr><td>caud</td><td>1</td><td>AA</td><td> 48982</td><td> 68981</td><td>/dcs04/lieber/statsgen/shizhong/michael/mwas/gwas//libd_chr1.pgen</td><td>/dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/caud/out/chr1_AA.rda</td><td>248918358</td><td>1069461</td><td>2202702</td><td>⋯</td><td>/dcs04/lieber/statsgen/mnagle/mwas/pheno/caud/out/chr1_AA_48982-68981.rds  </td><td>..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-48982-68981-dynamic-1corestotal-allcorepera-20240510-122147.rds  </td><td>caud_AA_1_48982_68981  </td><td>..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-48982-68981-dynamic-1corestotal-allcorepera-caud-20240510-122147.rds  </td><td>20240510</td><td>122147</td><td>2024-05-10 12:21:47</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-48982-68981-dynamic-1corestotal-allcorepera-caud-20240510-122147_gwas_stat_bp_alleleprocessed_a3_results-101324.csv  </td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-48982-68981-dynamic-1corestotal-allcorepera-caud-20240510-122147_gwas_stat_scz_alleleprocessed_a3_results-101324.csv  </td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-48982-68981-dynamic-1corestotal-allcorepera-caud-20240510-122147_gwas_stat_mdd_alleleprocessed_a3_results-101324.csv  </td></tr>\n",
       "\t<tr><td>caud</td><td>1</td><td>AA</td><td> 68982</td><td> 88981</td><td>/dcs04/lieber/statsgen/shizhong/michael/mwas/gwas//libd_chr1.pgen</td><td>/dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/caud/out/chr1_AA.rda</td><td>248918358</td><td>1069461</td><td>2202702</td><td>⋯</td><td>/dcs04/lieber/statsgen/mnagle/mwas/pheno/caud/out/chr1_AA_68982-88981.rds  </td><td>..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-68982-88981-dynamic-1corestotal-allcorepera-20240510-122249.rds  </td><td>caud_AA_1_68982_88981  </td><td>..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-68982-88981-dynamic-1corestotal-allcorepera-caud-20240510-122249.rds  </td><td>20240510</td><td>122249</td><td>2024-05-10 12:22:49</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-68982-88981-dynamic-1corestotal-allcorepera-caud-20240510-122249_gwas_stat_bp_alleleprocessed_a3_results-101324.csv  </td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-68982-88981-dynamic-1corestotal-allcorepera-caud-20240510-122249_gwas_stat_scz_alleleprocessed_a3_results-101324.csv  </td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-68982-88981-dynamic-1corestotal-allcorepera-caud-20240510-122249_gwas_stat_mdd_alleleprocessed_a3_results-101324.csv  </td></tr>\n",
       "\t<tr><td>caud</td><td>1</td><td>AA</td><td> 88982</td><td>108981</td><td>/dcs04/lieber/statsgen/shizhong/michael/mwas/gwas//libd_chr1.pgen</td><td>/dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/caud/out/chr1_AA.rda</td><td>248918358</td><td>1069461</td><td>2202702</td><td>⋯</td><td>/dcs04/lieber/statsgen/mnagle/mwas/pheno/caud/out/chr1_AA_88982-108981.rds </td><td>..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-88982-108981-dynamic-1corestotal-allcorepera-20240510-122350.rds </td><td>caud_AA_1_88982_108981 </td><td>..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-88982-108981-dynamic-1corestotal-allcorepera-caud-20240510-122350.rds </td><td>20240510</td><td>122350</td><td>2024-05-10 12:23:50</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-88982-108981-dynamic-1corestotal-allcorepera-caud-20240510-122350_gwas_stat_bp_alleleprocessed_a3_results-101324.csv </td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-88982-108981-dynamic-1corestotal-allcorepera-caud-20240510-122350_gwas_stat_scz_alleleprocessed_a3_results-101324.csv </td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-88982-108981-dynamic-1corestotal-allcorepera-caud-20240510-122350_gwas_stat_mdd_alleleprocessed_a3_results-101324.csv </td></tr>\n",
       "\t<tr><td>caud</td><td>1</td><td>AA</td><td>108982</td><td>128981</td><td>/dcs04/lieber/statsgen/shizhong/michael/mwas/gwas//libd_chr1.pgen</td><td>/dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/caud/out/chr1_AA.rda</td><td>248918358</td><td>1069461</td><td>2202702</td><td>⋯</td><td>/dcs04/lieber/statsgen/mnagle/mwas/pheno/caud/out/chr1_AA_108982-128981.rds</td><td>..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-108982-128981-dynamic-1corestotal-allcorepera-20240510-122451.rds</td><td>caud_AA_1_108982_128981</td><td>..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-108982-128981-dynamic-1corestotal-allcorepera-caud-20240510-122451.rds</td><td>20240510</td><td>122451</td><td>2024-05-10 12:24:51</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-108982-128981-dynamic-1corestotal-allcorepera-caud-20240510-122451_gwas_stat_bp_alleleprocessed_a3_results-101324.csv</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-108982-128981-dynamic-1corestotal-allcorepera-caud-20240510-122451_gwas_stat_scz_alleleprocessed_a3_results-101324.csv</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-108982-128981-dynamic-1corestotal-allcorepera-caud-20240510-122451_gwas_stat_mdd_alleleprocessed_a3_results-101324.csv</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 6 × 22\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " region & Chr & population & chunk\\_start & chunk\\_end & SNP\\_data & methylation\\_data & last\\_meth\\_value\\_with\\_SNP\\_coverage & first\\_meth\\_value\\_with\\_SNP\\_coverage & last\\_meth\\_index\\_with\\_SNP\\_coverage & ⋯ & modified\\_methylation\\_data & path & tag & path\\_unmodified & date & time & datetime & result\\_bp & result\\_scz & result\\_mdd\\\\\n",
       " <chr> & <int> & <chr> & <int> & <dbl> & <chr> & <chr> & <int> & <int> & <int> & ⋯ & <chr> & <chr> & <chr> & <chr> & <dbl> & <dbl> & <dttm> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t caud & 1 & AA &   8982 &  28981 & /dcs04/lieber/statsgen/shizhong/michael/mwas/gwas//libd\\_chr1.pgen & /dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/caud/out/chr1\\_AA.rda & 248918358 & 1069461 & 2202702 & ⋯ & /dcs04/lieber/statsgen/mnagle/mwas/pheno/caud/out/chr1\\_AA\\_8982-28981.rds    & ..//output\\_EXPANSE\\_a2\\_caud/libd\\_chr1-chr1\\_AA-libd\\_chr1-chr1\\_AA-8982-28981-dynamic-1corestotal-allcorepera-20240510-121942.rds    & caud\\_AA\\_1\\_8982\\_28981    & ..//output\\_EXPANSE\\_a2\\_caud/libd\\_chr1-chr1\\_AA-libd\\_chr1-chr1\\_AA-8982-28981-dynamic-1corestotal-allcorepera-caud-20240510-121942.rds    & 20240510 & 121942 & 2024-05-10 12:19:42 & /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output\\_EXPANSE\\_dt\\_caud//libd\\_chr1-chr1\\_AA-libd\\_chr1-chr1\\_AA-8982-28981-dynamic-1corestotal-allcorepera-caud-20240510-121942\\_gwas\\_stat\\_bp\\_alleleprocessed\\_a3\\_results-101324.csv    & /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output\\_EXPANSE\\_dt\\_caud//libd\\_chr1-chr1\\_AA-libd\\_chr1-chr1\\_AA-8982-28981-dynamic-1corestotal-allcorepera-caud-20240510-121942\\_gwas\\_stat\\_scz\\_alleleprocessed\\_a3\\_results-101324.csv    & /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output\\_EXPANSE\\_dt\\_caud//libd\\_chr1-chr1\\_AA-libd\\_chr1-chr1\\_AA-8982-28981-dynamic-1corestotal-allcorepera-caud-20240510-121942\\_gwas\\_stat\\_mdd\\_alleleprocessed\\_a3\\_results-101324.csv   \\\\\n",
       "\t caud & 1 & AA &  28982 &  48981 & /dcs04/lieber/statsgen/shizhong/michael/mwas/gwas//libd\\_chr1.pgen & /dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/caud/out/chr1\\_AA.rda & 248918358 & 1069461 & 2202702 & ⋯ & /dcs04/lieber/statsgen/mnagle/mwas/pheno/caud/out/chr1\\_AA\\_28982-48981.rds   & ..//output\\_EXPANSE\\_a2\\_caud/libd\\_chr1-chr1\\_AA-libd\\_chr1-chr1\\_AA-28982-48981-dynamic-1corestotal-allcorepera-20240510-122045.rds   & caud\\_AA\\_1\\_28982\\_48981   & ..//output\\_EXPANSE\\_a2\\_caud/libd\\_chr1-chr1\\_AA-libd\\_chr1-chr1\\_AA-28982-48981-dynamic-1corestotal-allcorepera-caud-20240510-122045.rds   & 20240510 & 122045 & 2024-05-10 12:20:45 & /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output\\_EXPANSE\\_dt\\_caud//libd\\_chr1-chr1\\_AA-libd\\_chr1-chr1\\_AA-28982-48981-dynamic-1corestotal-allcorepera-caud-20240510-122045\\_gwas\\_stat\\_bp\\_alleleprocessed\\_a3\\_results-101324.csv   & /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output\\_EXPANSE\\_dt\\_caud//libd\\_chr1-chr1\\_AA-libd\\_chr1-chr1\\_AA-28982-48981-dynamic-1corestotal-allcorepera-caud-20240510-122045\\_gwas\\_stat\\_scz\\_alleleprocessed\\_a3\\_results-101324.csv   & /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output\\_EXPANSE\\_dt\\_caud//libd\\_chr1-chr1\\_AA-libd\\_chr1-chr1\\_AA-28982-48981-dynamic-1corestotal-allcorepera-caud-20240510-122045\\_gwas\\_stat\\_mdd\\_alleleprocessed\\_a3\\_results-101324.csv  \\\\\n",
       "\t caud & 1 & AA &  48982 &  68981 & /dcs04/lieber/statsgen/shizhong/michael/mwas/gwas//libd\\_chr1.pgen & /dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/caud/out/chr1\\_AA.rda & 248918358 & 1069461 & 2202702 & ⋯ & /dcs04/lieber/statsgen/mnagle/mwas/pheno/caud/out/chr1\\_AA\\_48982-68981.rds   & ..//output\\_EXPANSE\\_a2\\_caud/libd\\_chr1-chr1\\_AA-libd\\_chr1-chr1\\_AA-48982-68981-dynamic-1corestotal-allcorepera-20240510-122147.rds   & caud\\_AA\\_1\\_48982\\_68981   & ..//output\\_EXPANSE\\_a2\\_caud/libd\\_chr1-chr1\\_AA-libd\\_chr1-chr1\\_AA-48982-68981-dynamic-1corestotal-allcorepera-caud-20240510-122147.rds   & 20240510 & 122147 & 2024-05-10 12:21:47 & /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output\\_EXPANSE\\_dt\\_caud//libd\\_chr1-chr1\\_AA-libd\\_chr1-chr1\\_AA-48982-68981-dynamic-1corestotal-allcorepera-caud-20240510-122147\\_gwas\\_stat\\_bp\\_alleleprocessed\\_a3\\_results-101324.csv   & /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output\\_EXPANSE\\_dt\\_caud//libd\\_chr1-chr1\\_AA-libd\\_chr1-chr1\\_AA-48982-68981-dynamic-1corestotal-allcorepera-caud-20240510-122147\\_gwas\\_stat\\_scz\\_alleleprocessed\\_a3\\_results-101324.csv   & /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output\\_EXPANSE\\_dt\\_caud//libd\\_chr1-chr1\\_AA-libd\\_chr1-chr1\\_AA-48982-68981-dynamic-1corestotal-allcorepera-caud-20240510-122147\\_gwas\\_stat\\_mdd\\_alleleprocessed\\_a3\\_results-101324.csv  \\\\\n",
       "\t caud & 1 & AA &  68982 &  88981 & /dcs04/lieber/statsgen/shizhong/michael/mwas/gwas//libd\\_chr1.pgen & /dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/caud/out/chr1\\_AA.rda & 248918358 & 1069461 & 2202702 & ⋯ & /dcs04/lieber/statsgen/mnagle/mwas/pheno/caud/out/chr1\\_AA\\_68982-88981.rds   & ..//output\\_EXPANSE\\_a2\\_caud/libd\\_chr1-chr1\\_AA-libd\\_chr1-chr1\\_AA-68982-88981-dynamic-1corestotal-allcorepera-20240510-122249.rds   & caud\\_AA\\_1\\_68982\\_88981   & ..//output\\_EXPANSE\\_a2\\_caud/libd\\_chr1-chr1\\_AA-libd\\_chr1-chr1\\_AA-68982-88981-dynamic-1corestotal-allcorepera-caud-20240510-122249.rds   & 20240510 & 122249 & 2024-05-10 12:22:49 & /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output\\_EXPANSE\\_dt\\_caud//libd\\_chr1-chr1\\_AA-libd\\_chr1-chr1\\_AA-68982-88981-dynamic-1corestotal-allcorepera-caud-20240510-122249\\_gwas\\_stat\\_bp\\_alleleprocessed\\_a3\\_results-101324.csv   & /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output\\_EXPANSE\\_dt\\_caud//libd\\_chr1-chr1\\_AA-libd\\_chr1-chr1\\_AA-68982-88981-dynamic-1corestotal-allcorepera-caud-20240510-122249\\_gwas\\_stat\\_scz\\_alleleprocessed\\_a3\\_results-101324.csv   & /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output\\_EXPANSE\\_dt\\_caud//libd\\_chr1-chr1\\_AA-libd\\_chr1-chr1\\_AA-68982-88981-dynamic-1corestotal-allcorepera-caud-20240510-122249\\_gwas\\_stat\\_mdd\\_alleleprocessed\\_a3\\_results-101324.csv  \\\\\n",
       "\t caud & 1 & AA &  88982 & 108981 & /dcs04/lieber/statsgen/shizhong/michael/mwas/gwas//libd\\_chr1.pgen & /dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/caud/out/chr1\\_AA.rda & 248918358 & 1069461 & 2202702 & ⋯ & /dcs04/lieber/statsgen/mnagle/mwas/pheno/caud/out/chr1\\_AA\\_88982-108981.rds  & ..//output\\_EXPANSE\\_a2\\_caud/libd\\_chr1-chr1\\_AA-libd\\_chr1-chr1\\_AA-88982-108981-dynamic-1corestotal-allcorepera-20240510-122350.rds  & caud\\_AA\\_1\\_88982\\_108981  & ..//output\\_EXPANSE\\_a2\\_caud/libd\\_chr1-chr1\\_AA-libd\\_chr1-chr1\\_AA-88982-108981-dynamic-1corestotal-allcorepera-caud-20240510-122350.rds  & 20240510 & 122350 & 2024-05-10 12:23:50 & /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output\\_EXPANSE\\_dt\\_caud//libd\\_chr1-chr1\\_AA-libd\\_chr1-chr1\\_AA-88982-108981-dynamic-1corestotal-allcorepera-caud-20240510-122350\\_gwas\\_stat\\_bp\\_alleleprocessed\\_a3\\_results-101324.csv  & /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output\\_EXPANSE\\_dt\\_caud//libd\\_chr1-chr1\\_AA-libd\\_chr1-chr1\\_AA-88982-108981-dynamic-1corestotal-allcorepera-caud-20240510-122350\\_gwas\\_stat\\_scz\\_alleleprocessed\\_a3\\_results-101324.csv  & /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output\\_EXPANSE\\_dt\\_caud//libd\\_chr1-chr1\\_AA-libd\\_chr1-chr1\\_AA-88982-108981-dynamic-1corestotal-allcorepera-caud-20240510-122350\\_gwas\\_stat\\_mdd\\_alleleprocessed\\_a3\\_results-101324.csv \\\\\n",
       "\t caud & 1 & AA & 108982 & 128981 & /dcs04/lieber/statsgen/shizhong/michael/mwas/gwas//libd\\_chr1.pgen & /dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/caud/out/chr1\\_AA.rda & 248918358 & 1069461 & 2202702 & ⋯ & /dcs04/lieber/statsgen/mnagle/mwas/pheno/caud/out/chr1\\_AA\\_108982-128981.rds & ..//output\\_EXPANSE\\_a2\\_caud/libd\\_chr1-chr1\\_AA-libd\\_chr1-chr1\\_AA-108982-128981-dynamic-1corestotal-allcorepera-20240510-122451.rds & caud\\_AA\\_1\\_108982\\_128981 & ..//output\\_EXPANSE\\_a2\\_caud/libd\\_chr1-chr1\\_AA-libd\\_chr1-chr1\\_AA-108982-128981-dynamic-1corestotal-allcorepera-caud-20240510-122451.rds & 20240510 & 122451 & 2024-05-10 12:24:51 & /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output\\_EXPANSE\\_dt\\_caud//libd\\_chr1-chr1\\_AA-libd\\_chr1-chr1\\_AA-108982-128981-dynamic-1corestotal-allcorepera-caud-20240510-122451\\_gwas\\_stat\\_bp\\_alleleprocessed\\_a3\\_results-101324.csv & /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output\\_EXPANSE\\_dt\\_caud//libd\\_chr1-chr1\\_AA-libd\\_chr1-chr1\\_AA-108982-128981-dynamic-1corestotal-allcorepera-caud-20240510-122451\\_gwas\\_stat\\_scz\\_alleleprocessed\\_a3\\_results-101324.csv & /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output\\_EXPANSE\\_dt\\_caud//libd\\_chr1-chr1\\_AA-libd\\_chr1-chr1\\_AA-108982-128981-dynamic-1corestotal-allcorepera-caud-20240510-122451\\_gwas\\_stat\\_mdd\\_alleleprocessed\\_a3\\_results-101324.csv\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 6 × 22\n",
       "\n",
       "| region &lt;chr&gt; | Chr &lt;int&gt; | population &lt;chr&gt; | chunk_start &lt;int&gt; | chunk_end &lt;dbl&gt; | SNP_data &lt;chr&gt; | methylation_data &lt;chr&gt; | last_meth_value_with_SNP_coverage &lt;int&gt; | first_meth_value_with_SNP_coverage &lt;int&gt; | last_meth_index_with_SNP_coverage &lt;int&gt; | ⋯ ⋯ | modified_methylation_data &lt;chr&gt; | path &lt;chr&gt; | tag &lt;chr&gt; | path_unmodified &lt;chr&gt; | date &lt;dbl&gt; | time &lt;dbl&gt; | datetime &lt;dttm&gt; | result_bp &lt;chr&gt; | result_scz &lt;chr&gt; | result_mdd &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| caud | 1 | AA |   8982 |  28981 | /dcs04/lieber/statsgen/shizhong/michael/mwas/gwas//libd_chr1.pgen | /dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/caud/out/chr1_AA.rda | 248918358 | 1069461 | 2202702 | ⋯ | /dcs04/lieber/statsgen/mnagle/mwas/pheno/caud/out/chr1_AA_8982-28981.rds    | ..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-8982-28981-dynamic-1corestotal-allcorepera-20240510-121942.rds    | caud_AA_1_8982_28981    | ..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-8982-28981-dynamic-1corestotal-allcorepera-caud-20240510-121942.rds    | 20240510 | 121942 | 2024-05-10 12:19:42 | /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-8982-28981-dynamic-1corestotal-allcorepera-caud-20240510-121942_gwas_stat_bp_alleleprocessed_a3_results-101324.csv    | /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-8982-28981-dynamic-1corestotal-allcorepera-caud-20240510-121942_gwas_stat_scz_alleleprocessed_a3_results-101324.csv    | /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-8982-28981-dynamic-1corestotal-allcorepera-caud-20240510-121942_gwas_stat_mdd_alleleprocessed_a3_results-101324.csv    |\n",
       "| caud | 1 | AA |  28982 |  48981 | /dcs04/lieber/statsgen/shizhong/michael/mwas/gwas//libd_chr1.pgen | /dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/caud/out/chr1_AA.rda | 248918358 | 1069461 | 2202702 | ⋯ | /dcs04/lieber/statsgen/mnagle/mwas/pheno/caud/out/chr1_AA_28982-48981.rds   | ..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-28982-48981-dynamic-1corestotal-allcorepera-20240510-122045.rds   | caud_AA_1_28982_48981   | ..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-28982-48981-dynamic-1corestotal-allcorepera-caud-20240510-122045.rds   | 20240510 | 122045 | 2024-05-10 12:20:45 | /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-28982-48981-dynamic-1corestotal-allcorepera-caud-20240510-122045_gwas_stat_bp_alleleprocessed_a3_results-101324.csv   | /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-28982-48981-dynamic-1corestotal-allcorepera-caud-20240510-122045_gwas_stat_scz_alleleprocessed_a3_results-101324.csv   | /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-28982-48981-dynamic-1corestotal-allcorepera-caud-20240510-122045_gwas_stat_mdd_alleleprocessed_a3_results-101324.csv   |\n",
       "| caud | 1 | AA |  48982 |  68981 | /dcs04/lieber/statsgen/shizhong/michael/mwas/gwas//libd_chr1.pgen | /dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/caud/out/chr1_AA.rda | 248918358 | 1069461 | 2202702 | ⋯ | /dcs04/lieber/statsgen/mnagle/mwas/pheno/caud/out/chr1_AA_48982-68981.rds   | ..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-48982-68981-dynamic-1corestotal-allcorepera-20240510-122147.rds   | caud_AA_1_48982_68981   | ..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-48982-68981-dynamic-1corestotal-allcorepera-caud-20240510-122147.rds   | 20240510 | 122147 | 2024-05-10 12:21:47 | /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-48982-68981-dynamic-1corestotal-allcorepera-caud-20240510-122147_gwas_stat_bp_alleleprocessed_a3_results-101324.csv   | /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-48982-68981-dynamic-1corestotal-allcorepera-caud-20240510-122147_gwas_stat_scz_alleleprocessed_a3_results-101324.csv   | /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-48982-68981-dynamic-1corestotal-allcorepera-caud-20240510-122147_gwas_stat_mdd_alleleprocessed_a3_results-101324.csv   |\n",
       "| caud | 1 | AA |  68982 |  88981 | /dcs04/lieber/statsgen/shizhong/michael/mwas/gwas//libd_chr1.pgen | /dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/caud/out/chr1_AA.rda | 248918358 | 1069461 | 2202702 | ⋯ | /dcs04/lieber/statsgen/mnagle/mwas/pheno/caud/out/chr1_AA_68982-88981.rds   | ..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-68982-88981-dynamic-1corestotal-allcorepera-20240510-122249.rds   | caud_AA_1_68982_88981   | ..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-68982-88981-dynamic-1corestotal-allcorepera-caud-20240510-122249.rds   | 20240510 | 122249 | 2024-05-10 12:22:49 | /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-68982-88981-dynamic-1corestotal-allcorepera-caud-20240510-122249_gwas_stat_bp_alleleprocessed_a3_results-101324.csv   | /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-68982-88981-dynamic-1corestotal-allcorepera-caud-20240510-122249_gwas_stat_scz_alleleprocessed_a3_results-101324.csv   | /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-68982-88981-dynamic-1corestotal-allcorepera-caud-20240510-122249_gwas_stat_mdd_alleleprocessed_a3_results-101324.csv   |\n",
       "| caud | 1 | AA |  88982 | 108981 | /dcs04/lieber/statsgen/shizhong/michael/mwas/gwas//libd_chr1.pgen | /dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/caud/out/chr1_AA.rda | 248918358 | 1069461 | 2202702 | ⋯ | /dcs04/lieber/statsgen/mnagle/mwas/pheno/caud/out/chr1_AA_88982-108981.rds  | ..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-88982-108981-dynamic-1corestotal-allcorepera-20240510-122350.rds  | caud_AA_1_88982_108981  | ..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-88982-108981-dynamic-1corestotal-allcorepera-caud-20240510-122350.rds  | 20240510 | 122350 | 2024-05-10 12:23:50 | /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-88982-108981-dynamic-1corestotal-allcorepera-caud-20240510-122350_gwas_stat_bp_alleleprocessed_a3_results-101324.csv  | /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-88982-108981-dynamic-1corestotal-allcorepera-caud-20240510-122350_gwas_stat_scz_alleleprocessed_a3_results-101324.csv  | /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-88982-108981-dynamic-1corestotal-allcorepera-caud-20240510-122350_gwas_stat_mdd_alleleprocessed_a3_results-101324.csv  |\n",
       "| caud | 1 | AA | 108982 | 128981 | /dcs04/lieber/statsgen/shizhong/michael/mwas/gwas//libd_chr1.pgen | /dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/caud/out/chr1_AA.rda | 248918358 | 1069461 | 2202702 | ⋯ | /dcs04/lieber/statsgen/mnagle/mwas/pheno/caud/out/chr1_AA_108982-128981.rds | ..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-108982-128981-dynamic-1corestotal-allcorepera-20240510-122451.rds | caud_AA_1_108982_128981 | ..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-108982-128981-dynamic-1corestotal-allcorepera-caud-20240510-122451.rds | 20240510 | 122451 | 2024-05-10 12:24:51 | /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-108982-128981-dynamic-1corestotal-allcorepera-caud-20240510-122451_gwas_stat_bp_alleleprocessed_a3_results-101324.csv | /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-108982-128981-dynamic-1corestotal-allcorepera-caud-20240510-122451_gwas_stat_scz_alleleprocessed_a3_results-101324.csv | /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-108982-128981-dynamic-1corestotal-allcorepera-caud-20240510-122451_gwas_stat_mdd_alleleprocessed_a3_results-101324.csv |\n",
       "\n"
      ],
      "text/plain": [
       "  region Chr population chunk_start chunk_end\n",
       "1 caud   1   AA           8982       28981   \n",
       "2 caud   1   AA          28982       48981   \n",
       "3 caud   1   AA          48982       68981   \n",
       "4 caud   1   AA          68982       88981   \n",
       "5 caud   1   AA          88982      108981   \n",
       "6 caud   1   AA         108982      128981   \n",
       "  SNP_data                                                         \n",
       "1 /dcs04/lieber/statsgen/shizhong/michael/mwas/gwas//libd_chr1.pgen\n",
       "2 /dcs04/lieber/statsgen/shizhong/michael/mwas/gwas//libd_chr1.pgen\n",
       "3 /dcs04/lieber/statsgen/shizhong/michael/mwas/gwas//libd_chr1.pgen\n",
       "4 /dcs04/lieber/statsgen/shizhong/michael/mwas/gwas//libd_chr1.pgen\n",
       "5 /dcs04/lieber/statsgen/shizhong/michael/mwas/gwas//libd_chr1.pgen\n",
       "6 /dcs04/lieber/statsgen/shizhong/michael/mwas/gwas//libd_chr1.pgen\n",
       "  methylation_data                                                       \n",
       "1 /dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/caud/out/chr1_AA.rda\n",
       "2 /dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/caud/out/chr1_AA.rda\n",
       "3 /dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/caud/out/chr1_AA.rda\n",
       "4 /dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/caud/out/chr1_AA.rda\n",
       "5 /dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/caud/out/chr1_AA.rda\n",
       "6 /dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/caud/out/chr1_AA.rda\n",
       "  last_meth_value_with_SNP_coverage first_meth_value_with_SNP_coverage\n",
       "1 248918358                         1069461                           \n",
       "2 248918358                         1069461                           \n",
       "3 248918358                         1069461                           \n",
       "4 248918358                         1069461                           \n",
       "5 248918358                         1069461                           \n",
       "6 248918358                         1069461                           \n",
       "  last_meth_index_with_SNP_coverage ⋯\n",
       "1 2202702                           ⋯\n",
       "2 2202702                           ⋯\n",
       "3 2202702                           ⋯\n",
       "4 2202702                           ⋯\n",
       "5 2202702                           ⋯\n",
       "6 2202702                           ⋯\n",
       "  modified_methylation_data                                                  \n",
       "1 /dcs04/lieber/statsgen/mnagle/mwas/pheno/caud/out/chr1_AA_8982-28981.rds   \n",
       "2 /dcs04/lieber/statsgen/mnagle/mwas/pheno/caud/out/chr1_AA_28982-48981.rds  \n",
       "3 /dcs04/lieber/statsgen/mnagle/mwas/pheno/caud/out/chr1_AA_48982-68981.rds  \n",
       "4 /dcs04/lieber/statsgen/mnagle/mwas/pheno/caud/out/chr1_AA_68982-88981.rds  \n",
       "5 /dcs04/lieber/statsgen/mnagle/mwas/pheno/caud/out/chr1_AA_88982-108981.rds \n",
       "6 /dcs04/lieber/statsgen/mnagle/mwas/pheno/caud/out/chr1_AA_108982-128981.rds\n",
       "  path                                                                                                                            \n",
       "1 ..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-8982-28981-dynamic-1corestotal-allcorepera-20240510-121942.rds   \n",
       "2 ..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-28982-48981-dynamic-1corestotal-allcorepera-20240510-122045.rds  \n",
       "3 ..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-48982-68981-dynamic-1corestotal-allcorepera-20240510-122147.rds  \n",
       "4 ..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-68982-88981-dynamic-1corestotal-allcorepera-20240510-122249.rds  \n",
       "5 ..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-88982-108981-dynamic-1corestotal-allcorepera-20240510-122350.rds \n",
       "6 ..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-108982-128981-dynamic-1corestotal-allcorepera-20240510-122451.rds\n",
       "  tag                    \n",
       "1 caud_AA_1_8982_28981   \n",
       "2 caud_AA_1_28982_48981  \n",
       "3 caud_AA_1_48982_68981  \n",
       "4 caud_AA_1_68982_88981  \n",
       "5 caud_AA_1_88982_108981 \n",
       "6 caud_AA_1_108982_128981\n",
       "  path_unmodified                                                                                                                      \n",
       "1 ..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-8982-28981-dynamic-1corestotal-allcorepera-caud-20240510-121942.rds   \n",
       "2 ..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-28982-48981-dynamic-1corestotal-allcorepera-caud-20240510-122045.rds  \n",
       "3 ..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-48982-68981-dynamic-1corestotal-allcorepera-caud-20240510-122147.rds  \n",
       "4 ..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-68982-88981-dynamic-1corestotal-allcorepera-caud-20240510-122249.rds  \n",
       "5 ..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-88982-108981-dynamic-1corestotal-allcorepera-caud-20240510-122350.rds \n",
       "6 ..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-108982-128981-dynamic-1corestotal-allcorepera-caud-20240510-122451.rds\n",
       "  date     time   datetime           \n",
       "1 20240510 121942 2024-05-10 12:19:42\n",
       "2 20240510 122045 2024-05-10 12:20:45\n",
       "3 20240510 122147 2024-05-10 12:21:47\n",
       "4 20240510 122249 2024-05-10 12:22:49\n",
       "5 20240510 122350 2024-05-10 12:23:50\n",
       "6 20240510 122451 2024-05-10 12:24:51\n",
       "  result_bp                                                                                                                                                                                                                            \n",
       "1 /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-8982-28981-dynamic-1corestotal-allcorepera-caud-20240510-121942_gwas_stat_bp_alleleprocessed_a3_results-101324.csv   \n",
       "2 /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-28982-48981-dynamic-1corestotal-allcorepera-caud-20240510-122045_gwas_stat_bp_alleleprocessed_a3_results-101324.csv  \n",
       "3 /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-48982-68981-dynamic-1corestotal-allcorepera-caud-20240510-122147_gwas_stat_bp_alleleprocessed_a3_results-101324.csv  \n",
       "4 /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-68982-88981-dynamic-1corestotal-allcorepera-caud-20240510-122249_gwas_stat_bp_alleleprocessed_a3_results-101324.csv  \n",
       "5 /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-88982-108981-dynamic-1corestotal-allcorepera-caud-20240510-122350_gwas_stat_bp_alleleprocessed_a3_results-101324.csv \n",
       "6 /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-108982-128981-dynamic-1corestotal-allcorepera-caud-20240510-122451_gwas_stat_bp_alleleprocessed_a3_results-101324.csv\n",
       "  result_scz                                                                                                                                                                                                                            \n",
       "1 /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-8982-28981-dynamic-1corestotal-allcorepera-caud-20240510-121942_gwas_stat_scz_alleleprocessed_a3_results-101324.csv   \n",
       "2 /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-28982-48981-dynamic-1corestotal-allcorepera-caud-20240510-122045_gwas_stat_scz_alleleprocessed_a3_results-101324.csv  \n",
       "3 /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-48982-68981-dynamic-1corestotal-allcorepera-caud-20240510-122147_gwas_stat_scz_alleleprocessed_a3_results-101324.csv  \n",
       "4 /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-68982-88981-dynamic-1corestotal-allcorepera-caud-20240510-122249_gwas_stat_scz_alleleprocessed_a3_results-101324.csv  \n",
       "5 /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-88982-108981-dynamic-1corestotal-allcorepera-caud-20240510-122350_gwas_stat_scz_alleleprocessed_a3_results-101324.csv \n",
       "6 /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-108982-128981-dynamic-1corestotal-allcorepera-caud-20240510-122451_gwas_stat_scz_alleleprocessed_a3_results-101324.csv\n",
       "  result_mdd                                                                                                                                                                                                                            \n",
       "1 /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-8982-28981-dynamic-1corestotal-allcorepera-caud-20240510-121942_gwas_stat_mdd_alleleprocessed_a3_results-101324.csv   \n",
       "2 /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-28982-48981-dynamic-1corestotal-allcorepera-caud-20240510-122045_gwas_stat_mdd_alleleprocessed_a3_results-101324.csv  \n",
       "3 /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-48982-68981-dynamic-1corestotal-allcorepera-caud-20240510-122147_gwas_stat_mdd_alleleprocessed_a3_results-101324.csv  \n",
       "4 /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-68982-88981-dynamic-1corestotal-allcorepera-caud-20240510-122249_gwas_stat_mdd_alleleprocessed_a3_results-101324.csv  \n",
       "5 /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-88982-108981-dynamic-1corestotal-allcorepera-caud-20240510-122350_gwas_stat_mdd_alleleprocessed_a3_results-101324.csv \n",
       "6 /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-108982-128981-dynamic-1corestotal-allcorepera-caud-20240510-122451_gwas_stat_mdd_alleleprocessed_a3_results-101324.csv"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(result_table)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "c35aa293-9ae8-4c5c-8d8f-a43238c2baff",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>11421</li><li>22</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 11421\n",
       "\\item 22\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 11421\n",
       "2. 22\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 11421    22"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dim(result_table)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "9471c052-f3dd-4b42-996c-4b0140afb8ce",
   "metadata": {},
   "outputs": [],
   "source": [
    "fwrite(result_table, \"37-OUT/37-OUT_result_table_no_dups.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "a7e8694b-af4f-492b-ad30-f9f4ac2df216",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 6 × 22</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>region</th><th scope=col>Chr</th><th scope=col>population</th><th scope=col>chunk_start</th><th scope=col>chunk_end</th><th scope=col>SNP_data</th><th scope=col>methylation_data</th><th scope=col>last_meth_value_with_SNP_coverage</th><th scope=col>first_meth_value_with_SNP_coverage</th><th scope=col>last_meth_index_with_SNP_coverage</th><th scope=col>⋯</th><th scope=col>modified_methylation_data</th><th scope=col>path</th><th scope=col>tag</th><th scope=col>path_unmodified</th><th scope=col>date</th><th scope=col>time</th><th scope=col>datetime</th><th scope=col>result_bp</th><th scope=col>result_scz</th><th scope=col>result_mdd</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>⋯</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dttm&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>caud</td><td>1</td><td>AA</td><td>  8982</td><td> 28981</td><td>/dcs04/lieber/statsgen/shizhong/michael/mwas/gwas//libd_chr1.pgen</td><td>/dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/caud/out/chr1_AA.rda</td><td>248918358</td><td>1069461</td><td>2202702</td><td>⋯</td><td>/dcs04/lieber/statsgen/mnagle/mwas/pheno/caud/out/chr1_AA_8982-28981.rds   </td><td>..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-8982-28981-dynamic-1corestotal-allcorepera-20240510-121942.rds   </td><td>caud_AA_1_8982_28981   </td><td>..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-8982-28981-dynamic-1corestotal-allcorepera-caud-20240510-121942.rds   </td><td>20240510</td><td>121942</td><td>2024-05-10 12:19:42</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-8982-28981-dynamic-1corestotal-allcorepera-caud-20240510-121942_gwas_stat_bp_alleleprocessed_a3_results-101324.csv   </td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-8982-28981-dynamic-1corestotal-allcorepera-caud-20240510-121942_gwas_stat_scz_alleleprocessed_a3_results-101324.csv   </td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-8982-28981-dynamic-1corestotal-allcorepera-caud-20240510-121942_gwas_stat_mdd_alleleprocessed_a3_results-101324.csv   </td></tr>\n",
       "\t<tr><td>caud</td><td>1</td><td>AA</td><td> 28982</td><td> 48981</td><td>/dcs04/lieber/statsgen/shizhong/michael/mwas/gwas//libd_chr1.pgen</td><td>/dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/caud/out/chr1_AA.rda</td><td>248918358</td><td>1069461</td><td>2202702</td><td>⋯</td><td>/dcs04/lieber/statsgen/mnagle/mwas/pheno/caud/out/chr1_AA_28982-48981.rds  </td><td>..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-28982-48981-dynamic-1corestotal-allcorepera-20240510-122045.rds  </td><td>caud_AA_1_28982_48981  </td><td>..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-28982-48981-dynamic-1corestotal-allcorepera-caud-20240510-122045.rds  </td><td>20240510</td><td>122045</td><td>2024-05-10 12:20:45</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-28982-48981-dynamic-1corestotal-allcorepera-caud-20240510-122045_gwas_stat_bp_alleleprocessed_a3_results-101324.csv  </td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-28982-48981-dynamic-1corestotal-allcorepera-caud-20240510-122045_gwas_stat_scz_alleleprocessed_a3_results-101324.csv  </td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-28982-48981-dynamic-1corestotal-allcorepera-caud-20240510-122045_gwas_stat_mdd_alleleprocessed_a3_results-101324.csv  </td></tr>\n",
       "\t<tr><td>caud</td><td>1</td><td>AA</td><td> 48982</td><td> 68981</td><td>/dcs04/lieber/statsgen/shizhong/michael/mwas/gwas//libd_chr1.pgen</td><td>/dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/caud/out/chr1_AA.rda</td><td>248918358</td><td>1069461</td><td>2202702</td><td>⋯</td><td>/dcs04/lieber/statsgen/mnagle/mwas/pheno/caud/out/chr1_AA_48982-68981.rds  </td><td>..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-48982-68981-dynamic-1corestotal-allcorepera-20240510-122147.rds  </td><td>caud_AA_1_48982_68981  </td><td>..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-48982-68981-dynamic-1corestotal-allcorepera-caud-20240510-122147.rds  </td><td>20240510</td><td>122147</td><td>2024-05-10 12:21:47</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-48982-68981-dynamic-1corestotal-allcorepera-caud-20240510-122147_gwas_stat_bp_alleleprocessed_a3_results-101324.csv  </td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-48982-68981-dynamic-1corestotal-allcorepera-caud-20240510-122147_gwas_stat_scz_alleleprocessed_a3_results-101324.csv  </td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-48982-68981-dynamic-1corestotal-allcorepera-caud-20240510-122147_gwas_stat_mdd_alleleprocessed_a3_results-101324.csv  </td></tr>\n",
       "\t<tr><td>caud</td><td>1</td><td>AA</td><td> 68982</td><td> 88981</td><td>/dcs04/lieber/statsgen/shizhong/michael/mwas/gwas//libd_chr1.pgen</td><td>/dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/caud/out/chr1_AA.rda</td><td>248918358</td><td>1069461</td><td>2202702</td><td>⋯</td><td>/dcs04/lieber/statsgen/mnagle/mwas/pheno/caud/out/chr1_AA_68982-88981.rds  </td><td>..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-68982-88981-dynamic-1corestotal-allcorepera-20240510-122249.rds  </td><td>caud_AA_1_68982_88981  </td><td>..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-68982-88981-dynamic-1corestotal-allcorepera-caud-20240510-122249.rds  </td><td>20240510</td><td>122249</td><td>2024-05-10 12:22:49</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-68982-88981-dynamic-1corestotal-allcorepera-caud-20240510-122249_gwas_stat_bp_alleleprocessed_a3_results-101324.csv  </td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-68982-88981-dynamic-1corestotal-allcorepera-caud-20240510-122249_gwas_stat_scz_alleleprocessed_a3_results-101324.csv  </td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-68982-88981-dynamic-1corestotal-allcorepera-caud-20240510-122249_gwas_stat_mdd_alleleprocessed_a3_results-101324.csv  </td></tr>\n",
       "\t<tr><td>caud</td><td>1</td><td>AA</td><td> 88982</td><td>108981</td><td>/dcs04/lieber/statsgen/shizhong/michael/mwas/gwas//libd_chr1.pgen</td><td>/dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/caud/out/chr1_AA.rda</td><td>248918358</td><td>1069461</td><td>2202702</td><td>⋯</td><td>/dcs04/lieber/statsgen/mnagle/mwas/pheno/caud/out/chr1_AA_88982-108981.rds </td><td>..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-88982-108981-dynamic-1corestotal-allcorepera-20240510-122350.rds </td><td>caud_AA_1_88982_108981 </td><td>..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-88982-108981-dynamic-1corestotal-allcorepera-caud-20240510-122350.rds </td><td>20240510</td><td>122350</td><td>2024-05-10 12:23:50</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-88982-108981-dynamic-1corestotal-allcorepera-caud-20240510-122350_gwas_stat_bp_alleleprocessed_a3_results-101324.csv </td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-88982-108981-dynamic-1corestotal-allcorepera-caud-20240510-122350_gwas_stat_scz_alleleprocessed_a3_results-101324.csv </td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-88982-108981-dynamic-1corestotal-allcorepera-caud-20240510-122350_gwas_stat_mdd_alleleprocessed_a3_results-101324.csv </td></tr>\n",
       "\t<tr><td>caud</td><td>1</td><td>AA</td><td>108982</td><td>128981</td><td>/dcs04/lieber/statsgen/shizhong/michael/mwas/gwas//libd_chr1.pgen</td><td>/dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/caud/out/chr1_AA.rda</td><td>248918358</td><td>1069461</td><td>2202702</td><td>⋯</td><td>/dcs04/lieber/statsgen/mnagle/mwas/pheno/caud/out/chr1_AA_108982-128981.rds</td><td>..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-108982-128981-dynamic-1corestotal-allcorepera-20240510-122451.rds</td><td>caud_AA_1_108982_128981</td><td>..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-108982-128981-dynamic-1corestotal-allcorepera-caud-20240510-122451.rds</td><td>20240510</td><td>122451</td><td>2024-05-10 12:24:51</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-108982-128981-dynamic-1corestotal-allcorepera-caud-20240510-122451_gwas_stat_bp_alleleprocessed_a3_results-101324.csv</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-108982-128981-dynamic-1corestotal-allcorepera-caud-20240510-122451_gwas_stat_scz_alleleprocessed_a3_results-101324.csv</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-108982-128981-dynamic-1corestotal-allcorepera-caud-20240510-122451_gwas_stat_mdd_alleleprocessed_a3_results-101324.csv</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 6 × 22\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " region & Chr & population & chunk\\_start & chunk\\_end & SNP\\_data & methylation\\_data & last\\_meth\\_value\\_with\\_SNP\\_coverage & first\\_meth\\_value\\_with\\_SNP\\_coverage & last\\_meth\\_index\\_with\\_SNP\\_coverage & ⋯ & modified\\_methylation\\_data & path & tag & path\\_unmodified & date & time & datetime & result\\_bp & result\\_scz & result\\_mdd\\\\\n",
       " <chr> & <int> & <chr> & <int> & <dbl> & <chr> & <chr> & <int> & <int> & <int> & ⋯ & <chr> & <chr> & <chr> & <chr> & <dbl> & <dbl> & <dttm> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t caud & 1 & AA &   8982 &  28981 & /dcs04/lieber/statsgen/shizhong/michael/mwas/gwas//libd\\_chr1.pgen & /dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/caud/out/chr1\\_AA.rda & 248918358 & 1069461 & 2202702 & ⋯ & /dcs04/lieber/statsgen/mnagle/mwas/pheno/caud/out/chr1\\_AA\\_8982-28981.rds    & ..//output\\_EXPANSE\\_a2\\_caud/libd\\_chr1-chr1\\_AA-libd\\_chr1-chr1\\_AA-8982-28981-dynamic-1corestotal-allcorepera-20240510-121942.rds    & caud\\_AA\\_1\\_8982\\_28981    & ..//output\\_EXPANSE\\_a2\\_caud/libd\\_chr1-chr1\\_AA-libd\\_chr1-chr1\\_AA-8982-28981-dynamic-1corestotal-allcorepera-caud-20240510-121942.rds    & 20240510 & 121942 & 2024-05-10 12:19:42 & /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output\\_EXPANSE\\_dt\\_caud//libd\\_chr1-chr1\\_AA-libd\\_chr1-chr1\\_AA-8982-28981-dynamic-1corestotal-allcorepera-caud-20240510-121942\\_gwas\\_stat\\_bp\\_alleleprocessed\\_a3\\_results-101324.csv    & /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output\\_EXPANSE\\_dt\\_caud//libd\\_chr1-chr1\\_AA-libd\\_chr1-chr1\\_AA-8982-28981-dynamic-1corestotal-allcorepera-caud-20240510-121942\\_gwas\\_stat\\_scz\\_alleleprocessed\\_a3\\_results-101324.csv    & /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output\\_EXPANSE\\_dt\\_caud//libd\\_chr1-chr1\\_AA-libd\\_chr1-chr1\\_AA-8982-28981-dynamic-1corestotal-allcorepera-caud-20240510-121942\\_gwas\\_stat\\_mdd\\_alleleprocessed\\_a3\\_results-101324.csv   \\\\\n",
       "\t caud & 1 & AA &  28982 &  48981 & /dcs04/lieber/statsgen/shizhong/michael/mwas/gwas//libd\\_chr1.pgen & /dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/caud/out/chr1\\_AA.rda & 248918358 & 1069461 & 2202702 & ⋯ & /dcs04/lieber/statsgen/mnagle/mwas/pheno/caud/out/chr1\\_AA\\_28982-48981.rds   & ..//output\\_EXPANSE\\_a2\\_caud/libd\\_chr1-chr1\\_AA-libd\\_chr1-chr1\\_AA-28982-48981-dynamic-1corestotal-allcorepera-20240510-122045.rds   & caud\\_AA\\_1\\_28982\\_48981   & ..//output\\_EXPANSE\\_a2\\_caud/libd\\_chr1-chr1\\_AA-libd\\_chr1-chr1\\_AA-28982-48981-dynamic-1corestotal-allcorepera-caud-20240510-122045.rds   & 20240510 & 122045 & 2024-05-10 12:20:45 & /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output\\_EXPANSE\\_dt\\_caud//libd\\_chr1-chr1\\_AA-libd\\_chr1-chr1\\_AA-28982-48981-dynamic-1corestotal-allcorepera-caud-20240510-122045\\_gwas\\_stat\\_bp\\_alleleprocessed\\_a3\\_results-101324.csv   & /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output\\_EXPANSE\\_dt\\_caud//libd\\_chr1-chr1\\_AA-libd\\_chr1-chr1\\_AA-28982-48981-dynamic-1corestotal-allcorepera-caud-20240510-122045\\_gwas\\_stat\\_scz\\_alleleprocessed\\_a3\\_results-101324.csv   & /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output\\_EXPANSE\\_dt\\_caud//libd\\_chr1-chr1\\_AA-libd\\_chr1-chr1\\_AA-28982-48981-dynamic-1corestotal-allcorepera-caud-20240510-122045\\_gwas\\_stat\\_mdd\\_alleleprocessed\\_a3\\_results-101324.csv  \\\\\n",
       "\t caud & 1 & AA &  48982 &  68981 & /dcs04/lieber/statsgen/shizhong/michael/mwas/gwas//libd\\_chr1.pgen & /dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/caud/out/chr1\\_AA.rda & 248918358 & 1069461 & 2202702 & ⋯ & /dcs04/lieber/statsgen/mnagle/mwas/pheno/caud/out/chr1\\_AA\\_48982-68981.rds   & ..//output\\_EXPANSE\\_a2\\_caud/libd\\_chr1-chr1\\_AA-libd\\_chr1-chr1\\_AA-48982-68981-dynamic-1corestotal-allcorepera-20240510-122147.rds   & caud\\_AA\\_1\\_48982\\_68981   & ..//output\\_EXPANSE\\_a2\\_caud/libd\\_chr1-chr1\\_AA-libd\\_chr1-chr1\\_AA-48982-68981-dynamic-1corestotal-allcorepera-caud-20240510-122147.rds   & 20240510 & 122147 & 2024-05-10 12:21:47 & /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output\\_EXPANSE\\_dt\\_caud//libd\\_chr1-chr1\\_AA-libd\\_chr1-chr1\\_AA-48982-68981-dynamic-1corestotal-allcorepera-caud-20240510-122147\\_gwas\\_stat\\_bp\\_alleleprocessed\\_a3\\_results-101324.csv   & /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output\\_EXPANSE\\_dt\\_caud//libd\\_chr1-chr1\\_AA-libd\\_chr1-chr1\\_AA-48982-68981-dynamic-1corestotal-allcorepera-caud-20240510-122147\\_gwas\\_stat\\_scz\\_alleleprocessed\\_a3\\_results-101324.csv   & /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output\\_EXPANSE\\_dt\\_caud//libd\\_chr1-chr1\\_AA-libd\\_chr1-chr1\\_AA-48982-68981-dynamic-1corestotal-allcorepera-caud-20240510-122147\\_gwas\\_stat\\_mdd\\_alleleprocessed\\_a3\\_results-101324.csv  \\\\\n",
       "\t caud & 1 & AA &  68982 &  88981 & /dcs04/lieber/statsgen/shizhong/michael/mwas/gwas//libd\\_chr1.pgen & /dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/caud/out/chr1\\_AA.rda & 248918358 & 1069461 & 2202702 & ⋯ & /dcs04/lieber/statsgen/mnagle/mwas/pheno/caud/out/chr1\\_AA\\_68982-88981.rds   & ..//output\\_EXPANSE\\_a2\\_caud/libd\\_chr1-chr1\\_AA-libd\\_chr1-chr1\\_AA-68982-88981-dynamic-1corestotal-allcorepera-20240510-122249.rds   & caud\\_AA\\_1\\_68982\\_88981   & ..//output\\_EXPANSE\\_a2\\_caud/libd\\_chr1-chr1\\_AA-libd\\_chr1-chr1\\_AA-68982-88981-dynamic-1corestotal-allcorepera-caud-20240510-122249.rds   & 20240510 & 122249 & 2024-05-10 12:22:49 & /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output\\_EXPANSE\\_dt\\_caud//libd\\_chr1-chr1\\_AA-libd\\_chr1-chr1\\_AA-68982-88981-dynamic-1corestotal-allcorepera-caud-20240510-122249\\_gwas\\_stat\\_bp\\_alleleprocessed\\_a3\\_results-101324.csv   & /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output\\_EXPANSE\\_dt\\_caud//libd\\_chr1-chr1\\_AA-libd\\_chr1-chr1\\_AA-68982-88981-dynamic-1corestotal-allcorepera-caud-20240510-122249\\_gwas\\_stat\\_scz\\_alleleprocessed\\_a3\\_results-101324.csv   & /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output\\_EXPANSE\\_dt\\_caud//libd\\_chr1-chr1\\_AA-libd\\_chr1-chr1\\_AA-68982-88981-dynamic-1corestotal-allcorepera-caud-20240510-122249\\_gwas\\_stat\\_mdd\\_alleleprocessed\\_a3\\_results-101324.csv  \\\\\n",
       "\t caud & 1 & AA &  88982 & 108981 & /dcs04/lieber/statsgen/shizhong/michael/mwas/gwas//libd\\_chr1.pgen & /dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/caud/out/chr1\\_AA.rda & 248918358 & 1069461 & 2202702 & ⋯ & /dcs04/lieber/statsgen/mnagle/mwas/pheno/caud/out/chr1\\_AA\\_88982-108981.rds  & ..//output\\_EXPANSE\\_a2\\_caud/libd\\_chr1-chr1\\_AA-libd\\_chr1-chr1\\_AA-88982-108981-dynamic-1corestotal-allcorepera-20240510-122350.rds  & caud\\_AA\\_1\\_88982\\_108981  & ..//output\\_EXPANSE\\_a2\\_caud/libd\\_chr1-chr1\\_AA-libd\\_chr1-chr1\\_AA-88982-108981-dynamic-1corestotal-allcorepera-caud-20240510-122350.rds  & 20240510 & 122350 & 2024-05-10 12:23:50 & /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output\\_EXPANSE\\_dt\\_caud//libd\\_chr1-chr1\\_AA-libd\\_chr1-chr1\\_AA-88982-108981-dynamic-1corestotal-allcorepera-caud-20240510-122350\\_gwas\\_stat\\_bp\\_alleleprocessed\\_a3\\_results-101324.csv  & /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output\\_EXPANSE\\_dt\\_caud//libd\\_chr1-chr1\\_AA-libd\\_chr1-chr1\\_AA-88982-108981-dynamic-1corestotal-allcorepera-caud-20240510-122350\\_gwas\\_stat\\_scz\\_alleleprocessed\\_a3\\_results-101324.csv  & /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output\\_EXPANSE\\_dt\\_caud//libd\\_chr1-chr1\\_AA-libd\\_chr1-chr1\\_AA-88982-108981-dynamic-1corestotal-allcorepera-caud-20240510-122350\\_gwas\\_stat\\_mdd\\_alleleprocessed\\_a3\\_results-101324.csv \\\\\n",
       "\t caud & 1 & AA & 108982 & 128981 & /dcs04/lieber/statsgen/shizhong/michael/mwas/gwas//libd\\_chr1.pgen & /dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/caud/out/chr1\\_AA.rda & 248918358 & 1069461 & 2202702 & ⋯ & /dcs04/lieber/statsgen/mnagle/mwas/pheno/caud/out/chr1\\_AA\\_108982-128981.rds & ..//output\\_EXPANSE\\_a2\\_caud/libd\\_chr1-chr1\\_AA-libd\\_chr1-chr1\\_AA-108982-128981-dynamic-1corestotal-allcorepera-20240510-122451.rds & caud\\_AA\\_1\\_108982\\_128981 & ..//output\\_EXPANSE\\_a2\\_caud/libd\\_chr1-chr1\\_AA-libd\\_chr1-chr1\\_AA-108982-128981-dynamic-1corestotal-allcorepera-caud-20240510-122451.rds & 20240510 & 122451 & 2024-05-10 12:24:51 & /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output\\_EXPANSE\\_dt\\_caud//libd\\_chr1-chr1\\_AA-libd\\_chr1-chr1\\_AA-108982-128981-dynamic-1corestotal-allcorepera-caud-20240510-122451\\_gwas\\_stat\\_bp\\_alleleprocessed\\_a3\\_results-101324.csv & /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output\\_EXPANSE\\_dt\\_caud//libd\\_chr1-chr1\\_AA-libd\\_chr1-chr1\\_AA-108982-128981-dynamic-1corestotal-allcorepera-caud-20240510-122451\\_gwas\\_stat\\_scz\\_alleleprocessed\\_a3\\_results-101324.csv & /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output\\_EXPANSE\\_dt\\_caud//libd\\_chr1-chr1\\_AA-libd\\_chr1-chr1\\_AA-108982-128981-dynamic-1corestotal-allcorepera-caud-20240510-122451\\_gwas\\_stat\\_mdd\\_alleleprocessed\\_a3\\_results-101324.csv\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 6 × 22\n",
       "\n",
       "| region &lt;chr&gt; | Chr &lt;int&gt; | population &lt;chr&gt; | chunk_start &lt;int&gt; | chunk_end &lt;dbl&gt; | SNP_data &lt;chr&gt; | methylation_data &lt;chr&gt; | last_meth_value_with_SNP_coverage &lt;int&gt; | first_meth_value_with_SNP_coverage &lt;int&gt; | last_meth_index_with_SNP_coverage &lt;int&gt; | ⋯ ⋯ | modified_methylation_data &lt;chr&gt; | path &lt;chr&gt; | tag &lt;chr&gt; | path_unmodified &lt;chr&gt; | date &lt;dbl&gt; | time &lt;dbl&gt; | datetime &lt;dttm&gt; | result_bp &lt;chr&gt; | result_scz &lt;chr&gt; | result_mdd &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| caud | 1 | AA |   8982 |  28981 | /dcs04/lieber/statsgen/shizhong/michael/mwas/gwas//libd_chr1.pgen | /dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/caud/out/chr1_AA.rda | 248918358 | 1069461 | 2202702 | ⋯ | /dcs04/lieber/statsgen/mnagle/mwas/pheno/caud/out/chr1_AA_8982-28981.rds    | ..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-8982-28981-dynamic-1corestotal-allcorepera-20240510-121942.rds    | caud_AA_1_8982_28981    | ..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-8982-28981-dynamic-1corestotal-allcorepera-caud-20240510-121942.rds    | 20240510 | 121942 | 2024-05-10 12:19:42 | /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-8982-28981-dynamic-1corestotal-allcorepera-caud-20240510-121942_gwas_stat_bp_alleleprocessed_a3_results-101324.csv    | /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-8982-28981-dynamic-1corestotal-allcorepera-caud-20240510-121942_gwas_stat_scz_alleleprocessed_a3_results-101324.csv    | /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-8982-28981-dynamic-1corestotal-allcorepera-caud-20240510-121942_gwas_stat_mdd_alleleprocessed_a3_results-101324.csv    |\n",
       "| caud | 1 | AA |  28982 |  48981 | /dcs04/lieber/statsgen/shizhong/michael/mwas/gwas//libd_chr1.pgen | /dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/caud/out/chr1_AA.rda | 248918358 | 1069461 | 2202702 | ⋯ | /dcs04/lieber/statsgen/mnagle/mwas/pheno/caud/out/chr1_AA_28982-48981.rds   | ..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-28982-48981-dynamic-1corestotal-allcorepera-20240510-122045.rds   | caud_AA_1_28982_48981   | ..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-28982-48981-dynamic-1corestotal-allcorepera-caud-20240510-122045.rds   | 20240510 | 122045 | 2024-05-10 12:20:45 | /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-28982-48981-dynamic-1corestotal-allcorepera-caud-20240510-122045_gwas_stat_bp_alleleprocessed_a3_results-101324.csv   | /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-28982-48981-dynamic-1corestotal-allcorepera-caud-20240510-122045_gwas_stat_scz_alleleprocessed_a3_results-101324.csv   | /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-28982-48981-dynamic-1corestotal-allcorepera-caud-20240510-122045_gwas_stat_mdd_alleleprocessed_a3_results-101324.csv   |\n",
       "| caud | 1 | AA |  48982 |  68981 | /dcs04/lieber/statsgen/shizhong/michael/mwas/gwas//libd_chr1.pgen | /dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/caud/out/chr1_AA.rda | 248918358 | 1069461 | 2202702 | ⋯ | /dcs04/lieber/statsgen/mnagle/mwas/pheno/caud/out/chr1_AA_48982-68981.rds   | ..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-48982-68981-dynamic-1corestotal-allcorepera-20240510-122147.rds   | caud_AA_1_48982_68981   | ..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-48982-68981-dynamic-1corestotal-allcorepera-caud-20240510-122147.rds   | 20240510 | 122147 | 2024-05-10 12:21:47 | /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-48982-68981-dynamic-1corestotal-allcorepera-caud-20240510-122147_gwas_stat_bp_alleleprocessed_a3_results-101324.csv   | /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-48982-68981-dynamic-1corestotal-allcorepera-caud-20240510-122147_gwas_stat_scz_alleleprocessed_a3_results-101324.csv   | /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-48982-68981-dynamic-1corestotal-allcorepera-caud-20240510-122147_gwas_stat_mdd_alleleprocessed_a3_results-101324.csv   |\n",
       "| caud | 1 | AA |  68982 |  88981 | /dcs04/lieber/statsgen/shizhong/michael/mwas/gwas//libd_chr1.pgen | /dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/caud/out/chr1_AA.rda | 248918358 | 1069461 | 2202702 | ⋯ | /dcs04/lieber/statsgen/mnagle/mwas/pheno/caud/out/chr1_AA_68982-88981.rds   | ..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-68982-88981-dynamic-1corestotal-allcorepera-20240510-122249.rds   | caud_AA_1_68982_88981   | ..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-68982-88981-dynamic-1corestotal-allcorepera-caud-20240510-122249.rds   | 20240510 | 122249 | 2024-05-10 12:22:49 | /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-68982-88981-dynamic-1corestotal-allcorepera-caud-20240510-122249_gwas_stat_bp_alleleprocessed_a3_results-101324.csv   | /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-68982-88981-dynamic-1corestotal-allcorepera-caud-20240510-122249_gwas_stat_scz_alleleprocessed_a3_results-101324.csv   | /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-68982-88981-dynamic-1corestotal-allcorepera-caud-20240510-122249_gwas_stat_mdd_alleleprocessed_a3_results-101324.csv   |\n",
       "| caud | 1 | AA |  88982 | 108981 | /dcs04/lieber/statsgen/shizhong/michael/mwas/gwas//libd_chr1.pgen | /dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/caud/out/chr1_AA.rda | 248918358 | 1069461 | 2202702 | ⋯ | /dcs04/lieber/statsgen/mnagle/mwas/pheno/caud/out/chr1_AA_88982-108981.rds  | ..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-88982-108981-dynamic-1corestotal-allcorepera-20240510-122350.rds  | caud_AA_1_88982_108981  | ..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-88982-108981-dynamic-1corestotal-allcorepera-caud-20240510-122350.rds  | 20240510 | 122350 | 2024-05-10 12:23:50 | /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-88982-108981-dynamic-1corestotal-allcorepera-caud-20240510-122350_gwas_stat_bp_alleleprocessed_a3_results-101324.csv  | /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-88982-108981-dynamic-1corestotal-allcorepera-caud-20240510-122350_gwas_stat_scz_alleleprocessed_a3_results-101324.csv  | /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-88982-108981-dynamic-1corestotal-allcorepera-caud-20240510-122350_gwas_stat_mdd_alleleprocessed_a3_results-101324.csv  |\n",
       "| caud | 1 | AA | 108982 | 128981 | /dcs04/lieber/statsgen/shizhong/michael/mwas/gwas//libd_chr1.pgen | /dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/caud/out/chr1_AA.rda | 248918358 | 1069461 | 2202702 | ⋯ | /dcs04/lieber/statsgen/mnagle/mwas/pheno/caud/out/chr1_AA_108982-128981.rds | ..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-108982-128981-dynamic-1corestotal-allcorepera-20240510-122451.rds | caud_AA_1_108982_128981 | ..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-108982-128981-dynamic-1corestotal-allcorepera-caud-20240510-122451.rds | 20240510 | 122451 | 2024-05-10 12:24:51 | /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-108982-128981-dynamic-1corestotal-allcorepera-caud-20240510-122451_gwas_stat_bp_alleleprocessed_a3_results-101324.csv | /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-108982-128981-dynamic-1corestotal-allcorepera-caud-20240510-122451_gwas_stat_scz_alleleprocessed_a3_results-101324.csv | /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-108982-128981-dynamic-1corestotal-allcorepera-caud-20240510-122451_gwas_stat_mdd_alleleprocessed_a3_results-101324.csv |\n",
       "\n"
      ],
      "text/plain": [
       "  region Chr population chunk_start chunk_end\n",
       "1 caud   1   AA           8982       28981   \n",
       "2 caud   1   AA          28982       48981   \n",
       "3 caud   1   AA          48982       68981   \n",
       "4 caud   1   AA          68982       88981   \n",
       "5 caud   1   AA          88982      108981   \n",
       "6 caud   1   AA         108982      128981   \n",
       "  SNP_data                                                         \n",
       "1 /dcs04/lieber/statsgen/shizhong/michael/mwas/gwas//libd_chr1.pgen\n",
       "2 /dcs04/lieber/statsgen/shizhong/michael/mwas/gwas//libd_chr1.pgen\n",
       "3 /dcs04/lieber/statsgen/shizhong/michael/mwas/gwas//libd_chr1.pgen\n",
       "4 /dcs04/lieber/statsgen/shizhong/michael/mwas/gwas//libd_chr1.pgen\n",
       "5 /dcs04/lieber/statsgen/shizhong/michael/mwas/gwas//libd_chr1.pgen\n",
       "6 /dcs04/lieber/statsgen/shizhong/michael/mwas/gwas//libd_chr1.pgen\n",
       "  methylation_data                                                       \n",
       "1 /dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/caud/out/chr1_AA.rda\n",
       "2 /dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/caud/out/chr1_AA.rda\n",
       "3 /dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/caud/out/chr1_AA.rda\n",
       "4 /dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/caud/out/chr1_AA.rda\n",
       "5 /dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/caud/out/chr1_AA.rda\n",
       "6 /dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/caud/out/chr1_AA.rda\n",
       "  last_meth_value_with_SNP_coverage first_meth_value_with_SNP_coverage\n",
       "1 248918358                         1069461                           \n",
       "2 248918358                         1069461                           \n",
       "3 248918358                         1069461                           \n",
       "4 248918358                         1069461                           \n",
       "5 248918358                         1069461                           \n",
       "6 248918358                         1069461                           \n",
       "  last_meth_index_with_SNP_coverage ⋯\n",
       "1 2202702                           ⋯\n",
       "2 2202702                           ⋯\n",
       "3 2202702                           ⋯\n",
       "4 2202702                           ⋯\n",
       "5 2202702                           ⋯\n",
       "6 2202702                           ⋯\n",
       "  modified_methylation_data                                                  \n",
       "1 /dcs04/lieber/statsgen/mnagle/mwas/pheno/caud/out/chr1_AA_8982-28981.rds   \n",
       "2 /dcs04/lieber/statsgen/mnagle/mwas/pheno/caud/out/chr1_AA_28982-48981.rds  \n",
       "3 /dcs04/lieber/statsgen/mnagle/mwas/pheno/caud/out/chr1_AA_48982-68981.rds  \n",
       "4 /dcs04/lieber/statsgen/mnagle/mwas/pheno/caud/out/chr1_AA_68982-88981.rds  \n",
       "5 /dcs04/lieber/statsgen/mnagle/mwas/pheno/caud/out/chr1_AA_88982-108981.rds \n",
       "6 /dcs04/lieber/statsgen/mnagle/mwas/pheno/caud/out/chr1_AA_108982-128981.rds\n",
       "  path                                                                                                                            \n",
       "1 ..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-8982-28981-dynamic-1corestotal-allcorepera-20240510-121942.rds   \n",
       "2 ..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-28982-48981-dynamic-1corestotal-allcorepera-20240510-122045.rds  \n",
       "3 ..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-48982-68981-dynamic-1corestotal-allcorepera-20240510-122147.rds  \n",
       "4 ..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-68982-88981-dynamic-1corestotal-allcorepera-20240510-122249.rds  \n",
       "5 ..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-88982-108981-dynamic-1corestotal-allcorepera-20240510-122350.rds \n",
       "6 ..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-108982-128981-dynamic-1corestotal-allcorepera-20240510-122451.rds\n",
       "  tag                    \n",
       "1 caud_AA_1_8982_28981   \n",
       "2 caud_AA_1_28982_48981  \n",
       "3 caud_AA_1_48982_68981  \n",
       "4 caud_AA_1_68982_88981  \n",
       "5 caud_AA_1_88982_108981 \n",
       "6 caud_AA_1_108982_128981\n",
       "  path_unmodified                                                                                                                      \n",
       "1 ..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-8982-28981-dynamic-1corestotal-allcorepera-caud-20240510-121942.rds   \n",
       "2 ..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-28982-48981-dynamic-1corestotal-allcorepera-caud-20240510-122045.rds  \n",
       "3 ..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-48982-68981-dynamic-1corestotal-allcorepera-caud-20240510-122147.rds  \n",
       "4 ..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-68982-88981-dynamic-1corestotal-allcorepera-caud-20240510-122249.rds  \n",
       "5 ..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-88982-108981-dynamic-1corestotal-allcorepera-caud-20240510-122350.rds \n",
       "6 ..//output_EXPANSE_a2_caud/libd_chr1-chr1_AA-libd_chr1-chr1_AA-108982-128981-dynamic-1corestotal-allcorepera-caud-20240510-122451.rds\n",
       "  date     time   datetime           \n",
       "1 20240510 121942 2024-05-10 12:19:42\n",
       "2 20240510 122045 2024-05-10 12:20:45\n",
       "3 20240510 122147 2024-05-10 12:21:47\n",
       "4 20240510 122249 2024-05-10 12:22:49\n",
       "5 20240510 122350 2024-05-10 12:23:50\n",
       "6 20240510 122451 2024-05-10 12:24:51\n",
       "  result_bp                                                                                                                                                                                                                            \n",
       "1 /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-8982-28981-dynamic-1corestotal-allcorepera-caud-20240510-121942_gwas_stat_bp_alleleprocessed_a3_results-101324.csv   \n",
       "2 /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-28982-48981-dynamic-1corestotal-allcorepera-caud-20240510-122045_gwas_stat_bp_alleleprocessed_a3_results-101324.csv  \n",
       "3 /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-48982-68981-dynamic-1corestotal-allcorepera-caud-20240510-122147_gwas_stat_bp_alleleprocessed_a3_results-101324.csv  \n",
       "4 /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-68982-88981-dynamic-1corestotal-allcorepera-caud-20240510-122249_gwas_stat_bp_alleleprocessed_a3_results-101324.csv  \n",
       "5 /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-88982-108981-dynamic-1corestotal-allcorepera-caud-20240510-122350_gwas_stat_bp_alleleprocessed_a3_results-101324.csv \n",
       "6 /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-108982-128981-dynamic-1corestotal-allcorepera-caud-20240510-122451_gwas_stat_bp_alleleprocessed_a3_results-101324.csv\n",
       "  result_scz                                                                                                                                                                                                                            \n",
       "1 /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-8982-28981-dynamic-1corestotal-allcorepera-caud-20240510-121942_gwas_stat_scz_alleleprocessed_a3_results-101324.csv   \n",
       "2 /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-28982-48981-dynamic-1corestotal-allcorepera-caud-20240510-122045_gwas_stat_scz_alleleprocessed_a3_results-101324.csv  \n",
       "3 /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-48982-68981-dynamic-1corestotal-allcorepera-caud-20240510-122147_gwas_stat_scz_alleleprocessed_a3_results-101324.csv  \n",
       "4 /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-68982-88981-dynamic-1corestotal-allcorepera-caud-20240510-122249_gwas_stat_scz_alleleprocessed_a3_results-101324.csv  \n",
       "5 /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-88982-108981-dynamic-1corestotal-allcorepera-caud-20240510-122350_gwas_stat_scz_alleleprocessed_a3_results-101324.csv \n",
       "6 /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-108982-128981-dynamic-1corestotal-allcorepera-caud-20240510-122451_gwas_stat_scz_alleleprocessed_a3_results-101324.csv\n",
       "  result_mdd                                                                                                                                                                                                                            \n",
       "1 /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-8982-28981-dynamic-1corestotal-allcorepera-caud-20240510-121942_gwas_stat_mdd_alleleprocessed_a3_results-101324.csv   \n",
       "2 /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-28982-48981-dynamic-1corestotal-allcorepera-caud-20240510-122045_gwas_stat_mdd_alleleprocessed_a3_results-101324.csv  \n",
       "3 /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-48982-68981-dynamic-1corestotal-allcorepera-caud-20240510-122147_gwas_stat_mdd_alleleprocessed_a3_results-101324.csv  \n",
       "4 /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-68982-88981-dynamic-1corestotal-allcorepera-caud-20240510-122249_gwas_stat_mdd_alleleprocessed_a3_results-101324.csv  \n",
       "5 /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-88982-108981-dynamic-1corestotal-allcorepera-caud-20240510-122350_gwas_stat_mdd_alleleprocessed_a3_results-101324.csv \n",
       "6 /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_caud//libd_chr1-chr1_AA-libd_chr1-chr1_AA-108982-128981-dynamic-1corestotal-allcorepera-caud-20240510-122451_gwas_stat_mdd_alleleprocessed_a3_results-101324.csv"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(result_table)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "a1de9c4e-f34b-41ad-ac45-32663263678a",
   "metadata": {},
   "outputs": [],
   "source": [
    "### fast version modified to save combined_cor and comment out subsequent steps\n",
    "\n",
    "# Load necessary libraries\n",
    "library(data.table)\n",
    "library(parallel)\n",
    "library(qqman)\n",
    "library(ggplot2)\n",
    "library(stringr)\n",
    "library(foreach)\n",
    "library(doParallel)\n",
    "\n",
    "# Define variables\n",
    "desired_subpopulations <- c(\"AA\", \"EA\", \"all\")\n",
    "desired_regions <- c(\"caud\", \"hippo\", \"dlpfc\")\n",
    "traits <- c(\"scz\", \"bp\", \"mdd\")\n",
    "\n",
    "# Detect the number of cores and set up the parallel backend\n",
    "num_cores <- 32  # Use one less than the total number of cores\n",
    "cl <- makeCluster(num_cores)\n",
    "registerDoParallel(cl)\n",
    "\n",
    "# Create output directory once\n",
    "dir.create(\"37-OUT\", showWarnings = FALSE)\n",
    "\n",
    "# Loop over subpopulations, regions, and traits\n",
    "for (desired_subpopulation in desired_subpopulations) {\n",
    "  for (desired_region in desired_regions) {\n",
    "    for (trait in traits) {\n",
    "      \n",
    "      sub_df <- result_table[which(result_table$population == desired_subpopulation &\n",
    "                                   result_table$region == desired_region)]\n",
    "      \n",
    "      if(nrow(sub_df) == 0) stop(\"No rows in sub_df. Error.\")\n",
    "       \n",
    "      # Prepare variables for parallel processing\n",
    "      data_files_subset <- sub_df[[paste0(\"result_\", trait)]]\n",
    "      result_table_chr <- sub_df$Chr\n",
    "      \n",
    "      # Parallel data reading and processing\n",
    "      data_list <- foreach(i = seq_along(data_files_subset), .packages = 'data.table') %dopar% {\n",
    "        dt <- fread(data_files_subset[i], showProgress = FALSE)\n",
    "        dt[, chr := result_table_chr[i]]  # Add 'chr' column efficiently\n",
    "        dt\n",
    "      }\n",
    "\n",
    "      dir <- unique(dirname(data_files_subset))\n",
    "      \n",
    "      # Combine the data tables into one\n",
    "      combined_data <- rbindlist(data_list, use.names = TRUE, fill = TRUE)\n",
    "      #stop(\"debug here\")\n",
    "      \n",
    "      # Read and combine correlation files in parallel - This part is messed up. We should only read in cor files for corresponding sub_df paths modified with gsub\n",
    "      cor_files <- str_split_fixed(data_files_subset, \"_gwas\", 2)[, 1]\n",
    "      cor_files <- unique(cor_files)\n",
    "      cor_files <- paste0(cor_files, \"_cv-eval_combined.csv\")\n",
    "\n",
    "      data_files_subset <- result_table_chr <- NULL\n",
    " \n",
    "      cor_list <- foreach(file = cor_files, .packages = 'data.table') %dopar% {\n",
    "        fread(file, showProgress = FALSE)\n",
    "      }\n",
    "      \n",
    "      combined_cor <- rbindlist(cor_list, use.names = TRUE, fill = TRUE)\n",
    "\n",
    "      combined_cor <- unique(combined_cor)\n",
    "      \n",
    "      # Write out the combined_cor data frame with appropriate filename\n",
    "      fwrite(combined_cor, file.path(\"37-OUT\", paste0(\"combined_cor_\", trait, \"_results_a6_\", desired_subpopulation, \"_\", desired_region, \".csv\")))\n",
    "      \n",
    "      # Commenting out the rest of the processing steps\n",
    "      # \n",
    "      # # Ensure consistent column names for merging\n",
    "      # setnames(combined_data, old = \"bp\", new = \"cg\", skip_absent = TRUE)\n",
    "      # setnames(combined_cor, old = \"V4\", new = \"cg\", skip_absent = TRUE)\n",
    "      # \n",
    "      # # Set keys for efficient merging\n",
    "      # setkey(combined_data, cg)\n",
    "      # setkey(combined_cor, cg)\n",
    "      # \n",
    "      # # Merge datasets\n",
    "      # merged <- merge(combined_data, combined_cor, by = c(\"chr\", \"cg\"))\n",
    "      # \n",
    "      # combined_data <- combined_cor <- NULL\n",
    "      #   \n",
    "      # # Filter merged data\n",
    "      # merged <- merged[cor >= 0.1]\n",
    "      # merged_sample <- merged[sample(1:nrow(merged), 10000), ]\n",
    "      # merged_subset <- merged[p <= 0.0001]\n",
    "      # merged_subset[, SNP := paste0(chr, \":\", cg)]\n",
    "      # \n",
    "      # # Bonferroni correction\n",
    "      # bonferroni_threshold <- 0.05 / nrow(merged)\n",
    "      # \n",
    "      # bonferroni_pass <- merged[p <= bonferroni_threshold]\n",
    "      # fwrite(merged, file.path(\"37-OUT\", paste0(\"unfiltered_\", trait, \"_results_a6_\", desired_subpopulation, \"_\", desired_region, \".csv\")))\n",
    "      # merged <- NULL\n",
    "      # \n",
    "      # # Save filtered and unfiltered data\n",
    "      # fwrite(merged_subset, file.path(\"37-OUT\", paste0(\"filtered_\", trait, \"_results_a6_\", desired_subpopulation, \"_\", desired_region, \".csv\")))\n",
    "      # fwrite(bonferroni_pass, file.path(\"37-OUT\", paste0(\"filtered-bonf_\", trait, \"_results_a6_\", desired_subpopulation, \"_\", desired_region, \".csv\")))\n",
    "      # \n",
    "      # # Save Manhattan plot\n",
    "      # png(filename = file.path(\"37-OUT\", paste0(\"manhattan_\", trait, \"_\", desired_subpopulation, \"_\", desired_region, \".png\")))\n",
    "      # manhattan(merged_subset, chr = \"chr\", bp = \"cg\", p = \"p\", main = paste(\"Manhattan plot for\", trait, desired_subpopulation, desired_region))\n",
    "      # dev.off()\n",
    "      # \n",
    "      # # Create and save QQ plot\n",
    "      # png(filename = file.path(\"37-OUT\", paste0(\"qqplot_\", trait, \"_\", desired_subpopulation, \"_\", desired_region, \".png\")))\n",
    "      # qq(merged_sample$p, main = paste(\"QQ plot for 10k samples from\", trait, desired_subpopulation, desired_region))\n",
    "      # dev.off()\n",
    "      # \n",
    "      # gc()\n",
    "    }\n",
    "  }\n",
    "}\n",
    "\n",
    "# Stop the cluster\n",
    "stopCluster(cl)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "92b45b20-f10e-4b71-97fe-8b7c68b21952",
   "metadata": {},
   "outputs": [],
   "source": [
    "### fast version\n",
    "\n",
    "# Load necessary libraries\n",
    "library(data.table)\n",
    "library(parallel)\n",
    "library(qqman)\n",
    "library(ggplot2)\n",
    "library(stringr)\n",
    "library(foreach)\n",
    "library(doParallel)\n",
    "\n",
    "# Define variables\n",
    "desired_subpopulations <- c(\"AA\", \"EA\", \"all\")\n",
    "desired_regions <- c(\"caud\", \"hippo\", \"dlpfc\")\n",
    "traits <- c(\"scz\", \"bp\", \"mdd\")\n",
    "\n",
    "# Detect the number of cores and set up the parallel backend\n",
    "num_cores <- 32  # Use one less than the total number of cores\n",
    "cl <- makeCluster(num_cores)\n",
    "registerDoParallel(cl)\n",
    "\n",
    "# Create output directory once\n",
    "dir.create(\"37-OUT\", showWarnings = FALSE)\n",
    "\n",
    "# Loop over subpopulations, regions, and traits\n",
    "for (desired_subpopulation in desired_subpopulations) {\n",
    "  for (desired_region in desired_regions) {\n",
    "    for (trait in traits) {\n",
    "      \n",
    "      sub_df <- result_table[which(result_table$population == desired_subpopulation &\n",
    "                                   result_table$region == desired_region)]\n",
    "\n",
    "      if(nrow(sub_df) == 0) stop(\"No rows in sub_df. Error.\")\n",
    "       \n",
    "      # Prepare variables for parallel processing\n",
    "      data_files_subset <- sub_df[[paste0(\"result_\", trait)]]\n",
    "      result_table_chr <- sub_df$Chr\n",
    "      \n",
    "      # Parallel data reading and processing\n",
    "      data_list <- foreach(i = seq_along(data_files_subset), .packages = 'data.table') %dopar% {\n",
    "        dt <- fread(data_files_subset[i], showProgress = FALSE)\n",
    "        dt[, chr := result_table_chr[i]]  # Add 'chr' column efficiently\n",
    "        dt\n",
    "      }\n",
    "\n",
    "      dir <- unique(dirname(data_files_subset))\n",
    "      \n",
    "      # Combine the data tables into one\n",
    "      combined_data <- rbindlist(data_list, use.names = TRUE, fill = TRUE)\n",
    "      #stop(\"debug here\")\n",
    "      \n",
    "      # Read and combine correlation files in parallel - This part is messed up. We should only read in cor files for corresponding sub_df paths modiffied with gsub\n",
    "      cor_files <- str_split_fixed(data_files_subset, \"_gwas\", 2)[, 1]\n",
    "      cor_files <- unique(cor_files)\n",
    "      cor_files <- paste0(cor_files, \"_cv-eval_combined.csv\")\n",
    "\n",
    "      data_files_subset <- result_table_chr <- NULL\n",
    " \n",
    "      cor_list <- foreach(file = cor_files, .packages = 'data.table') %dopar% {\n",
    "        fread(file, showProgress = FALSE)\n",
    "      }\n",
    "      \n",
    "      combined_cor <- rbindlist(cor_list, use.names = TRUE, fill = TRUE)\n",
    "\n",
    "      combined_cor <- unique(combined_cor)\n",
    "      \n",
    "      # Ensure consistent column names for merging\n",
    "      setnames(combined_data, old = \"bp\", new = \"cg\", skip_absent = TRUE)\n",
    "      setnames(combined_cor, old = \"V4\", new = \"cg\", skip_absent = TRUE)\n",
    "      \n",
    "      # Set keys for efficient merging\n",
    "      setkey(combined_data, cg)\n",
    "      setkey(combined_cor, cg)\n",
    "      \n",
    "      # Merge datasets\n",
    "      merged <- merge(combined_data, combined_cor, by = c(\"chr\", \"cg\"))\n",
    "\n",
    "      combined_data <- combined_cor <- NULL\n",
    "        \n",
    "      # Filter merged data\n",
    "      merged <- merged[cor >= 0.1]\n",
    "      merged_sample <- merged[sample(1:nrow(merged), 10000), ]\n",
    "      merged_subset <- merged[p <= 0.0001]\n",
    "      merged_subset[, SNP := paste0(chr, \":\", cg)]\n",
    "      \n",
    "      # Bonferroni correction\n",
    "      bonferroni_threshold <- 0.05 / nrow(merged)\n",
    "\n",
    "      bonferroni_pass <- merged[p <= bonferroni_threshold]\n",
    "      fwrite(merged, file.path(\"37-OUT\", paste0(\"unfiltered_\", trait, \"_results_a6_\", desired_subpopulation, \"_\", desired_region, \".csv\")))\n",
    "      merged <- NULL\n",
    "      \n",
    "      # Save filtered and unfiltered data\n",
    "      fwrite(merged_subset, file.path(\"37-OUT\", paste0(\"filtered_\", trait, \"_results_a6_\", desired_subpopulation, \"_\", desired_region, \".csv\")))\n",
    "      fwrite(bonferroni_pass, file.path(\"37-OUT\", paste0(\"filtered-bonf_\", trait, \"_results_a6_\", desired_subpopulation, \"_\", desired_region, \".csv\")))\n",
    "      \n",
    "      # Save Manhattan plot\n",
    "      png(filename = file.path(\"37-OUT\", paste0(\"manhattan_\", trait, \"_\", desired_subpopulation, \"_\", desired_region, \".png\")))\n",
    "      manhattan(merged_subset, chr = \"chr\", bp = \"cg\", p = \"p\", main = paste(\"Manhattan plot for\", trait, desired_subpopulation, desired_region))\n",
    "      dev.off()\n",
    "      \n",
    "      # Create and save QQ plot\n",
    "      png(filename = file.path(\"37-OUT\", paste0(\"qqplot_\", trait, \"_\", desired_subpopulation, \"_\", desired_region, \".png\")))\n",
    "      qq(merged_sample$p, main = paste(\"QQ plot for 10k samples from\", trait, desired_subpopulation, desired_region))\n",
    "      dev.off()\n",
    "\n",
    "      gc()\n",
    "    }\n",
    "  }\n",
    "}\n",
    "\n",
    "# Stop the cluster\n",
    "stopCluster(cl)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "792406f7-ea14-4bd9-bd54-9c3abe6d708f",
   "metadata": {},
   "source": [
    "# What's up with duplicate entries?"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "45da0eb7-ab6b-4328-a923-f1cbb7ebb35e",
   "metadata": {},
   "source": [
    "Identify them"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "800560e9-9843-48b0-a53f-1305738f92ea",
   "metadata": {},
   "outputs": [],
   "source": [
    "# version with logging so we can see where duplicates are coming from\n",
    "\n",
    "# Load necessary libraries\n",
    "library(data.table)\n",
    "library(parallel)\n",
    "library(qqman)\n",
    "library(ggplot2)\n",
    "library(stringr)\n",
    "library(foreach)\n",
    "library(doParallel)\n",
    "\n",
    "# Define variables\n",
    "desired_subpopulations <- c(\"AA\", \"EA\", \"all\")\n",
    "desired_regions <- c(\"caud\", \"hippo\", \"dlpfc\")\n",
    "traits <- c(\"scz\", \"bp\", \"mdd\")\n",
    "\n",
    "# Detect the number of cores and set up the parallel backend\n",
    "num_cores <- 32  # Use one less than the total number of cores\n",
    "cl <- makeCluster(num_cores)\n",
    "registerDoParallel(cl)\n",
    "\n",
    "# Create output directory once\n",
    "dir.create(\"37-OUT\", showWarnings = FALSE)\n",
    "\n",
    "# Loop over subpopulations, regions, and traits\n",
    "for (desired_subpopulation in desired_subpopulations) {\n",
    "  for (desired_region in desired_regions) {\n",
    "    for (trait in traits) {\n",
    "      \n",
    "      sub_df <- result_table[which(result_table$population == desired_subpopulation &\n",
    "                                   result_table$region == desired_region)]\n",
    "\n",
    "      if(nrow(sub_df) == 0) stop(\"No rows in sub_df. Error.\")\n",
    "       \n",
    "      # Prepare variables for parallel processing\n",
    "      data_files_subset <- sub_df[[paste0(\"result_\", trait)]]\n",
    "      result_table_chr <- sub_df$Chr\n",
    "      \n",
    "      # Parallel data reading and processing\n",
    "      data_list <- foreach(i = seq_along(data_files_subset), .packages = 'data.table') %dopar% {\n",
    "        dt <- fread(data_files_subset[i], showProgress = FALSE)\n",
    "        dt[, chr := result_table_chr[i]]  # Add 'chr' column efficiently\n",
    "        dt[, source_file := data_files_subset[i]]  # Add source file column\n",
    "        dt\n",
    "      }\n",
    "\n",
    "      dir <- unique(dirname(data_files_subset))\n",
    "      \n",
    "      # Combine the data tables into one\n",
    "      combined_data <- rbindlist(data_list, use.names = TRUE, fill = TRUE)\n",
    "      #stop(\"debug here\")\n",
    "      \n",
    "      # Read and combine correlation files in parallel\n",
    "      cor_files <- str_split_fixed(data_files_subset, \"_gwas\", 2)[, 1]\n",
    "      cor_files <- unique(cor_files)\n",
    "      cor_files <- paste0(cor_files, \"_cv-eval_combined.csv\")\n",
    "\n",
    "      data_files_subset <- result_table_chr <- NULL\n",
    " \n",
    "      cor_list <- foreach(file = cor_files, .packages = 'data.table') %dopar% {\n",
    "        dt <- fread(file, showProgress = FALSE)\n",
    "        dt[, source_file := file]  # Add source file column\n",
    "        dt\n",
    "      }\n",
    "      \n",
    "      combined_cor <- rbindlist(cor_list, use.names = TRUE, fill = TRUE)\n",
    "\n",
    "      combined_cor <- unique(combined_cor)\n",
    "      \n",
    "      # Ensure consistent column names for merging\n",
    "      setnames(combined_data, old = \"bp\", new = \"cg\", skip_absent = TRUE)\n",
    "      setnames(combined_cor, old = \"V4\", new = \"cg\", skip_absent = TRUE)\n",
    "      \n",
    "      # Set keys for efficient merging\n",
    "      setkey(combined_data, chr, cg)\n",
    "      setkey(combined_cor, chr, cg)\n",
    "\n",
    "      # Check for duplicates in combined_data\n",
    "      dup_combined_data <- combined_data[duplicated(combined_data, by = c(\"chr\", \"cg\"))]\n",
    "      if (nrow(dup_combined_data) > 0) {\n",
    "          message(\"Duplicates found in combined_data.\")\n",
    "      }\n",
    "\n",
    "      # Check for duplicates in combined_cor\n",
    "      dup_combined_cor <- combined_cor[duplicated(combined_cor, by = c(\"chr\", \"cg\"))]\n",
    "      if (nrow(dup_combined_cor) > 0) {\n",
    "        message(\"Duplicates found in combined_cor.\")\n",
    "      }\n",
    "      \n",
    "      # Merge datasets\n",
    "      merged <- merge(combined_data, combined_cor, by = c(\"chr\", \"cg\"), suffixes = c(\".data\", \".cor\"))\n",
    "\n",
    "      combined_data <- combined_cor <- NULL\n",
    "        \n",
    "      # Logging duplicates\n",
    "      dup_rows <- merged[duplicated(merged, by = c(\"chr\", \"cg\")) | duplicated(merged, by = c(\"chr\", \"cg\"), fromLast = TRUE)]\n",
    "      if (nrow(dup_rows) > 0) {\n",
    "        log_file <- file.path(\"37-OUT\", paste0(\"duplicate_log_\", trait, \"_\", desired_subpopulation, \"_\", desired_region, \".txt\"))\n",
    "        fwrite(head(dup_rows, 1000), log_file)  # Limit to first 1000 duplicates\n",
    "      }\n",
    "        \n",
    "      # Filter merged data\n",
    "      merged <- merged[cor >= 0.1]\n",
    "      merged_sample <- merged[sample(1:nrow(merged), min(10000, nrow(merged))), ]\n",
    "      merged_subset <- merged[p <= 0.0001]\n",
    "      merged_subset[, SNP := paste0(chr, \":\", cg)]\n",
    "      \n",
    "      # Bonferroni correction\n",
    "      bonferroni_threshold <- 0.05 / nrow(merged)\n",
    "\n",
    "      bonferroni_pass <- merged[p <= bonferroni_threshold]\n",
    "      fwrite(merged, file.path(\"37-OUT\", paste0(\"unfiltered_\", trait, \"_results_a6_\", desired_subpopulation, \"_\", desired_region, \".csv\")))\n",
    "      merged <- NULL\n",
    "      \n",
    "      # Save filtered and unfiltered data\n",
    "      fwrite(merged_subset, file.path(\"37-OUT\", paste0(\"filtered_\", trait, \"_results_a6_\", desired_subpopulation, \"_\", desired_region, \".csv\")))\n",
    "      fwrite(bonferroni_pass, file.path(\"37-OUT\", paste0(\"filtered-bonf_\", trait, \"_results_a6_\", desired_subpopulation, \"_\", desired_region, \".csv\")))\n",
    "      \n",
    "      # Save Manhattan plot\n",
    "      png(filename = file.path(\"37-OUT\", paste0(\"manhattan_\", trait, \"_\", desired_subpopulation, \"_\", desired_region, \".png\")))\n",
    "      manhattan(merged_subset, chr = \"chr\", bp = \"cg\", p = \"p\", main = paste(\"Manhattan plot for\", trait, desired_subpopulation, desired_region))\n",
    "      dev.off()\n",
    "      \n",
    "      # Create and save QQ plot\n",
    "      png(filename = file.path(\"37-OUT\", paste0(\"qqplot_\", trait, \"_\", desired_subpopulation, \"_\", desired_region, \".png\")))\n",
    "      qq(merged_sample$p, main = paste(\"QQ plot for 10k samples from\", trait, desired_subpopulation, desired_region))\n",
    "      dev.off()\n",
    "\n",
    "      gc()\n",
    "    }\n",
    "  }\n",
    "}\n",
    "\n",
    "# Stop the cluster\n",
    "stopCluster(cl)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "60e5760d-c52b-486f-bf48-fcb91fc04cc9",
   "metadata": {},
   "source": [
    "### investigate further "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b5843797-fda0-4f9c-a16b-77bf1ea895fe",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0649edb5-6b68-4d33-a99b-2d8000795307",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8182e473-870a-4ad5-858a-420b57e12823",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ce76a176-62ad-4272-85be-5c5c4cb23e48",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "94b5be36-ab9c-4737-a75d-ffbdcfc844d4",
   "metadata": {},
   "outputs": [],
   "source": [
    "is.data.frame(sub_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "56671d17-a7b4-40d0-bdd4-68a9208efec8",
   "metadata": {},
   "outputs": [],
   "source": [
    "is.data.table(sub_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "15f3ba3f-f1f1-4489-970b-7054cf3fc41a",
   "metadata": {},
   "outputs": [],
   "source": [
    "data_files_subset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0a9cad0c-680c-4680-9d23-81b8015695fc",
   "metadata": {},
   "outputs": [],
   "source": [
    "dir"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "266591aa-e9a1-4341-8253-1d4dd42e13a6",
   "metadata": {},
   "outputs": [],
   "source": [
    "desired_subpopulation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "98d61d0e-2bc4-4b64-913d-aaa085089754",
   "metadata": {},
   "outputs": [],
   "source": [
    "desired_region"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "892e3260-d31f-41d1-896c-8d5bdcd03d59",
   "metadata": {},
   "outputs": [],
   "source": [
    "head(result_table)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0f88a679-9f62-4955-8822-6029feb408f1",
   "metadata": {},
   "outputs": [],
   "source": [
    "levels(factor(result_table$region))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "10936953-cdd3-458d-9d4c-e4778760c046",
   "metadata": {},
   "outputs": [],
   "source": [
    "head(combined_data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "601cccaf-23da-4b15-82ff-6330a37a2e5f",
   "metadata": {},
   "outputs": [],
   "source": [
    "head(combined_cor)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7ecc0fec-8a49-44c5-a206-6730dba4b352",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9d4c1f43-53f4-4182-9a43-2811e6e3c7e9",
   "metadata": {},
   "outputs": [],
   "source": [
    "head(sub_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bd37f63f-7cfb-4db9-981c-fcb2f5eab512",
   "metadata": {},
   "outputs": [],
   "source": [
    "head(data_files_subset)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "592fd834-3fc0-4226-af37-8925a30b00f4",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ebaf2d06-c3af-46e9-9fa2-a20722edd737",
   "metadata": {},
   "outputs": [],
   "source": [
    "desired_subpopulation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2730e1f9-f389-4cfa-a92b-52830b537652",
   "metadata": {},
   "outputs": [],
   "source": [
    "desired_region"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1feb0aff-a7e3-4258-9b4a-07ed6b4a1609",
   "metadata": {},
   "outputs": [],
   "source": [
    "trait"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "984738e1-3de2-4610-9aba-f579b85a7589",
   "metadata": {},
   "outputs": [],
   "source": [
    "head(result_table)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f1d04bc4-5dc8-45a1-927c-4abb35c84a34",
   "metadata": {},
   "outputs": [],
   "source": [
    "gead()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "441b16af-40de-4c8f-b796-52cc6c1b1dbe",
   "metadata": {},
   "outputs": [],
   "source": [
    "sub_df[, which(colnames(sub_df) == paste0(\"result_\", trait))]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5bd6c4e2-1c34-4654-86ab-18b95f9d124f",
   "metadata": {},
   "outputs": [],
   "source": [
    "sub_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b6818bd4-fd50-45bc-9561-5dde81daf5c7",
   "metadata": {},
   "outputs": [],
   "source": [
    "head(combined_data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "30dc3734-c4ad-4380-b68e-1298bc36bcad",
   "metadata": {},
   "outputs": [],
   "source": [
    "head(combined_cor)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5f44c8e1-4ecd-46de-94a4-c3020709aa28",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d25986c5-bf4a-4605-8c8e-7854182d38df",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8005707f-42c6-4069-a74a-8b63e202f09c",
   "metadata": {},
   "outputs": [],
   "source": [
    "# ## slow\n",
    "\n",
    "# # Load necessary libraries\n",
    "# library(data.table)\n",
    "# library(parallel)\n",
    "# library(stringr)\n",
    "\n",
    "# # Define variables\n",
    "# desired_subpopulations <- c(\"AA\", \"EA\", \"all\")\n",
    "# desired_regions <- c(\"caud\", \"hippo\", \"dlpfc\")\n",
    "# traits <- c(\"scz\", \"bp\", \"mdd\")\n",
    "\n",
    "# num_cores <- 6\n",
    "# chromosome_stats <- list()  # Store stats for each combination\n",
    "\n",
    "# # Loop over subpopulations, regions, and traits\n",
    "# for (desired_subpopulation in desired_subpopulations) {\n",
    "#   for (desired_region in desired_regions) {\n",
    "#     for (trait in traits) {\n",
    "      \n",
    "#       # Define directory path\n",
    "#       dir <- paste0(\"/expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/output_EXPANSE_dt_\", desired_region, \"/\")\n",
    "      \n",
    "#       # List and filter data files\n",
    "#       data_files <- list.files(dir, full.names = TRUE, pattern = paste0(\"gwas_stat_\", trait, \"_alleleprocessed_a3\"))\n",
    "#       data_files <- data_files[grepl(\"csv\", data_files)]\n",
    "#       data_files_alt_name <- gsub(paste0(\"-\", desired_region), \"\", data_files)\n",
    "      \n",
    "#       # Create result table\n",
    "#       result_table <- data.table(str_split_fixed(data_files_alt_name, \"[-_]\", n = Inf))\n",
    "#       result_table$path <- data_files\n",
    "#       result_table$chr <- as.numeric(gsub(\"chr\", \"\", result_table$V5))\n",
    "#       result_table <- result_table[which(result_table$V7 == desired_subpopulation)]\n",
    "      \n",
    "#       # Filter by most recent date\n",
    "#       result_table$region <- str_split_fixed(result_table$path, \"EXPANSE_dt_\", 2)[, 2]\n",
    "#       result_table$region <- str_split_fixed(result_table$region, \"/\", 2)[, 1]\n",
    "    \n",
    "#       result_table[, V17 := as.Date(V17, format=\"%Y%m%d\")]\n",
    "    \n",
    "#       # Filter to keep only the row with the most recent date for each V12\n",
    "    \n",
    "#       result_table$tag <- paste0(result_table$V11, \"_\", result_table$V21, \"_\", result_table$region, \"_\", result_table$chr, \"_\", result_table$V12)\n",
    "#       resuilt_table <- result_table[which(result_table$V23 == \"a3\"), ]\n",
    "      \n",
    "#       # Sanity check for duplicates\n",
    "#       dupes_check <- result_table[, .N, by=.(V12, V17)][N > 1]\n",
    "#       if (nrow(dupes_check) > 0) {\n",
    "#         stop(\"Sanity check failed: Multiple datasets found for the same V12 on the same date.\")\n",
    "#       }\n",
    "      \n",
    "#       # Combine data files\n",
    "#       data_files <- result_table$path\n",
    "#       data_list <- list()\n",
    "#       for (i in seq_along(data_files)) {\n",
    "#         data_list[[i]] <- fread(data_files[i])\n",
    "#         data_list[[i]]$chr <- result_table[i]$chr\n",
    "#       }\n",
    "#       combined_data <- rbindlist(data_list)\n",
    "      \n",
    "#       # Combine correlation files\n",
    "#       cor_files <- list.files(dir, full.names = TRUE, pattern = \"eval\")\n",
    "#       cor_list <- mclapply(cor_files, fread, mc.cores = num_cores)\n",
    "#       combined_cor <- rbindlist(cor_list)\n",
    "      \n",
    "#       # Merge datasets\n",
    "#       colnames(combined_data)[4] <- \"cg\"\n",
    "#       merged <- merge(combined_data, combined_cor)\n",
    "#       merged <- merged[which(merged$cor >= 0.1), ]\n",
    "#       merged_subset <- merged[which(merged$p <= 0.0001), ]\n",
    "#       merged_subset$SNP <- paste0(merged_subset$chr, \":\", merged_subset$cg)\n",
    "      \n",
    "#       # Bonferroni correction\n",
    "#       bonferroni_threshold <- 0.05 / nrow(merged)\n",
    "#       bonferroni_pass <- merged[merged$p <= bonferroni_threshold, ]\n",
    "      \n",
    "#       # Count rows by chromosome\n",
    "#       unfiltered_counts <- merged[, .N, by = chr]\n",
    "#       filtered_counts <- merged_subset[, .N, by = chr]\n",
    "      \n",
    "#       # Add to stats list\n",
    "#       chromosome_stats[[paste0(trait, \"_\", desired_subpopulation, \"_\", desired_region)]] <- list(\n",
    "#         unfiltered = unfiltered_counts,\n",
    "#         filtered = filtered_counts,\n",
    "#         bonferroni = bonferroni_pass[, .N, by = chr]\n",
    "#       )\n",
    "      \n",
    "#       # Print counts to console\n",
    "#       cat(\"Chromosome counts for trait:\", trait, \", subpopulation:\", desired_subpopulation, \", region:\", desired_region, \"\\n\")\n",
    "#       cat(\"Unfiltered counts:\\n\")\n",
    "#       print(unfiltered_counts)\n",
    "#       cat(\"Filtered counts (p <= 0.01):\\n\")\n",
    "#       print(filtered_counts)\n",
    "#       cat(\"Bonferroni-filtered counts:\\n\")\n",
    "#       print(chromosome_stats[[paste0(trait, \"_\", desired_subpopulation, \"_\", desired_region)]]$bonferroni)\n",
    "#       gc()\n",
    "#     }\n",
    "#   }\n",
    "# }\n",
    "\n",
    "# # Write CSV files for stats\n",
    "# for (name in names(chromosome_stats)) {\n",
    "#   write.csv(cbind(\"Chromosome\" = chromosome_stats[[name]]$unfiltered$chr, \n",
    "#                   \"Unfiltered_Count\" = chromosome_stats[[name]]$unfiltered$N, \n",
    "#                   \"Filtered_Count\" = chromosome_stats[[name]]$filtered$N, \n",
    "#                   \"Bonferroni_Count\" = chromosome_stats[[name]]$bonferroni$N), \n",
    "#             file = paste0(name, \"_chromosome_counts.csv\"), row.names = FALSE)\n",
    "# }"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a81012f7-b5a4-4294-9518-5a8e4012ce0a",
   "metadata": {},
   "outputs": [],
   "source": [
    "head(merged)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1ae0f009-0806-4532-8701-d7fbb4c53394",
   "metadata": {},
   "outputs": [],
   "source": [
    "dim(merged)"
   ]
  },
  {
   "cell_type": "raw",
   "id": "a270a647-3ef8-409a-b7fa-541ceff83c42",
   "metadata": {},
   "source": [
    "head(merged_subset)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b5aef0d1-d44e-4744-a0b9-13b6c73fe7ed",
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_subset[which(merged_subset$`chr.x` != merged_subset$`chr.y`), ]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "71b465eb-fbeb-4504-a392-40c97ef2411b",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.3.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
