{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "cf486ff8-1c46-4d4b-8891-7cc3250fd190",
   "metadata": {},
   "source": [
    "# Clarify my confusion with the outer loop"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "22c161f6-39ec-4b00-baa2-40285616330a",
   "metadata": {},
   "source": [
    "Get data ready"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "16bf7ecd-d162-4a98-baf2-87947378270b",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: Matrix\n",
      "\n",
      "Loaded glmnet 4.1-8\n",
      "\n",
      "Loading required package: foreach\n",
      "\n",
      "Loading required package: iterators\n",
      "\n",
      "Loading required package: parallel\n",
      "\n",
      "Loading required package: future\n",
      "\n",
      "Loading required package: BiocGenerics\n",
      "\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,\n",
      "    match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,\n",
      "    Position, rank, rbind, Reduce, rownames, sapply, setdiff, sort,\n",
      "    table, tapply, union, unique, unsplit, which.max, which.min\n",
      "\n",
      "\n",
      "Loading required package: GenomicRanges\n",
      "\n",
      "Loading required package: stats4\n",
      "\n",
      "Loading required package: S4Vectors\n",
      "\n",
      "\n",
      "Attaching package: ‘S4Vectors’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:Matrix’:\n",
      "\n",
      "    expand, unname\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:data.table’:\n",
      "\n",
      "    first, second\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:utils’:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Loading required package: IRanges\n",
      "\n",
      "\n",
      "Attaching package: ‘IRanges’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:data.table’:\n",
      "\n",
      "    shift\n",
      "\n",
      "\n",
      "Loading required package: GenomeInfoDb\n",
      "\n",
      "Loading required package: SummarizedExperiment\n",
      "\n",
      "Loading required package: MatrixGenerics\n",
      "\n",
      "Loading required package: matrixStats\n",
      "\n",
      "\n",
      "Attaching package: ‘MatrixGenerics’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:matrixStats’:\n",
      "\n",
      "    colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,\n",
      "    colCounts, colCummaxs, colCummins, colCumprods, colCumsums,\n",
      "    colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,\n",
      "    colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,\n",
      "    colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,\n",
      "    colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,\n",
      "    colWeightedMeans, colWeightedMedians, colWeightedSds,\n",
      "    colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,\n",
      "    rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,\n",
      "    rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,\n",
      "    rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,\n",
      "    rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,\n",
      "    rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,\n",
      "    rowWeightedMads, rowWeightedMeans, rowWeightedMedians,\n",
      "    rowWeightedSds, rowWeightedVars\n",
      "\n",
      "\n",
      "Loading required package: Biobase\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘Biobase’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:MatrixGenerics’:\n",
      "\n",
      "    rowMedians\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:matrixStats’:\n",
      "\n",
      "    anyMissing, rowMedians\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(CpGWAS)\n",
    "library(data.table)\n",
    "library(glmnet)\n",
    "library(e1071)\n",
    "library(doParallel)\n",
    "library(future.apply)\n",
    "library(optparse)\n",
    "library(pgenlibr)\n",
    "library(bsseq)\n",
    "library(tools)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9adb3e54-7b9e-4395-a20a-48530e9404ca",
   "metadata": {},
   "source": [
    "## Prepare input arguments"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "871e2dfb-0062-41bd-8215-062d9b8c75bb",
   "metadata": {},
   "outputs": [],
   "source": [
    "if(Sys.getenv(\"RSTUDIO\") == 0){\n",
    "\n",
    "  # Define command line options\n",
    "  option_list <- list(\n",
    "    make_option(c(\"-o\", \"--outdir\"), type = \"character\", default = \"./output/\",\n",
    "                help = \"Output directory, default is './output/'\"),\n",
    "    make_option(c(\"-c1\", \"--chunk1\"), type = \"integer\", default = 1,\n",
    "                help = \"Starting methylation site index for processing\"),\n",
    "    make_option(c(\"-c2\", \"--chunk2\"), type = \"integer\", default = NA,\n",
    "                help = \"Ending methylation site index for processing\"),\n",
    "    make_option(c(\"-s\", \"--snp_data_path\"), type = \"character\", default = NULL,\n",
    "                help = \"Path to SNP data (required)\"),\n",
    "    make_option(c(\"-m\", \"--methylation_data_path\"), type = \"character\", default = NULL,\n",
    "                help = \"Path to methylation data (required)\")\n",
    "  )\n",
    "\n",
    "  # Parse options\n",
    "  args <- parse_args(OptionParser(option_list = option_list))\n",
    "} else {\n",
    "  args <- list(\n",
    "    outdir = \"./output/\",\n",
    "    chunk1 = 1000000,\n",
    "    chunk2 = 1000100,\n",
    "    snp_data_path = \"/Users/michaelnagle/code/mwas/gwas/libd_chr1.pgen\",\n",
    "    methylation_data_path = \"/Users/michaelnagle/code/mwas/pheno/dlpfc/out/chr1_AA.rda\"\n",
    "  )\n",
    "}\n",
    "\n",
    "if(!dir.exists(args$outdir)) {\n",
    "  dir.create(args$outdir)\n",
    "}\n",
    "\n",
    "# Check required arguments\n",
    "if (is.null(args$snp_data_path) || is.null(args$methylation_data_path)) {\n",
    "  stop(\"Paths to both SNP and methylation data are required.\")\n",
    "}\n",
    "\n",
    "load(args$methylation_data_path)\n",
    "\n",
    "# Pt. 2: Initialize MethylationInput object -------------------------------\n",
    "\n",
    "methInput <- new(\"MethylationInput\",\n",
    "                 BSseq_obj = BSobj2,\n",
    "                 snp_data_path = args$snp_data_path,\n",
    "                 args = args)\n",
    "\n",
    "# Pt. 3: Main loop to process SNP data for each methylation site ----------\n",
    "\n",
    "start_time <- Sys.time()  # Start time capture\n",
    "\n",
    "window_sizes <- c(1000, 2000, 5000, 10000, 20000, 50000, 100000, 200000, 500000)\n",
    "\n",
    "scaffoldIdentifier <- paste0(tools::file_path_sans_ext(basename(args$snp_data_path)),\n",
    "                             \"-\",\n",
    "                             tools::file_path_sans_ext(basename(args$methylation_data_path)))\n",
    "\n",
    "scaffold_name <- scaffoldIdentifier"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "73cca8fe-bb54-4f54-b402-c8f69ca859f7",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true
   },
   "source": [
    "## Load helper functions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "310e809e-11e5-4fb4-ae0e-2cb2adf12d86",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Function to reorder and filter geno matrix based on a vector of accepted genotype IDs\n",
    "reorder_and_filter_geno <- function(geno, genotype_IDs) {\n",
    "  # Filter out rows from geno that are not in genotype_IDs\n",
    "  idx_geno <- which(row.names(geno) %in% genotype_IDs)\n",
    "  geno_filtered <- geno[idx_geno, ]\n",
    "  \n",
    "  # Check if both geno and genotype_IDs have the same row names\n",
    "  if (length(genotype_IDs) != nrow(geno_filtered) || !all(genotype_IDs %in% row.names(geno_filtered))) {\n",
    "    stop(\"Row names do not match 100% between geno and genotype_IDs\")\n",
    "  }\n",
    "  \n",
    "  # Find the matching indices for reordering\n",
    "  match_indices <- match(genotype_IDs, row.names(geno_filtered))\n",
    "  \n",
    "  # Reorder geno matrix to match the order of genotype_IDs\n",
    "  reordered_geno <- geno_filtered[match_indices, ]\n",
    "  \n",
    "  return(reordered_geno)\n",
    "}\n",
    "\n",
    "regress_out_cov <- function(methylations, cov, n_benchmarks = NULL) {\n",
    "  print(\"We just entered regress_out_cov()\")\n",
    "  \n",
    "  if(is.null(methylations)){\n",
    "    stop(\"Error: methylation data not found\")\n",
    "  }\n",
    "  \n",
    "  cat(\"Dimensions of methylations: \", dim(methylations), \"\\n\")\n",
    "  \n",
    "  # Creating the model formula\n",
    "  colnames(cov) <- gsub(\"\\\\(Intercept\\\\)\", \"Intercept\", colnames(cov))\n",
    "  cov <- as.data.frame(cov)\n",
    "  model_formula <- as.formula(paste(\"y ~ \", paste(colnames(cov), collapse=\" + \")))\n",
    "  \n",
    "  n_tests <- if (is.null(n_benchmarks)) ncol(methylations) else n_benchmarks\n",
    "  residuals_matrix <- matrix(NA, nrow = nrow(methylations), ncol = n_tests)\n",
    "  \n",
    "  for(i in 1:n_tests) {\n",
    "    y <- methylations[, i]\n",
    "    lm_model <- lm(model_formula, data = cbind(y, cov))\n",
    "    residuals_matrix[, i] <- residuals(lm_model)\n",
    "  }\n",
    "  \n",
    "  cat(\"Residuals computed for \", n_tests, \" tests.\\n\")\n",
    "  return(residuals_matrix)\n",
    "}\n",
    "\n",
    "regress_out_cov_parallel <- function(methylations, cov_matrix, n_benchmarks = NULL) {\n",
    "  print(\"We just entered regress_out_cov_parallel()\")\n",
    "  no_cores <- detectCores() / 2\n",
    "  plan(sequential)\n",
    "  \n",
    "  if(is.null(methylations)){\n",
    "    stop(\"Error: methylation data not found\")\n",
    "  }\n",
    "  \n",
    "  cat(\"Dimensions of methylations: \", dim(methylations), \"\\n\")\n",
    "  \n",
    "  cat(\"Dimensions of cov_matrix: \", dim(cov_matrix), \"\\n\")\n",
    "  \n",
    "  pseudoinv <- solve(t(cov_matrix) %*% cov_matrix) %*% t(cov_matrix)\n",
    "  cat(\"Dimensions of pseudoinv: \", dim(pseudoinv), \"\\n\")\n",
    "  \n",
    "  n_tests <- if (is.null(n_benchmarks)) ncol(methylations) else n_benchmarks\n",
    "  chunk_size <- ceiling(n_tests / no_cores)\n",
    "  chunks <- lapply(1:no_cores, function(i) {\n",
    "    start_col <- (i - 1) * chunk_size + 1\n",
    "    end_col <- min(i * chunk_size, n_tests)\n",
    "    methylations[, start_col:end_col]\n",
    "  })\n",
    "  \n",
    "  residuals_computation <- function(chunk, cov_matrix, pseudoinv) {\n",
    "    tryCatch({\n",
    "      # Compute the fitted values for the entire matrix\n",
    "      fitted_values <- cov_matrix %*% (pseudoinv %*% chunk)\n",
    "      # Compute the residuals for the entire matrix\n",
    "      residuals <- chunk - fitted_values\n",
    "      return(residuals)\n",
    "    }, error = function(e) {\n",
    "      cat(\"Error occurred: \", e$message, \"\\n\")\n",
    "      cat(\"Dimensions of cov_matrix: \", dim(cov_matrix), \"\\n\")\n",
    "      cat(\"Dimensions of pseudoinv: \", dim(pseudoinv), \"\\n\")\n",
    "      cat(\"Dimensions of chunk: \", dim(chunk), \"\\n\")\n",
    "      matrix(NA, nrow = nrow(chunk), ncol = ncol(chunk))\n",
    "    })\n",
    "  }\n",
    "  \n",
    "  start_time <- Sys.time()\n",
    "  results <- future_lapply(chunks, residuals_computation, cov_matrix, pseudoinv)\n",
    "  residuals_matrix <- do.call(cbind, results)\n",
    "  \n",
    "  if (!is.null(n_benchmarks)) {\n",
    "    end_time <- Sys.time()\n",
    "    elapsed_time <- end_time - start_time\n",
    "    cat(\"Elapsed time: \", elapsed_time, \"\\n\")\n",
    "  }\n",
    "  \n",
    "  return(residuals_matrix)\n",
    "}\n",
    "\n",
    "processCovariates <- function(dataFrame, \n",
    "                              colsToExclude = c(\"ID.\", \"DNum\", \"BrNum\",\n",
    "                                                \"brnum\", \"brnumerical\"),\n",
    "                              genotype_IDs = NULL) {\n",
    "  \n",
    "  if(is.null(genotype_IDs)){\n",
    "    stop(paste0(\"`processCovariates must receive a vector of genotype IDs\",\n",
    "                \" with order matching methylation trait file.\"))\n",
    "  }\n",
    "  \n",
    "  # Sanity check: ensure rownames match brnum\n",
    "  if (!all(rownames(dataFrame) == dataFrame$brnum)) {\n",
    "    stop(\"Row names do not match 'brnum'\")\n",
    "  }\n",
    "  \n",
    "  # Exclude specific columns\n",
    "  dataFrame <- dataFrame[, !(names(dataFrame) %in% colsToExclude)]\n",
    "  \n",
    "  # Exclude columns with only one factor level and print a warning\n",
    "  single_factor <- sapply(dataFrame, function(x) length(levels(factor(x))) == 1)\n",
    "  \n",
    "  if (any(single_factor)) {\n",
    "    # warning(\"Removing columns with only one factor level: \",\n",
    "    #         paste(names(dataFrame)[single_factor], collapse = \", \"))\n",
    "    dataFrame <- dataFrame[, !single_factor]\n",
    "  }\n",
    "  \n",
    "  dataFrame <- dataFrame[which(rownames(dataFrame) %in% genotype_IDs), ]\n",
    "  dataFrame <- dataFrame[order(rownames(dataFrame)),]\n",
    "  \n",
    "  if(!all(rownames(dataFrame) == genotype_IDs)){\n",
    "    stop(\"Mismatch with covariate data and genotype IDs\")\n",
    "  }\n",
    "  \n",
    "  # Get the indices of the factor columns\n",
    "  factor_columns <- sapply(dataFrame, is.factor)\n",
    "  \n",
    "  # One-hot encode the factor columns using model.matrix\n",
    "  one_hot_encoded <- model.matrix(~., data = dataFrame[, factor_columns])\n",
    "  \n",
    "  # Combine the one-hot encoded columns with the numeric columns\n",
    "  final_data <- as.matrix(cbind(one_hot_encoded, dataFrame[, !factor_columns]))\n",
    "  \n",
    "  return(final_data)\n",
    "}\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d520b71d-5114-4e15-80e9-59db770aafd4",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true
   },
   "source": [
    "## Load data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "74c51d71-6a56-4d69-a4a8-8e4241d55018",
   "metadata": {
    "jupyter": {
     "source_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "\n",
    "# Pt. 2: Prepare input data -----------------------------------------------\n",
    "\n",
    "# Load methylation data\n",
    "suppressWarnings(load(args$methylation_data_path))\n",
    "methylations <- t((as.matrix(getMeth(BSobj2,\n",
    "                                     type = \"smooth\", what  = \"perBase\"))))\n",
    "colnames(methylations) <- paste0(\"pos_\", start(ranges(rowRanges(BSobj2))))\n",
    "# # Extract only once then subset\n",
    "\n",
    "scaffold_name <- tools::file_path_sans_ext(basename(scaffold_name))\n",
    "pgen_path <- gsub(args$snp_data_path, pattern = \"pvar\", replacement = \"pgen\")\n",
    "pvar_path <- gsub(args$snp_data_path, pattern = \"pgen\", replacement = \"pvar\")\n",
    "psam_path <- gsub(pvar_path, pattern = \"pvar\", replacement = \"psam\")\n",
    "\n",
    "pvar.1 <- pgenlibr::NewPvar(pvar_path)\n",
    "pvar.2 <- fread(pvar_path)\n",
    "pgen <- pgenlibr::NewPgen(pgen_path, pvar=pvar.1)\n",
    "psam <- fread(psam_path)\n",
    "psam_in_wgbs <- psam[which(psam$`#IID` %in% rownames(methylations))]\n",
    "genotype_IDs <- psam_in_wgbs$`#IID`\n",
    "genotype_IDs <- intersect(rownames(methylations), genotype_IDs)\n",
    "genotype_IDs <- genotype_IDs[order(genotype_IDs)]\n",
    "\n",
    "\n",
    "cov <- processCovariates(dataFrame = colData(BSobj2),\n",
    "                         colsToExclude = c(\"ID.\", \"DNum\", \"brnum\",\n",
    "                                           \"BrNum\", \"brnumerical\"),\n",
    "                         genotype_IDs = genotype_IDs)\n",
    "\n",
    "# Matching indices for genotypes in 'methylations'\n",
    "# Note, it's possible not all found in methylations are also in SNP data.\n",
    "#idx_methylations <- match(genotype_IDs, colnames(methylations))\n",
    "methylations <- methylations[which(rownames(methylations) %in% genotype_IDs), ]\n",
    "print(\"Do methylations and genotype_IDs match?\")\n",
    "print(all(rownames(methylations) == genotype_IDs))\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "63287f88-bcb1-4a28-9794-315b33d1ed54",
   "metadata": {},
   "source": [
    "## Extract methylation, SNPs for one test"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cc5353b2-fa74-47bd-aff8-7801f23a53af",
   "metadata": {},
   "outputs": [],
   "source": [
    "wind <- c(1000, 2000, 5000, 10000, 20000, 50000, 100000, 200000, 500000)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "56a4d42a-3352-4cb1-b3a3-9eece0f5b0c8",
   "metadata": {},
   "outputs": [],
   "source": [
    "i <- 2000000\n",
    "\n",
    "\n",
    "w <- wind[9]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bcd8d2e9-702c-4af0-a53f-2acd3bde7c34",
   "metadata": {},
   "outputs": [],
   "source": [
    "methylation <- methylations[, i]  # Extracting methylation for current site\n",
    "\n",
    "# Get base position for the current methylation site\n",
    "\n",
    "meth_site_pos <- start(ranges(granges(BSobj2)))[i]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9837bbcd-5a9a-4731-b49b-9ecf034821f7",
   "metadata": {},
   "outputs": [],
   "source": [
    "cat(\"Processing window size: \", w, \"\\n\")\n",
    "\n",
    "# Calculate window bounds\n",
    "lower_bound <- meth_site_pos - w - 1\n",
    "upper_bound <- meth_site_pos + w - 1\n",
    "  \n",
    "# Find SNP indices within the window\n",
    "snp_indices <- which(pvar.2$POS >= lower_bound & pvar.2$POS <= upper_bound)\n",
    "\n",
    "if (length(snp_indices) == 0) {\n",
    "  next\n",
    "}\n",
    "\n",
    "# Get SNPs for desired indices\n",
    "SNPs <- pgenlibr::ReadList(pgen, variant_subset = snp_indices)\n",
    "\n",
    "rownames(SNPs) <- psam$`#IID`\n",
    "\n",
    "SNPs <- reorder_and_filter_geno(geno = SNPs, genotype_IDs)\n",
    "\n",
    "#cv.pred(geno, methylation, fold=5)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4d431c9f-e243-418a-ae27-e2f50e31ab9a",
   "metadata": {},
   "source": [
    "## Load original version of `glmnet.tune.alpha`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d3834107-5ed6-4e39-ada7-a11683f1b382",
   "metadata": {},
   "outputs": [],
   "source": [
    "glmnet.tune.alpha <- function(X,y,fold=5){\n",
    "        if(nrow(X)!=length(y)){\n",
    "                stop(\"Number of samples is differerent\")\n",
    "        } \n",
    "\n",
    "        print(paste0(\"Inside `glmnet.tune.alpha`, X and y have \", length(y), \" samples\"))\n",
    "\n",
    "        # remove missing data\n",
    "        idx <- !is.na(y)\n",
    "        y <- y[idx]\n",
    "        X <- X[idx,]\n",
    "\n",
    "        # tune alpha  \n",
    "        a <- seq(0, 1, 0.1)\n",
    "        set.seed(2023)\n",
    "        fold.id <- cut(sample(1:length(y)),breaks=fold,labels=FALSE)\n",
    "\n",
    "        print(\"Folds have this many samples: \")\n",
    "        print(table(fold.id))\n",
    "    \n",
    "    tune <- foreach(ai = a, .combine = rbind) %dopar% {\n",
    "        cv <- cv.glmnet(\n",
    "                        X,\n",
    "                        y,\n",
    "                        foldid = fold.id,\n",
    "                        type.measure = \"mse\",\n",
    "                        paralle = TRUE,\n",
    "                        alpha = ai\n",
    "                        )\n",
    "        data.frame(\n",
    "                        cvm = cv$cvm[cv$lambda == cv$lambda.1se],\n",
    "                        lambda = cv$lambda.1se,\n",
    "                        alpha = ai\n",
    "                        )\n",
    "        } \n",
    "    cv.opt <- tune[which.min(tune$cvm),] \n",
    "\n",
    "        # fit final model\n",
    "        fit = glmnet(\n",
    "        X,\n",
    "        y,\n",
    "        lambda = cv.opt$lambda,\n",
    "        alpha = cv.opt$alpha\n",
    "                )\n",
    "\n",
    "        # extract feature names and effects\n",
    "        coef <- coef(fit)\n",
    "        coef.names <- rownames(coef)[-1] # drop intercept\n",
    "        coef.values <- coef[-1]\n",
    "\n",
    "        # keep features with non-zero effects\n",
    "        coef.names <- coef.names[coef.values!=0]\n",
    "        coef.values <- coef.values[coef.values!=0]\n",
    "        fs <- data.frame(v=coef.names,coefs=coef.values)\n",
    "\n",
    "        # train sample prediction\n",
    "        predY <- predict(fit, X)\n",
    "        if(var(predY) > 0){\n",
    "                r <- cor(y,predY)\n",
    "        } else{\n",
    "                r <- NA\n",
    "        }\n",
    "        return(list(model=fit,features=fs,para=cv.opt,cor=r))\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f86e115a-dcdb-4b09-9975-ef5fa7561121",
   "metadata": {},
   "source": [
    "## Go inside `cv.pred`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "51297801-d7e4-4564-968b-420b8adc9959",
   "metadata": {},
   "outputs": [],
   "source": [
    "X <- SNPs\n",
    "y <- methylation\n",
    "fold <- 5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f6b5b077-06d7-467d-84a6-f42883f8b470",
   "metadata": {},
   "outputs": [],
   "source": [
    "set.seed(2018)\n",
    "if(nrow(X)!=length(y)){\n",
    "        stop(\"Number of observations is differerent\")\n",
    "}\n",
    "\n",
    "# impute missing values\n",
    "if(sum(is.na(X))>0){\n",
    "        m <- colMeans(X,na.rm=TRUE)\n",
    "        id <- apply(X,2,anyNA)\n",
    "        id <- which(id==TRUE)\n",
    "        for(j in 1:length(id)){\n",
    "                X[is.na(X[,id[j]]),id[j]] <- m[id[j]]\n",
    "        }\n",
    "}\n",
    "\n",
    "# segment dataset into n folds\n",
    "N <- nrow(X)\n",
    "idx <- sample(1:N)\n",
    "X<-X[idx,]\n",
    "y<-y[idx]\n",
    "fold.id <- cut(1:N,breaks=fold,labels=FALSE)\n",
    "\n",
    "# cross validation\n",
    "cv <- matrix(NA,nrow=fold,ncol=2)\n",
    "for(f in 1:fold){\n",
    "        paste0(\"Fold: \", fold)\n",
    "    \n",
    "        testIndexes <- which(fold.id==f,arr.ind=TRUE)\n",
    "        trainx <- X[-testIndexes,]\n",
    "        trainy <- y[-testIndexes]\n",
    "        testx <- X[testIndexes,]\n",
    "        testy <- y[testIndexes]\n",
    "        fit <- glmnet.tune.alpha(trainx,trainy,fold=5)\n",
    "\n",
    "        print(fit$para)\n",
    "    \n",
    "        pred <- predict(fit$model, testx)\n",
    "        cv[f,1] <- cor(pred,testy)\n",
    "        cv[f,2] <- mean((pred-testy)^2)\n",
    "\n",
    "        cat(\"\\n\\n\")\n",
    "}\n",
    "cvm <- apply(cv,2,mean)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e7757a5e-d69f-43c3-bdb8-cce1b9219246",
   "metadata": {},
   "source": [
    "## Make a small change to `cv.pred`"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6e874d01-36db-47bd-b638-7228bde13005",
   "metadata": {},
   "source": [
    "Call `glmnet.tune.alpha` outside of loop, use it on full dataset, then extract optimal parameters, then run evaluation loop."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f3b1ac1d-da97-446a-90b4-3804d59c15cd",
   "metadata": {},
   "outputs": [],
   "source": [
    "set.seed(2018)\n",
    "if(nrow(X)!=length(y)){\n",
    "        stop(\"Number of observations is differerent\")\n",
    "}\n",
    "\n",
    "# impute missing values\n",
    "if(sum(is.na(X))>0){\n",
    "        m <- colMeans(X,na.rm=TRUE)\n",
    "        id <- apply(X,2,anyNA)\n",
    "        id <- which(id==TRUE)\n",
    "        for(j in 1:length(id)){\n",
    "                X[is.na(X[,id[j]]),id[j]] <- m[id[j]]\n",
    "        }\n",
    "}\n",
    "\n",
    "# segment dataset into n folds\n",
    "N <- nrow(X)\n",
    "idx <- sample(1:N)\n",
    "X<-X[idx,]\n",
    "y<-y[idx]\n",
    "fold.id <- cut(1:N,breaks=fold,labels=FALSE)\n",
    "\n",
    "### Moved this line and changed trainx and trainy to X and Y\n",
    "fit <- glmnet.tune.alpha(X,y,fold=5)\n",
    "best_lambda <- fit$para$lambda\n",
    "best_alpha <- fit$para$alpha\n",
    "\n",
    "# cross validation\n",
    "cv <- matrix(NA,nrow=fold,ncol=2)\n",
    "for(f in 1:fold){\n",
    "        paste0(\"Fold: \", fold)\n",
    "    \n",
    "        testIndexes <- which(fold.id==f,arr.ind=TRUE)\n",
    "        trainx <- X[-testIndexes,]\n",
    "        trainy <- y[-testIndexes]\n",
    "        testx <- X[testIndexes,]\n",
    "        testy <- y[testIndexes]\n",
    "\n",
    "        fit <- glmnet.tune.alpha(trainx,trainy,fold=5)\n",
    "    \n",
    "        print(fit$para)\n",
    "    \n",
    "        pred <- predict(fit$model, testx)\n",
    "        cv[f,1] <- cor(pred,testy)\n",
    "        cv[f,2] <- mean((pred-testy)^2)\n",
    "\n",
    "        cat(\"\\n\\n\")\n",
    "}\n",
    "cvm <- apply(cv,2,mean)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3ce7c951-1ad8-46be-a722-058b9216c568",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.3.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
