{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "f02d8dcb-ce43-410f-a9f5-cd2fdb8d5f8a",
   "metadata": {},
   "source": [
    "# Match up SNP, methylation and covariate files to be analyzed together"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "890c92dc-0bf6-4a7c-978c-2237c91bcdb5",
   "metadata": {},
   "source": [
    "As part of this, we will determine which methylation sites have coverage in the SNP data and should be analyzed."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "11ab38a4-cbb6-4843-8d2a-23761d2af9d3",
   "metadata": {},
   "source": [
    "## Match SNP and methylation files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "a43638a7-5919-4575-a93b-1892081d386a",
   "metadata": {},
   "outputs": [],
   "source": [
    "library(stringr)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "bf6b0783-a2ac-44cf-a9ce-1e34ad46093a",
   "metadata": {},
   "outputs": [],
   "source": [
    "library(data.table)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "02110f3e-3256-4fa6-8433-ea4874857cc6",
   "metadata": {},
   "outputs": [],
   "source": [
    "SNP_files <- list.files(\"/expanse/lustre/projects/jhu152/naglemi/mwas/gwas/\",\n",
    "                        #\"/dcs04/lieber/statsgen/shizhong/michael/mwas/gwas\",\n",
    "                        pattern = \"pgen\", full.names = TRUE)\n",
    "SNP_files <- SNP_files[ordered(SNP_files)]\n",
    "\n",
    "SNP_files_libd <- data.frame(SNPs_libd = SNP_files[grepl(\"libd\", SNP_files)])\n",
    "SNP_files_ref_EUR <- data.frame(SNPs_ref_EUR = SNP_files[grepl(\"ref_EUR\", SNP_files)])\n",
    "\n",
    "SNP_files_libd$Chr <- as.numeric(stringr::str_extract(SNP_files_libd$SNPs_libd, \"(?<=chr)\\\\d+\"))\n",
    "SNP_files_ref_EUR$Chr <- as.numeric(stringr::str_extract(SNP_files_ref_EUR$SNPs_ref_EUR, \"(?<=chr)\\\\d+\"))\n",
    "\n",
    "SNP_files <- merge(SNP_files_libd, SNP_files_ref_EUR)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "6a111d49-fe5f-4cca-a6aa-eaf87f18d66f",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "meth_files <- list.files(\"/expanse/lustre/projects/jhu152/naglemi/mwas/pheno\", recursive = TRUE, pattern = \"rda\", full.names = TRUE)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "48d044b2-2146-4221-b670-56351ddc26c0",
   "metadata": {},
   "source": [
    "We will bring in original dlpfc and hippo methylation files, and the caudate files that were reprocessed to store everything in given files (where previously those ones had h5 pointers)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "597fc11a-9af9-45e5-85bd-332ce7c21fb1",
   "metadata": {},
   "outputs": [],
   "source": [
    "#meth_caud <- data.frame(meth_caud = meth_files[grepl(\"caud\", meth_files)])\n",
    "meth_dlpfc <- data.frame(meth_dlpfc = meth_files[grepl(\"dlpfc\", meth_files)])\n",
    "meth_hippo <- data.frame(meth_hippo = meth_files[grepl(\"hippo\", meth_files)])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "af6d76ed-a58f-48e6-b528-6eec20184763",
   "metadata": {},
   "outputs": [],
   "source": [
    "meth_caud_files <- list.files(\"/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud\",\n",
    "                              recursive = TRUE, pattern = \"rda\", full.names = TRUE)\n",
    "meth_caud_files <- meth_caud_files[!grepl(\"pointers\", meth_caud_files)]\n",
    "meth_caud <- data.frame(meth_caud = meth_caud_files[grepl(\"caud\", meth_caud_files)])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "17f7df0e-753a-44a7-9eb4-a128d2daf977",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>66</li><li>1</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 66\n",
       "\\item 1\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 66\n",
       "2. 1\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 66  1"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dim(meth_caud)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "91ef29dd-8f93-4cdb-a023-aba560aee00f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 1</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>meth_dlpfc</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr1_AA.rda  </td></tr>\n",
       "\t<tr><th scope=row>2</th><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr1_all.rda </td></tr>\n",
       "\t<tr><th scope=row>3</th><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr1_EA.rda  </td></tr>\n",
       "\t<tr><th scope=row>4</th><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr10_AA.rda </td></tr>\n",
       "\t<tr><th scope=row>5</th><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr10_all.rda</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr10_EA.rda </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 1\n",
       "\\begin{tabular}{r|l}\n",
       "  & meth\\_dlpfc\\\\\n",
       "  & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr1\\_AA.rda  \\\\\n",
       "\t2 & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr1\\_all.rda \\\\\n",
       "\t3 & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr1\\_EA.rda  \\\\\n",
       "\t4 & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr10\\_AA.rda \\\\\n",
       "\t5 & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr10\\_all.rda\\\\\n",
       "\t6 & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr10\\_EA.rda \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 1\n",
       "\n",
       "| <!--/--> | meth_dlpfc &lt;chr&gt; |\n",
       "|---|---|\n",
       "| 1 | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr1_AA.rda   |\n",
       "| 2 | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr1_all.rda  |\n",
       "| 3 | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr1_EA.rda   |\n",
       "| 4 | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr10_AA.rda  |\n",
       "| 5 | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr10_all.rda |\n",
       "| 6 | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr10_EA.rda  |\n",
       "\n"
      ],
      "text/plain": [
       "  meth_dlpfc                                                                \n",
       "1 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr1_AA.rda  \n",
       "2 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr1_all.rda \n",
       "3 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr1_EA.rda  \n",
       "4 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr10_AA.rda \n",
       "5 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr10_all.rda\n",
       "6 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr10_EA.rda "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(meth_dlpfc)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "5d964444-1d4d-4ada-a8aa-2ffc8ce688e1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 1</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>meth_caud</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr1_AA.rda  </td></tr>\n",
       "\t<tr><th scope=row>2</th><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr1_all.rda </td></tr>\n",
       "\t<tr><th scope=row>3</th><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr1_EA.rda  </td></tr>\n",
       "\t<tr><th scope=row>4</th><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr10_AA.rda </td></tr>\n",
       "\t<tr><th scope=row>5</th><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr10_all.rda</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr10_EA.rda </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 1\n",
       "\\begin{tabular}{r|l}\n",
       "  & meth\\_caud\\\\\n",
       "  & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr1\\_AA.rda  \\\\\n",
       "\t2 & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr1\\_all.rda \\\\\n",
       "\t3 & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr1\\_EA.rda  \\\\\n",
       "\t4 & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr10\\_AA.rda \\\\\n",
       "\t5 & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr10\\_all.rda\\\\\n",
       "\t6 & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr10\\_EA.rda \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 1\n",
       "\n",
       "| <!--/--> | meth_caud &lt;chr&gt; |\n",
       "|---|---|\n",
       "| 1 | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr1_AA.rda   |\n",
       "| 2 | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr1_all.rda  |\n",
       "| 3 | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr1_EA.rda   |\n",
       "| 4 | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr10_AA.rda  |\n",
       "| 5 | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr10_all.rda |\n",
       "| 6 | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr10_EA.rda  |\n",
       "\n"
      ],
      "text/plain": [
       "  meth_caud                                                                \n",
       "1 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr1_AA.rda  \n",
       "2 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr1_all.rda \n",
       "3 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr1_EA.rda  \n",
       "4 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr10_AA.rda \n",
       "5 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr10_all.rda\n",
       "6 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr10_EA.rda "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(meth_caud)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "7ea3c702-d849-499c-b861-db7a57743e55",
   "metadata": {},
   "outputs": [],
   "source": [
    "meth_caud$Chr <- as.numeric(str_extract(meth_caud$meth_caud, \"(?<=chr)\\\\d+(?=_)\"))\n",
    "meth_dlpfc$Chr <- as.numeric(str_extract(meth_dlpfc$meth_dlpfc, \"(?<=chr)\\\\d+(?=_)\"))\n",
    "meth_hippo$Chr <- as.numeric(str_extract(meth_hippo$meth_hippo, \"(?<=chr)\\\\d+(?=_)\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "ff859ff9-1233-4931-86ef-d4f33c3feb19",
   "metadata": {},
   "outputs": [],
   "source": [
    "reshape_and_rename_corrected <- function(data, region) {\n",
    "  # Extracting the type (AA, EA, all) based on the filename pattern\n",
    "  data$Type <- gsub(paste0(\".*\", region, \"/out/chr\\\\d+_(AA|EA|all)\\\\.rda\"), \"\\\\1\", data[[1]])\n",
    "  \n",
    "  # Creating a wide dataframe with separate columns for AA, EA, and all\n",
    "  wide_data <- reshape(data, idvar = \"Chr\", timevar = \"Type\", direction = \"wide\")\n",
    "  \n",
    "  # Correcting column names to include only the region and type without redundant parts\n",
    "  colnames(wide_data)[-1] <- sapply(colnames(wide_data)[-1], function(x) {\n",
    "    paste(region, gsub(\".*\\\\.(AA|EA|all)$\", \"\\\\1\", x), sep = \"_\")\n",
    "  })\n",
    "  \n",
    "  return(wide_data)\n",
    "}\n",
    "\n",
    "# Applying the corrected function to each dataset\n",
    "meth_caud_wide <- reshape_and_rename_corrected(meth_caud, \"caud\")\n",
    "meth_dlpfc_wide <- reshape_and_rename_corrected(meth_dlpfc, \"dlpfc\")\n",
    "meth_hippo_wide <- reshape_and_rename_corrected(meth_hippo, \"hippo\")\n",
    "\n",
    "merged_data <- Reduce(function(x, y) merge(x, y, by = \"Chr\", all = TRUE), list(meth_caud_wide, meth_dlpfc_wide, meth_hippo_wide))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "d4eeee20-fa45-4c7f-af37-0d2a8edc1bc8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 10</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Chr</th><th scope=col>caud_AA</th><th scope=col>caud_all</th><th scope=col>caud_EA</th><th scope=col>dlpfc_AA</th><th scope=col>dlpfc_all</th><th scope=col>dlpfc_EA</th><th scope=col>hippo_AA</th><th scope=col>hippo_all</th><th scope=col>hippo_EA</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>1</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr1_AA.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr1_all.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr1_EA.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr1_AA.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr1_all.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr1_EA.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr1_AA.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr1_all.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr1_EA.rda</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>2</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr2_AA.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr2_all.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr2_EA.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr2_AA.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr2_all.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr2_EA.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr2_AA.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr2_all.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr2_EA.rda</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>3</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr3_AA.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr3_all.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr3_EA.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr3_AA.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr3_all.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr3_EA.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr3_AA.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr3_all.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr3_EA.rda</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>4</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr4_AA.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr4_all.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr4_EA.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr4_AA.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr4_all.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr4_EA.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr4_AA.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr4_all.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr4_EA.rda</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>5</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr5_AA.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr5_all.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr5_EA.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr5_AA.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr5_all.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr5_EA.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr5_AA.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr5_all.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr5_EA.rda</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>6</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr6_AA.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr6_all.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr6_EA.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr6_AA.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr6_all.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr6_EA.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr6_AA.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr6_all.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr6_EA.rda</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 10\n",
       "\\begin{tabular}{r|llllllllll}\n",
       "  & Chr & caud\\_AA & caud\\_all & caud\\_EA & dlpfc\\_AA & dlpfc\\_all & dlpfc\\_EA & hippo\\_AA & hippo\\_all & hippo\\_EA\\\\\n",
       "  & <dbl> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & 1 & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr1\\_AA.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr1\\_all.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr1\\_EA.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr1\\_AA.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr1\\_all.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr1\\_EA.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr1\\_AA.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr1\\_all.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr1\\_EA.rda\\\\\n",
       "\t2 & 2 & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr2\\_AA.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr2\\_all.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr2\\_EA.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr2\\_AA.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr2\\_all.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr2\\_EA.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr2\\_AA.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr2\\_all.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr2\\_EA.rda\\\\\n",
       "\t3 & 3 & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr3\\_AA.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr3\\_all.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr3\\_EA.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr3\\_AA.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr3\\_all.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr3\\_EA.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr3\\_AA.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr3\\_all.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr3\\_EA.rda\\\\\n",
       "\t4 & 4 & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr4\\_AA.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr4\\_all.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr4\\_EA.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr4\\_AA.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr4\\_all.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr4\\_EA.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr4\\_AA.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr4\\_all.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr4\\_EA.rda\\\\\n",
       "\t5 & 5 & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr5\\_AA.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr5\\_all.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr5\\_EA.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr5\\_AA.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr5\\_all.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr5\\_EA.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr5\\_AA.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr5\\_all.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr5\\_EA.rda\\\\\n",
       "\t6 & 6 & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr6\\_AA.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr6\\_all.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr6\\_EA.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr6\\_AA.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr6\\_all.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr6\\_EA.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr6\\_AA.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr6\\_all.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr6\\_EA.rda\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 10\n",
       "\n",
       "| <!--/--> | Chr &lt;dbl&gt; | caud_AA &lt;chr&gt; | caud_all &lt;chr&gt; | caud_EA &lt;chr&gt; | dlpfc_AA &lt;chr&gt; | dlpfc_all &lt;chr&gt; | dlpfc_EA &lt;chr&gt; | hippo_AA &lt;chr&gt; | hippo_all &lt;chr&gt; | hippo_EA &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | 1 | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr1_AA.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr1_all.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr1_EA.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr1_AA.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr1_all.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr1_EA.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr1_AA.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr1_all.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr1_EA.rda |\n",
       "| 2 | 2 | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr2_AA.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr2_all.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr2_EA.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr2_AA.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr2_all.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr2_EA.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr2_AA.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr2_all.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr2_EA.rda |\n",
       "| 3 | 3 | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr3_AA.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr3_all.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr3_EA.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr3_AA.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr3_all.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr3_EA.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr3_AA.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr3_all.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr3_EA.rda |\n",
       "| 4 | 4 | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr4_AA.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr4_all.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr4_EA.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr4_AA.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr4_all.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr4_EA.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr4_AA.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr4_all.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr4_EA.rda |\n",
       "| 5 | 5 | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr5_AA.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr5_all.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr5_EA.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr5_AA.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr5_all.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr5_EA.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr5_AA.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr5_all.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr5_EA.rda |\n",
       "| 6 | 6 | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr6_AA.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr6_all.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr6_EA.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr6_AA.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr6_all.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr6_EA.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr6_AA.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr6_all.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr6_EA.rda |\n",
       "\n"
      ],
      "text/plain": [
       "  Chr caud_AA                                                                \n",
       "1 1   /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr1_AA.rda\n",
       "2 2   /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr2_AA.rda\n",
       "3 3   /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr3_AA.rda\n",
       "4 4   /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr4_AA.rda\n",
       "5 5   /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr5_AA.rda\n",
       "6 6   /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr6_AA.rda\n",
       "  caud_all                                                                \n",
       "1 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr1_all.rda\n",
       "2 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr2_all.rda\n",
       "3 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr3_all.rda\n",
       "4 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr4_all.rda\n",
       "5 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr5_all.rda\n",
       "6 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr6_all.rda\n",
       "  caud_EA                                                                \n",
       "1 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr1_EA.rda\n",
       "2 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr2_EA.rda\n",
       "3 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr3_EA.rda\n",
       "4 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr4_EA.rda\n",
       "5 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr5_EA.rda\n",
       "6 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr6_EA.rda\n",
       "  dlpfc_AA                                                                \n",
       "1 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr1_AA.rda\n",
       "2 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr2_AA.rda\n",
       "3 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr3_AA.rda\n",
       "4 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr4_AA.rda\n",
       "5 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr5_AA.rda\n",
       "6 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr6_AA.rda\n",
       "  dlpfc_all                                                                \n",
       "1 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr1_all.rda\n",
       "2 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr2_all.rda\n",
       "3 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr3_all.rda\n",
       "4 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr4_all.rda\n",
       "5 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr5_all.rda\n",
       "6 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr6_all.rda\n",
       "  dlpfc_EA                                                                \n",
       "1 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr1_EA.rda\n",
       "2 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr2_EA.rda\n",
       "3 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr3_EA.rda\n",
       "4 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr4_EA.rda\n",
       "5 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr5_EA.rda\n",
       "6 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr6_EA.rda\n",
       "  hippo_AA                                                                \n",
       "1 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr1_AA.rda\n",
       "2 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr2_AA.rda\n",
       "3 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr3_AA.rda\n",
       "4 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr4_AA.rda\n",
       "5 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr5_AA.rda\n",
       "6 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr6_AA.rda\n",
       "  hippo_all                                                                \n",
       "1 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr1_all.rda\n",
       "2 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr2_all.rda\n",
       "3 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr3_all.rda\n",
       "4 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr4_all.rda\n",
       "5 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr5_all.rda\n",
       "6 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr6_all.rda\n",
       "  hippo_EA                                                                \n",
       "1 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr1_EA.rda\n",
       "2 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr2_EA.rda\n",
       "3 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr3_EA.rda\n",
       "4 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr4_EA.rda\n",
       "5 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr5_EA.rda\n",
       "6 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr6_EA.rda"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(merged_data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "628695cb-e1e9-4682-817a-a4eb74891085",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Chr</th><th scope=col>SNPs_libd</th><th scope=col>SNPs_ref_EUR</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>1</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr1.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//ref_EUR_chr1.pgen</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>2</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr2.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//ref_EUR_chr2.pgen</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>3</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr3.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//ref_EUR_chr3.pgen</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>4</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr4.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//ref_EUR_chr4.pgen</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>5</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr5.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//ref_EUR_chr5.pgen</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>6</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr6.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//ref_EUR_chr6.pgen</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 3\n",
       "\\begin{tabular}{r|lll}\n",
       "  & Chr & SNPs\\_libd & SNPs\\_ref\\_EUR\\\\\n",
       "  & <dbl> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & 1 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr1.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//ref\\_EUR\\_chr1.pgen\\\\\n",
       "\t2 & 2 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr2.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//ref\\_EUR\\_chr2.pgen\\\\\n",
       "\t3 & 3 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr3.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//ref\\_EUR\\_chr3.pgen\\\\\n",
       "\t4 & 4 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr4.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//ref\\_EUR\\_chr4.pgen\\\\\n",
       "\t5 & 5 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr5.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//ref\\_EUR\\_chr5.pgen\\\\\n",
       "\t6 & 6 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr6.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//ref\\_EUR\\_chr6.pgen\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 3\n",
       "\n",
       "| <!--/--> | Chr &lt;dbl&gt; | SNPs_libd &lt;chr&gt; | SNPs_ref_EUR &lt;chr&gt; |\n",
       "|---|---|---|---|\n",
       "| 1 | 1 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr1.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//ref_EUR_chr1.pgen |\n",
       "| 2 | 2 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr2.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//ref_EUR_chr2.pgen |\n",
       "| 3 | 3 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr3.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//ref_EUR_chr3.pgen |\n",
       "| 4 | 4 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr4.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//ref_EUR_chr4.pgen |\n",
       "| 5 | 5 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr5.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//ref_EUR_chr5.pgen |\n",
       "| 6 | 6 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr6.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//ref_EUR_chr6.pgen |\n",
       "\n"
      ],
      "text/plain": [
       "  Chr SNPs_libd                                                        \n",
       "1 1   /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr1.pgen\n",
       "2 2   /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr2.pgen\n",
       "3 3   /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr3.pgen\n",
       "4 4   /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr4.pgen\n",
       "5 5   /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr5.pgen\n",
       "6 6   /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr6.pgen\n",
       "  SNPs_ref_EUR                                                        \n",
       "1 /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//ref_EUR_chr1.pgen\n",
       "2 /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//ref_EUR_chr2.pgen\n",
       "3 /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//ref_EUR_chr3.pgen\n",
       "4 /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//ref_EUR_chr4.pgen\n",
       "5 /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//ref_EUR_chr5.pgen\n",
       "6 /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//ref_EUR_chr6.pgen"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(SNP_files)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "81dda542-2bdd-47b9-b8e7-b68693713941",
   "metadata": {},
   "outputs": [],
   "source": [
    "merged <- merge(SNP_files, merged_data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "f3f7cfe2-8f6d-4614-b8e8-522e4bc00527",
   "metadata": {},
   "outputs": [],
   "source": [
    "data.table::fwrite(merged, \"09-OUT_matched_SNP_meth_files.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7adcd524-536f-444c-8ed2-92bdce8ca0c6",
   "metadata": {},
   "source": [
    "## Determine first, last position for each SNP, methylation file pair"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "565e2e57-710b-4b00-99b7-4ec322ed3098",
   "metadata": {},
   "outputs": [],
   "source": [
    "merged$SNPs_ref_EUR <- NULL"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "68518ba4-c5b5-4532-a9d6-41255e0858ff",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 11</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Chr</th><th scope=col>SNPs_libd</th><th scope=col>caud_AA</th><th scope=col>caud_all</th><th scope=col>caud_EA</th><th scope=col>dlpfc_AA</th><th scope=col>dlpfc_all</th><th scope=col>dlpfc_EA</th><th scope=col>hippo_AA</th><th scope=col>hippo_all</th><th scope=col>hippo_EA</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>1</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr1.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr1_AA.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr1_all.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr1_EA.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr1_AA.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr1_all.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr1_EA.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr1_AA.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr1_all.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr1_EA.rda</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>2</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr2.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr2_AA.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr2_all.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr2_EA.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr2_AA.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr2_all.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr2_EA.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr2_AA.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr2_all.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr2_EA.rda</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>3</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr3.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr3_AA.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr3_all.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr3_EA.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr3_AA.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr3_all.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr3_EA.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr3_AA.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr3_all.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr3_EA.rda</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>4</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr4.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr4_AA.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr4_all.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr4_EA.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr4_AA.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr4_all.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr4_EA.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr4_AA.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr4_all.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr4_EA.rda</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>5</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr5.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr5_AA.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr5_all.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr5_EA.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr5_AA.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr5_all.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr5_EA.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr5_AA.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr5_all.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr5_EA.rda</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>6</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr6.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr6_AA.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr6_all.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr6_EA.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr6_AA.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr6_all.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr6_EA.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr6_AA.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr6_all.rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr6_EA.rda</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 11\n",
       "\\begin{tabular}{r|lllllllllll}\n",
       "  & Chr & SNPs\\_libd & caud\\_AA & caud\\_all & caud\\_EA & dlpfc\\_AA & dlpfc\\_all & dlpfc\\_EA & hippo\\_AA & hippo\\_all & hippo\\_EA\\\\\n",
       "  & <dbl> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & 1 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr1.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr1\\_AA.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr1\\_all.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr1\\_EA.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr1\\_AA.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr1\\_all.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr1\\_EA.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr1\\_AA.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr1\\_all.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr1\\_EA.rda\\\\\n",
       "\t2 & 2 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr2.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr2\\_AA.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr2\\_all.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr2\\_EA.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr2\\_AA.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr2\\_all.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr2\\_EA.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr2\\_AA.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr2\\_all.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr2\\_EA.rda\\\\\n",
       "\t3 & 3 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr3.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr3\\_AA.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr3\\_all.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr3\\_EA.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr3\\_AA.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr3\\_all.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr3\\_EA.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr3\\_AA.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr3\\_all.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr3\\_EA.rda\\\\\n",
       "\t4 & 4 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr4.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr4\\_AA.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr4\\_all.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr4\\_EA.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr4\\_AA.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr4\\_all.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr4\\_EA.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr4\\_AA.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr4\\_all.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr4\\_EA.rda\\\\\n",
       "\t5 & 5 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr5.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr5\\_AA.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr5\\_all.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr5\\_EA.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr5\\_AA.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr5\\_all.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr5\\_EA.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr5\\_AA.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr5\\_all.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr5\\_EA.rda\\\\\n",
       "\t6 & 6 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr6.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr6\\_AA.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr6\\_all.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr6\\_EA.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr6\\_AA.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr6\\_all.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr6\\_EA.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr6\\_AA.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr6\\_all.rda & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr6\\_EA.rda\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 11\n",
       "\n",
       "| <!--/--> | Chr &lt;dbl&gt; | SNPs_libd &lt;chr&gt; | caud_AA &lt;chr&gt; | caud_all &lt;chr&gt; | caud_EA &lt;chr&gt; | dlpfc_AA &lt;chr&gt; | dlpfc_all &lt;chr&gt; | dlpfc_EA &lt;chr&gt; | hippo_AA &lt;chr&gt; | hippo_all &lt;chr&gt; | hippo_EA &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | 1 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr1.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr1_AA.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr1_all.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr1_EA.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr1_AA.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr1_all.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr1_EA.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr1_AA.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr1_all.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr1_EA.rda |\n",
       "| 2 | 2 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr2.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr2_AA.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr2_all.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr2_EA.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr2_AA.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr2_all.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr2_EA.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr2_AA.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr2_all.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr2_EA.rda |\n",
       "| 3 | 3 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr3.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr3_AA.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr3_all.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr3_EA.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr3_AA.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr3_all.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr3_EA.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr3_AA.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr3_all.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr3_EA.rda |\n",
       "| 4 | 4 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr4.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr4_AA.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr4_all.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr4_EA.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr4_AA.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr4_all.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr4_EA.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr4_AA.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr4_all.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr4_EA.rda |\n",
       "| 5 | 5 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr5.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr5_AA.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr5_all.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr5_EA.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr5_AA.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr5_all.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr5_EA.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr5_AA.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr5_all.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr5_EA.rda |\n",
       "| 6 | 6 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr6.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr6_AA.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr6_all.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr6_EA.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr6_AA.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr6_all.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr6_EA.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr6_AA.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr6_all.rda | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr6_EA.rda |\n",
       "\n"
      ],
      "text/plain": [
       "  Chr SNPs_libd                                                        \n",
       "1 1   /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr1.pgen\n",
       "2 2   /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr2.pgen\n",
       "3 3   /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr3.pgen\n",
       "4 4   /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr4.pgen\n",
       "5 5   /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr5.pgen\n",
       "6 6   /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr6.pgen\n",
       "  caud_AA                                                                \n",
       "1 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr1_AA.rda\n",
       "2 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr2_AA.rda\n",
       "3 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr3_AA.rda\n",
       "4 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr4_AA.rda\n",
       "5 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr5_AA.rda\n",
       "6 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr6_AA.rda\n",
       "  caud_all                                                                \n",
       "1 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr1_all.rda\n",
       "2 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr2_all.rda\n",
       "3 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr3_all.rda\n",
       "4 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr4_all.rda\n",
       "5 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr5_all.rda\n",
       "6 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr6_all.rda\n",
       "  caud_EA                                                                \n",
       "1 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr1_EA.rda\n",
       "2 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr2_EA.rda\n",
       "3 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr3_EA.rda\n",
       "4 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr4_EA.rda\n",
       "5 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr5_EA.rda\n",
       "6 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr6_EA.rda\n",
       "  dlpfc_AA                                                                \n",
       "1 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr1_AA.rda\n",
       "2 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr2_AA.rda\n",
       "3 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr3_AA.rda\n",
       "4 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr4_AA.rda\n",
       "5 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr5_AA.rda\n",
       "6 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr6_AA.rda\n",
       "  dlpfc_all                                                                \n",
       "1 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr1_all.rda\n",
       "2 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr2_all.rda\n",
       "3 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr3_all.rda\n",
       "4 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr4_all.rda\n",
       "5 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr5_all.rda\n",
       "6 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr6_all.rda\n",
       "  dlpfc_EA                                                                \n",
       "1 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr1_EA.rda\n",
       "2 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr2_EA.rda\n",
       "3 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr3_EA.rda\n",
       "4 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr4_EA.rda\n",
       "5 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr5_EA.rda\n",
       "6 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/dlpfc/out/chr6_EA.rda\n",
       "  hippo_AA                                                                \n",
       "1 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr1_AA.rda\n",
       "2 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr2_AA.rda\n",
       "3 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr3_AA.rda\n",
       "4 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr4_AA.rda\n",
       "5 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr5_AA.rda\n",
       "6 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr6_AA.rda\n",
       "  hippo_all                                                                \n",
       "1 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr1_all.rda\n",
       "2 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr2_all.rda\n",
       "3 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr3_all.rda\n",
       "4 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr4_all.rda\n",
       "5 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr5_all.rda\n",
       "6 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr6_all.rda\n",
       "  hippo_EA                                                                \n",
       "1 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr1_EA.rda\n",
       "2 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr2_EA.rda\n",
       "3 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr3_EA.rda\n",
       "4 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr4_EA.rda\n",
       "5 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr5_EA.rda\n",
       "6 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/hippo/out/chr6_EA.rda"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(merged)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "82e0be2d-c9df-42cf-868a-5eab22408dfa",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 6 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Chr</th><th scope=col>type</th><th scope=col>SNP_data</th><th scope=col>methylation_data</th></tr>\n",
       "\t<tr><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>1</td><td>rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr1.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr1_AA.rda</td></tr>\n",
       "\t<tr><td>2</td><td>rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr2.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr2_AA.rda</td></tr>\n",
       "\t<tr><td>3</td><td>rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr3.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr3_AA.rda</td></tr>\n",
       "\t<tr><td>4</td><td>rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr4.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr4_AA.rda</td></tr>\n",
       "\t<tr><td>5</td><td>rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr5.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr5_AA.rda</td></tr>\n",
       "\t<tr><td>6</td><td>rda</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr6.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr6_AA.rda</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 6 × 4\n",
       "\\begin{tabular}{llll}\n",
       " Chr & type & SNP\\_data & methylation\\_data\\\\\n",
       " <dbl> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t 1 & rda & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr1.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr1\\_AA.rda\\\\\n",
       "\t 2 & rda & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr2.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr2\\_AA.rda\\\\\n",
       "\t 3 & rda & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr3.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr3\\_AA.rda\\\\\n",
       "\t 4 & rda & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr4.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr4\\_AA.rda\\\\\n",
       "\t 5 & rda & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr5.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr5\\_AA.rda\\\\\n",
       "\t 6 & rda & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr6.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr6\\_AA.rda\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 6 × 4\n",
       "\n",
       "| Chr &lt;dbl&gt; | type &lt;chr&gt; | SNP_data &lt;chr&gt; | methylation_data &lt;chr&gt; |\n",
       "|---|---|---|---|\n",
       "| 1 | rda | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr1.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr1_AA.rda |\n",
       "| 2 | rda | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr2.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr2_AA.rda |\n",
       "| 3 | rda | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr3.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr3_AA.rda |\n",
       "| 4 | rda | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr4.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr4_AA.rda |\n",
       "| 5 | rda | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr5.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr5_AA.rda |\n",
       "| 6 | rda | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr6.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr6_AA.rda |\n",
       "\n"
      ],
      "text/plain": [
       "  Chr type SNP_data                                                         \n",
       "1 1   rda  /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr1.pgen\n",
       "2 2   rda  /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr2.pgen\n",
       "3 3   rda  /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr3.pgen\n",
       "4 4   rda  /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr4.pgen\n",
       "5 5   rda  /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr5.pgen\n",
       "6 6   rda  /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr6.pgen\n",
       "  methylation_data                                                       \n",
       "1 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr1_AA.rda\n",
       "2 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr2_AA.rda\n",
       "3 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr3_AA.rda\n",
       "4 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr4_AA.rda\n",
       "5 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr5_AA.rda\n",
       "6 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr6_AA.rda"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Load data.table\n",
    "library(data.table)\n",
    "\n",
    "# Assuming 'merged' is your data.table with the wide format\n",
    "# Set it as a data.table if it's not already\n",
    "setDT(merged)\n",
    "\n",
    "# Melt the data.table from wide to long format\n",
    "# id.vars = \"Chr\" to keep the chromosome column as identifier\n",
    "# measure.vars patterns to match SNP data and methylation data columns\n",
    "long_format <- melt(\n",
    "  merged,\n",
    "  id.vars = \"Chr\",\n",
    "  measure.vars = patterns(\"^SNPs_libd\", \"caud_.*|dlpfc_.*|hippo_.*\"),\n",
    "  value.name = c(\"SNP_data\", \"methylation_data\"),\n",
    "  variable.name = \"type\"\n",
    ")\n",
    "\n",
    "# Adjust 'type' to distinguish between SNP and methylation data if needed\n",
    "long_format[, type := ifelse(grepl(\"SNPs_libd\", type), \"pgen\", \"rda\")]\n",
    "\n",
    "# View the head of the long format table\n",
    "head(long_format)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "918c4c3a-417b-4005-a8ad-6273adc9957f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 6 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Chr</th><th scope=col>SNP_data</th><th scope=col>methylation_data</th></tr>\n",
       "\t<tr><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>1</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr1.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr1_AA.rda</td></tr>\n",
       "\t<tr><td>2</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr2.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr2_AA.rda</td></tr>\n",
       "\t<tr><td>3</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr3.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr3_AA.rda</td></tr>\n",
       "\t<tr><td>4</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr4.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr4_AA.rda</td></tr>\n",
       "\t<tr><td>5</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr5.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr5_AA.rda</td></tr>\n",
       "\t<tr><td>6</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr6.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr6_AA.rda</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 6 × 3\n",
       "\\begin{tabular}{lll}\n",
       " Chr & SNP\\_data & methylation\\_data\\\\\n",
       " <dbl> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t 1 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr1.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr1\\_AA.rda\\\\\n",
       "\t 2 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr2.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr2\\_AA.rda\\\\\n",
       "\t 3 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr3.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr3\\_AA.rda\\\\\n",
       "\t 4 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr4.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr4\\_AA.rda\\\\\n",
       "\t 5 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr5.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr5\\_AA.rda\\\\\n",
       "\t 6 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr6.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr6\\_AA.rda\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 6 × 3\n",
       "\n",
       "| Chr &lt;dbl&gt; | SNP_data &lt;chr&gt; | methylation_data &lt;chr&gt; |\n",
       "|---|---|---|\n",
       "| 1 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr1.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr1_AA.rda |\n",
       "| 2 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr2.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr2_AA.rda |\n",
       "| 3 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr3.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr3_AA.rda |\n",
       "| 4 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr4.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr4_AA.rda |\n",
       "| 5 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr5.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr5_AA.rda |\n",
       "| 6 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr6.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr6_AA.rda |\n",
       "\n"
      ],
      "text/plain": [
       "  Chr SNP_data                                                         \n",
       "1 1   /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr1.pgen\n",
       "2 2   /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr2.pgen\n",
       "3 3   /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr3.pgen\n",
       "4 4   /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr4.pgen\n",
       "5 5   /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr5.pgen\n",
       "6 6   /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr6.pgen\n",
       "  methylation_data                                                       \n",
       "1 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr1_AA.rda\n",
       "2 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr2_AA.rda\n",
       "3 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr3_AA.rda\n",
       "4 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr4_AA.rda\n",
       "5 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr5_AA.rda\n",
       "6 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr6_AA.rda"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Load the data.table library\n",
    "library(data.table)\n",
    "\n",
    "# Assuming 'merged' is already a data.table\n",
    "# If it's not, convert it using setDT(merged)\n",
    "\n",
    "# Select SNP column and reshape methylation columns into long format\n",
    "# Extract column names for methylation data\n",
    "methylation_columns <- grep(\"caud_|dlpfc_|hippo_\", names(merged), value = TRUE)\n",
    "\n",
    "# Create a long format table\n",
    "long_format <- melt(merged, id.vars = \"Chr\", measure.vars = methylation_columns, variable.name = \"Methylation_Type\", value.name = \"methylation_data\")\n",
    "\n",
    "# Add SNP data to the long format data.table\n",
    "long_format[, SNP_data := merged[.SD, on = \"Chr\", SNPs_libd]]\n",
    "\n",
    "# Remove Methylation_Type column and arrange columns as specified\n",
    "long_format[, `Methylation_Type` := NULL]\n",
    "setcolorder(long_format, c(\"Chr\", \"SNP_data\", \"methylation_data\"))\n",
    "\n",
    "# Result\n",
    "head(long_format)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "81140524-1678-4fee-b24e-9d3feec39f32",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: BiocGenerics\n",
      "\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,\n",
      "    match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,\n",
      "    Position, rank, rbind, Reduce, rownames, sapply, setdiff, sort,\n",
      "    table, tapply, union, unique, unsplit, which.max, which.min\n",
      "\n",
      "\n",
      "Loading required package: GenomicRanges\n",
      "\n",
      "Loading required package: stats4\n",
      "\n",
      "Loading required package: S4Vectors\n",
      "\n",
      "\n",
      "Attaching package: ‘S4Vectors’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:data.table’:\n",
      "\n",
      "    first, second\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:utils’:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Loading required package: IRanges\n",
      "\n",
      "\n",
      "Attaching package: ‘IRanges’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:data.table’:\n",
      "\n",
      "    shift\n",
      "\n",
      "\n",
      "Loading required package: GenomeInfoDb\n",
      "\n",
      "Loading required package: SummarizedExperiment\n",
      "\n",
      "Loading required package: MatrixGenerics\n",
      "\n",
      "Loading required package: matrixStats\n",
      "\n",
      "\n",
      "Attaching package: ‘MatrixGenerics’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:matrixStats’:\n",
      "\n",
      "    colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,\n",
      "    colCounts, colCummaxs, colCummins, colCumprods, colCumsums,\n",
      "    colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,\n",
      "    colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,\n",
      "    colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,\n",
      "    colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,\n",
      "    colWeightedMeans, colWeightedMedians, colWeightedSds,\n",
      "    colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,\n",
      "    rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,\n",
      "    rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,\n",
      "    rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,\n",
      "    rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,\n",
      "    rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,\n",
      "    rowWeightedMads, rowWeightedMeans, rowWeightedMedians,\n",
      "    rowWeightedSds, rowWeightedVars\n",
      "\n",
      "\n",
      "Loading required package: Biobase\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘Biobase’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:MatrixGenerics’:\n",
      "\n",
      "    rowMedians\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:matrixStats’:\n",
      "\n",
      "    anyMissing, rowMedians\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(bsseq)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "aa0f94b6-4948-43e5-8d9b-20f13058594d",
   "metadata": {},
   "source": [
    "### Test on first row"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "3844cf92-a5ce-441b-adad-ec80046cdbc3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>198</li><li>3</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 198\n",
       "\\item 3\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 198\n",
       "2. 3\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 198   3"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 6 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Chr</th><th scope=col>SNP_data</th><th scope=col>methylation_data</th></tr>\n",
       "\t<tr><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>1</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr1.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr1_AA.rda</td></tr>\n",
       "\t<tr><td>2</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr2.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr2_AA.rda</td></tr>\n",
       "\t<tr><td>3</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr3.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr3_AA.rda</td></tr>\n",
       "\t<tr><td>4</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr4.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr4_AA.rda</td></tr>\n",
       "\t<tr><td>5</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr5.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr5_AA.rda</td></tr>\n",
       "\t<tr><td>6</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr6.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr6_AA.rda</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 6 × 3\n",
       "\\begin{tabular}{lll}\n",
       " Chr & SNP\\_data & methylation\\_data\\\\\n",
       " <dbl> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t 1 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr1.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr1\\_AA.rda\\\\\n",
       "\t 2 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr2.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr2\\_AA.rda\\\\\n",
       "\t 3 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr3.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr3\\_AA.rda\\\\\n",
       "\t 4 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr4.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr4\\_AA.rda\\\\\n",
       "\t 5 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr5.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr5\\_AA.rda\\\\\n",
       "\t 6 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr6.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr6\\_AA.rda\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 6 × 3\n",
       "\n",
       "| Chr &lt;dbl&gt; | SNP_data &lt;chr&gt; | methylation_data &lt;chr&gt; |\n",
       "|---|---|---|\n",
       "| 1 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr1.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr1_AA.rda |\n",
       "| 2 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr2.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr2_AA.rda |\n",
       "| 3 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr3.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr3_AA.rda |\n",
       "| 4 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr4.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr4_AA.rda |\n",
       "| 5 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr5.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr5_AA.rda |\n",
       "| 6 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr6.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr6_AA.rda |\n",
       "\n"
      ],
      "text/plain": [
       "  Chr SNP_data                                                         \n",
       "1 1   /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr1.pgen\n",
       "2 2   /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr2.pgen\n",
       "3 3   /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr3.pgen\n",
       "4 4   /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr4.pgen\n",
       "5 5   /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr5.pgen\n",
       "6 6   /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr6.pgen\n",
       "  methylation_data                                                       \n",
       "1 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr1_AA.rda\n",
       "2 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr2_AA.rda\n",
       "3 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr3_AA.rda\n",
       "4 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr4_AA.rda\n",
       "5 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr5_AA.rda\n",
       "6 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr6_AA.rda"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "[1] \"2024-04-08 13:19:23 PDT\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dim(long_format)\n",
    "head(long_format)\n",
    "Sys.time()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "f329fd1c-aaca-45c2-b08c-8d9488bb9502",
   "metadata": {},
   "outputs": [],
   "source": [
    "i <- 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "fad236f6-ba91-4f0a-a806-63c0dfc9065d",
   "metadata": {},
   "outputs": [],
   "source": [
    "pvar <- fread(gsub(\"pgen\", \"pvar\", long_format$SNP_data[i]))[, 1:3]\n",
    "load(long_format$methylation_data[i])\n",
    "first_SNP_position <- min(pvar$POS)\n",
    "last_SNP_position <- max(pvar$POS)\n",
    "first_meth_position <- min(start(BSobj2))\n",
    "last_meth_position <- max(start(BSobj2))\n",
    "window_size <- 10000\n",
    "# Find first methylation index with SNP coverage\n",
    "meth_indices_w_SNP_coverage <- which(start(BSobj2) >= (first_SNP_position - window_size) &\n",
    "                                   start(BSobj2) <= (last_SNP_position + window_size))\n",
    "first_meth_index_with_SNP_coverage <- min(meth_indices_w_SNP_coverage)\n",
    "# Correcting the approach to find the last methylation index with SNP coverage\n",
    "last_meth_index_with_SNP_coverage <- max(meth_indices_w_SNP_coverage)\n",
    "\n",
    "first_meth_value_with_SNP_coverage <- start(BSobj2)[first_meth_index_with_SNP_coverage]\n",
    "last_meth_value_with_SNP_coverage <- start(BSobj2)[last_meth_index_with_SNP_coverage]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "2e965697-be77-4de4-84e3-98f36fdff697",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "8982"
      ],
      "text/latex": [
       "8982"
      ],
      "text/markdown": [
       "8982"
      ],
      "text/plain": [
       "[1] 8982"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "first_meth_index_with_SNP_coverage"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "7fa75a0c-1d0f-4a9f-9fc2-9c2d0a96da93",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "1079456"
      ],
      "text/latex": [
       "1079456"
      ],
      "text/markdown": [
       "1079456"
      ],
      "text/plain": [
       "[1] 1079456"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "first_SNP_position"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "0c77d224-7910-4d4a-81c5-f3e3af28ce23",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "248908368"
      ],
      "text/latex": [
       "248908368"
      ],
      "text/markdown": [
       "248908368"
      ],
      "text/plain": [
       "[1] 248908368"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "max(pvar$POS)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "64d98af3-2fd4-46f8-96ed-d736a2730803",
   "metadata": {},
   "outputs": [],
   "source": [
    "if(!identical(start(BSobj2), sort(start(BSobj2)))) {\n",
    "  stop(\"start positions in BSobj2 are not ordered\")\n",
    "}\n",
    "\n",
    "BSobj2 <- NULL"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "76194ec3-f4f5-4129-93b5-86dad0dacd3d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[1] \"2024-04-08 13:22:10 PDT\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "Sys.time()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "acf2c578-b996-4afc-aac4-66254ccbe8d8",
   "metadata": {},
   "source": [
    "### Sanity checks"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "303055c1-6c10-4a50-9e3f-9aa398b43270",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "1069461"
      ],
      "text/latex": [
       "1069461"
      ],
      "text/markdown": [
       "1069461"
      ],
      "text/plain": [
       "[1] 1069461"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "first_meth_value_with_SNP_coverage"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "af109676-cb39-4210-acfa-db564a592074",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "248918358"
      ],
      "text/latex": [
       "248918358"
      ],
      "text/markdown": [
       "248918358"
      ],
      "text/plain": [
       "[1] 248918358"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "last_meth_value_with_SNP_coverage"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "e6266bac-06a3-4d4d-9508-258e46c8915c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "8982"
      ],
      "text/latex": [
       "8982"
      ],
      "text/markdown": [
       "8982"
      ],
      "text/plain": [
       "[1] 8982"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "first_meth_index_with_SNP_coverage"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "c18aa399-e950-4276-90cb-d1df93f536b5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "2202702"
      ],
      "text/latex": [
       "2202702"
      ],
      "text/markdown": [
       "2202702"
      ],
      "text/plain": [
       "[1] 2202702"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "last_meth_index_with_SNP_coverage"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f18e3dc3-90c1-4308-8c0f-9aacf87b1a6b",
   "metadata": {},
   "source": [
    "### Deploy"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "edcdd58b-fc66-43b0-9f29-4b0c2adb5d2a",
   "metadata": {},
   "outputs": [],
   "source": [
    "i"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ac6d160c-b5bb-4e4f-a07c-5f6712fab8af",
   "metadata": {},
   "outputs": [],
   "source": [
    "BSobj2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bb91286f-813d-4057-abdd-c8cda78e6bd9",
   "metadata": {},
   "outputs": [],
   "source": [
    "long_format$first_snp_position <-\n",
    "long_format$last_snp_position <-\n",
    "long_format$last_meth_index <-\n",
    "long_format$first_meth_position <- \n",
    "long_format$last_meth_position <-\n",
    "long_format$first_meth_index_with_SNP_coverage <-\n",
    "long_format$last_meth_index_with_SNP_coverage <- \n",
    "long_format$first_meth_value_with_SNP_coverage <-\n",
    "long_format$last_meth_value_with_SNP_coverage <- \n",
    "NA\n",
    "\n",
    "start_time <- Sys.time()\n",
    "\n",
    "for(i in 1:nrow(long_format)){\n",
    "    print(i)\n",
    "    iter_start_time <- Sys.time()\n",
    "    \n",
    "    pvar <- fread(gsub(\"pgen\", \"pvar\", long_format$SNP_data[i]))[, 1:3]\n",
    "    load(long_format$methylation_data[i])\n",
    "\n",
    "    if(!identical(start(BSobj2), sort(start(BSobj2)))) {\n",
    "      stop(\"start positions in BSobj2 are not ordered\")\n",
    "    }\n",
    "    \n",
    "    first_SNP_position <- min(pvar$POS)\n",
    "    last_SNP_position <- max(pvar$POS)\n",
    "    first_meth_position <- min(start(BSobj2))\n",
    "    last_meth_position <- max(start(BSobj2))\n",
    "    window_size <- 10000\n",
    "    # Find first methylation index with SNP coverage\n",
    "    meth_indices_w_SNP_coverage <- which(start(BSobj2) >= (first_SNP_position - window_size) &\n",
    "                                         start(BSobj2) <= (last_SNP_position + window_size))\n",
    "    first_meth_index_with_SNP_coverage <- min(meth_indices_w_SNP_coverage)\n",
    "    # Correcting the approach to find the last methylation index with SNP coverage\n",
    "    last_meth_index_with_SNP_coverage <- max(meth_indices_w_SNP_coverage)\n",
    "\n",
    "    first_meth_value_with_SNP_coverage <- start(BSobj2)[first_meth_index_with_SNP_coverage]\n",
    "    last_meth_value_with_SNP_coverage <- start(BSobj2)[last_meth_index_with_SNP_coverage]\n",
    "\n",
    "    long_format$first_snp_position[i] <- first_SNP_position\n",
    "    long_format$last_snp_position[i] <- last_SNP_position\n",
    "    long_format$last_meth_index[i] <- length(start(BSobj2))\n",
    "    BSobj2 <- NULL\n",
    "    long_format$first_meth_position[i] <- first_meth_position\n",
    "    long_format$last_meth_position[i] <- last_meth_position\n",
    "    \n",
    "    long_format$first_meth_index_with_SNP_coverage[i] <- first_meth_index_with_SNP_coverage\n",
    "    long_format$last_meth_index_with_SNP_coverage[i] <- last_meth_index_with_SNP_coverage\n",
    "    long_format$first_meth_value_with_SNP_coverage[i] <- first_meth_value_with_SNP_coverage\n",
    "    long_format$last_meth_value_with_SNP_coverage[i] <- last_meth_value_with_SNP_coverage\n",
    "    \n",
    "    iter_end_time <- Sys.time()\n",
    "    elapsed_time_per_iter <- iter_end_time - iter_start_time\n",
    "    total_elapsed_time <- Sys.time() - start_time\n",
    "    average_time_per_iter <- total_elapsed_time / i\n",
    "    estimated_total_time <- average_time_per_iter * nrow(long_format)\n",
    "    estimated_time_remaining <- estimated_total_time - total_elapsed_time\n",
    "\n",
    "    print(sprintf(\"Ran %d of %d so far. Estimated time remaining: %s minutes\\n\", \n",
    "                i, nrow(long_format), round(as.numeric(estimated_time_remaining) / 60, 2)))\n",
    "\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bd0ef4bf-141a-4852-baae-0f5090276302",
   "metadata": {},
   "outputs": [],
   "source": [
    "fwrite(long_format, \"09-OUT_matched_files_and_indices_to_test_a3.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "e360b4b3-97c9-4d00-87bd-61e5ad0a074a",
   "metadata": {},
   "outputs": [],
   "source": [
    "long_format_a2 <- fread(\"09-OUT_matched_files_and_indices_to_test_a2.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "b57c9ac5-0de9-483d-bcae-064cc6d93119",
   "metadata": {},
   "outputs": [],
   "source": [
    "long_format <- fread(\"09-OUT_matched_files_and_indices_to_test_a3.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "00c7ebbb-6962-4fcb-87a7-10cb8632b8bd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 6 × 7</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Chr</th><th scope=col>SNP_data</th><th scope=col>methylation_data</th><th scope=col>last_meth_value_with_SNP_coverage</th><th scope=col>first_meth_value_with_SNP_coverage</th><th scope=col>last_meth_index_with_SNP_coverage</th><th scope=col>first_meth_index_with_SNP_coverage</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>1</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr1.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr1_AA.rda</td><td>248918358</td><td>1069461</td><td>2202702</td><td>8982</td></tr>\n",
       "\t<tr><td>2</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr2.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr2_AA.rda</td><td>241863783</td><td>  10001</td><td>2019984</td><td>   1</td></tr>\n",
       "\t<tr><td>3</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr3.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr3_AA.rda</td><td>198099789</td><td>  11602</td><td>1538467</td><td>   1</td></tr>\n",
       "\t<tr><td>4</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr4.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr4_AA.rda</td><td>189877411</td><td>  69399</td><td>1387731</td><td>   1</td></tr>\n",
       "\t<tr><td>5</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr5.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr5_AA.rda</td><td>181172584</td><td>  44104</td><td>1409038</td><td>   1</td></tr>\n",
       "\t<tr><td>6</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr6.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr6_AA.rda</td><td>170619093</td><td> 192453</td><td>1412543</td><td>1138</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 6 × 7\n",
       "\\begin{tabular}{lllllll}\n",
       " Chr & SNP\\_data & methylation\\_data & last\\_meth\\_value\\_with\\_SNP\\_coverage & first\\_meth\\_value\\_with\\_SNP\\_coverage & last\\_meth\\_index\\_with\\_SNP\\_coverage & first\\_meth\\_index\\_with\\_SNP\\_coverage\\\\\n",
       " <int> & <chr> & <chr> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t 1 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr1.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr1\\_AA.rda & 248918358 & 1069461 & 2202702 & 8982\\\\\n",
       "\t 2 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr2.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr2\\_AA.rda & 241863783 &   10001 & 2019984 &    1\\\\\n",
       "\t 3 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr3.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr3\\_AA.rda & 198099789 &   11602 & 1538467 &    1\\\\\n",
       "\t 4 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr4.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr4\\_AA.rda & 189877411 &   69399 & 1387731 &    1\\\\\n",
       "\t 5 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr5.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr5\\_AA.rda & 181172584 &   44104 & 1409038 &    1\\\\\n",
       "\t 6 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr6.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr6\\_AA.rda & 170619093 &  192453 & 1412543 & 1138\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 6 × 7\n",
       "\n",
       "| Chr &lt;int&gt; | SNP_data &lt;chr&gt; | methylation_data &lt;chr&gt; | last_meth_value_with_SNP_coverage &lt;int&gt; | first_meth_value_with_SNP_coverage &lt;int&gt; | last_meth_index_with_SNP_coverage &lt;int&gt; | first_meth_index_with_SNP_coverage &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|\n",
       "| 1 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr1.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr1_AA.rda | 248918358 | 1069461 | 2202702 | 8982 |\n",
       "| 2 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr2.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr2_AA.rda | 241863783 |   10001 | 2019984 |    1 |\n",
       "| 3 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr3.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr3_AA.rda | 198099789 |   11602 | 1538467 |    1 |\n",
       "| 4 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr4.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr4_AA.rda | 189877411 |   69399 | 1387731 |    1 |\n",
       "| 5 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr5.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr5_AA.rda | 181172584 |   44104 | 1409038 |    1 |\n",
       "| 6 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr6.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr6_AA.rda | 170619093 |  192453 | 1412543 | 1138 |\n",
       "\n"
      ],
      "text/plain": [
       "  Chr SNP_data                                                         \n",
       "1 1   /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr1.pgen\n",
       "2 2   /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr2.pgen\n",
       "3 3   /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr3.pgen\n",
       "4 4   /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr4.pgen\n",
       "5 5   /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr5.pgen\n",
       "6 6   /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr6.pgen\n",
       "  methylation_data                                                       \n",
       "1 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr1_AA.rda\n",
       "2 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr2_AA.rda\n",
       "3 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr3_AA.rda\n",
       "4 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr4_AA.rda\n",
       "5 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr5_AA.rda\n",
       "6 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr6_AA.rda\n",
       "  last_meth_value_with_SNP_coverage first_meth_value_with_SNP_coverage\n",
       "1 248918358                         1069461                           \n",
       "2 241863783                           10001                           \n",
       "3 198099789                           11602                           \n",
       "4 189877411                           69399                           \n",
       "5 181172584                           44104                           \n",
       "6 170619093                          192453                           \n",
       "  last_meth_index_with_SNP_coverage first_meth_index_with_SNP_coverage\n",
       "1 2202702                           8982                              \n",
       "2 2019984                              1                              \n",
       "3 1538467                              1                              \n",
       "4 1387731                              1                              \n",
       "5 1409038                              1                              \n",
       "6 1412543                           1138                              "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(long_format_a2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "e563bc84-f3e5-42ca-b230-e8aca5d1611d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 6 × 12</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Chr</th><th scope=col>SNP_data</th><th scope=col>methylation_data</th><th scope=col>last_meth_value_with_SNP_coverage</th><th scope=col>first_meth_value_with_SNP_coverage</th><th scope=col>last_meth_index_with_SNP_coverage</th><th scope=col>first_meth_index_with_SNP_coverage</th><th scope=col>last_meth_position</th><th scope=col>first_meth_position</th><th scope=col>last_meth_index</th><th scope=col>last_snp_position</th><th scope=col>first_snp_position</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>1</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr1.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr1_AA.rda</td><td>248918358</td><td>1069461</td><td>2202702</td><td>8982</td><td>248932459</td><td>792731</td><td>2202819</td><td>248908368</td><td>1079456</td></tr>\n",
       "\t<tr><td>2</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr2.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr2_AA.rda</td><td>241863783</td><td>  10001</td><td>2019984</td><td>   1</td><td>242110081</td><td> 10001</td><td>2025108</td><td>241853790</td><td>  10797</td></tr>\n",
       "\t<tr><td>3</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr3.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr3_AA.rda</td><td>198099789</td><td>  11602</td><td>1538467</td><td>   1</td><td>198099789</td><td> 11602</td><td>1538467</td><td>198114462</td><td>  18519</td></tr>\n",
       "\t<tr><td>4</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr4.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr4_AA.rda</td><td>189877411</td><td>  69399</td><td>1387731</td><td>   1</td><td>190048422</td><td> 69399</td><td>1388072</td><td>189974231</td><td>  68894</td></tr>\n",
       "\t<tr><td>5</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr5.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr5_AA.rda</td><td>181172584</td><td>  44104</td><td>1409038</td><td>   1</td><td>181172584</td><td> 44104</td><td>1409038</td><td>181286423</td><td>  25361</td></tr>\n",
       "\t<tr><td>6</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr6.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr6_AA.rda</td><td>170619093</td><td> 192453</td><td>1412543</td><td>1138</td><td>170745956</td><td> 60470</td><td>1413875</td><td>170609176</td><td> 202452</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 6 × 12\n",
       "\\begin{tabular}{llllllllllll}\n",
       " Chr & SNP\\_data & methylation\\_data & last\\_meth\\_value\\_with\\_SNP\\_coverage & first\\_meth\\_value\\_with\\_SNP\\_coverage & last\\_meth\\_index\\_with\\_SNP\\_coverage & first\\_meth\\_index\\_with\\_SNP\\_coverage & last\\_meth\\_position & first\\_meth\\_position & last\\_meth\\_index & last\\_snp\\_position & first\\_snp\\_position\\\\\n",
       " <int> & <chr> & <chr> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t 1 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr1.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr1\\_AA.rda & 248918358 & 1069461 & 2202702 & 8982 & 248932459 & 792731 & 2202819 & 248908368 & 1079456\\\\\n",
       "\t 2 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr2.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr2\\_AA.rda & 241863783 &   10001 & 2019984 &    1 & 242110081 &  10001 & 2025108 & 241853790 &   10797\\\\\n",
       "\t 3 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr3.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr3\\_AA.rda & 198099789 &   11602 & 1538467 &    1 & 198099789 &  11602 & 1538467 & 198114462 &   18519\\\\\n",
       "\t 4 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr4.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr4\\_AA.rda & 189877411 &   69399 & 1387731 &    1 & 190048422 &  69399 & 1388072 & 189974231 &   68894\\\\\n",
       "\t 5 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr5.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr5\\_AA.rda & 181172584 &   44104 & 1409038 &    1 & 181172584 &  44104 & 1409038 & 181286423 &   25361\\\\\n",
       "\t 6 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr6.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr6\\_AA.rda & 170619093 &  192453 & 1412543 & 1138 & 170745956 &  60470 & 1413875 & 170609176 &  202452\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 6 × 12\n",
       "\n",
       "| Chr &lt;int&gt; | SNP_data &lt;chr&gt; | methylation_data &lt;chr&gt; | last_meth_value_with_SNP_coverage &lt;int&gt; | first_meth_value_with_SNP_coverage &lt;int&gt; | last_meth_index_with_SNP_coverage &lt;int&gt; | first_meth_index_with_SNP_coverage &lt;int&gt; | last_meth_position &lt;int&gt; | first_meth_position &lt;int&gt; | last_meth_index &lt;int&gt; | last_snp_position &lt;int&gt; | first_snp_position &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr1.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr1_AA.rda | 248918358 | 1069461 | 2202702 | 8982 | 248932459 | 792731 | 2202819 | 248908368 | 1079456 |\n",
       "| 2 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr2.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr2_AA.rda | 241863783 |   10001 | 2019984 |    1 | 242110081 |  10001 | 2025108 | 241853790 |   10797 |\n",
       "| 3 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr3.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr3_AA.rda | 198099789 |   11602 | 1538467 |    1 | 198099789 |  11602 | 1538467 | 198114462 |   18519 |\n",
       "| 4 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr4.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr4_AA.rda | 189877411 |   69399 | 1387731 |    1 | 190048422 |  69399 | 1388072 | 189974231 |   68894 |\n",
       "| 5 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr5.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr5_AA.rda | 181172584 |   44104 | 1409038 |    1 | 181172584 |  44104 | 1409038 | 181286423 |   25361 |\n",
       "| 6 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr6.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr6_AA.rda | 170619093 |  192453 | 1412543 | 1138 | 170745956 |  60470 | 1413875 | 170609176 |  202452 |\n",
       "\n"
      ],
      "text/plain": [
       "  Chr SNP_data                                                         \n",
       "1 1   /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr1.pgen\n",
       "2 2   /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr2.pgen\n",
       "3 3   /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr3.pgen\n",
       "4 4   /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr4.pgen\n",
       "5 5   /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr5.pgen\n",
       "6 6   /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr6.pgen\n",
       "  methylation_data                                                       \n",
       "1 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr1_AA.rda\n",
       "2 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr2_AA.rda\n",
       "3 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr3_AA.rda\n",
       "4 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr4_AA.rda\n",
       "5 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr5_AA.rda\n",
       "6 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr6_AA.rda\n",
       "  last_meth_value_with_SNP_coverage first_meth_value_with_SNP_coverage\n",
       "1 248918358                         1069461                           \n",
       "2 241863783                           10001                           \n",
       "3 198099789                           11602                           \n",
       "4 189877411                           69399                           \n",
       "5 181172584                           44104                           \n",
       "6 170619093                          192453                           \n",
       "  last_meth_index_with_SNP_coverage first_meth_index_with_SNP_coverage\n",
       "1 2202702                           8982                              \n",
       "2 2019984                              1                              \n",
       "3 1538467                              1                              \n",
       "4 1387731                              1                              \n",
       "5 1409038                              1                              \n",
       "6 1412543                           1138                              \n",
       "  last_meth_position first_meth_position last_meth_index last_snp_position\n",
       "1 248932459          792731              2202819         248908368        \n",
       "2 242110081           10001              2025108         241853790        \n",
       "3 198099789           11602              1538467         198114462        \n",
       "4 190048422           69399              1388072         189974231        \n",
       "5 181172584           44104              1409038         181286423        \n",
       "6 170745956           60470              1413875         170609176        \n",
       "  first_snp_position\n",
       "1 1079456           \n",
       "2   10797           \n",
       "3   18519           \n",
       "4   68894           \n",
       "5   25361           \n",
       "6  202452           "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(long_format)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "44f8909b-4518-4d47-a124-f9b8764c647e",
   "metadata": {},
   "source": [
    "Started around 2pm"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0f01e623-7e5d-4e5b-a5e5-21325c033602",
   "metadata": {},
   "source": [
    "## Prepare covariate files"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8e06cac7-a8a6-4289-8632-a5fcaa8cd641",
   "metadata": {},
   "source": [
    "### Get paths"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "70e73d93-b862-4d05-8a89-e69ef94b494e",
   "metadata": {
    "jupyter": {
     "source_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "# genotype_pc <- list.files(\"/dcs04/lieber/statsgen/shizhong/database/libd/genotype/postmortem/topmed/merge_H650_1M_2.5M_5M/EA_AA/all/pca_wgbs\",\n",
    "#                           recursive = TRUE,\n",
    "#                           full.names = TRUE,\n",
    "#                           pattern = \"pca.eigenvec\")\n",
    "\n",
    "# genotype_pc <- data.frame(genotype_pc_path = genotype_pc, brain_region = c(\"caud\", \"dlpfc\", \"hippo\"))\n",
    "\n",
    "# genotype_pc <- rbind(genotype_pc, genotype_pc, genotype_pc)\n",
    "\n",
    "# genotype_pc$subpopulation <- c(\"EA\", \"EA\", \"EA\", \"AA\", \"AA\", \"AA\", \"all\", \"all\", \"all\")\n",
    "\n",
    "# genotype_pc\n",
    "\n",
    "# meth_pc <- list.files(\"/dcs04/lieber/statsgen/shizhong/michael/mwas/pheno/\",\n",
    "#                       recursive = TRUE,\n",
    "#                       full.names = TRUE,\n",
    "#                       pattern = \"pc_\")\n",
    "\n",
    "# meth_pc <- data.table(meth_pc_path = meth_pc)\n",
    "\n",
    "# meth_pc[, subpopulation := sub(\".*pc_(.*)\\\\.csv\", \"\\\\1\", meth_pc_path)]\n",
    "# meth_pc[, brain_region := sub(\".*/pheno//(.*)/out/.*\", \"\\\\1\", meth_pc_path)]\n",
    "\n",
    "# meth_pc\n",
    "\n",
    "# cov3 <- \"/dcs04/lieber/statsgen/shizhong/database/libd/genotype/postmortem/phenotype/pheno_PC\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a4803857-0d71-408b-a38e-ef4cff575b77",
   "metadata": {},
   "source": [
    "### Format, merge with data frame"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a29858b1-4db7-4127-a52b-407d08d595df",
   "metadata": {
    "jupyter": {
     "source_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "# long_format[, subpopulation := sub(\".*_(.*)\\\\.rda\", \"\\\\1\", methylation_data)]\n",
    "# long_format[, brain_region := sub(\".*/pheno/(.*)/out/.*\", \"\\\\1\", methylation_data)]\n",
    "\n",
    "# merged <- merge(long_format, genotype_pc)\n",
    "\n",
    "# merged <- merge(merged, meth_pc)\n",
    "\n",
    "# merged$pheno_pc <- cov3\n",
    "\n",
    "# head(merged)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "982f7b87-da03-4f0e-93e3-53588e368388",
   "metadata": {},
   "source": [
    "### Unite covariate files for each row into a single file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5c870563-6a76-417e-be7f-1683c29cc012",
   "metadata": {},
   "outputs": [],
   "source": [
    "# outdir <- \"/dcs04/lieber/statsgen/mnagle/mwas/full_covariates/\"\n",
    "# #dir.create(outdir)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4c22cd0f-c0ec-4d20-ad60-447bf0ce2eba",
   "metadata": {},
   "source": [
    "#### test for first row"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "07b31074-c093-42de-b02f-bdd0a58c6179",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# colnames(merged)\n",
    "\n",
    "# dim(merged)\n",
    "\n",
    "\n",
    "\n",
    "# merged_pc_only <- unique(merged[, c(1, 2, 10:12)])\n",
    "\n",
    "# merged_pc_only\n",
    "\n",
    "# i <- 1\n",
    "\n",
    "# geno_pc <- fread(merged$genotype_pc_path[i])\n",
    "# meth_pc <- fread(merged$meth_pc_path[i])\n",
    "# pheno_pc <- fread(merged$pheno_pc[i])\n",
    "\n",
    "# meth_pc <- meth_pc[, 1:11]\n",
    "\n",
    "# geno_pc <- geno_pc[, 1:4]\n",
    "\n",
    "# pheno_pc <- pheno_pc[, c(2, 6, 7, 8)]\n",
    "\n",
    "# colnames(meth_pc)[1] <- colnames(pheno_pc)[1] <- colnames(geno_pc)[1] <- \"ID\"\n",
    "\n",
    "# colnames(meth_pc)[2:11] <- paste0(\"methPC\", 1:10)\n",
    "\n",
    "# colnames(geno_pc)[2:4] <- paste0(\"genoPC\", 1:3)\n",
    "\n",
    "# geno_pc$ID <- gsub(\"Br0\", \"Br\", geno_pc$ID)\n",
    "# pheno_pc$ID <- gsub(\"Br0\", \"Br\", pheno_pc$ID)\n",
    "# meth_pc$ID <- gsub(\"Br0\", \"Br\", meth_pc$ID)\n",
    "\n",
    "# #dim(meth_pc)\n",
    "# #dim(geno_pc)\n",
    "# #dim(pheno_pc)\n",
    "\n",
    "# unified_pc <- merge(geno_pc, pheno_pc)\n",
    "# #dim(unified_pc)\n",
    "# unified_pc <- merge(unified_pc, meth_pc)\n",
    "# #dim(unified_pc)\n",
    "\n",
    "# if(nrow(unified_pc) != nrow(meth_pc)){\n",
    "#     stop(paste0(\"We have a mismatch for row \", i))\n",
    "#     }\n",
    "\n",
    "# outname <- paste0(outdir, merged$subpopulation[i], \"_\", merged$brain_region[i], \".csv\")\n",
    "# fwrite(merged_pc_only, outname)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5ec1d2e7-ced9-4a1b-890e-89124650ffd0",
   "metadata": {},
   "source": [
    "#### Run for all"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "16635bdb-fa8f-4051-92a3-3ba7d18b72d7",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# for(i in 1:nrow(merged_pc_only)){\n",
    "#     geno_pc <- fread(merged_pc_only$genotype_pc_path[i])\n",
    "#     meth_pc <- fread(merged_pc_only$meth_pc_path[i])\n",
    "#     pheno_pc <- fread(merged_pc_only$pheno_pc[i])\n",
    "\n",
    "#     meth_pc <- meth_pc[, 1:11]\n",
    "\n",
    "#     geno_pc <- geno_pc[, 1:4]\n",
    "\n",
    "#     pheno_pc <- pheno_pc[, c(2, 6, 7, 8)]\n",
    "\n",
    "#     colnames(meth_pc)[1] <- colnames(pheno_pc)[1] <- colnames(geno_pc)[1] <- \"ID\"\n",
    "\n",
    "#     colnames(meth_pc)[2:11] <- paste0(\"methPC\", 1:10)\n",
    "\n",
    "#     colnames(geno_pc)[2:4] <- paste0(\"genoPC\", 1:3)\n",
    "\n",
    "#     geno_pc$ID <- gsub(\"Br0\", \"Br\", geno_pc$ID)\n",
    "#     pheno_pc$ID <- gsub(\"Br0\", \"Br\", pheno_pc$ID)\n",
    "#     pheno_pc$Dx <- gsub(\"chizo\", \"CZ\", pheno_pc$Dx)\n",
    "#     meth_pc$ID <- gsub(\"Br0\", \"Br\", meth_pc$ID)\n",
    "\n",
    "#     #dim(meth_pc)\n",
    "#     #dim(geno_pc)\n",
    "#     #dim(pheno_pc)\n",
    "\n",
    "#     unified_pc <- merge(geno_pc, meth_pc)\n",
    "#     #dim(unified_pc)\n",
    "#     unified_pc <- merge(unified_pc, pheno_pc)\n",
    "#     #dim(unified_pc)\n",
    "\n",
    "#     if(nrow(unified_pc) != nrow(meth_pc)){\n",
    "#         print(setdiff(meth_pc$ID, unified_pc$ID))\n",
    "#         warning(paste0(\"We have a mismatch for row \", i))\n",
    "#         print(merged_pc_only[i, ])\n",
    "#         }\n",
    "#     outname <- paste0(outdir, merged_pc_only$subpopulation[i], \"_\", merged_pc_only$brain_region[i], \".csv\")\n",
    "#     fwrite(unified_pc, outname)\n",
    "    \n",
    "#     }\n",
    "\n",
    "\n",
    "# outname\n",
    "\n",
    "# merged$full_covariate_path <-\n",
    "# paste0(outdir, merged$subpopulation, \"_\", merged$brain_region, \".csv\")\n",
    "\n",
    "# fread(\"/dcs04/lieber/statsgen/mnagle/mwas/full_covariates/AA_caud.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f120e93d-62b7-48be-8aee-952f8dfd2507",
   "metadata": {},
   "source": [
    "## Match covariate files to corresponding datasets"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "074dd342-820d-49ae-8bcf-91bf4284acdc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'AA_caud.csv'</li><li>'AA_dlpfc.csv'</li><li>'AA_hippo.csv'</li><li>'all_caud.csv'</li><li>'all_dlpfc.csv'</li><li>'all_hippo.csv'</li><li>'EA_caud.csv'</li><li>'EA_dlpfc.csv'</li><li>'EA_hippo.csv'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'AA\\_caud.csv'\n",
       "\\item 'AA\\_dlpfc.csv'\n",
       "\\item 'AA\\_hippo.csv'\n",
       "\\item 'all\\_caud.csv'\n",
       "\\item 'all\\_dlpfc.csv'\n",
       "\\item 'all\\_hippo.csv'\n",
       "\\item 'EA\\_caud.csv'\n",
       "\\item 'EA\\_dlpfc.csv'\n",
       "\\item 'EA\\_hippo.csv'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'AA_caud.csv'\n",
       "2. 'AA_dlpfc.csv'\n",
       "3. 'AA_hippo.csv'\n",
       "4. 'all_caud.csv'\n",
       "5. 'all_dlpfc.csv'\n",
       "6. 'all_hippo.csv'\n",
       "7. 'EA_caud.csv'\n",
       "8. 'EA_dlpfc.csv'\n",
       "9. 'EA_hippo.csv'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"AA_caud.csv\"   \"AA_dlpfc.csv\"  \"AA_hippo.csv\"  \"all_caud.csv\" \n",
       "[5] \"all_dlpfc.csv\" \"all_hippo.csv\" \"EA_caud.csv\"   \"EA_dlpfc.csv\" \n",
       "[9] \"EA_hippo.csv\" "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "list.files(\"../../full_covariates/\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "ba4246df-a7df-4c2c-be03-62aa57383140",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 6 × 7</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Chr</th><th scope=col>SNP_data</th><th scope=col>methylation_data</th><th scope=col>last_meth_value_with_SNP_coverage</th><th scope=col>first_meth_value_with_SNP_coverage</th><th scope=col>last_meth_index_with_SNP_coverage</th><th scope=col>first_meth_index_with_SNP_coverage</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>1</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr1.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr1_AA.rda</td><td>248918358</td><td>1069461</td><td>2202702</td><td>8982</td></tr>\n",
       "\t<tr><td>2</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr2.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr2_AA.rda</td><td>241863783</td><td>  10001</td><td>2019984</td><td>   1</td></tr>\n",
       "\t<tr><td>3</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr3.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr3_AA.rda</td><td>198099789</td><td>  11602</td><td>1538467</td><td>   1</td></tr>\n",
       "\t<tr><td>4</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr4.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr4_AA.rda</td><td>189877411</td><td>  69399</td><td>1387731</td><td>   1</td></tr>\n",
       "\t<tr><td>5</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr5.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr5_AA.rda</td><td>181172584</td><td>  44104</td><td>1409038</td><td>   1</td></tr>\n",
       "\t<tr><td>6</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr6.pgen</td><td>/expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr6_AA.rda</td><td>170619093</td><td> 192453</td><td>1412543</td><td>1138</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 6 × 7\n",
       "\\begin{tabular}{lllllll}\n",
       " Chr & SNP\\_data & methylation\\_data & last\\_meth\\_value\\_with\\_SNP\\_coverage & first\\_meth\\_value\\_with\\_SNP\\_coverage & last\\_meth\\_index\\_with\\_SNP\\_coverage & first\\_meth\\_index\\_with\\_SNP\\_coverage\\\\\n",
       " <int> & <chr> & <chr> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t 1 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr1.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr1\\_AA.rda & 248918358 & 1069461 & 2202702 & 8982\\\\\n",
       "\t 2 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr2.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr2\\_AA.rda & 241863783 &   10001 & 2019984 &    1\\\\\n",
       "\t 3 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr3.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr3\\_AA.rda & 198099789 &   11602 & 1538467 &    1\\\\\n",
       "\t 4 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr4.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr4\\_AA.rda & 189877411 &   69399 & 1387731 &    1\\\\\n",
       "\t 5 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr5.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr5\\_AA.rda & 181172584 &   44104 & 1409038 &    1\\\\\n",
       "\t 6 & /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd\\_chr6.pgen & /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr6\\_AA.rda & 170619093 &  192453 & 1412543 & 1138\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 6 × 7\n",
       "\n",
       "| Chr &lt;int&gt; | SNP_data &lt;chr&gt; | methylation_data &lt;chr&gt; | last_meth_value_with_SNP_coverage &lt;int&gt; | first_meth_value_with_SNP_coverage &lt;int&gt; | last_meth_index_with_SNP_coverage &lt;int&gt; | first_meth_index_with_SNP_coverage &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|\n",
       "| 1 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr1.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr1_AA.rda | 248918358 | 1069461 | 2202702 | 8982 |\n",
       "| 2 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr2.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr2_AA.rda | 241863783 |   10001 | 2019984 |    1 |\n",
       "| 3 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr3.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr3_AA.rda | 198099789 |   11602 | 1538467 |    1 |\n",
       "| 4 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr4.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr4_AA.rda | 189877411 |   69399 | 1387731 |    1 |\n",
       "| 5 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr5.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr5_AA.rda | 181172584 |   44104 | 1409038 |    1 |\n",
       "| 6 | /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr6.pgen | /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr6_AA.rda | 170619093 |  192453 | 1412543 | 1138 |\n",
       "\n"
      ],
      "text/plain": [
       "  Chr SNP_data                                                         \n",
       "1 1   /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr1.pgen\n",
       "2 2   /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr2.pgen\n",
       "3 3   /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr3.pgen\n",
       "4 4   /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr4.pgen\n",
       "5 5   /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr5.pgen\n",
       "6 6   /expanse/lustre/projects/jhu152/naglemi/mwas/gwas//libd_chr6.pgen\n",
       "  methylation_data                                                       \n",
       "1 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr1_AA.rda\n",
       "2 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr2_AA.rda\n",
       "3 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr3_AA.rda\n",
       "4 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr4_AA.rda\n",
       "5 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr5_AA.rda\n",
       "6 /expanse/lustre/projects/jhu152/naglemi/mwas/pheno/caud/out/chr6_AA.rda\n",
       "  last_meth_value_with_SNP_coverage first_meth_value_with_SNP_coverage\n",
       "1 248918358                         1069461                           \n",
       "2 241863783                           10001                           \n",
       "3 198099789                           11602                           \n",
       "4 189877411                           69399                           \n",
       "5 181172584                           44104                           \n",
       "6 170619093                          192453                           \n",
       "  last_meth_index_with_SNP_coverage first_meth_index_with_SNP_coverage\n",
       "1 2202702                           8982                              \n",
       "2 2019984                              1                              \n",
       "3 1538467                              1                              \n",
       "4 1387731                              1                              \n",
       "5 1409038                              1                              \n",
       "6 1412543                           1138                              "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(long_format)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "583e407c-3d69-4785-a293-baaec93fafa4",
   "metadata": {},
   "outputs": [],
   "source": [
    "long_format$population <- str_split_fixed(long_format$methylation_data,\n",
    "                                          pattern = \"_\",\n",
    "                                          2)[, 2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "5a9f23ee-328a-4e5e-a4e9-ac9510362663",
   "metadata": {},
   "outputs": [],
   "source": [
    "long_format$population <- gsub(\"\\\\.rda\", \"\", long_format$population)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "f792791d-afbb-443f-8b2e-f17eff21ed4a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'AA'</li><li>'AA'</li><li>'AA'</li><li>'AA'</li><li>'AA'</li><li>'AA'</li><li>'AA'</li><li>'AA'</li><li>'AA'</li><li>'AA'</li><li>'AA'</li><li>'AA'</li><li>'AA'</li><li>'AA'</li><li>'AA'</li><li>'AA'</li><li>'AA'</li><li>'AA'</li><li>'AA'</li><li>'AA'</li><li>'AA'</li><li>'AA'</li><li>'all'</li><li>'all'</li><li>'all'</li><li>'all'</li><li>'all'</li><li>'all'</li><li>'all'</li><li>'all'</li><li>'all'</li><li>'all'</li><li>'all'</li><li>'all'</li><li>'all'</li><li>'all'</li><li>'all'</li><li>'all'</li><li>'all'</li><li>'all'</li><li>'all'</li><li>'all'</li><li>'all'</li><li>'all'</li><li>'EA'</li><li>'EA'</li><li>'EA'</li><li>'EA'</li><li>'EA'</li><li>'EA'</li><li>'EA'</li><li>'EA'</li><li>'EA'</li><li>'EA'</li><li>'EA'</li><li>'EA'</li><li>'EA'</li><li>'EA'</li><li>'EA'</li><li>'EA'</li><li>'EA'</li><li>'EA'</li><li>'EA'</li><li>'EA'</li><li>'EA'</li><li>'EA'</li><li>'AA'</li><li>'AA'</li><li>'AA'</li><li>'AA'</li><li>'AA'</li><li>'AA'</li><li>'AA'</li><li>'AA'</li><li>'AA'</li><li>'AA'</li><li>'AA'</li><li>'AA'</li><li>'AA'</li><li>'AA'</li><li>'AA'</li><li>'AA'</li><li>'AA'</li><li>'AA'</li><li>'AA'</li><li>'AA'</li><li>'AA'</li><li>'AA'</li><li>'all'</li><li>'all'</li><li>'all'</li><li>'all'</li><li>'all'</li><li>'all'</li><li>'all'</li><li>'all'</li><li>'all'</li><li>'all'</li><li>'all'</li><li>'all'</li><li>'all'</li><li>'all'</li><li>'all'</li><li>'all'</li><li>'all'</li><li>'all'</li><li>'all'</li><li>'all'</li><li>'all'</li><li>'all'</li><li>'EA'</li><li>'EA'</li><li>'EA'</li><li>'EA'</li><li>'EA'</li><li>'EA'</li><li>'EA'</li><li>'EA'</li><li>'EA'</li><li>'EA'</li><li>'EA'</li><li>'EA'</li><li>'EA'</li><li>'EA'</li><li>'EA'</li><li>'EA'</li><li>'EA'</li><li>'EA'</li><li>'EA'</li><li>'EA'</li><li>'EA'</li><li>'EA'</li><li>'AA'</li><li>'AA'</li><li>'AA'</li><li>'AA'</li><li>'AA'</li><li>'AA'</li><li>'AA'</li><li>'AA'</li><li>'AA'</li><li>'AA'</li><li>'AA'</li><li>'AA'</li><li>'AA'</li><li>'AA'</li><li>'AA'</li><li>'AA'</li><li>'AA'</li><li>'AA'</li><li>'AA'</li><li>'AA'</li><li>'AA'</li><li>'AA'</li><li>'all'</li><li>'all'</li><li>'all'</li><li>'all'</li><li>'all'</li><li>'all'</li><li>'all'</li><li>'all'</li><li>'all'</li><li>'all'</li><li>'all'</li><li>'all'</li><li>'all'</li><li>'all'</li><li>'all'</li><li>'all'</li><li>'all'</li><li>'all'</li><li>'all'</li><li>'all'</li><li>'all'</li><li>'all'</li><li>'EA'</li><li>'EA'</li><li>'EA'</li><li>'EA'</li><li>'EA'</li><li>'EA'</li><li>'EA'</li><li>'EA'</li><li>'EA'</li><li>'EA'</li><li>'EA'</li><li>'EA'</li><li>'EA'</li><li>'EA'</li><li>'EA'</li><li>'EA'</li><li>'EA'</li><li>'EA'</li><li>'EA'</li><li>'EA'</li><li>'EA'</li><li>'EA'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'AA'\n",
       "\\item 'AA'\n",
       "\\item 'AA'\n",
       "\\item 'AA'\n",
       "\\item 'AA'\n",
       "\\item 'AA'\n",
       "\\item 'AA'\n",
       "\\item 'AA'\n",
       "\\item 'AA'\n",
       "\\item 'AA'\n",
       "\\item 'AA'\n",
       "\\item 'AA'\n",
       "\\item 'AA'\n",
       "\\item 'AA'\n",
       "\\item 'AA'\n",
       "\\item 'AA'\n",
       "\\item 'AA'\n",
       "\\item 'AA'\n",
       "\\item 'AA'\n",
       "\\item 'AA'\n",
       "\\item 'AA'\n",
       "\\item 'AA'\n",
       "\\item 'all'\n",
       "\\item 'all'\n",
       "\\item 'all'\n",
       "\\item 'all'\n",
       "\\item 'all'\n",
       "\\item 'all'\n",
       "\\item 'all'\n",
       "\\item 'all'\n",
       "\\item 'all'\n",
       "\\item 'all'\n",
       "\\item 'all'\n",
       "\\item 'all'\n",
       "\\item 'all'\n",
       "\\item 'all'\n",
       "\\item 'all'\n",
       "\\item 'all'\n",
       "\\item 'all'\n",
       "\\item 'all'\n",
       "\\item 'all'\n",
       "\\item 'all'\n",
       "\\item 'all'\n",
       "\\item 'all'\n",
       "\\item 'EA'\n",
       "\\item 'EA'\n",
       "\\item 'EA'\n",
       "\\item 'EA'\n",
       "\\item 'EA'\n",
       "\\item 'EA'\n",
       "\\item 'EA'\n",
       "\\item 'EA'\n",
       "\\item 'EA'\n",
       "\\item 'EA'\n",
       "\\item 'EA'\n",
       "\\item 'EA'\n",
       "\\item 'EA'\n",
       "\\item 'EA'\n",
       "\\item 'EA'\n",
       "\\item 'EA'\n",
       "\\item 'EA'\n",
       "\\item 'EA'\n",
       "\\item 'EA'\n",
       "\\item 'EA'\n",
       "\\item 'EA'\n",
       "\\item 'EA'\n",
       "\\item 'AA'\n",
       "\\item 'AA'\n",
       "\\item 'AA'\n",
       "\\item 'AA'\n",
       "\\item 'AA'\n",
       "\\item 'AA'\n",
       "\\item 'AA'\n",
       "\\item 'AA'\n",
       "\\item 'AA'\n",
       "\\item 'AA'\n",
       "\\item 'AA'\n",
       "\\item 'AA'\n",
       "\\item 'AA'\n",
       "\\item 'AA'\n",
       "\\item 'AA'\n",
       "\\item 'AA'\n",
       "\\item 'AA'\n",
       "\\item 'AA'\n",
       "\\item 'AA'\n",
       "\\item 'AA'\n",
       "\\item 'AA'\n",
       "\\item 'AA'\n",
       "\\item 'all'\n",
       "\\item 'all'\n",
       "\\item 'all'\n",
       "\\item 'all'\n",
       "\\item 'all'\n",
       "\\item 'all'\n",
       "\\item 'all'\n",
       "\\item 'all'\n",
       "\\item 'all'\n",
       "\\item 'all'\n",
       "\\item 'all'\n",
       "\\item 'all'\n",
       "\\item 'all'\n",
       "\\item 'all'\n",
       "\\item 'all'\n",
       "\\item 'all'\n",
       "\\item 'all'\n",
       "\\item 'all'\n",
       "\\item 'all'\n",
       "\\item 'all'\n",
       "\\item 'all'\n",
       "\\item 'all'\n",
       "\\item 'EA'\n",
       "\\item 'EA'\n",
       "\\item 'EA'\n",
       "\\item 'EA'\n",
       "\\item 'EA'\n",
       "\\item 'EA'\n",
       "\\item 'EA'\n",
       "\\item 'EA'\n",
       "\\item 'EA'\n",
       "\\item 'EA'\n",
       "\\item 'EA'\n",
       "\\item 'EA'\n",
       "\\item 'EA'\n",
       "\\item 'EA'\n",
       "\\item 'EA'\n",
       "\\item 'EA'\n",
       "\\item 'EA'\n",
       "\\item 'EA'\n",
       "\\item 'EA'\n",
       "\\item 'EA'\n",
       "\\item 'EA'\n",
       "\\item 'EA'\n",
       "\\item 'AA'\n",
       "\\item 'AA'\n",
       "\\item 'AA'\n",
       "\\item 'AA'\n",
       "\\item 'AA'\n",
       "\\item 'AA'\n",
       "\\item 'AA'\n",
       "\\item 'AA'\n",
       "\\item 'AA'\n",
       "\\item 'AA'\n",
       "\\item 'AA'\n",
       "\\item 'AA'\n",
       "\\item 'AA'\n",
       "\\item 'AA'\n",
       "\\item 'AA'\n",
       "\\item 'AA'\n",
       "\\item 'AA'\n",
       "\\item 'AA'\n",
       "\\item 'AA'\n",
       "\\item 'AA'\n",
       "\\item 'AA'\n",
       "\\item 'AA'\n",
       "\\item 'all'\n",
       "\\item 'all'\n",
       "\\item 'all'\n",
       "\\item 'all'\n",
       "\\item 'all'\n",
       "\\item 'all'\n",
       "\\item 'all'\n",
       "\\item 'all'\n",
       "\\item 'all'\n",
       "\\item 'all'\n",
       "\\item 'all'\n",
       "\\item 'all'\n",
       "\\item 'all'\n",
       "\\item 'all'\n",
       "\\item 'all'\n",
       "\\item 'all'\n",
       "\\item 'all'\n",
       "\\item 'all'\n",
       "\\item 'all'\n",
       "\\item 'all'\n",
       "\\item 'all'\n",
       "\\item 'all'\n",
       "\\item 'EA'\n",
       "\\item 'EA'\n",
       "\\item 'EA'\n",
       "\\item 'EA'\n",
       "\\item 'EA'\n",
       "\\item 'EA'\n",
       "\\item 'EA'\n",
       "\\item 'EA'\n",
       "\\item 'EA'\n",
       "\\item 'EA'\n",
       "\\item 'EA'\n",
       "\\item 'EA'\n",
       "\\item 'EA'\n",
       "\\item 'EA'\n",
       "\\item 'EA'\n",
       "\\item 'EA'\n",
       "\\item 'EA'\n",
       "\\item 'EA'\n",
       "\\item 'EA'\n",
       "\\item 'EA'\n",
       "\\item 'EA'\n",
       "\\item 'EA'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'AA'\n",
       "2. 'AA'\n",
       "3. 'AA'\n",
       "4. 'AA'\n",
       "5. 'AA'\n",
       "6. 'AA'\n",
       "7. 'AA'\n",
       "8. 'AA'\n",
       "9. 'AA'\n",
       "10. 'AA'\n",
       "11. 'AA'\n",
       "12. 'AA'\n",
       "13. 'AA'\n",
       "14. 'AA'\n",
       "15. 'AA'\n",
       "16. 'AA'\n",
       "17. 'AA'\n",
       "18. 'AA'\n",
       "19. 'AA'\n",
       "20. 'AA'\n",
       "21. 'AA'\n",
       "22. 'AA'\n",
       "23. 'all'\n",
       "24. 'all'\n",
       "25. 'all'\n",
       "26. 'all'\n",
       "27. 'all'\n",
       "28. 'all'\n",
       "29. 'all'\n",
       "30. 'all'\n",
       "31. 'all'\n",
       "32. 'all'\n",
       "33. 'all'\n",
       "34. 'all'\n",
       "35. 'all'\n",
       "36. 'all'\n",
       "37. 'all'\n",
       "38. 'all'\n",
       "39. 'all'\n",
       "40. 'all'\n",
       "41. 'all'\n",
       "42. 'all'\n",
       "43. 'all'\n",
       "44. 'all'\n",
       "45. 'EA'\n",
       "46. 'EA'\n",
       "47. 'EA'\n",
       "48. 'EA'\n",
       "49. 'EA'\n",
       "50. 'EA'\n",
       "51. 'EA'\n",
       "52. 'EA'\n",
       "53. 'EA'\n",
       "54. 'EA'\n",
       "55. 'EA'\n",
       "56. 'EA'\n",
       "57. 'EA'\n",
       "58. 'EA'\n",
       "59. 'EA'\n",
       "60. 'EA'\n",
       "61. 'EA'\n",
       "62. 'EA'\n",
       "63. 'EA'\n",
       "64. 'EA'\n",
       "65. 'EA'\n",
       "66. 'EA'\n",
       "67. 'AA'\n",
       "68. 'AA'\n",
       "69. 'AA'\n",
       "70. 'AA'\n",
       "71. 'AA'\n",
       "72. 'AA'\n",
       "73. 'AA'\n",
       "74. 'AA'\n",
       "75. 'AA'\n",
       "76. 'AA'\n",
       "77. 'AA'\n",
       "78. 'AA'\n",
       "79. 'AA'\n",
       "80. 'AA'\n",
       "81. 'AA'\n",
       "82. 'AA'\n",
       "83. 'AA'\n",
       "84. 'AA'\n",
       "85. 'AA'\n",
       "86. 'AA'\n",
       "87. 'AA'\n",
       "88. 'AA'\n",
       "89. 'all'\n",
       "90. 'all'\n",
       "91. 'all'\n",
       "92. 'all'\n",
       "93. 'all'\n",
       "94. 'all'\n",
       "95. 'all'\n",
       "96. 'all'\n",
       "97. 'all'\n",
       "98. 'all'\n",
       "99. 'all'\n",
       "100. 'all'\n",
       "101. 'all'\n",
       "102. 'all'\n",
       "103. 'all'\n",
       "104. 'all'\n",
       "105. 'all'\n",
       "106. 'all'\n",
       "107. 'all'\n",
       "108. 'all'\n",
       "109. 'all'\n",
       "110. 'all'\n",
       "111. 'EA'\n",
       "112. 'EA'\n",
       "113. 'EA'\n",
       "114. 'EA'\n",
       "115. 'EA'\n",
       "116. 'EA'\n",
       "117. 'EA'\n",
       "118. 'EA'\n",
       "119. 'EA'\n",
       "120. 'EA'\n",
       "121. 'EA'\n",
       "122. 'EA'\n",
       "123. 'EA'\n",
       "124. 'EA'\n",
       "125. 'EA'\n",
       "126. 'EA'\n",
       "127. 'EA'\n",
       "128. 'EA'\n",
       "129. 'EA'\n",
       "130. 'EA'\n",
       "131. 'EA'\n",
       "132. 'EA'\n",
       "133. 'AA'\n",
       "134. 'AA'\n",
       "135. 'AA'\n",
       "136. 'AA'\n",
       "137. 'AA'\n",
       "138. 'AA'\n",
       "139. 'AA'\n",
       "140. 'AA'\n",
       "141. 'AA'\n",
       "142. 'AA'\n",
       "143. 'AA'\n",
       "144. 'AA'\n",
       "145. 'AA'\n",
       "146. 'AA'\n",
       "147. 'AA'\n",
       "148. 'AA'\n",
       "149. 'AA'\n",
       "150. 'AA'\n",
       "151. 'AA'\n",
       "152. 'AA'\n",
       "153. 'AA'\n",
       "154. 'AA'\n",
       "155. 'all'\n",
       "156. 'all'\n",
       "157. 'all'\n",
       "158. 'all'\n",
       "159. 'all'\n",
       "160. 'all'\n",
       "161. 'all'\n",
       "162. 'all'\n",
       "163. 'all'\n",
       "164. 'all'\n",
       "165. 'all'\n",
       "166. 'all'\n",
       "167. 'all'\n",
       "168. 'all'\n",
       "169. 'all'\n",
       "170. 'all'\n",
       "171. 'all'\n",
       "172. 'all'\n",
       "173. 'all'\n",
       "174. 'all'\n",
       "175. 'all'\n",
       "176. 'all'\n",
       "177. 'EA'\n",
       "178. 'EA'\n",
       "179. 'EA'\n",
       "180. 'EA'\n",
       "181. 'EA'\n",
       "182. 'EA'\n",
       "183. 'EA'\n",
       "184. 'EA'\n",
       "185. 'EA'\n",
       "186. 'EA'\n",
       "187. 'EA'\n",
       "188. 'EA'\n",
       "189. 'EA'\n",
       "190. 'EA'\n",
       "191. 'EA'\n",
       "192. 'EA'\n",
       "193. 'EA'\n",
       "194. 'EA'\n",
       "195. 'EA'\n",
       "196. 'EA'\n",
       "197. 'EA'\n",
       "198. 'EA'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "  [1] \"AA\"  \"AA\"  \"AA\"  \"AA\"  \"AA\"  \"AA\"  \"AA\"  \"AA\"  \"AA\"  \"AA\"  \"AA\"  \"AA\" \n",
       " [13] \"AA\"  \"AA\"  \"AA\"  \"AA\"  \"AA\"  \"AA\"  \"AA\"  \"AA\"  \"AA\"  \"AA\"  \"all\" \"all\"\n",
       " [25] \"all\" \"all\" \"all\" \"all\" \"all\" \"all\" \"all\" \"all\" \"all\" \"all\" \"all\" \"all\"\n",
       " [37] \"all\" \"all\" \"all\" \"all\" \"all\" \"all\" \"all\" \"all\" \"EA\"  \"EA\"  \"EA\"  \"EA\" \n",
       " [49] \"EA\"  \"EA\"  \"EA\"  \"EA\"  \"EA\"  \"EA\"  \"EA\"  \"EA\"  \"EA\"  \"EA\"  \"EA\"  \"EA\" \n",
       " [61] \"EA\"  \"EA\"  \"EA\"  \"EA\"  \"EA\"  \"EA\"  \"AA\"  \"AA\"  \"AA\"  \"AA\"  \"AA\"  \"AA\" \n",
       " [73] \"AA\"  \"AA\"  \"AA\"  \"AA\"  \"AA\"  \"AA\"  \"AA\"  \"AA\"  \"AA\"  \"AA\"  \"AA\"  \"AA\" \n",
       " [85] \"AA\"  \"AA\"  \"AA\"  \"AA\"  \"all\" \"all\" \"all\" \"all\" \"all\" \"all\" \"all\" \"all\"\n",
       " [97] \"all\" \"all\" \"all\" \"all\" \"all\" \"all\" \"all\" \"all\" \"all\" \"all\" \"all\" \"all\"\n",
       "[109] \"all\" \"all\" \"EA\"  \"EA\"  \"EA\"  \"EA\"  \"EA\"  \"EA\"  \"EA\"  \"EA\"  \"EA\"  \"EA\" \n",
       "[121] \"EA\"  \"EA\"  \"EA\"  \"EA\"  \"EA\"  \"EA\"  \"EA\"  \"EA\"  \"EA\"  \"EA\"  \"EA\"  \"EA\" \n",
       "[133] \"AA\"  \"AA\"  \"AA\"  \"AA\"  \"AA\"  \"AA\"  \"AA\"  \"AA\"  \"AA\"  \"AA\"  \"AA\"  \"AA\" \n",
       "[145] \"AA\"  \"AA\"  \"AA\"  \"AA\"  \"AA\"  \"AA\"  \"AA\"  \"AA\"  \"AA\"  \"AA\"  \"all\" \"all\"\n",
       "[157] \"all\" \"all\" \"all\" \"all\" \"all\" \"all\" \"all\" \"all\" \"all\" \"all\" \"all\" \"all\"\n",
       "[169] \"all\" \"all\" \"all\" \"all\" \"all\" \"all\" \"all\" \"all\" \"EA\"  \"EA\"  \"EA\"  \"EA\" \n",
       "[181] \"EA\"  \"EA\"  \"EA\"  \"EA\"  \"EA\"  \"EA\"  \"EA\"  \"EA\"  \"EA\"  \"EA\"  \"EA\"  \"EA\" \n",
       "[193] \"EA\"  \"EA\"  \"EA\"  \"EA\"  \"EA\"  \"EA\" "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "long_format$population"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "e39fe345-ebbb-4657-b901-ff4fca1bdb0b",
   "metadata": {},
   "outputs": [],
   "source": [
    "long_format$region <- str_split_fixed(long_format$methylation_data,\n",
    "                                      pattern = \"pheno/\",\n",
    "                                      2)[, 2]\n",
    "long_format$region <- str_split_fixed(long_format$region,\n",
    "                                      pattern = \"/out\",\n",
    "                                      2)[, 1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "d861e897-0d6a-4a18-876c-7e3cf741d9fd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'caud'</li><li>'caud'</li><li>'caud'</li><li>'caud'</li><li>'caud'</li><li>'caud'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'caud'\n",
       "\\item 'caud'\n",
       "\\item 'caud'\n",
       "\\item 'caud'\n",
       "\\item 'caud'\n",
       "\\item 'caud'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'caud'\n",
       "2. 'caud'\n",
       "3. 'caud'\n",
       "4. 'caud'\n",
       "5. 'caud'\n",
       "6. 'caud'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"caud\" \"caud\" \"caud\" \"caud\" \"caud\" \"caud\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(long_format$region)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "9d3b06aa-5c53-4cac-aee9-ea6be884065e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'AA_caud.csv'</li><li>'AA_dlpfc.csv'</li><li>'AA_hippo.csv'</li><li>'all_caud.csv'</li><li>'all_dlpfc.csv'</li><li>'all_hippo.csv'</li><li>'EA_caud.csv'</li><li>'EA_dlpfc.csv'</li><li>'EA_hippo.csv'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'AA\\_caud.csv'\n",
       "\\item 'AA\\_dlpfc.csv'\n",
       "\\item 'AA\\_hippo.csv'\n",
       "\\item 'all\\_caud.csv'\n",
       "\\item 'all\\_dlpfc.csv'\n",
       "\\item 'all\\_hippo.csv'\n",
       "\\item 'EA\\_caud.csv'\n",
       "\\item 'EA\\_dlpfc.csv'\n",
       "\\item 'EA\\_hippo.csv'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'AA_caud.csv'\n",
       "2. 'AA_dlpfc.csv'\n",
       "3. 'AA_hippo.csv'\n",
       "4. 'all_caud.csv'\n",
       "5. 'all_dlpfc.csv'\n",
       "6. 'all_hippo.csv'\n",
       "7. 'EA_caud.csv'\n",
       "8. 'EA_dlpfc.csv'\n",
       "9. 'EA_hippo.csv'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"AA_caud.csv\"   \"AA_dlpfc.csv\"  \"AA_hippo.csv\"  \"all_caud.csv\" \n",
       "[5] \"all_dlpfc.csv\" \"all_hippo.csv\" \"EA_caud.csv\"   \"EA_dlpfc.csv\" \n",
       "[9] \"EA_hippo.csv\" "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "list.files(\"/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "85c07eac-9ee0-437d-8a4b-0c77ada921b1",
   "metadata": {},
   "outputs": [],
   "source": [
    "long_format$cov_file <- paste0(\"/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/\",\n",
    "                               long_format$population,\n",
    "                               \"_\",\n",
    "                               long_format$region,\n",
    "                               \".csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "d20b1ae6-a8f1-4ad5-ae8e-723d354690e2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "TRUE"
      ],
      "text/latex": [
       "TRUE"
      ],
      "text/markdown": [
       "TRUE"
      ],
      "text/plain": [
       "[1] TRUE"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "file.exists(long_format$cov_file[1])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "02d6c296-a0bd-4af6-8972-60b3343fbbf6",
   "metadata": {},
   "outputs": [],
   "source": [
    "fwrite(long_format, \"09-OUT_matched_SNP_meth_cov.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "553b719a-29b2-4ef3-8c74-f54a201e3500",
   "metadata": {},
   "source": [
    "## Inspect files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "a46afeb2-e4ea-4201-9dde-9b9a5078747c",
   "metadata": {},
   "outputs": [],
   "source": [
    "files <- unique(long_format$cov_file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "f323ccde-c07e-43e5-85a2-858041d4e736",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_caud.csv'</li><li>'/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_caud.csv'</li><li>'/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_caud.csv'</li><li>'/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_dlpfc.csv'</li><li>'/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_dlpfc.csv'</li><li>'/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_dlpfc.csv'</li><li>'/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_hippo.csv'</li><li>'/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_hippo.csv'</li><li>'/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_hippo.csv'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item '/expanse/lustre/projects/jhu152/naglemi/mwas/full\\_covariates/AA\\_caud.csv'\n",
       "\\item '/expanse/lustre/projects/jhu152/naglemi/mwas/full\\_covariates/all\\_caud.csv'\n",
       "\\item '/expanse/lustre/projects/jhu152/naglemi/mwas/full\\_covariates/EA\\_caud.csv'\n",
       "\\item '/expanse/lustre/projects/jhu152/naglemi/mwas/full\\_covariates/AA\\_dlpfc.csv'\n",
       "\\item '/expanse/lustre/projects/jhu152/naglemi/mwas/full\\_covariates/all\\_dlpfc.csv'\n",
       "\\item '/expanse/lustre/projects/jhu152/naglemi/mwas/full\\_covariates/EA\\_dlpfc.csv'\n",
       "\\item '/expanse/lustre/projects/jhu152/naglemi/mwas/full\\_covariates/AA\\_hippo.csv'\n",
       "\\item '/expanse/lustre/projects/jhu152/naglemi/mwas/full\\_covariates/all\\_hippo.csv'\n",
       "\\item '/expanse/lustre/projects/jhu152/naglemi/mwas/full\\_covariates/EA\\_hippo.csv'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. '/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_caud.csv'\n",
       "2. '/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_caud.csv'\n",
       "3. '/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_caud.csv'\n",
       "4. '/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_dlpfc.csv'\n",
       "5. '/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_dlpfc.csv'\n",
       "6. '/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_dlpfc.csv'\n",
       "7. '/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_hippo.csv'\n",
       "8. '/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_hippo.csv'\n",
       "9. '/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_hippo.csv'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_caud.csv\"  \n",
       "[2] \"/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_caud.csv\" \n",
       "[3] \"/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_caud.csv\"  \n",
       "[4] \"/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_dlpfc.csv\" \n",
       "[5] \"/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_dlpfc.csv\"\n",
       "[6] \"/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_dlpfc.csv\" \n",
       "[7] \"/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_hippo.csv\" \n",
       "[8] \"/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_hippo.csv\"\n",
       "[9] \"/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_hippo.csv\" "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "fd251868-712d-42fe-b8c7-0c6679e19388",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_caud.csv\"\n",
      "         ID   genoPC1     genoPC2    genoPC3    methPC1    methPC2     methPC3\n",
      "     <char>     <num>       <num>      <num>      <num>      <num>       <num>\n",
      "  1: Br1003 0.0591231 -0.03980770  0.0205308 -157.04647   22.43150   -3.613153\n",
      "  2: Br1004 0.0507512  0.01670030 -0.0235019 -207.97793   14.56387  -48.109417\n",
      "  3: Br1007 0.0683202  0.00502359 -0.0595899  119.29338 -184.28704 -127.987516\n",
      "  4: Br1017 0.0685061 -0.00844357 -0.0109324  148.35140  -89.71558  -97.742602\n",
      "  5: Br1021 0.0596699  0.00367841  0.0479552  453.47100 -132.38714   35.349695\n",
      " ---                                                                          \n",
      "160:  Br948 0.0639954  0.00695405 -0.0100903  413.58583 -263.70724    4.226894\n",
      "161:  Br949 0.0712634 -0.02716360  0.0572070 -200.93232  265.80379   56.822449\n",
      "162:  Br963 0.0120633  0.01905100 -0.0894283  -27.92241 -144.83283  -66.482063\n",
      "163:  Br991 0.0436291  0.01022270 -0.0078481  -42.06156 -116.78523 -130.518346\n",
      "164:  Br993 0.0646887  0.01978440 -0.0513997  -19.51001 -109.27932  -76.972489\n",
      "         methPC4    methPC5     methPC6     methPC7     methPC8   methPC9\n",
      "           <num>      <num>       <num>       <num>       <num>     <num>\n",
      "  1:   28.587802 -165.83079   15.491404   27.347098   68.146346 -83.14472\n",
      "  2:  -48.368597   65.47954 -174.436205  -40.363495 -103.813378  11.08724\n",
      "  3:    8.047659  100.51821  112.404812  -44.079714  -13.331837 109.55059\n",
      "  4:    7.072870  -78.40472   77.029617  -33.334885  -34.948204  73.88633\n",
      "  5:   80.759933  123.87860  -39.433812 -122.191818   64.627124 -62.89107\n",
      " ---                                                                     \n",
      "160:   89.551899   99.99237   49.852902  -83.637729  -21.617751  60.49754\n",
      "161:   55.415802   85.21831   30.102630  -35.354101    8.835777  68.86163\n",
      "162:   -1.418709 -178.63591   22.689742   12.902745   43.021195  18.93760\n",
      "163: -121.402176  142.89865 -116.952419   42.139084  -77.185461 -27.19868\n",
      "164:  -29.740514  -91.02831   -3.745881   -8.245914  -86.432950  20.17579\n",
      "       methPC10      Dx   Age    Sex\n",
      "          <num>  <char> <num> <char>\n",
      "  1:   74.40552 Control 52.57      F\n",
      "  2:  -84.67051 Control 21.01      M\n",
      "  3:   55.89200 Control 57.10      M\n",
      "  4:   65.62252 Control 48.42      F\n",
      "  5:  -25.93989     SCZ 63.18      M\n",
      " ---                                \n",
      "160:  -77.51143 Control 45.91      M\n",
      "161: -100.62874 Control 18.05      M\n",
      "162:   25.76437     SCZ 62.63      F\n",
      "163:  -91.46101 Control 35.24      M\n",
      "164:   54.41147 Control 30.48      F\n",
      "\n",
      "\n",
      "[1] \"/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_caud.csv\"\n",
      "         ID    genoPC1     genoPC2    genoPC3     methPC1     methPC2\n",
      "     <char>      <num>       <num>      <num>       <num>       <num>\n",
      "  1: Br1003  0.0591231 -0.03980770  0.0205308  143.209364  128.591799\n",
      "  2: Br1004  0.0507512  0.01670030 -0.0235019  194.371327   81.154099\n",
      "  3: Br1007  0.0683202  0.00502359 -0.0595899 -123.938202  154.705592\n",
      "  4: Br1016 -0.0639844 -0.04362820  0.0334612  364.038425 -227.750105\n",
      "  5: Br1017  0.0685061 -0.00844357 -0.0109324 -155.012135  131.998283\n",
      " ---                                                                 \n",
      "293:  Br949  0.0712634 -0.02716360  0.0572070  185.578863    9.321499\n",
      "294:  Br963  0.0120633  0.01905100 -0.0894283   18.242075  138.563662\n",
      "295:  Br983 -0.0632810  0.05693800  0.0498407  161.899115  -69.416598\n",
      "296:  Br991  0.0436291  0.01022270 -0.0078481   33.524233   88.576510\n",
      "297:  Br993  0.0646887  0.01978440 -0.0513997    9.149205  162.418650\n",
      "        methPC3      methPC4    methPC5    methPC6   methPC7    methPC8\n",
      "          <num>        <num>      <num>      <num>     <num>      <num>\n",
      "  1:  -88.15445 -113.8687194 -19.873129 -117.74383 -41.25133  52.943663\n",
      "  2:  -40.36260  -94.5687221  83.616459   33.95542 160.87447 -32.969432\n",
      "  3:  157.18145 -119.7367842 -17.233759  111.03051 -67.78135 -56.681392\n",
      "  4:   70.33994   -0.3303182  72.028156   27.09302 188.58356 -44.239040\n",
      "  5:   56.22611 -153.6865000  -8.949595  -24.21156 -70.99105 -86.086464\n",
      " ---                                                                   \n",
      "293: -284.85512 -124.8564948 -43.044284  166.83053  36.32648  67.345472\n",
      "294:   97.43856  -38.3780078 -27.275614 -144.47756 -87.38696  34.348382\n",
      "295:  132.52277   88.8594729 -63.825596   41.82027  64.48729  -5.826039\n",
      "296:  131.64191  -73.4129952 155.392517   67.00646 105.92564 -20.286185\n",
      "297:   54.86754 -115.6506241  22.101912  -41.49506 -27.70562 -54.334400\n",
      "        methPC9    methPC10      Dx   Age    Sex\n",
      "          <num>       <num>  <char> <num> <char>\n",
      "  1: -50.511411   63.607456 Control 52.57      F\n",
      "  2:  57.579260 -105.135625 Control 21.01      M\n",
      "  3: -48.163955   40.716193 Control 57.10      M\n",
      "  4:  36.884147  -50.579984     SCZ 20.22      M\n",
      "  5: -55.229823  -69.932589 Control 48.42      F\n",
      " ---                                            \n",
      "293:  -7.295453 -110.519839 Control 18.05      M\n",
      "294: -65.159030  -76.230717     SCZ 62.63      F\n",
      "295: -19.926292  -17.980898     SCZ 43.38      M\n",
      "296:  88.397023    5.188198 Control 35.24      M\n",
      "297: -28.847538  -60.373896 Control 30.48      F\n",
      "\n",
      "\n",
      "[1] \"/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_caud.csv\"\n",
      "         ID    genoPC1     genoPC2     genoPC3    methPC1    methPC2\n",
      "     <char>      <num>       <num>       <num>      <num>      <num>\n",
      "  1: Br1016 -0.0639844 -0.04362820  0.03346120 -362.98061  -45.12000\n",
      "  2: Br1023 -0.0545722 -0.32745900 -0.03046360  375.80159  268.63608\n",
      "  3: Br1033 -0.0632346 -0.00120511  0.00466806  -29.33084 -256.66808\n",
      "  4: Br1034 -0.0631741 -0.00447357  0.01718670   56.85657  102.79692\n",
      "  5: Br1092 -0.0625556  0.00244388  0.01672360  668.32882   88.05402\n",
      " ---                                                                \n",
      "129: Br5422 -0.0634322  0.02188450 -0.00751211 -189.26668 -389.74008\n",
      "130: Br5488 -0.0630627  0.01326140  0.01109260 -139.86621 -272.29997\n",
      "131: Br5584 -0.0622708  0.03558350  0.01842430  614.51488  105.05376\n",
      "132: Br5590 -0.0623503  0.02232040  0.01516640  167.13790 -469.34565\n",
      "133:  Br983 -0.0632810  0.05693800  0.04984070 -152.57606   87.87503\n",
      "         methPC3    methPC4    methPC5    methPC6    methPC7    methPC8\n",
      "           <num>      <num>      <num>      <num>      <num>      <num>\n",
      "  1:  113.748708  -99.14896  -28.24952  209.58378 -108.19134   18.31989\n",
      "  2:    1.834617   56.97634  -66.99901  -31.53365  -12.25377  -54.31259\n",
      "  3:   66.793944  -68.22094 -149.17240   89.20109  157.02823   21.81519\n",
      "  4:  -72.870736   90.07828   51.51396  240.37175   92.46728  -46.89330\n",
      "  5:   31.415066   84.15608   19.30624   81.39193 -195.06984  -24.47723\n",
      " ---                                                                   \n",
      "129:  -58.435708  -98.54374  -54.25851 -394.23302   16.70346  -52.80071\n",
      "130:   24.916181  -54.81268  -84.67826   23.18821  -34.69417   17.46165\n",
      "131:   52.900792  146.87375  -62.89305   33.71024  -12.19326  -13.19995\n",
      "132: -196.539204 -250.83733   48.29170  -85.22061  308.62688 -812.00649\n",
      "133:  -45.050627   51.27018  -74.06954   78.85572  -32.53474   16.65463\n",
      "        methPC9   methPC10      Dx   Age    Sex\n",
      "          <num>      <num>  <char> <num> <char>\n",
      "  1:   32.70283   17.19026     SCZ 20.22      M\n",
      "  2:  -59.79824   62.90355 Control 63.00      M\n",
      "  3:   47.29677   89.27475 Control 26.51      F\n",
      "  4:  167.12336  -25.76810 Control 57.60      M\n",
      "  5:  -71.81823  133.07974 Control 48.95      M\n",
      " ---                                           \n",
      "129:  297.31001  194.77587 Control 46.50      M\n",
      "130:   63.47559 -101.16536     SCZ 33.36      M\n",
      "131:  -81.90924 -131.36285 Control 64.55      M\n",
      "132: -348.84082 -128.43221     SCZ 51.95      F\n",
      "133:  -12.75479   85.91127     SCZ 43.38      M\n",
      "\n",
      "\n",
      "[1] \"/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_dlpfc.csv\"\n",
      "         ID   genoPC1     genoPC2     genoPC3     methPC1    methPC2\n",
      "     <char>     <num>       <num>       <num>       <num>      <num>\n",
      "  1: Br1003 0.0649632  0.07491190 -0.12977100 -1080.75747 -183.50828\n",
      "  2: Br1004 0.0521374  0.00724347 -0.14008400   117.24850  149.10851\n",
      "  3: Br1007 0.0778691  0.04240600  0.01276880  -263.01460  347.66967\n",
      "  4: Br1017 0.0791192  0.08267400  0.04123100    37.93715  242.18930\n",
      "  5: Br1021 0.0648035  0.08840540  0.05633170  -200.32431  261.42153\n",
      " ---                                                                \n",
      "107: Br2053 0.0621001 -0.06673570 -0.04526590   269.48427  112.59525\n",
      "108: Br2064 0.0180528 -0.08424130 -0.00597791   337.14428  112.38708\n",
      "109: Br2074 0.0687298 -0.01040090 -0.01942440   381.57927  117.86133\n",
      "110: Br2076 0.0378210 -0.08667300  0.03923520   570.60024   21.19848\n",
      "111: Br2263 0.0585964  0.09701510  0.08510750    97.83190  218.34212\n",
      "         methPC3    methPC4      methPC5    methPC6    methPC7     methPC8\n",
      "           <num>      <num>        <num>      <num>      <num>       <num>\n",
      "  1:   43.997443  -23.78873   73.3547148  -65.13058   68.97964  -76.186697\n",
      "  2:  210.480053  116.41197  -18.6921154   -7.44167   58.69192   92.345865\n",
      "  3: -151.630658 -155.46098 -161.9570809   71.83581  -16.46575   -4.039417\n",
      "  4: -255.331803 -169.15786  -30.6088998 -160.60365 -119.68724 -203.422434\n",
      "  5:  159.267846  138.19444  149.5032646  159.26241   27.37276  -15.746279\n",
      " ---                                                                      \n",
      "107:   -4.301287   28.83286   15.2701569  112.98358   55.32636  -49.846509\n",
      "108: -129.266661  -72.27720  -49.9386708   37.12828  105.05280   58.891251\n",
      "109: -152.076204  -83.10759   -0.2539756   55.46671  110.84727  -23.714433\n",
      "110: -276.534292 -219.81808 -143.9289858   78.10422 -274.28639  158.548798\n",
      "111: -123.082327 -123.38397 -164.2116923 -155.37632  -17.70599  -56.663012\n",
      "        methPC9   methPC10      Dx   Age    Sex\n",
      "          <num>      <num>  <char> <num> <char>\n",
      "  1:   84.14746   35.76471 Control 52.57      F\n",
      "  2:   57.05211   13.87637 Control 21.01      M\n",
      "  3:   20.07848  -83.17815 Control 57.10      M\n",
      "  4:   98.54120 -202.96072 Control 48.42      F\n",
      "  5:   33.30777   32.49216     SCZ 63.18      M\n",
      " ---                                           \n",
      "107:   10.61061  -72.10424 Control 65.58      M\n",
      "108:   67.59786   40.37271 Control 40.24      M\n",
      "109:  -19.80550   34.75501 Control 44.82      M\n",
      "110: -192.24843  -49.86376 Control 56.48      M\n",
      "111:  -31.10210  176.22890 Control 40.31      F\n",
      "\n",
      "\n",
      "[1] \"/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_dlpfc.csv\"\n",
      "         ID    genoPC1     genoPC2     genoPC3     methPC1    methPC2   methPC3\n",
      "     <char>      <num>       <num>       <num>       <num>      <num>     <num>\n",
      "  1: Br1003  0.0649632  0.07491190 -0.12977100 -1063.85637 -178.16658 -122.6163\n",
      "  2: Br1004  0.0521374  0.00724347 -0.14008400   124.24599  145.08204  -45.8196\n",
      "  3: Br1007  0.0778691  0.04240600  0.01276880  -242.20924  337.35835 -102.6885\n",
      "  4: Br1016 -0.1131540  0.02141440  0.01341990   210.62733 -677.01118  234.9817\n",
      "  5: Br1017  0.0791192  0.08267400  0.04123100    56.75471  237.67957 -196.4897\n",
      " ---                                                                           \n",
      "159: Br2263  0.0585964  0.09701510  0.08510750   114.10469  210.87989 -105.2660\n",
      "160: Br2271 -0.1134390  0.00287223  0.02055880   332.45815  112.26493  211.5714\n",
      "161: Br2285 -0.1117670 -0.05757400  0.00189044   257.01609  149.96425  197.1481\n",
      "162: Br2288 -0.1117880  0.00246579  0.02149540   385.29869   36.14148  204.0244\n",
      "163: Br2292 -0.1132940 -0.00214188  0.03040180   223.89452  137.53033  233.1251\n",
      "         methPC4     methPC5     methPC6     methPC7    methPC8     methPC9\n",
      "           <num>       <num>       <num>       <num>      <num>       <num>\n",
      "  1:   41.019397  -15.628191   25.712962  -93.254520  34.981711   -1.678003\n",
      "  2:  236.922395  -23.319253  -78.783541  -27.489030 -18.863886   60.377535\n",
      "  3: -180.217203 -180.319796  -27.880320   74.632757 -31.780362    5.828085\n",
      "  4:    2.795834   -4.344111  -38.000925  -56.424583 -42.914173  -90.919412\n",
      "  5: -245.901087 -103.081975   51.141895   -7.924028 136.443338 -115.072485\n",
      " ---                                                                       \n",
      "159: -129.659558 -177.636468  -64.658767  -61.233364 114.832854  -36.695199\n",
      "160: -149.401256  -34.416805   19.141696  161.676819 -35.340706  -32.887638\n",
      "161:  -89.960365   29.605450   -6.285973  -41.587304 -60.969489   30.871495\n",
      "162:   -5.713373   28.223047  -16.686731    0.579782  -3.752676   -1.716414\n",
      "163:  -25.201612 -128.562878 -178.440875 -269.096274  78.524488   13.494552\n",
      "       methPC10      Dx   Age    Sex\n",
      "          <num>  <char> <num> <char>\n",
      "  1: 133.010687 Control 52.57      F\n",
      "  2:  -6.303476 Control 21.01      M\n",
      "  3: -34.318495 Control 57.10      M\n",
      "  4:  22.720535     SCZ 20.22      M\n",
      "  5: 124.219645 Control 48.42      F\n",
      " ---                                \n",
      "159:  24.114723 Control 40.31      F\n",
      "160:  20.179643     SCZ 50.80      M\n",
      "161:  71.441875 Control 53.29      M\n",
      "162:  67.355583 Control 44.70      M\n",
      "163:  37.479896     SCZ 20.34      M\n",
      "\n",
      "\n",
      "[1] \"/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_dlpfc.csv\"\n",
      "        ID    genoPC1      genoPC2      genoPC3      methPC1     methPC2\n",
      "    <char>      <num>        <num>        <num>        <num>       <num>\n",
      " 1: Br1016 -0.1131540  0.021414400  0.013419900 -240.6170337  716.996486\n",
      " 2: Br1023 -0.1004800  0.223684000 -0.069715200 -214.6398790  -95.602467\n",
      " 3: Br1034 -0.1127480 -0.000420085 -0.032424500  -12.7597732  608.189028\n",
      " 4: Br1093 -0.1121340 -0.001252560  0.032796100   68.3422717 -247.580861\n",
      " 5: Br1214 -0.1120750 -0.027417000  0.027889200 -413.5490180  -55.239519\n",
      " 6: Br1245 -0.1029510  0.053666100 -0.033747200  -90.3494382 -130.890717\n",
      " 7: Br1279 -0.1112940  0.005234130 -0.000853104  152.0150695  588.503930\n",
      " 8: Br1339 -0.1132870  0.010877300  0.014949200 1227.4138317 -769.918025\n",
      " 9: Br1350 -0.1063070  0.030046700 -0.021749700   63.7420269 -189.230398\n",
      "10: Br1370 -0.1139960 -0.024187300  0.025577800  557.5978258 -429.070832\n",
      "11: Br1378 -0.1082100 -0.009348570 -0.017150900  174.8536946 -276.420318\n",
      "12: Br1383 -0.1118790 -0.017615800 -0.009568290   53.7263291 -235.931535\n",
      "13: Br1389 -0.1068190  0.007721730 -0.027255200 -441.1937579    3.755752\n",
      "14: Br1416 -0.1122680 -0.022115700  0.022651700 -732.8184687  127.442333\n",
      "15: Br1425 -0.1141170 -0.043226300 -0.035242900  160.4554499 -235.111010\n",
      "16: Br1446 -0.1087730  0.027536400 -0.047564300   41.5894562 -296.444779\n",
      "17: Br1453 -0.0955063  0.045087300  0.013898000 -144.5635759 -133.842659\n",
      "18: Br1463 -0.1109110 -0.030383200  0.022017300 -514.7207076   50.845581\n",
      "19: Br1497 -0.1074600  0.025419600  0.008188180   29.5632898  615.923338\n",
      "20: Br1498 -0.1119290 -0.036150700  0.032639300 -326.4666560 -181.500294\n",
      "21: Br1499 -0.1101320  0.000941522 -0.002120940    0.1417225 -239.225002\n",
      "22: Br1529 -0.1132830 -0.076525900 -0.030374100  315.6824331  492.165617\n",
      "23: Br1565 -0.1115410 -0.029686400  0.003242870 1065.3944071 -690.755394\n",
      "24: Br1568 -0.1129450  0.005383120 -0.005288100 -159.2745990 -135.995659\n",
      "25: Br1578 -0.1127770 -0.040505100  0.038831100 -441.1626765  -33.032212\n",
      "26: Br1581 -0.1106110 -0.049336800 -0.012629600 1141.0776375  149.335005\n",
      "27: Br1594 -0.1127870 -0.037553900  0.014319600 -337.2880387 -127.323440\n",
      "28: Br1602 -0.1129940 -0.065389400  0.002065240 -357.9079883 -128.594724\n",
      "29: Br1637 -0.1132100  0.027626600  0.004689860  384.3917661  482.811006\n",
      "30: Br1648 -0.1119030 -0.007235530  0.019586700   73.0007555 -217.614710\n",
      "31: Br1654 -0.1036780  0.256365000 -0.130772000  339.7603783 -372.315017\n",
      "32: Br1669 -0.1117640 -0.065826900  0.017098500  581.0999258  381.667040\n",
      "33: Br1676 -0.1043850  0.195621000 -0.042725700  319.8230181  492.874692\n",
      "34: Br1678 -0.1124520 -0.026777500  0.006011970  706.7192937  334.307052\n",
      "35: Br1738 -0.1137770 -0.012229300 -0.021233900 -327.8245501  -44.948508\n",
      "36: Br1753 -0.1129660 -0.027554000  0.043013300 -329.2165098  -76.662867\n",
      "37: Br1761 -0.1155610 -0.037423000  0.016387200 -210.0041665 -174.255585\n",
      "38: Br1764 -0.1104700 -0.027194700 -0.030860000  485.2271468  418.678692\n",
      "39: Br1880 -0.1107490 -0.015736700  0.017495300 -162.1260222 -207.054415\n",
      "40: Br1936 -0.1118660 -0.028052300 -0.019938300  815.2848401  275.699203\n",
      "41: Br1946 -0.1128160 -0.013788900 -0.017039500   88.7814737  600.533730\n",
      "42: Br1958 -0.1115910  0.006644430  0.014573200 -757.2844460   75.788053\n",
      "43: Br1960 -0.1115080 -0.012101700  0.030657400  215.0421072  542.217990\n",
      "44: Br2009 -0.1110700 -0.000812338 -0.021688900 -408.9504445 -107.524305\n",
      "45: Br2044 -0.1108720 -0.060961900  0.012569200 -297.7280701 -171.397224\n",
      "46: Br2084 -0.1056440  0.027108600 -0.041478300 -317.8696335 -170.338509\n",
      "47: Br2242 -0.1129820 -0.024477900 -0.003141490 -134.9505331 -210.773953\n",
      "48: Br2257 -0.1037410  0.225214000 -0.028392400 -329.2817354 -149.298062\n",
      "49: Br2271 -0.1134390  0.002872230  0.020558800 -382.2355018 -105.636879\n",
      "50: Br2285 -0.1117670 -0.057574000  0.001890440 -294.9701667 -151.218487\n",
      "51: Br2288 -0.1117880  0.002465790  0.021495400 -423.8799276  -31.109076\n",
      "52: Br2292 -0.1132940 -0.002141880  0.030401800 -257.0928328 -135.877084\n",
      "        ID    genoPC1      genoPC2      genoPC3      methPC1     methPC2\n",
      "        methPC3     methPC4     methPC5       methPC6     methPC7      methPC8\n",
      "          <num>       <num>       <num>         <num>       <num>        <num>\n",
      " 1:  -43.275506  -65.613340   47.654509  -40.73966536    4.554927  -67.8646220\n",
      " 2: -277.067015   28.361260  -91.094486 -101.33762301  -13.220663  117.2267183\n",
      " 3:    6.664371   70.906831  -24.208597  -19.05397322  -63.675807  -61.4843334\n",
      " 4:  -89.600457  145.968871  -18.340009   -0.03581199   24.965400   12.7789248\n",
      " 5:  -60.934469  131.393273   44.329258   11.79189818   86.834125   86.8917433\n",
      " 6: -186.443577  -81.885887  -31.572190   -3.59645598  -46.717735   54.7781547\n",
      " 7:  -63.668438 -167.038301  137.586099  162.09500437  -60.464441   40.9155463\n",
      " 8:  -18.113917   21.407917   16.136609  233.27613947    5.385740  -78.6112788\n",
      " 9: -214.137087  -83.453867  -99.651341  -23.34071914   42.917244   83.7249845\n",
      "10: -189.724615  -99.549760  -61.405018   63.72331492   -8.240192   74.4189080\n",
      "11: -109.219423   46.318224   16.030665   12.48780209  -21.662339  -50.6254091\n",
      "12:  210.349127 -188.440140 -374.553755 -132.44915656   54.992959 -172.1423626\n",
      "13: -257.944887  -71.093071   51.349469   53.67993213   36.918612   60.5666942\n",
      "14: -261.755254  -10.713915  134.947824   -6.69589311    6.971228  -23.4676232\n",
      "15: -158.998989   36.671390  -89.277486  -23.73316462  -49.564167  -86.9215960\n",
      "16:  224.872046 -119.932512  106.580708  -78.28946628 -106.532928 -171.0267916\n",
      "17: -238.670088  -54.979254   38.869424   -8.13309744    9.032989   15.1171467\n",
      "18: -255.589014   82.795458   33.687387  -38.39887739  -17.987924  -25.6996866\n",
      "19:  -12.936169  151.891114    3.776715   56.39185944  -58.844494   47.3533957\n",
      "20:  188.799015  597.572035  274.755391   -2.62709167  -42.461945   57.1273567\n",
      "21:  -28.382705  136.631553  -26.386355  -42.34192610  -24.196018  -19.7119714\n",
      "22:   14.281884   70.076176  -51.869970   18.95059305  -43.806410   25.5431831\n",
      "23: -190.566745   73.883196  237.582116  -66.00394280  -40.309404   67.7859892\n",
      "24:  -58.788862    6.577189   28.997793   -9.29621349   33.745328  -42.3320344\n",
      "25:  -28.089668 -172.649575  -83.551795  -28.41753884  -24.876538   -4.5984118\n",
      "26:   60.545783   88.280908  129.482792  123.43059954  -40.618985   60.3554269\n",
      "27:  189.168305  211.583819  -64.764384  130.89905502  -12.447914  -36.6495569\n",
      "28:  -39.660737  243.384697  -16.385931  -92.92773850  -27.084039 -111.5298923\n",
      "29:  -22.167546  -67.031891  -40.125817   63.85819000  -25.287682   28.3170947\n",
      "30:  -80.524383    2.253366  -95.951945  -97.00992936  -61.001121  -99.7620711\n",
      "31:   94.635907   23.625653 -215.408059 -281.16518071  -91.936928  161.8553105\n",
      "32:  107.326649  153.129220  -47.579193   38.85433447   -3.985338  -87.0956813\n",
      "33:   17.114114  117.618778  -76.336123   83.40527842  -47.317596   96.0118784\n",
      "34:   67.372197  -79.760904   19.511359   54.92777760   37.115362   -2.2521055\n",
      "35: -183.287650   53.492754  -55.396279  -19.61106676  -31.733929   -0.6789288\n",
      "36:  -93.408635  -12.919710  -93.480712  -57.65430193  -31.758340 -124.4902826\n",
      "37:   54.265887 -177.159164 -125.362174   43.18230066   77.858955 -136.6737680\n",
      "38:   35.626427  107.484206  -63.691733  -41.70529234   32.204175   20.2513038\n",
      "39:  229.897826 -393.759858  208.803657  249.60785499  -10.424133  256.4626697\n",
      "40:  160.798932 -214.998426  152.964284 -463.59193753  318.780336  -34.4031869\n",
      "41:  -31.146747  -69.750647  -52.466615   41.05063329  -49.224966  -37.8141400\n",
      "42:   62.031562  -46.703400  212.052500   34.34523648  102.077682  -29.8954291\n",
      "43:    3.965522  -25.389348  -77.773839  -39.79915787   16.776981  -14.7652309\n",
      "44:  377.929880  -79.374014  274.381464 -400.65432983 -406.512938  224.5272521\n",
      "45:  302.323720  140.851235  122.367009   54.62936587  561.296580 -215.4576558\n",
      "46:  355.236666   -5.504654 -274.231643  312.94998679 -266.946255    6.0718205\n",
      "47:  236.871107  -30.893242   70.367647  141.52554596 -125.347375 -251.1697342\n",
      "48:  170.238844   11.786428 -175.570218  103.47915049  331.044393  606.9153728\n",
      "49:  152.671077  -43.742430 -270.808697 -104.80289904    8.733961  -58.0711925\n",
      "50:    2.149804  -44.483663   35.168781  -21.33030570   29.673075   18.5010379\n",
      "51:  -78.848884  -26.408089  -16.922543   14.75469642   50.849335  -35.0548328\n",
      "52:  -52.185184 -320.716490  316.783448  141.44620694  -18.540842 -143.2481031\n",
      "        methPC3     methPC4     methPC5       methPC6     methPC7      methPC8\n",
      "        methPC9    methPC10      Dx   Age    Sex\n",
      "          <num>       <num>  <char> <num> <char>\n",
      " 1:   24.234744  -57.503466     SCZ 20.22      M\n",
      " 2:  -51.194671   -3.371285 Control 63.00      M\n",
      " 3:    7.424395  -43.309677 Control 57.60      M\n",
      " 4:  -93.194492   65.578568     SCZ 71.81      F\n",
      " 5:  -92.341174    1.975586 Control 61.13      M\n",
      " 6:  -21.414496   19.186623     SCZ 42.62      M\n",
      " 7:  100.247988   10.726255 Control 18.41      M\n",
      " 8: -197.155872  288.386946     SCZ 60.08      F\n",
      " 9:  -75.695893   -7.582738 Control 24.61      M\n",
      "10:   93.185760  -50.201452     SCZ 30.86      M\n",
      "11:   58.355413   57.515323 Control 41.01      M\n",
      "12:  -67.172493   19.080850     SCZ 58.96      M\n",
      "13:  -31.766054  -15.824194 Control 28.35      M\n",
      "14:  -84.378214  -63.891029     SCZ 37.16      M\n",
      "15:    5.452471  -49.816186     SCZ 48.08      M\n",
      "16:  130.810012  115.632469 Control 34.44      M\n",
      "17:  -53.553379   -9.709870     SCZ 32.01      F\n",
      "18:   25.307936  -42.933963 Control 49.53      M\n",
      "19:   80.764320  -20.956058     SCZ 43.96      M\n",
      "20: -126.037398 -166.542547     SCZ 92.10      F\n",
      "21:  -26.638234   54.120127     SCZ 96.92      F\n",
      "22:   35.660919   57.541525     SCZ 70.24      M\n",
      "23:   12.726563 -236.095015 Control 51.00      M\n",
      "24:   16.274720  -12.889498 Control 39.64      M\n",
      "25:  -11.164945    9.687507     SCZ 42.06      M\n",
      "26:   33.862715  -28.278868 Control 46.08      M\n",
      "27:  248.452181  145.345899     SCZ 68.97      M\n",
      "28:  -27.474962   43.202762     SCZ 83.14      F\n",
      "29:   31.496061   24.880063     SCZ 46.19      M\n",
      "30:  -22.778903   70.199076     SCZ 48.09      F\n",
      "31:  567.923185 -174.276095 Control 62.27      M\n",
      "32:   33.478066   46.654840 Control 83.59      F\n",
      "33:  -23.581813   21.949575     SCZ 68.00      M\n",
      "34:   61.654785   40.324655     SCZ 42.71      M\n",
      "35:  -16.406459  -53.402134     SCZ 55.68      M\n",
      "36:   41.676702   -3.449391 Control 48.99      F\n",
      "37:  -70.250898  -71.160863 Control 30.01      M\n",
      "38:  -72.206612  -30.431116 Control 56.05      M\n",
      "39:   19.268795 -342.807603 Control 18.72      M\n",
      "40: -333.400724  -37.765361 Control 52.49      M\n",
      "41:    4.498615   10.840847     SCZ 46.53      F\n",
      "42:   16.267194  256.769105     SCZ 41.09      F\n",
      "43:    1.266151   -7.799876 Control 55.51      M\n",
      "44:  -81.576149  135.917481     SCZ 55.68      M\n",
      "45:  174.622150 -229.143586 Control 55.91      M\n",
      "46: -326.556933 -316.995390 Control 48.07      M\n",
      "47:   19.057632  271.119273     SCZ 50.73      F\n",
      "48:   -4.828014  323.613548 Control 59.22      M\n",
      "49:  -30.716847  -54.531828     SCZ 50.80      M\n",
      "50:  -44.974544  -35.270885 Control 53.29      M\n",
      "51:  -34.388473   -3.715787 Control 44.70      M\n",
      "52:  176.879173   79.406857     SCZ 20.34      M\n",
      "        methPC9    methPC10      Dx   Age    Sex\n",
      "\n",
      "\n",
      "[1] \"/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/AA_hippo.csv\"\n",
      "         ID   genoPC1     genoPC2     genoPC3    methPC1    methPC2     methPC3\n",
      "     <char>     <num>       <num>       <num>      <num>      <num>       <num>\n",
      "  1: Br1003 0.0671444 -0.03723470  0.07277280 -750.33742   32.21923  -38.616053\n",
      "  2: Br1004 0.0549911 -0.01358010  0.05291970   59.12224 -155.71734   92.497935\n",
      "  3: Br1007 0.0784743 -0.00547966  0.10094100 -120.64815 -406.33444  -56.429329\n",
      "  4: Br1017 0.0790859 -0.02720780  0.06569900  609.35078  252.23812  318.144211\n",
      "  5: Br1021 0.0663847 -0.00491380  0.08978490  173.60171 -200.20486   -2.157365\n",
      " ---                                                                           \n",
      "112: Br2046 0.0632065 -0.03789640  0.03100930   78.67129 -223.70830 -198.697269\n",
      "113: Br2064 0.0222938  0.02400700 -0.11872300  309.29996   -2.49657   99.857024\n",
      "114: Br2074 0.0698603  0.01633750 -0.00164279  444.77498  158.33634 -309.096299\n",
      "115: Br2076 0.0409088  0.01335690 -0.09504010  613.58934  393.87173 -657.024069\n",
      "116: Br2263 0.0604473  0.01276770  0.06225040 -464.10697 -772.04444 -264.301631\n",
      "        methPC4     methPC5    methPC6    methPC7    methPC8    methPC9\n",
      "          <num>       <num>      <num>      <num>      <num>      <num>\n",
      "  1:   90.63044  -57.256359   12.00985  -49.42859   37.26774  -32.40091\n",
      "  2: -226.00309 -141.920945  -31.08281  -28.55716   18.93465  -40.14265\n",
      "  3: -122.36312  140.126551  -88.62768  -84.10880  113.29933   95.50827\n",
      "  4:  432.98565 -302.300736  182.67443   45.52894  -83.53054  166.92437\n",
      "  5:  135.52729   66.616546 -104.18816  -60.47998  -55.13264 -227.85149\n",
      " ---                                                                   \n",
      "112:  100.80465   55.291361  -71.96352  -73.97691 -121.09078  -67.25683\n",
      "113:  -41.95379   59.319352  -31.83172  -38.65752  -32.40069  -61.08129\n",
      "114:   24.27494  168.689447  -57.19434  -33.51027  -93.16363  -56.01161\n",
      "115:   31.63707  149.477426   93.27688 -205.68590  181.75919  132.65194\n",
      "116: -162.25856   -3.563789  -21.92140  -74.30818  166.81708  -39.01282\n",
      "        methPC10      Dx   Age    Sex\n",
      "           <num>  <char> <num> <char>\n",
      "  1:  107.757255 Control 52.57      F\n",
      "  2:  -81.371943 Control 21.01      M\n",
      "  3:   21.781729 Control 57.10      M\n",
      "  4:   78.149548 Control 48.42      F\n",
      "  5:  121.940050     SCZ 63.18      M\n",
      " ---                                 \n",
      "112: -228.704459 Control 44.72      M\n",
      "113:  -51.803343 Control 40.24      M\n",
      "114:    2.774008 Control 44.82      M\n",
      "115:  -45.296585 Control 56.48      M\n",
      "116:   23.570366 Control 40.31      F\n",
      "\n",
      "\n",
      "[1] \"/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/all_hippo.csv\"\n",
      "         ID    genoPC1     genoPC2     genoPC3    methPC1     methPC2\n",
      "     <char>      <num>       <num>       <num>      <num>       <num>\n",
      "  1: Br1003  0.0671444 -0.03723470 0.072772800 -741.52347   -7.633219\n",
      "  2: Br1004  0.0549911 -0.01358010 0.052919700   61.36371 -151.938629\n",
      "  3: Br1007  0.0784743 -0.00547966 0.100941000  -94.95856 -398.668461\n",
      "  4: Br1016 -0.1027300 -0.00787423 0.002258500 -463.22801  354.857300\n",
      "  5: Br1017  0.0790859 -0.02720780 0.065699000  552.23559  258.134883\n",
      " ---                                                                 \n",
      "174: Br2263  0.0604473  0.01276770 0.062250400 -405.33756 -770.778364\n",
      "175: Br2271 -0.1021800  0.05378080 0.044594500   57.44792 -258.807717\n",
      "176: Br2285 -0.1009750  0.07137340 0.000263522  451.89056  148.652270\n",
      "177: Br2288 -0.1012690  0.01869000 0.005307350  243.90711  -53.972998\n",
      "178: Br2292 -0.1018720  0.03922100 0.005475250  -76.54187 -404.828130\n",
      "        methPC3    methPC4    methPC5     methPC6    methPC7     methPC8\n",
      "          <num>      <num>      <num>       <num>      <num>       <num>\n",
      "  1:  106.26609  -94.38499   62.03134  -37.796088    9.62500  -61.623860\n",
      "  2:  -24.98168  185.30929  142.90778 -181.925481   65.08744   -5.219542\n",
      "  3:   85.83211   93.15735  121.97178  132.153691  140.77204  -38.120997\n",
      "  4: -139.94406  165.25847 -168.28460    8.663533  190.68687   69.134257\n",
      "  5: -278.54876 -434.75292  105.41046 -316.855047  -72.22147    7.598522\n",
      " ---                                                                    \n",
      "174:  304.97921  131.76085   96.83040   21.528228   80.93795  -54.020848\n",
      "175:  -23.62020  123.52510 -170.25673  233.900244  163.83693   32.625060\n",
      "176: -263.52528   55.08424 -101.27832  213.277769 -160.81250 -137.987547\n",
      "177: -168.04429  106.55103 -123.00738  101.513211  -13.04860  -55.378786\n",
      "178:  139.06917  205.03758 -178.45426  174.472722  193.73576  -34.534556\n",
      "         methPC9   methPC10      Dx   Age    Sex\n",
      "           <num>      <num>  <char> <num> <char>\n",
      "  1:    5.881620  -12.24595 Control 52.57      F\n",
      "  2:  -26.188442   40.76255 Control 21.01      M\n",
      "  3:  -41.247657 -100.17968 Control 57.10      M\n",
      "  4:   25.225274  -44.47614     SCZ 20.22      M\n",
      "  5:  135.869066 -182.25286 Control 48.42      F\n",
      " ---                                            \n",
      "174: -116.648372  -11.72287 Control 40.31      F\n",
      "175:  -69.180598   49.87293     SCZ 50.80      M\n",
      "176:  130.236008  122.94180 Control 53.29      M\n",
      "177:   -7.563657   12.94201 Control 44.70      M\n",
      "178:    6.874370   28.55279     SCZ 20.34      M\n",
      "\n",
      "\n",
      "[1] \"/expanse/lustre/projects/jhu152/naglemi/mwas/full_covariates/EA_hippo.csv\"\n",
      "        ID    genoPC1     genoPC2      genoPC3    methPC1      methPC2\n",
      "    <char>      <num>       <num>        <num>      <num>        <num>\n",
      " 1: Br1016 -0.1027300 -0.00787423  0.002258500 -582.25041  -195.066376\n",
      " 2: Br1023 -0.0906693 -0.33101200  0.001277980  182.96059   -84.161823\n",
      " 3: Br1034 -0.1018990  0.01366040  0.005231300 -548.37626  -191.801205\n",
      " 4: Br1092 -0.1009730 -0.00094918 -0.000945211  -41.94443   568.769599\n",
      " 5: Br1093 -0.1011000  0.03486130  0.029724800  101.05825   355.618952\n",
      " 6: Br1113 -0.0854901 -0.00882451 -0.039126600  143.23793   339.165330\n",
      " 7: Br1214 -0.1008120  0.03555980 -0.022338600  337.06740  -199.400979\n",
      " 8: Br1245 -0.0922120 -0.10017800  0.002183760  248.69857  -225.792557\n",
      " 9: Br1279 -0.1004230  0.03902360  0.003548450 -512.08919  -361.925898\n",
      "10: Br1339 -0.1024740  0.02810540 -0.002562430  176.92323   143.287642\n",
      "11: Br1350 -0.0961430 -0.03097440 -0.014126100  270.94977  -274.625994\n",
      "12: Br1370 -0.1041680  0.05004620  0.006850120  107.49395   111.765055\n",
      "13: Br1378 -0.0970794 -0.03051090 -0.017293400   38.76623   443.036200\n",
      "14: Br1383 -0.1001970  0.01315300  0.003655210  175.79177    71.249064\n",
      "15: Br1389 -0.0962472  0.01670170  0.023479300  143.96501   -42.469646\n",
      "16: Br1416 -0.1013830  0.06147150  0.024308100  234.60066  -132.279731\n",
      "17: Br1425 -0.1036300  0.04835540 -0.024441500  102.31471   108.498348\n",
      "18: Br1446 -0.0980070 -0.01928200  0.004661170  151.62370   -14.346290\n",
      "19: Br1453 -0.0857703  0.02711000  0.057006000  -58.66985   630.367281\n",
      "20: Br1463 -0.0993881  0.03941320 -0.023119900  167.68343   -20.453697\n",
      "21: Br1474 -0.0916084 -0.04832380  0.005950190 -528.55585  -255.701707\n",
      "22: Br1497 -0.0967251 -0.10966200 -0.034163700 -569.53604  -130.269035\n",
      "23: Br1498 -0.1007390  0.05147130 -0.015587900   59.83001   396.270942\n",
      "24: Br1499 -0.0991484  0.00478075 -0.009938150  372.46205  -253.678693\n",
      "25: Br1524 -0.1023180  0.07143290 -0.019154800  128.47224    93.529180\n",
      "26: Br1529 -0.1015250  0.07694270 -0.018659500 -537.03856  -245.014382\n",
      "27: Br1565 -0.1008490  0.06882650 -0.007512950  194.36988   121.546316\n",
      "28: Br1568 -0.1019500  0.03064910 -0.011984300  169.97621   111.059058\n",
      "29: Br1578 -0.1014430  0.05776300  0.003972810  218.64184    78.454752\n",
      "30: Br1581 -0.0982663  0.05185630 -0.034525300 -422.83707  -504.940440\n",
      "31: Br1587 -0.1006940  0.06591400  0.011976400  -64.13092   629.247221\n",
      "32: Br1594 -0.1022510  0.04235760  0.000458669  234.58622   -31.872645\n",
      "33: Br1602 -0.1030550  0.07210760 -0.031802700  371.63170  -208.260354\n",
      "34: Br1613 -0.1020210  0.02386620 -0.008531080  133.66060   240.930873\n",
      "35: Br1637 -0.1021140  0.05267540  0.037110600 -473.49806  -385.335141\n",
      "36: Br1648 -0.1012240  0.04606280  0.048101300  131.73138   226.888935\n",
      "37: Br1649 -0.0908569  0.01123590  0.038409800  758.32224 -1237.563005\n",
      "38: Br1654 -0.0925862 -0.35445700 -0.015651900  133.93552   253.929372\n",
      "39: Br1669 -0.1008880  0.06479820 -0.030542900 -481.58707  -407.918003\n",
      "40: Br1676 -0.0940465 -0.26755300  0.006746240 -630.96621   -44.485691\n",
      "41: Br1678 -0.1020560  0.05324430  0.003468490 -636.96406     9.175191\n",
      "42: Br1682 -0.1013810  0.04752950 -0.021304400  238.54404  -100.248349\n",
      "43: Br1684 -0.0923002 -0.46806400 -0.103611000 -613.01026    19.684534\n",
      "44: Br1695 -0.0926539 -0.32203900 -0.060851600  134.22546   318.241812\n",
      "45: Br1738 -0.1024910  0.05029640 -0.019461400  293.55748   -95.853870\n",
      "46: Br1753 -0.1018300  0.06089810  0.000952445  205.64537   -43.748835\n",
      "47: Br1761 -0.1038320  0.06851160  0.006817800  167.31678   196.119530\n",
      "48: Br1764 -0.0990592  0.05010820  0.011133700 -657.61505   238.262911\n",
      "49: Br1880 -0.0996120  0.05669170  0.031124500  144.66460    91.776766\n",
      "50: Br1936 -0.1003250  0.06337560 -0.019228600 -413.37075  -517.375000\n",
      "51: Br1946 -0.1017370  0.06878660  0.023906000 -742.59522   314.271127\n",
      "52: Br1958 -0.1011290  0.03321760  0.026686700  194.12563    23.010581\n",
      "53: Br1960 -0.1006380  0.04887210  0.012954700 -464.94512  -453.925969\n",
      "54: Br2009 -0.0997165  0.02657730  0.023145700   76.83688   374.172477\n",
      "55: Br2044 -0.0999900  0.06247590 -0.013440400  226.93992   234.461186\n",
      "56: Br2084 -0.0957608 -0.02415290  0.004926590  342.05110  -160.364399\n",
      "57: Br2242 -0.1015410  0.02374360 -0.005642010  468.44528  -716.805042\n",
      "58: Br2257 -0.0929668 -0.30607400 -0.013746700  119.85276   473.446630\n",
      "59: Br2271 -0.1021800  0.05378080  0.044594500  151.70920   228.287780\n",
      "60: Br2285 -0.1009750  0.07137340  0.000263522  381.42860  -292.480862\n",
      "61: Br2288 -0.1012690  0.01869000  0.005307350  242.83282   -26.771138\n",
      "62: Br2292 -0.1018720  0.03922100  0.005475250  101.04938   410.412113\n",
      "        ID    genoPC1     genoPC2      genoPC3    methPC1      methPC2\n",
      "        methPC3     methPC4      methPC5      methPC6     methPC7     methPC8\n",
      "          <num>       <num>        <num>        <num>       <num>       <num>\n",
      " 1:   14.540472 -155.757748   88.6097039 -259.8419670   -5.449045   66.233979\n",
      " 2:  111.237437 -121.846469 -108.6757029    9.2013555  114.092966  149.898635\n",
      " 3: -114.255203  197.384645  -32.9009625  -49.0105515    9.324397  -35.902978\n",
      " 4:  198.333064  -22.722300 -234.3282171   -3.2496194   35.845048   34.789159\n",
      " 5:   64.023480  161.421514 -132.1731676   78.8760307   -1.594505  -20.586247\n",
      " 6:    1.992726  245.581819  -10.5033520   30.2706637  -68.666633 -234.235737\n",
      " 7: -337.894321  354.039826  425.5176097 -247.2367802  369.044853  495.578158\n",
      " 8:  125.047837  -16.826207 -155.5555318 -117.1457760  -18.620158   24.138941\n",
      " 9:  109.848260 -147.799061   18.1965489  -32.7692444  -31.100968   24.028410\n",
      "10: -148.344418  241.853766   55.6154907 -113.1722635  225.756575 -335.788259\n",
      "11: -183.048746 -182.924437  -87.9950311 -108.1193358  -11.672785   -2.771749\n",
      "12:  -26.041753 -196.004336 -112.6980619  -68.0977295  117.696521  -58.681104\n",
      "13:   90.061273    9.043227 -134.5459824   21.4092546   68.215514  -10.087425\n",
      "14:  -46.367415 -277.228616  226.6349182    8.8970630  -97.587533   15.885120\n",
      "15:  -48.492811 -275.813045 -178.6792611 -201.3130684  180.071705   46.341492\n",
      "16:  301.107244  -73.920979  -77.1879159   18.7431394   26.686483  163.336945\n",
      "17:   39.767161 -164.506695 -130.8301959   49.0661344  -62.600244  120.808624\n",
      "18:  -87.759864 -238.585611 -166.4379959   -0.3229409  -16.373402   16.913015\n",
      "19:  284.143057 -115.021541 -111.1939957  -13.7456433   82.608649  -23.266353\n",
      "20:  -54.034694  -76.116778 -157.1891916   81.7435062  100.676782  130.929601\n",
      "21:   21.879361 -115.600208   12.0818994  -47.6934160  -60.046067  -47.938904\n",
      "22:   -5.228694   87.431746   -5.5083052   -8.7525451   31.371455 -114.588836\n",
      "23:  133.031881  133.043054  -79.2409276  468.2435336  311.881576   58.026330\n",
      "24:   86.280662  189.878980  165.0556309  251.2905762  141.398937  -90.317058\n",
      "25:   35.013679 -235.162485 -236.1705871 -107.5947287  119.852960   -3.685486\n",
      "26:   -4.533821  -83.562840   53.9029058   80.5273768   64.128529   23.490179\n",
      "27: -152.513312  323.008029 -233.1122209  -70.0705331 -129.299327  140.209307\n",
      "28:  -21.363639  -42.845027   55.0871522   42.4803816   -6.892909  132.729782\n",
      "29: -222.811266  283.186059  -23.9843223 -382.8091237 -161.125759 -187.502670\n",
      "30: -271.560283  -10.436050  -62.6541400   68.8860477   19.834747 -122.165452\n",
      "31:  257.816539 -177.744545 -116.5928661 -144.7301731  -60.711966  -14.818794\n",
      "32: -175.760852 -129.318376  116.1010482   99.2825318  284.488812 -152.988695\n",
      "33: -351.559320  123.099850 -153.1923099  227.1630216   74.297372 -215.201319\n",
      "34:  -10.241629  149.035076   -4.4943656  216.1722822  -46.933449   20.015127\n",
      "35:  -38.953918   56.245793   21.9815686   10.1341992  -96.496005  181.078896\n",
      "36:   14.823867 -123.382410  115.9730717  -16.4138207   24.436559   41.287252\n",
      "37: 1164.507985  238.326989   19.4246888 -101.4089620  -18.457874 -142.812480\n",
      "38: -114.958126  241.005358  222.3501145 -137.3247318  -16.759815 -206.631167\n",
      "39:  -57.894248   69.058524   36.5694613   68.3493041  147.891417  117.686516\n",
      "40:  -30.969052  107.351926   72.8595774  -54.5318254  208.043614 -178.424710\n",
      "41:   81.726377   67.644362  -27.0103896  -82.5284138  -27.917464   34.222010\n",
      "42: -199.345973 -152.845218   31.2608310   17.9082447 -185.207212   -7.749106\n",
      "43:  227.902388  -15.768336  119.4266480  294.0448592  142.791981  -61.726141\n",
      "44:   65.582654   58.153835  181.6036652  201.1506684 -274.855527  244.687062\n",
      "45: -207.338635    5.591203  -70.3618159   59.4731755 -139.443945  -25.307344\n",
      "46: -179.940854 -199.010573 -117.2023111  129.1008230  -64.137937 -149.539353\n",
      "47:   24.337212 -300.564413  175.1987340   63.9302339 -194.445384    3.362312\n",
      "48:  112.016160  376.121489  -45.8590777   10.1945955 -123.743488   12.382286\n",
      "49:  -32.991154 -245.395012   36.2129016 -325.2002788  110.275358  -54.893816\n",
      "50: -206.727276 -101.153217  -14.3194914  121.7473843 -181.525944   -7.126310\n",
      "51:  255.950730   69.242670   -0.8199151    5.8845411 -160.562529  -17.216893\n",
      "52: -123.350833  132.287879 -256.6898943 -132.1802154   66.318645   36.430516\n",
      "53:  -89.421161  -80.180097   17.3331448  -28.5711037  -51.934040   59.203524\n",
      "54:   55.691332  287.756653 -291.6527436  -70.8123892 -123.356589   96.710688\n",
      "55:  -95.214465  398.059210  101.2334937 -103.4069923 -183.823891  121.533188\n",
      "56:   12.127419  -40.465721  343.4967256   54.5502804 -210.309486 -172.138188\n",
      "57: -130.389476   30.537342 -208.2946763  108.7489193  -50.309615  192.247453\n",
      "58:  174.445713  -37.248409  310.7036578  117.6941059  -15.216166  138.143681\n",
      "59:   12.713137 -106.799900  342.9407758  -67.2075593   96.068552  -85.357252\n",
      "60: -350.548805  -46.500814   55.3399631  181.0460132 -226.090040  -61.037143\n",
      "61: -182.525469  -97.835512   20.8797626    8.2717761   16.417894  -80.474813\n",
      "62:  226.432384 -228.497841  336.4632317 -109.2202908  -66.250199  -21.366404\n",
      "        methPC3     methPC4      methPC5      methPC6     methPC7     methPC8\n",
      "        methPC9    methPC10      Dx   Age    Sex\n",
      "          <num>       <num>  <char> <num> <char>\n",
      " 1:   -9.111428  113.843382     SCZ 20.22      M\n",
      " 2:  -74.053124  -19.005887 Control 63.00      M\n",
      " 3:  -76.886305  -46.958898 Control 57.60      M\n",
      " 4:   77.225962 -106.518608 Control 48.95      M\n",
      " 5:  -11.694494   62.468243     SCZ 71.81      F\n",
      " 6:   -5.345080   50.438819 Control 51.91      F\n",
      " 7:  191.539158 -376.761498 Control 61.13      M\n",
      " 8:   61.277693   25.546106     SCZ 42.62      M\n",
      " 9:  109.316610  114.353724 Control 18.41      M\n",
      "10: -166.975823  -65.443447     SCZ 60.08      F\n",
      "11:   29.821288   36.726363 Control 24.61      M\n",
      "12:  121.631000  -47.841167     SCZ 30.86      M\n",
      "13:   11.494909  -24.972315 Control 41.01      M\n",
      "14: -499.987935 -303.416547     SCZ 58.96      M\n",
      "15:  -17.804942   48.407651 Control 28.35      M\n",
      "16:   22.326263   63.211894     SCZ 37.16      M\n",
      "17: -151.377575  -79.642054     SCZ 48.08      M\n",
      "18:  -26.496734  -45.080724 Control 34.44      M\n",
      "19:   29.616977 -143.584157     SCZ 32.01      F\n",
      "20: -161.367973   69.281996 Control 49.53      M\n",
      "21:    5.193001  -62.257441 Control 38.39      F\n",
      "22:  115.111501  -11.042529     SCZ 43.96      M\n",
      "23:   92.199592   73.972340     SCZ 92.10      F\n",
      "24: -246.862499  194.906404     SCZ 96.92      F\n",
      "25:  119.396232   -8.296911 Control 19.26      F\n",
      "26:   15.329947   69.676691     SCZ 70.24      M\n",
      "27:  -17.747144  101.224934 Control 51.00      M\n",
      "28:   32.082988  171.123500 Control 39.64      M\n",
      "29: -151.738194  221.269624     SCZ 42.06      M\n",
      "30:  190.035289   -8.744906 Control 46.08      M\n",
      "31:  -50.825653 -220.892762 Control 33.97      F\n",
      "32:   47.910208  159.214981     SCZ 68.97      M\n",
      "33:   88.783007 -158.350549     SCZ 83.14      F\n",
      "34:  -56.081678   65.811194 Control 77.94      M\n",
      "35: -125.596296  159.865664     SCZ 46.19      M\n",
      "36: -179.516092  -34.802183     SCZ 48.09      F\n",
      "37:   42.462676  -95.593952 Control 46.94      M\n",
      "38:  -82.566920 -131.556325 Control 62.27      M\n",
      "39: -136.486862   66.494799 Control 83.59      F\n",
      "40:   -4.406158  -46.735110     SCZ 68.00      M\n",
      "41:   -7.685975  -64.068151     SCZ 42.71      M\n",
      "42:  117.775842   26.487066     SCZ 41.33      M\n",
      "43:   87.518910  -11.452758     SCZ 81.05      F\n",
      "44: -107.532868  165.700235 Control 63.20      M\n",
      "45:  115.728214  120.582598     SCZ 55.68      M\n",
      "46:  -30.815582 -190.567840 Control 48.99      F\n",
      "47:   57.215565 -117.407731 Control 30.01      M\n",
      "48:   -4.628713  -48.783068 Control 56.05      M\n",
      "49:  118.097820  200.526425 Control 18.72      M\n",
      "50:  -11.991251  -88.915321 Control 52.49      M\n",
      "51:   25.327254  -99.295012     SCZ 46.53      F\n",
      "52:  -49.648594   94.715593     SCZ 41.09      F\n",
      "53:  -96.415860  -19.755639 Control 55.51      M\n",
      "54:   28.077767  -97.619083     SCZ 55.68      M\n",
      "55:  163.512848   62.851624 Control 55.91      M\n",
      "56:  141.443685 -137.834873 Control 48.07      M\n",
      "57: -116.890996    1.448893     SCZ 50.73      F\n",
      "58:  128.422184   59.037981 Control 59.22      M\n",
      "59: -209.587458  176.103074     SCZ 50.80      M\n",
      "60:  226.178587 -126.056902 Control 53.29      M\n",
      "61:   58.950930   53.134436 Control 44.70      M\n",
      "62:  217.122297  210.828115     SCZ 20.34      M\n",
      "        methPC9    methPC10      Dx   Age    Sex\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "for(file in files){\n",
    "    print(file)\n",
    "    file_in <- fread(file)\n",
    "    print(file_in)\n",
    "    cat(\"\\n\\n\")\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "08cf5397-7ebc-463e-85ae-cb1f6cbd70a1",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.3.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
