{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "c335908f-1e2d-4119-9354-7005840a588b",
   "metadata": {},
   "source": [
    "# Look at previous code for GCTA, run with our data, become familiar"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "079b1871-1cf4-405d-a1e7-86299a9103d4",
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in setwd(paste0(\"out/\", outdir)): cannot change working directory\n",
     "output_type": "error",
     "traceback": [
      "Error in setwd(paste0(\"out/\", outdir)): cannot change working directory\nTraceback:\n",
      "1. setwd(paste0(\"out/\", outdir))"
     ]
    }
   ],
   "source": [
    "args = commandArgs(trailingOnly=TRUE)\n",
    "\n",
    "outdir <- args[1]\n",
    "chunk1 <- as.numeric(args[2])\n",
    "chunk2 <- as.numeric(args[3])\n",
    "\n",
    "wind <- as.numeric(c(\"1000\",\"2000\",\"5000\",\"10000\",\"20000\",\"50000\",\"100000\",\"200000\",\"500000\"))\n",
    "\n",
    "# This directory contains vmr files, which are...\n",
    "indir <- \"/dcs04/lieber/statsgen/shizhong/AANRI/VMR2/99/caud/aa/\"\n",
    "\n",
    "# path to dir containing pgen, psam files for a given population/subpopulation\n",
    "gwas <- \"/dcs04/lieber/statsgen/shizhong/database/libd/genotype/postmortem/topmed/merge_H650_1M_2.5M_5M/AA/all/plink/\"\n",
    "gcta <- \"/dcs04/lieber/statsgen/shizhong/software/gcta/gcta-1.94.1-linux-kernel-3-x86_64/gcta-1.94.1\"\n",
    "\n",
    "setwd(paste0(\"out/\",outdir))\n",
    "\n",
    "# load vmrs\n",
    "load(paste0(indir,\"out/chr1_vmr.rda\"))\n",
    "meth2 <- meth\n",
    "vmrs2 <- vmrs\n",
    "for(i in 2:22){\n",
    "        load(paste0(indir,\"out/chr\",i,\"_vmr.rda\"))\n",
    "        meth2 <- rbind(meth2,meth)\n",
    "        vmrs2 <- rbind(vmrs2,vmrs)\n",
    "}\n",
    "p <- t(meth2)\n",
    "ind <- rownames(p)\n",
    "\n",
    "# covariates\n",
    "f_demo <- \"/dcs04/lieber/statsgen/shizhong/database/libd/genotype/postmortem/phenotype/pheno_PC\"\n",
    "f_pc <- paste0(indir,\"/out/sva.csv\")\n",
    "f_ances <- \"/dcs04/lieber/statsgen/shizhong/AANRI/structure/structure_CEU_AFR/structure.out_ancestry_proportion_raceDemo_compare\"\n",
    "demo  <- read.table(f_demo,header=T)\n",
    "pc <- read.csv(f_pc)\n",
    "ances <- read.table(f_ances,header=T)\n",
    "\n",
    "# align samples\n",
    "id <- intersect(intersect(demo$BrNum,ind),pc$ind)\n",
    "demo <- demo[match(id,demo$BrNum),]\n",
    "pc <- pc[match(id,pc$ind),]\n",
    "p <- p[match(id,ind),]\n",
    "ances <- ances[match(id,ances$id),]\n",
    "\n",
    "# covariates\n",
    "#covs <- as.data.frame(cbind(Age=demo$Age,Sex=demo$Sex,demo[,11:20],pc[,3:12]))\n",
    "covs <- as.data.frame(cbind(Age=demo$Age,Sex=demo$Sex,Afr=ances$Afr,pc[,3:12]))\n",
    "covs$Sex[covs$Sex==\"M\"] <- 0\n",
    "covs$Sex[covs$Sex==\"F\"] <- 1\n",
    "\n",
    "# overlap samples with genotype data\n",
    "fam <- paste0(gwas,\"AA_chr1.psam\")\n",
    "fam <- read.table(fam,skip = 1, header = F)\n",
    "id <- intersect(fam[,1],demo$ID) # genotype ID\n",
    "write.table(id,\"id\",col.names=F,row.names=F,quote=F,sep=\"\\t\")\n",
    "\n",
    "# align samples\n",
    "idx <- match(id,demo$ID)\n",
    "p <- p[idx,]\n",
    "covs <- covs[idx,]\n",
    "covs <- cbind(0,id,covs)\n",
    "write.table(covs,\"covs\",col.names=F,row.names=F,quote=F,sep=\"\\t\")\n",
    "\n",
    "# loop over vmr between two chunks\n",
    "res <- c()\n",
    "for(i in chunk1:chunk2){\n",
    "        cat(i,\"\\n\")\n",
    "        chr <- gsub(\"chr\",\"\",vmrs2[i,1])\n",
    "        # loop over each window size\n",
    "        for(w in 1:length(wind)){\n",
    "        # plink subset\n",
    "        p1 <- ifelse(vmrs2[i,2] - wind[w] > 0, vmrs2[i,2] - wind[w],0)\n",
    "        p2 <- vmrs2[i,3] + wind[w]\n",
    "        gwas_prefix <- paste0(gwas,\"AA_chr\",chr)\n",
    "        command <- paste(\"plink2 --pfile \", gwas_prefix, \"--silent --keep id\", \"--chr \",chr,\"--from-bp\",p1,\"--to-bp\",p2,\"--snps-only 'just-acgt' --make-bed --out temp\",sep=\" \")\n",
    "        system(command)\n",
    "        if(!file.exists(\"temp.bim\")){\n",
    "                next;\n",
    "        }\n",
    "        # phenotype file \n",
    "        pheno <- cbind(0,id,p[,i])\n",
    "        write.table(pheno,\"pheno\",col.names=F,row.names=F,quote=F,sep=\"\\t\")\n",
    "        # grm\n",
    "        command <- paste(gcta, \"--bfile temp --make-grm-bin --out temp\", sep=\" \")\n",
    "        system(command)\n",
    "        # h2 estimation\n",
    "        command <- paste(gcta, \"--reml --grm-bin temp --pheno pheno --mpheno 1 --qcovar covs --out temp\", sep=\" \")\n",
    "        system(command)\n",
    "        # collect results\n",
    "        if(!file.exists(\"temp.hsq\")){\n",
    "                next;\n",
    "        }\n",
    "        temp <- read.table(\"temp.hsq\",header=T, fill=TRUE)\n",
    "        vmr <- paste0(\"chr\",chr,\"_\",vmrs2[i,2],\"_\",vmrs2[i,3])\n",
    "        temp$vmr <- vmr\n",
    "        temp$wind <- wind[w]\n",
    "        res <- rbind(res,temp)\n",
    "        # remove temp files\n",
    "        system(\"rm temp*\")\n",
    "        }\n",
    "\n",
    "}\n",
    "write.table(res,\"res.txt\",col.names=T,row.names=F,quote=F,sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3c01d32b-ffa9-4d16-a66c-d87cd9df1ece",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.3.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
