{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "7bf1026b-e8a8-40e9-9adc-fe45911863a3",
   "metadata": {},
   "source": [
    "# Try running complete stage 2 inside Jupyter with parallelization"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "27648b98-5637-4d6e-b395-708f73cb8137",
   "metadata": {},
   "outputs": [],
   "source": [
    "library(CpGWAS)\n",
    "library(data.table)\n",
    "library(stringr)\n",
    "library(optparse)\n",
    "\n",
    "# Command line options\n",
    "# option_list <- list(\n",
    "#   make_option(c(\"-g\", \"--genome_file_index\"), type = \"integer\", default = 1,\n",
    "#               help = \"Index of genome file to process\"),\n",
    "#   make_option(c(\"-d\", \"--data_file\"), type = \"character\", default = \"/expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/scripts/12-OUT_matched_SNP_meth_cov_outputs_1rowonly.csv\",\n",
    "#               help = \"Path to data file\")\n",
    "# )\n",
    "\n",
    "# opt <- parse_args(OptionParser(option_list = option_list))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "d0dc82e4-e170-4771-acde-58d661cc1c82",
   "metadata": {},
   "outputs": [],
   "source": [
    "opt <- list(genome_file_index = 1,\n",
    "            data_file = \"/expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/scripts/12-OUT_matched_SNP_meth_cov_outputs.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "7095b1bb-4847-430f-8d75-8108acf919a9",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Starting genome file processing\"\n",
      "[1] \"Processing genome file index: 1\"\n",
      "[1] \"Loaded SNP data\"\n",
      "[1] \"Files for this Chr:\"\n",
      "[1] 1965\n"
     ]
    }
   ],
   "source": [
    "# Load genome files\n",
    "genome_files <- list.files(\"/expanse/lustre/projects/jhu152/naglemi/mwas/gwas\",\n",
    "                           pattern = \"EUR\", full.names = TRUE)\n",
    "genome_files <- genome_files[grepl(\"pvar\", genome_files)]\n",
    "\n",
    "genome_files <- data.table(path = genome_files, Chr = NA)\n",
    "\n",
    "genome_files$Chr <- str_split_fixed(genome_files$path, \"chr\", 2)[, 2]\n",
    "genome_files$Chr <- gsub(\".pvar\", \"\", genome_files$Chr)\n",
    "\n",
    "genome_files$Chr <- as.integer(genome_files$Chr)\n",
    "genome_files <- genome_files[order(genome_files$Chr), ]\n",
    "\n",
    "df <- fread(opt$data_file)\n",
    "\n",
    "summary_stats_list <- list.files(\"/expanse/lustre/projects/jhu152/naglemi/mwas/gwas\", pattern = \"stat\", full.names = TRUE)\n",
    "\n",
    "# Pre-load all summary stats files into a list and clean/standardize column names\n",
    "summary_stats_data <- lapply(summary_stats_list, function(path) {\n",
    "  stats <- suppressWarnings(data.table::fread(path))\n",
    "  colnames(stats) <- gsub(\"#CHROM\", \"CHR\", colnames(stats))\n",
    "  clean_and_standardize_colnames(stats)\n",
    "})\n",
    "\n",
    "print(\"Starting genome file processing\")\n",
    "# Process the specified genome file\n",
    "g <- opt$genome_file_index\n",
    "print(paste(\"Processing genome file index:\", g))\n",
    "\n",
    "paths <- list(\n",
    "  pvar_path = genome_files[g]$path,\n",
    "  pgen_path = gsub(\"pvar\", \"pgen\", genome_files[g]$path),\n",
    "  psam_path = gsub(\"pvar\", \"psam\", genome_files[g]$path)\n",
    ")\n",
    "\n",
    "my_SNPs <- CpGWAS::loadSNPData(paths$pvar_path, paths$pgen_path, paths$psam_path)\n",
    "setkey(my_SNPs$pvar_dt, `#CHROM`, POS)\n",
    "df_this_chr <- df[which(df$Chr == genome_files[g]$Chr), ]\n",
    "\n",
    "summary_stats_data <- lapply(summary_stats_data, function(stats) stats[`CHR` == genome_files[g]$Chr])\n",
    "\n",
    "print(\"Loaded SNP data\")\n",
    "print(\"Files for this Chr:\")\n",
    "print(nrow(df_this_chr))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "65238e19-213d-432e-ba8d-93f0bf15e1ae",
   "metadata": {},
   "outputs": [],
   "source": [
    "for(j in 1:nrow(df_this_chr)){\n",
    "  print(paste0(\"File number: \", j))\n",
    "  if (grepl(\"empty\", df_this_chr$path[j])) {\n",
    "    message(paste0(\"no model for \", df_this_chr$path[j]))\n",
    "    next\n",
    "  }\n",
    "\n",
    "  all_files_exist <- TRUE\n",
    "  outnames <- vector(\"character\", length(summary_stats_list))\n",
    "\n",
    "  for (k in 1:length(summary_stats_list)) {\n",
    "    outnames[k] <- gsub(\"\\\\.rds$\", paste0(\"_\", basename(tools::file_path_sans_ext(summary_stats_list[[k]])), \"_results-a2jup.rds\"), df_this_chr$path[j])\n",
    "    if (!file.exists(outnames[k])) {\n",
    "      all_files_exist <- FALSE\n",
    "      break  # Exit the loop early if any file does not exist\n",
    "    }\n",
    "  }\n",
    "\n",
    "  if (all_files_exist) {\n",
    "    print(paste(\"All output files already exist for\", df_this_chr$path[j], \"- Skipping\"))\n",
    "    next  # Skip to the next file if all outputs already exist\n",
    "  }\n",
    "  #\n",
    "  my_rds <- tryCatch({\n",
    "    readRDS(df_this_chr$path[j])\n",
    "  }, error = function(e) {\n",
    "    # Print an error message and skip this iteration\n",
    "    message(\"ALERT!!! Error reading RDS file: \", e$message)\n",
    "    return(NULL)  # Return NULL to signal failure\n",
    "  })\n",
    "    \n",
    "  # Check if the readRDS call returned NULL (which indicates an error)\n",
    "  if (is.null(my_rds)) {\n",
    "    next  # Skip the rest of this loop iteration\n",
    "  }\n",
    "\n",
    "  print(paste(\"Loaded RDS file:\", df_this_chr$path[j]))\n",
    "\n",
    "  for (k in 1:length(summary_stats_list)) {\n",
    "    outname <- gsub(\"\\\\.rds$\", paste0(\"_\", basename(tools::file_path_sans_ext(summary_stats_list[[k]])), \"_results-a2jup.rds\"), df_this_chr$path[j])\n",
    "    if(file.exists(outname)) next\n",
    "    summary_stats <- summary_stats_data[[k]]\n",
    "\n",
    "    MWASmodels <- vector(\"list\", length(my_rds@models))\n",
    "    if (is.null(summary_stats)) {\n",
    "      summary_stats <- suppressWarnings(fread(summary_stats_list[[k]]))\n",
    "      summary_stats <- clean_and_standardize_colnames(summary_stats)\n",
    "    }\n",
    "\n",
    "    print(k)\n",
    "    print(\"Starting inner loop\")\n",
    "\n",
    "    for (i in seq_along(my_rds@models)) {\n",
    "      this_MethylationBase <- my_rds@models[[i]]\n",
    "      SNP_split <- stringr::str_split_fixed(names(this_MethylationBase@snpWeights), \":\", 4)\n",
    "      SNP_split[, 1] <- gsub(\"chr\", \"\", SNP_split[, 1])\n",
    "      SNP_split_dt <- data.table::as.data.table(SNP_split)\n",
    "      data.table::setnames(SNP_split_dt, c(\"chr\", \"post\", \"ref\", \"alt\"))\n",
    "      SNP_split_dt[, `:=`(chr = as.integer(chr), post = as.integer(post))]\n",
    "      data.table::setkey(SNP_split_dt, chr, post)\n",
    "\n",
    "      relevant_SNP_indices <- my_SNPs$pvar_dt[SNP_split_dt, on = .(`#CHROM` = chr, POS = post), which = TRUE, nomatch = 0]\n",
    "      relevant_ids <- my_SNPs$pvar_dt$ID[relevant_SNP_indices]\n",
    "      summary_stats_sub <- summary_stats[relevant_ids, nomatch = 0]\n",
    "\n",
    "      if (!identical(summary_stats_sub$BP, SNP_split_dt$post)) {\n",
    "        summary_stats_sub <- summary_stats_sub[order(summary_stats_sub$BP), ]\n",
    "        if (!identical(summary_stats_sub$BP, SNP_split_dt$post)) {\n",
    "          unmatched_positions <- !SNP_split_dt$post %in% summary_stats_sub$BP\n",
    "          if (any(unmatched_positions)) {\n",
    "            SNP_split_dt <- SNP_split_dt[!unmatched_positions, ]\n",
    "            this_MethylationBase@snpWeights <- this_MethylationBase@snpWeights[!unmatched_positions]\n",
    "\n",
    "            relevant_SNP_indices <- my_SNPs$pvar_dt[SNP_split_dt, on = .(`#CHROM` = chr, POS = post), which = TRUE, nomatch = 0]\n",
    "            if (!identical(summary_stats_sub$BP, SNP_split_dt$post)) {\n",
    "              stop(\"SNP order does not match even after removing unmatched positions. This should not happen. Code is broken.\")\n",
    "            }\n",
    "          }\n",
    "        }\n",
    "      }\n",
    "\n",
    "      if (!identical(SNP_split_dt$alt, summary_stats_sub$A2) | !identical(SNP_split_dt$ref, summary_stats_sub$A1)) {\n",
    "        not_matching <- which(SNP_split_dt$alt != summary_stats_sub$A2)\n",
    "        summary_stats_ref_flipped <- SNP_split_dt$ref[not_matching]\n",
    "        summary_stats_alt_flipped <- SNP_split_dt$alt[not_matching]\n",
    "        SNP_split_dt[not_matching, `:=`(ref = summary_stats_alt_flipped, alt = summary_stats_ref_flipped)]\n",
    "        this_MethylationBase@snpWeights[not_matching] <- this_MethylationBase@snpWeights[not_matching] * -1\n",
    "      }\n",
    "\n",
    "      G <- pgenlibr::ReadList(my_SNPs$pgen, variant_subset = relevant_SNP_indices)\n",
    "      #print(paste(\"Performing MWAS for model index:\", i))\n",
    "      mwas_out <- mwas(z = summary_stats_sub$BETA, w = this_MethylationBase@snpWeights, G = G)\n",
    "      mwas_out['pos'] <- this_MethylationBase@methylationPosition\n",
    "        \n",
    "      MWASmodels[[i]] <- mwas_out\n",
    "    }\n",
    "\n",
    "    results <- MWASresults(MWASmodels, paths$pvar_path, paths$pgen_path, paths$psam_path, summary_stats_list[[k]], df_this_chr$path[j])\n",
    "    saveRDS(results, outname)\n",
    "    print(paste(\"Saved results to:\", outname))\n",
    "  }\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "09242eb7-339d-4746-8322-466fcb2079eb",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.3.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
