{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "750ddea7-3a85-4c69-85cc-a871b54a59eb",
   "metadata": {},
   "source": [
    "# Verify successful flipping of alleles (make ref match libd) and identify any alleles that still don't match."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "4f23d846-7b9a-4df9-8fdb-e684b5bbe65e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "'/expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/scripts'"
      ],
      "text/latex": [
       "'/expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/scripts'"
      ],
      "text/markdown": [
       "'/expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/scripts'"
      ],
      "text/plain": [
       "[1] \"/expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/scripts\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "getwd()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "8c894339-d969-4cd0-ac66-09faeb53ac6f",
   "metadata": {},
   "outputs": [],
   "source": [
    "libd_files <- list.files(\"/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped/\", pattern = \"libd\", full.names = TRUE)\n",
    "ref_files <- list.files(\"/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped/\", pattern = \"flipped\", full.names = TRUE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "2619657e-424b-471d-aa00-8710c942d6ba",
   "metadata": {},
   "outputs": [],
   "source": [
    "libd_files <- libd_files[grepl(\"pvar\", libd_files)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "9921af7c-34cd-4732-9011-811a838b15e9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//libd_chr1.pvar'</li><li>'/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//libd_chr10.pvar'</li><li>'/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//libd_chr11.pvar'</li><li>'/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//libd_chr12.pvar'</li><li>'/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//libd_chr13.pvar'</li><li>'/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//libd_chr14.pvar'</li><li>'/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//libd_chr15.pvar'</li><li>'/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//libd_chr16.pvar'</li><li>'/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//libd_chr17.pvar'</li><li>'/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//libd_chr18.pvar'</li><li>'/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//libd_chr19.pvar'</li><li>'/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//libd_chr2.pvar'</li><li>'/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//libd_chr20.pvar'</li><li>'/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//libd_chr21.pvar'</li><li>'/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//libd_chr22.pvar'</li><li>'/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//libd_chr3.pvar'</li><li>'/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//libd_chr4.pvar'</li><li>'/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//libd_chr5.pvar'</li><li>'/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//libd_chr6.pvar'</li><li>'/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//libd_chr7.pvar'</li><li>'/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//libd_chr8.pvar'</li><li>'/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//libd_chr9.pvar'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas\\_flipped//libd\\_chr1.pvar'\n",
       "\\item '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas\\_flipped//libd\\_chr10.pvar'\n",
       "\\item '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas\\_flipped//libd\\_chr11.pvar'\n",
       "\\item '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas\\_flipped//libd\\_chr12.pvar'\n",
       "\\item '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas\\_flipped//libd\\_chr13.pvar'\n",
       "\\item '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas\\_flipped//libd\\_chr14.pvar'\n",
       "\\item '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas\\_flipped//libd\\_chr15.pvar'\n",
       "\\item '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas\\_flipped//libd\\_chr16.pvar'\n",
       "\\item '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas\\_flipped//libd\\_chr17.pvar'\n",
       "\\item '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas\\_flipped//libd\\_chr18.pvar'\n",
       "\\item '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas\\_flipped//libd\\_chr19.pvar'\n",
       "\\item '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas\\_flipped//libd\\_chr2.pvar'\n",
       "\\item '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas\\_flipped//libd\\_chr20.pvar'\n",
       "\\item '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas\\_flipped//libd\\_chr21.pvar'\n",
       "\\item '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas\\_flipped//libd\\_chr22.pvar'\n",
       "\\item '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas\\_flipped//libd\\_chr3.pvar'\n",
       "\\item '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas\\_flipped//libd\\_chr4.pvar'\n",
       "\\item '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas\\_flipped//libd\\_chr5.pvar'\n",
       "\\item '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas\\_flipped//libd\\_chr6.pvar'\n",
       "\\item '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas\\_flipped//libd\\_chr7.pvar'\n",
       "\\item '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas\\_flipped//libd\\_chr8.pvar'\n",
       "\\item '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas\\_flipped//libd\\_chr9.pvar'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//libd_chr1.pvar'\n",
       "2. '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//libd_chr10.pvar'\n",
       "3. '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//libd_chr11.pvar'\n",
       "4. '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//libd_chr12.pvar'\n",
       "5. '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//libd_chr13.pvar'\n",
       "6. '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//libd_chr14.pvar'\n",
       "7. '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//libd_chr15.pvar'\n",
       "8. '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//libd_chr16.pvar'\n",
       "9. '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//libd_chr17.pvar'\n",
       "10. '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//libd_chr18.pvar'\n",
       "11. '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//libd_chr19.pvar'\n",
       "12. '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//libd_chr2.pvar'\n",
       "13. '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//libd_chr20.pvar'\n",
       "14. '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//libd_chr21.pvar'\n",
       "15. '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//libd_chr22.pvar'\n",
       "16. '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//libd_chr3.pvar'\n",
       "17. '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//libd_chr4.pvar'\n",
       "18. '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//libd_chr5.pvar'\n",
       "19. '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//libd_chr6.pvar'\n",
       "20. '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//libd_chr7.pvar'\n",
       "21. '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//libd_chr8.pvar'\n",
       "22. '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//libd_chr9.pvar'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//libd_chr1.pvar\" \n",
       " [2] \"/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//libd_chr10.pvar\"\n",
       " [3] \"/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//libd_chr11.pvar\"\n",
       " [4] \"/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//libd_chr12.pvar\"\n",
       " [5] \"/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//libd_chr13.pvar\"\n",
       " [6] \"/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//libd_chr14.pvar\"\n",
       " [7] \"/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//libd_chr15.pvar\"\n",
       " [8] \"/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//libd_chr16.pvar\"\n",
       " [9] \"/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//libd_chr17.pvar\"\n",
       "[10] \"/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//libd_chr18.pvar\"\n",
       "[11] \"/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//libd_chr19.pvar\"\n",
       "[12] \"/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//libd_chr2.pvar\" \n",
       "[13] \"/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//libd_chr20.pvar\"\n",
       "[14] \"/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//libd_chr21.pvar\"\n",
       "[15] \"/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//libd_chr22.pvar\"\n",
       "[16] \"/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//libd_chr3.pvar\" \n",
       "[17] \"/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//libd_chr4.pvar\" \n",
       "[18] \"/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//libd_chr5.pvar\" \n",
       "[19] \"/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//libd_chr6.pvar\" \n",
       "[20] \"/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//libd_chr7.pvar\" \n",
       "[21] \"/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//libd_chr8.pvar\" \n",
       "[22] \"/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//libd_chr9.pvar\" "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "libd_files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "5d9d9958-2db4-4594-8b0f-7757c41ce446",
   "metadata": {},
   "outputs": [],
   "source": [
    "ref_files <- ref_files[grepl(\"pvar\", ref_files)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "87595eff-d813-451d-acdc-d0a46384fe74",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//ref_EUR_flipped_chr1.pvar'</li><li>'/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//ref_EUR_flipped_chr10.pvar'</li><li>'/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//ref_EUR_flipped_chr11.pvar'</li><li>'/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//ref_EUR_flipped_chr12.pvar'</li><li>'/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//ref_EUR_flipped_chr13.pvar'</li><li>'/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//ref_EUR_flipped_chr14.pvar'</li><li>'/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//ref_EUR_flipped_chr15.pvar'</li><li>'/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//ref_EUR_flipped_chr16.pvar'</li><li>'/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//ref_EUR_flipped_chr17.pvar'</li><li>'/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//ref_EUR_flipped_chr18.pvar'</li><li>'/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//ref_EUR_flipped_chr19.pvar'</li><li>'/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//ref_EUR_flipped_chr2.pvar'</li><li>'/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//ref_EUR_flipped_chr20.pvar'</li><li>'/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//ref_EUR_flipped_chr21.pvar'</li><li>'/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//ref_EUR_flipped_chr22.pvar'</li><li>'/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//ref_EUR_flipped_chr3.pvar'</li><li>'/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//ref_EUR_flipped_chr4.pvar'</li><li>'/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//ref_EUR_flipped_chr5.pvar'</li><li>'/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//ref_EUR_flipped_chr6.pvar'</li><li>'/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//ref_EUR_flipped_chr7.pvar'</li><li>'/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//ref_EUR_flipped_chr8.pvar'</li><li>'/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//ref_EUR_flipped_chr9.pvar'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas\\_flipped//ref\\_EUR\\_flipped\\_chr1.pvar'\n",
       "\\item '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas\\_flipped//ref\\_EUR\\_flipped\\_chr10.pvar'\n",
       "\\item '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas\\_flipped//ref\\_EUR\\_flipped\\_chr11.pvar'\n",
       "\\item '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas\\_flipped//ref\\_EUR\\_flipped\\_chr12.pvar'\n",
       "\\item '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas\\_flipped//ref\\_EUR\\_flipped\\_chr13.pvar'\n",
       "\\item '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas\\_flipped//ref\\_EUR\\_flipped\\_chr14.pvar'\n",
       "\\item '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas\\_flipped//ref\\_EUR\\_flipped\\_chr15.pvar'\n",
       "\\item '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas\\_flipped//ref\\_EUR\\_flipped\\_chr16.pvar'\n",
       "\\item '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas\\_flipped//ref\\_EUR\\_flipped\\_chr17.pvar'\n",
       "\\item '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas\\_flipped//ref\\_EUR\\_flipped\\_chr18.pvar'\n",
       "\\item '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas\\_flipped//ref\\_EUR\\_flipped\\_chr19.pvar'\n",
       "\\item '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas\\_flipped//ref\\_EUR\\_flipped\\_chr2.pvar'\n",
       "\\item '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas\\_flipped//ref\\_EUR\\_flipped\\_chr20.pvar'\n",
       "\\item '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas\\_flipped//ref\\_EUR\\_flipped\\_chr21.pvar'\n",
       "\\item '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas\\_flipped//ref\\_EUR\\_flipped\\_chr22.pvar'\n",
       "\\item '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas\\_flipped//ref\\_EUR\\_flipped\\_chr3.pvar'\n",
       "\\item '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas\\_flipped//ref\\_EUR\\_flipped\\_chr4.pvar'\n",
       "\\item '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas\\_flipped//ref\\_EUR\\_flipped\\_chr5.pvar'\n",
       "\\item '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas\\_flipped//ref\\_EUR\\_flipped\\_chr6.pvar'\n",
       "\\item '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas\\_flipped//ref\\_EUR\\_flipped\\_chr7.pvar'\n",
       "\\item '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas\\_flipped//ref\\_EUR\\_flipped\\_chr8.pvar'\n",
       "\\item '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas\\_flipped//ref\\_EUR\\_flipped\\_chr9.pvar'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//ref_EUR_flipped_chr1.pvar'\n",
       "2. '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//ref_EUR_flipped_chr10.pvar'\n",
       "3. '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//ref_EUR_flipped_chr11.pvar'\n",
       "4. '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//ref_EUR_flipped_chr12.pvar'\n",
       "5. '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//ref_EUR_flipped_chr13.pvar'\n",
       "6. '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//ref_EUR_flipped_chr14.pvar'\n",
       "7. '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//ref_EUR_flipped_chr15.pvar'\n",
       "8. '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//ref_EUR_flipped_chr16.pvar'\n",
       "9. '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//ref_EUR_flipped_chr17.pvar'\n",
       "10. '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//ref_EUR_flipped_chr18.pvar'\n",
       "11. '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//ref_EUR_flipped_chr19.pvar'\n",
       "12. '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//ref_EUR_flipped_chr2.pvar'\n",
       "13. '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//ref_EUR_flipped_chr20.pvar'\n",
       "14. '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//ref_EUR_flipped_chr21.pvar'\n",
       "15. '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//ref_EUR_flipped_chr22.pvar'\n",
       "16. '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//ref_EUR_flipped_chr3.pvar'\n",
       "17. '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//ref_EUR_flipped_chr4.pvar'\n",
       "18. '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//ref_EUR_flipped_chr5.pvar'\n",
       "19. '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//ref_EUR_flipped_chr6.pvar'\n",
       "20. '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//ref_EUR_flipped_chr7.pvar'\n",
       "21. '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//ref_EUR_flipped_chr8.pvar'\n",
       "22. '/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//ref_EUR_flipped_chr9.pvar'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//ref_EUR_flipped_chr1.pvar\" \n",
       " [2] \"/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//ref_EUR_flipped_chr10.pvar\"\n",
       " [3] \"/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//ref_EUR_flipped_chr11.pvar\"\n",
       " [4] \"/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//ref_EUR_flipped_chr12.pvar\"\n",
       " [5] \"/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//ref_EUR_flipped_chr13.pvar\"\n",
       " [6] \"/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//ref_EUR_flipped_chr14.pvar\"\n",
       " [7] \"/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//ref_EUR_flipped_chr15.pvar\"\n",
       " [8] \"/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//ref_EUR_flipped_chr16.pvar\"\n",
       " [9] \"/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//ref_EUR_flipped_chr17.pvar\"\n",
       "[10] \"/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//ref_EUR_flipped_chr18.pvar\"\n",
       "[11] \"/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//ref_EUR_flipped_chr19.pvar\"\n",
       "[12] \"/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//ref_EUR_flipped_chr2.pvar\" \n",
       "[13] \"/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//ref_EUR_flipped_chr20.pvar\"\n",
       "[14] \"/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//ref_EUR_flipped_chr21.pvar\"\n",
       "[15] \"/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//ref_EUR_flipped_chr22.pvar\"\n",
       "[16] \"/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//ref_EUR_flipped_chr3.pvar\" \n",
       "[17] \"/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//ref_EUR_flipped_chr4.pvar\" \n",
       "[18] \"/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//ref_EUR_flipped_chr5.pvar\" \n",
       "[19] \"/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//ref_EUR_flipped_chr6.pvar\" \n",
       "[20] \"/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//ref_EUR_flipped_chr7.pvar\" \n",
       "[21] \"/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//ref_EUR_flipped_chr8.pvar\" \n",
       "[22] \"/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped//ref_EUR_flipped_chr9.pvar\" "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "ref_files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "5c340579-b06f-4e1c-a17e-7e40666a3c23",
   "metadata": {},
   "outputs": [],
   "source": [
    "library(data.table)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "68958374-209c-4aa9-b8ef-50e7c00409db",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 6 × 7</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>#CHROM</th><th scope=col>POS</th><th scope=col>ID</th><th scope=col>REF</th><th scope=col>ALT</th><th scope=col>FILTER</th><th scope=col>INFO</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>1</td><td>1079456</td><td>chr1:1079456:A:G</td><td>A</td><td>G</td><td>PASS</td><td>AF=0.47992;MAF=0.47992;R2=0.98236;IMPUTED;AN=4378;AC=2268                  </td></tr>\n",
       "\t<tr><td>1</td><td>1079484</td><td>chr1:1079484:T:A</td><td>T</td><td>A</td><td>PASS</td><td>AF=0.53765;MAF=0.46235;R2=0.98207;IMPUTED;AN=4378;AC=2440                  </td></tr>\n",
       "\t<tr><td>1</td><td>1079746</td><td>chr1:1079746:A:G</td><td>A</td><td>G</td><td>PASS</td><td>AF=0.54145;MAF=0.45855;R2=0.98178;IMPUTED;ER2=0.97555;TYPED;AN=4378;AC=2452</td></tr>\n",
       "\t<tr><td>1</td><td>1079877</td><td>chr1:1079877:A:G</td><td>A</td><td>G</td><td>PASS</td><td>AF=0.5594;MAF=0.4406;R2=0.98131;IMPUTED;AN=4378;AC=2492                    </td></tr>\n",
       "\t<tr><td>1</td><td>1080171</td><td>chr1:1080171:C:T</td><td>C</td><td>T</td><td>PASS</td><td>AF=0.46064;MAF=0.46064;R2=0.97025;IMPUTED;AN=4378;AC=2233                  </td></tr>\n",
       "\t<tr><td>1</td><td>1080437</td><td>chr1:1080437:G:A</td><td>G</td><td>A</td><td>PASS</td><td>AF=0.68092;MAF=0.31908;R2=0.9755;IMPUTED;AN=4378;AC=3085                   </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 6 × 7\n",
       "\\begin{tabular}{lllllll}\n",
       " \\#CHROM & POS & ID & REF & ALT & FILTER & INFO\\\\\n",
       " <int> & <int> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t 1 & 1079456 & chr1:1079456:A:G & A & G & PASS & AF=0.47992;MAF=0.47992;R2=0.98236;IMPUTED;AN=4378;AC=2268                  \\\\\n",
       "\t 1 & 1079484 & chr1:1079484:T:A & T & A & PASS & AF=0.53765;MAF=0.46235;R2=0.98207;IMPUTED;AN=4378;AC=2440                  \\\\\n",
       "\t 1 & 1079746 & chr1:1079746:A:G & A & G & PASS & AF=0.54145;MAF=0.45855;R2=0.98178;IMPUTED;ER2=0.97555;TYPED;AN=4378;AC=2452\\\\\n",
       "\t 1 & 1079877 & chr1:1079877:A:G & A & G & PASS & AF=0.5594;MAF=0.4406;R2=0.98131;IMPUTED;AN=4378;AC=2492                    \\\\\n",
       "\t 1 & 1080171 & chr1:1080171:C:T & C & T & PASS & AF=0.46064;MAF=0.46064;R2=0.97025;IMPUTED;AN=4378;AC=2233                  \\\\\n",
       "\t 1 & 1080437 & chr1:1080437:G:A & G & A & PASS & AF=0.68092;MAF=0.31908;R2=0.9755;IMPUTED;AN=4378;AC=3085                   \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 6 × 7\n",
       "\n",
       "| #CHROM &lt;int&gt; | POS &lt;int&gt; | ID &lt;chr&gt; | REF &lt;chr&gt; | ALT &lt;chr&gt; | FILTER &lt;chr&gt; | INFO &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|\n",
       "| 1 | 1079456 | chr1:1079456:A:G | A | G | PASS | AF=0.47992;MAF=0.47992;R2=0.98236;IMPUTED;AN=4378;AC=2268                   |\n",
       "| 1 | 1079484 | chr1:1079484:T:A | T | A | PASS | AF=0.53765;MAF=0.46235;R2=0.98207;IMPUTED;AN=4378;AC=2440                   |\n",
       "| 1 | 1079746 | chr1:1079746:A:G | A | G | PASS | AF=0.54145;MAF=0.45855;R2=0.98178;IMPUTED;ER2=0.97555;TYPED;AN=4378;AC=2452 |\n",
       "| 1 | 1079877 | chr1:1079877:A:G | A | G | PASS | AF=0.5594;MAF=0.4406;R2=0.98131;IMPUTED;AN=4378;AC=2492                     |\n",
       "| 1 | 1080171 | chr1:1080171:C:T | C | T | PASS | AF=0.46064;MAF=0.46064;R2=0.97025;IMPUTED;AN=4378;AC=2233                   |\n",
       "| 1 | 1080437 | chr1:1080437:G:A | G | A | PASS | AF=0.68092;MAF=0.31908;R2=0.9755;IMPUTED;AN=4378;AC=3085                    |\n",
       "\n"
      ],
      "text/plain": [
       "  #CHROM POS     ID               REF ALT FILTER\n",
       "1 1      1079456 chr1:1079456:A:G A   G   PASS  \n",
       "2 1      1079484 chr1:1079484:T:A T   A   PASS  \n",
       "3 1      1079746 chr1:1079746:A:G A   G   PASS  \n",
       "4 1      1079877 chr1:1079877:A:G A   G   PASS  \n",
       "5 1      1080171 chr1:1080171:C:T C   T   PASS  \n",
       "6 1      1080437 chr1:1080437:G:A G   A   PASS  \n",
       "  INFO                                                                       \n",
       "1 AF=0.47992;MAF=0.47992;R2=0.98236;IMPUTED;AN=4378;AC=2268                  \n",
       "2 AF=0.53765;MAF=0.46235;R2=0.98207;IMPUTED;AN=4378;AC=2440                  \n",
       "3 AF=0.54145;MAF=0.45855;R2=0.98178;IMPUTED;ER2=0.97555;TYPED;AN=4378;AC=2452\n",
       "4 AF=0.5594;MAF=0.4406;R2=0.98131;IMPUTED;AN=4378;AC=2492                    \n",
       "5 AF=0.46064;MAF=0.46064;R2=0.97025;IMPUTED;AN=4378;AC=2233                  \n",
       "6 AF=0.68092;MAF=0.31908;R2=0.9755;IMPUTED;AN=4378;AC=3085                   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(fread(libd_files[1]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "b9d55b38-5f88-43a6-aef8-d0a27584280c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 6 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>#CHROM</th><th scope=col>POS</th><th scope=col>ID</th><th scope=col>REF</th><th scope=col>ALT</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>1</td><td>1079456</td><td>rs12401605</td><td>A</td><td>G</td></tr>\n",
       "\t<tr><td>1</td><td>1079484</td><td>rs12411041</td><td>T</td><td>A</td></tr>\n",
       "\t<tr><td>1</td><td>1079746</td><td>rs36027499</td><td>A</td><td>G</td></tr>\n",
       "\t<tr><td>1</td><td>1079877</td><td>rs9442369 </td><td>A</td><td>G</td></tr>\n",
       "\t<tr><td>1</td><td>1080171</td><td>rs9442370 </td><td>C</td><td>T</td></tr>\n",
       "\t<tr><td>1</td><td>1080437</td><td>rs12746483</td><td>G</td><td>A</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 6 × 5\n",
       "\\begin{tabular}{lllll}\n",
       " \\#CHROM & POS & ID & REF & ALT\\\\\n",
       " <int> & <int> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t 1 & 1079456 & rs12401605 & A & G\\\\\n",
       "\t 1 & 1079484 & rs12411041 & T & A\\\\\n",
       "\t 1 & 1079746 & rs36027499 & A & G\\\\\n",
       "\t 1 & 1079877 & rs9442369  & A & G\\\\\n",
       "\t 1 & 1080171 & rs9442370  & C & T\\\\\n",
       "\t 1 & 1080437 & rs12746483 & G & A\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 6 × 5\n",
       "\n",
       "| #CHROM &lt;int&gt; | POS &lt;int&gt; | ID &lt;chr&gt; | REF &lt;chr&gt; | ALT &lt;chr&gt; |\n",
       "|---|---|---|---|---|\n",
       "| 1 | 1079456 | rs12401605 | A | G |\n",
       "| 1 | 1079484 | rs12411041 | T | A |\n",
       "| 1 | 1079746 | rs36027499 | A | G |\n",
       "| 1 | 1079877 | rs9442369  | A | G |\n",
       "| 1 | 1080171 | rs9442370  | C | T |\n",
       "| 1 | 1080437 | rs12746483 | G | A |\n",
       "\n"
      ],
      "text/plain": [
       "  #CHROM POS     ID         REF ALT\n",
       "1 1      1079456 rs12401605 A   G  \n",
       "2 1      1079484 rs12411041 T   A  \n",
       "3 1      1079746 rs36027499 A   G  \n",
       "4 1      1079877 rs9442369  A   G  \n",
       "5 1      1080171 rs9442370  C   T  \n",
       "6 1      1080437 rs12746483 G   A  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(fread(ref_files[1]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "faabbe4b-6298-4b10-9b6c-9ed6ea9d4e79",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Chromosome 1: Tables are not identical\n",
      "\n",
      "Chromosome 2: Tables are not identical\n",
      "\n",
      "Chromosome 3: Tables are identical\n",
      "\n",
      "Chromosome 4: Tables are identical\n",
      "\n",
      "Chromosome 5: Tables are not identical\n",
      "\n",
      "Chromosome 6: Tables are not identical\n",
      "\n",
      "Chromosome 7: Tables are identical\n",
      "\n",
      "Chromosome 8: Tables are identical\n",
      "\n",
      "Chromosome 9: Tables are identical\n",
      "\n",
      "Chromosome 10: Tables are not identical\n",
      "\n",
      "Chromosome 11: Tables are identical\n",
      "\n",
      "Chromosome 12: Tables are not identical\n",
      "\n",
      "Chromosome 13: Tables are not identical\n",
      "\n",
      "Chromosome 14: Tables are identical\n",
      "\n",
      "Chromosome 15: Tables are not identical\n",
      "\n",
      "Chromosome 16: Tables are identical\n",
      "\n",
      "Chromosome 17: Tables are identical\n",
      "\n",
      "Chromosome 18: Tables are not identical\n",
      "\n",
      "Chromosome 19: Tables are identical\n",
      "\n",
      "Chromosome 20: Tables are identical\n",
      "\n",
      "Chromosome 21: Tables are identical\n",
      "\n",
      "Chromosome 22: Tables are not identical\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(data.table)\n",
    "\n",
    "# List of chromosomes\n",
    "chromosomes <- 1:22  # Assuming chromosomes 1 through 22\n",
    "\n",
    "# Base paths (adjust as necessary)\n",
    "path_libd <- \"/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped/\"\n",
    "path_ref <- \"/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped/\"\n",
    "\n",
    "# Loop through each chromosome\n",
    "for (chr in chromosomes) {\n",
    "  # Construct file names\n",
    "  file_libd <- paste0(path_libd, \"libd_chr\", chr, \".pvar\")\n",
    "  file_ref <- paste0(path_ref, \"ref_EUR_flipped_chr\", chr, \".pvar\")\n",
    "\n",
    "  # Read files\n",
    "  dt_libd <- fread(file_libd)[, .(CHROM = `#CHROM`, POS, REF, ALT)]\n",
    "  dt_ref <- fread(file_ref)[, .(CHROM = `#CHROM`, POS, REF, ALT)]\n",
    "\n",
    "  # Check if they are identical\n",
    "  identical_flag <- identical(dt_libd, dt_ref)\n",
    "\n",
    "  # Print results\n",
    "  cat(sprintf(\"Chromosome %d: Tables are %s\\n\\n\", chr, if (identical_flag) \"identical\" else \"not identical\"))\n",
    "}\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "f1dd3621-d84c-4088-bea2-3fdf838d57c6",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4 SNPs on chromosome 1 don't match, as shown below:\n",
      "[1] \"Rows in libd not in ref:\"\n",
      "   CHROM       POS    REF    ALT\n",
      "   <int>     <int> <char> <char>\n",
      "1:     1  47352558      G      T\n",
      "2:     1 202939940      A      C\n",
      "[1] \"Rows in ref not in libd:\"\n",
      "   CHROM       POS    REF    ALT\n",
      "   <int>     <int> <char> <char>\n",
      "1:     1  47352558      G      C\n",
      "2:     1 202939940      A      T\n",
      "\n",
      "2 SNPs on chromosome 2 don't match, as shown below:\n",
      "[1] \"Rows in libd not in ref:\"\n",
      "   CHROM      POS    REF    ALT\n",
      "   <int>    <int> <char> <char>\n",
      "1:     2 97349751      C      T\n",
      "[1] \"Rows in ref not in libd:\"\n",
      "   CHROM      POS    REF    ALT\n",
      "   <int>    <int> <char> <char>\n",
      "1:     2 97349751      G      A\n",
      "\n",
      "All SNPs on chromosome 3 match perfectly.\n",
      "\n",
      "\n",
      "All SNPs on chromosome 4 match perfectly.\n",
      "\n",
      "\n",
      "2 SNPs on chromosome 5 don't match, as shown below:\n",
      "[1] \"Rows in libd not in ref:\"\n",
      "   CHROM      POS    REF    ALT\n",
      "   <int>    <int> <char> <char>\n",
      "1:     5 29127615      T      C\n",
      "[1] \"Rows in ref not in libd:\"\n",
      "   CHROM      POS    REF    ALT\n",
      "   <int>    <int> <char> <char>\n",
      "1:     5 29127615      T      G\n",
      "\n",
      "20 SNPs on chromosome 6 don't match, as shown below:\n",
      "[1] \"Rows in libd not in ref:\"\n",
      "    CHROM      POS    REF    ALT\n",
      "    <int>    <int> <char> <char>\n",
      " 1:     6 61140626      T      C\n",
      " 2:     6 61150043      A      G\n",
      " 3:     6 61170701      C      T\n",
      " 4:     6 61173330      C      A\n",
      " 5:     6 61185008      A      G\n",
      " 6:     6 61187712      G      T\n",
      " 7:     6 61190696      C      T\n",
      " 8:     6 61197843      T      C\n",
      " 9:     6 61250730      G      T\n",
      "10:     6 61286007      G      T\n",
      "[1] \"Rows in ref not in libd:\"\n",
      "    CHROM      POS    REF    ALT\n",
      "    <int>    <int> <char> <char>\n",
      " 1:     6 61140626      A      G\n",
      " 2:     6 61150043      T      C\n",
      " 3:     6 61170701      G      A\n",
      " 4:     6 61173330      G      T\n",
      " 5:     6 61185008      T      C\n",
      " 6:     6 61187712      C      A\n",
      " 7:     6 61190696      G      A\n",
      " 8:     6 61197843      A      G\n",
      " 9:     6 61250730      C      A\n",
      "10:     6 61286007      C      A\n",
      "\n",
      "All SNPs on chromosome 7 match perfectly.\n",
      "\n",
      "\n",
      "All SNPs on chromosome 8 match perfectly.\n",
      "\n",
      "\n",
      "All SNPs on chromosome 9 match perfectly.\n",
      "\n",
      "\n",
      "2 SNPs on chromosome 10 don't match, as shown below:\n",
      "[1] \"Rows in libd not in ref:\"\n",
      "   CHROM       POS    REF    ALT\n",
      "   <int>     <int> <char> <char>\n",
      "1:    10 108423160      C      G\n",
      "[1] \"Rows in ref not in libd:\"\n",
      "   CHROM       POS    REF    ALT\n",
      "   <int>     <int> <char> <char>\n",
      "1:    10 108423160      C      A\n",
      "\n",
      "All SNPs on chromosome 11 match perfectly.\n",
      "\n",
      "\n",
      "2 SNPs on chromosome 12 don't match, as shown below:\n",
      "[1] \"Rows in libd not in ref:\"\n",
      "   CHROM       POS    REF    ALT\n",
      "   <int>     <int> <char> <char>\n",
      "1:    12 119689666      C      A\n",
      "[1] \"Rows in ref not in libd:\"\n",
      "   CHROM       POS    REF    ALT\n",
      "   <int>     <int> <char> <char>\n",
      "1:    12 119689666      C      T\n",
      "\n",
      "4 SNPs on chromosome 13 don't match, as shown below:\n",
      "[1] \"Rows in libd not in ref:\"\n",
      "   CHROM       POS    REF    ALT\n",
      "   <int>     <int> <char> <char>\n",
      "1:    13 101704483      G      T\n",
      "2:    13 106007683      T      C\n",
      "[1] \"Rows in ref not in libd:\"\n",
      "   CHROM       POS    REF    ALT\n",
      "   <int>     <int> <char> <char>\n",
      "1:    13 101704483      G      A\n",
      "2:    13 106007683      T      A\n",
      "\n",
      "All SNPs on chromosome 14 match perfectly.\n",
      "\n",
      "\n",
      "2 SNPs on chromosome 15 don't match, as shown below:\n",
      "[1] \"Rows in libd not in ref:\"\n",
      "   CHROM      POS    REF    ALT\n",
      "   <int>    <int> <char> <char>\n",
      "1:    15 93140187      C      T\n",
      "[1] \"Rows in ref not in libd:\"\n",
      "   CHROM      POS    REF    ALT\n",
      "   <int>    <int> <char> <char>\n",
      "1:    15 93140187      C      A\n",
      "\n",
      "All SNPs on chromosome 16 match perfectly.\n",
      "\n",
      "\n",
      "All SNPs on chromosome 17 match perfectly.\n",
      "\n",
      "\n",
      "2 SNPs on chromosome 18 don't match, as shown below:\n",
      "[1] \"Rows in libd not in ref:\"\n",
      "   CHROM      POS    REF    ALT\n",
      "   <int>    <int> <char> <char>\n",
      "1:    18 10699712      T      C\n",
      "[1] \"Rows in ref not in libd:\"\n",
      "   CHROM      POS    REF    ALT\n",
      "   <int>    <int> <char> <char>\n",
      "1:    18 10699712      T      A\n",
      "\n",
      "All SNPs on chromosome 19 match perfectly.\n",
      "\n",
      "\n",
      "All SNPs on chromosome 20 match perfectly.\n",
      "\n",
      "\n",
      "All SNPs on chromosome 21 match perfectly.\n",
      "\n",
      "\n",
      "2 SNPs on chromosome 22 don't match, as shown below:\n",
      "[1] \"Rows in libd not in ref:\"\n",
      "   CHROM      POS    REF    ALT\n",
      "   <int>    <int> <char> <char>\n",
      "1:    22 47177732      G      C\n",
      "[1] \"Rows in ref not in libd:\"\n",
      "   CHROM      POS    REF    ALT\n",
      "   <int>    <int> <char> <char>\n",
      "1:    22 47177732      G      A\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(data.table)\n",
    "library(dplyr)\n",
    "\n",
    "# List of chromosomes\n",
    "chromosomes <- 1:22  # Assuming chromosomes 1 through 22\n",
    "\n",
    "# Base paths (adjust as necessary)\n",
    "path_libd <- \"/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped/\"\n",
    "path_ref <- \"/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped/\"\n",
    "\n",
    "# Loop through each chromosome\n",
    "for (chr in chromosomes) {\n",
    "  # Construct file names\n",
    "  file_libd <- paste0(path_libd, \"libd_chr\", chr, \".pvar\")\n",
    "  file_ref <- paste0(path_ref, \"ref_EUR_flipped_chr\", chr, \".pvar\")\n",
    "\n",
    "  # Read files\n",
    "  dt_libd <- fread(file_libd)[, .(CHROM = `#CHROM`, POS, REF, ALT)]\n",
    "  dt_ref <- fread(file_ref)[, .(CHROM = `#CHROM`, POS, REF, ALT)]\n",
    "\n",
    "  # Find non-matching rows\n",
    "  non_matching_libd <- anti_join(dt_libd, dt_ref, by = c(\"CHROM\", \"POS\", \"REF\", \"ALT\"))\n",
    "  non_matching_ref <- anti_join(dt_ref, dt_libd, by = c(\"CHROM\", \"POS\", \"REF\", \"ALT\"))\n",
    "\n",
    "  # Report non-matching details\n",
    "  if (nrow(non_matching_libd) > 0 || nrow(non_matching_ref) > 0) {\n",
    "    cat(sprintf(\"%d SNPs on chromosome %d don't match, as shown below:\\n\", nrow(non_matching_libd) + nrow(non_matching_ref), chr))\n",
    "    \n",
    "    if (nrow(non_matching_libd) > 0) {\n",
    "      print(\"Rows in libd not in ref:\")\n",
    "      print(non_matching_libd)\n",
    "    }\n",
    "    \n",
    "    if (nrow(non_matching_ref) > 0) {\n",
    "      print(\"Rows in ref not in libd:\")\n",
    "      print(non_matching_ref)\n",
    "    }\n",
    "  } else {\n",
    "    cat(sprintf(\"All SNPs on chromosome %d match perfectly.\\n\\n\", chr))\n",
    "  }\n",
    "    cat(\"\\n\")\n",
    "}\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "d14273de-8984-4f69-9aec-229d114ff53a",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4 SNPs on chromosome 1 don't match, as shown below:\n",
      "[1] \"Rows in libd not in ref:\"\n",
      "   CHROM       POS    REF    ALT\n",
      "   <int>     <int> <char> <char>\n",
      "1:     1  47352558      G      T\n",
      "2:     1 202939940      A      C\n",
      "[1] \"Rows in ref not in libd:\"\n",
      "   CHROM       POS          ID    REF    ALT\n",
      "   <int>     <int>      <char> <char> <char>\n",
      "1:     1  47352558   rs2622925      G      C\n",
      "2:     1 202939940 rs114652390      A      T\n",
      "\n",
      "2 SNPs on chromosome 2 don't match, as shown below:\n",
      "[1] \"Rows in libd not in ref:\"\n",
      "   CHROM      POS    REF    ALT\n",
      "   <int>    <int> <char> <char>\n",
      "1:     2 97349751      C      T\n",
      "[1] \"Rows in ref not in libd:\"\n",
      "   CHROM      POS        ID    REF    ALT\n",
      "   <int>    <int>    <char> <char> <char>\n",
      "1:     2 97349751 rs9308891      G      A\n",
      "\n",
      "All SNPs on chromosome 3 match perfectly.\n",
      "\n",
      "\n",
      "All SNPs on chromosome 4 match perfectly.\n",
      "\n",
      "\n",
      "2 SNPs on chromosome 5 don't match, as shown below:\n",
      "[1] \"Rows in libd not in ref:\"\n",
      "   CHROM      POS    REF    ALT\n",
      "   <int>    <int> <char> <char>\n",
      "1:     5 29127615      T      C\n",
      "[1] \"Rows in ref not in libd:\"\n",
      "   CHROM      POS          ID    REF    ALT\n",
      "   <int>    <int>      <char> <char> <char>\n",
      "1:     5 29127615 rs116564546      T      G\n",
      "\n",
      "20 SNPs on chromosome 6 don't match, as shown below:\n",
      "[1] \"Rows in libd not in ref:\"\n",
      "    CHROM      POS    REF    ALT\n",
      "    <int>    <int> <char> <char>\n",
      " 1:     6 61140626      T      C\n",
      " 2:     6 61150043      A      G\n",
      " 3:     6 61170701      C      T\n",
      " 4:     6 61173330      C      A\n",
      " 5:     6 61185008      A      G\n",
      " 6:     6 61187712      G      T\n",
      " 7:     6 61190696      C      T\n",
      " 8:     6 61197843      T      C\n",
      " 9:     6 61250730      G      T\n",
      "10:     6 61286007      G      T\n",
      "[1] \"Rows in ref not in libd:\"\n",
      "    CHROM      POS          ID    REF    ALT\n",
      "    <int>    <int>      <char> <char> <char>\n",
      " 1:     6 61140626   rs1932643      A      G\n",
      " 2:     6 61150043  rs62424078      T      C\n",
      " 3:     6 61170701   rs9378089      G      A\n",
      " 4:     6 61173330 rs147899989      G      T\n",
      " 5:     6 61185008   rs9404914      T      C\n",
      " 6:     6 61187712 rs115703744      C      A\n",
      " 7:     6 61190696  rs62423559      G      A\n",
      " 8:     6 61197843  rs73469798      A      G\n",
      " 9:     6 61250730   rs2201271      C      A\n",
      "10:     6 61286007 rs115442350      C      A\n",
      "\n",
      "All SNPs on chromosome 7 match perfectly.\n",
      "\n",
      "\n",
      "All SNPs on chromosome 8 match perfectly.\n",
      "\n",
      "\n",
      "All SNPs on chromosome 9 match perfectly.\n",
      "\n",
      "\n",
      "2 SNPs on chromosome 10 don't match, as shown below:\n",
      "[1] \"Rows in libd not in ref:\"\n",
      "   CHROM       POS    REF    ALT\n",
      "   <int>     <int> <char> <char>\n",
      "1:    10 108423160      C      G\n",
      "[1] \"Rows in ref not in libd:\"\n",
      "   CHROM       POS        ID    REF    ALT\n",
      "   <int>     <int>    <char> <char> <char>\n",
      "1:    10 108423160 rs1576349      C      A\n",
      "\n",
      "All SNPs on chromosome 11 match perfectly.\n",
      "\n",
      "\n",
      "2 SNPs on chromosome 12 don't match, as shown below:\n",
      "[1] \"Rows in libd not in ref:\"\n",
      "   CHROM       POS    REF    ALT\n",
      "   <int>     <int> <char> <char>\n",
      "1:    12 119689666      C      A\n",
      "[1] \"Rows in ref not in libd:\"\n",
      "   CHROM       POS        ID    REF    ALT\n",
      "   <int>     <int>    <char> <char> <char>\n",
      "1:    12 119689666 rs7961828      C      T\n",
      "\n",
      "4 SNPs on chromosome 13 don't match, as shown below:\n",
      "[1] \"Rows in libd not in ref:\"\n",
      "   CHROM       POS    REF    ALT\n",
      "   <int>     <int> <char> <char>\n",
      "1:    13 101704483      G      T\n",
      "2:    13 106007683      T      C\n",
      "[1] \"Rows in ref not in libd:\"\n",
      "   CHROM       POS          ID    REF    ALT\n",
      "   <int>     <int>      <char> <char> <char>\n",
      "1:    13 101704483 rs117617139      G      A\n",
      "2:    13 106007683  rs16967945      T      A\n",
      "\n",
      "All SNPs on chromosome 14 match perfectly.\n",
      "\n",
      "\n",
      "2 SNPs on chromosome 15 don't match, as shown below:\n",
      "[1] \"Rows in libd not in ref:\"\n",
      "   CHROM      POS    REF    ALT\n",
      "   <int>    <int> <char> <char>\n",
      "1:    15 93140187      C      T\n",
      "[1] \"Rows in ref not in libd:\"\n",
      "   CHROM      POS          ID    REF    ALT\n",
      "   <int>    <int>      <char> <char> <char>\n",
      "1:    15 93140187 rs113095242      C      A\n",
      "\n",
      "All SNPs on chromosome 16 match perfectly.\n",
      "\n",
      "\n",
      "All SNPs on chromosome 17 match perfectly.\n",
      "\n",
      "\n",
      "2 SNPs on chromosome 18 don't match, as shown below:\n",
      "[1] \"Rows in libd not in ref:\"\n",
      "   CHROM      POS    REF    ALT\n",
      "   <int>    <int> <char> <char>\n",
      "1:    18 10699712      T      C\n",
      "[1] \"Rows in ref not in libd:\"\n",
      "   CHROM      POS        ID    REF    ALT\n",
      "   <int>    <int>    <char> <char> <char>\n",
      "1:    18 10699712 rs9807703      T      A\n",
      "\n",
      "All SNPs on chromosome 19 match perfectly.\n",
      "\n",
      "\n",
      "All SNPs on chromosome 20 match perfectly.\n",
      "\n",
      "\n",
      "All SNPs on chromosome 21 match perfectly.\n",
      "\n",
      "\n",
      "2 SNPs on chromosome 22 don't match, as shown below:\n",
      "[1] \"Rows in libd not in ref:\"\n",
      "   CHROM      POS    REF    ALT\n",
      "   <int>    <int> <char> <char>\n",
      "1:    22 47177732      G      C\n",
      "[1] \"Rows in ref not in libd:\"\n",
      "   CHROM      POS       ID    REF    ALT\n",
      "   <int>    <int>   <char> <char> <char>\n",
      "1:    22 47177732 rs135364      G      A\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(data.table)\n",
    "library(dplyr)\n",
    "\n",
    "# Prepare to collect all non-matching entries\n",
    "all_non_matching <- list()\n",
    "\n",
    "# List of chromosomes\n",
    "chromosomes <- 1:22  # Assuming chromosomes 1 through 22\n",
    "\n",
    "# Base paths (adjust as necessary)\n",
    "path_libd <- \"/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped/\"\n",
    "path_ref <- \"/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped/\"\n",
    "\n",
    "# Loop through each chromosome\n",
    "for (chr in chromosomes) {\n",
    "  # Construct file names\n",
    "  file_libd <- paste0(path_libd, \"libd_chr\", chr, \".pvar\")\n",
    "  file_ref <- paste0(path_ref, \"ref_EUR_flipped_chr\", chr, \".pvar\")\n",
    "\n",
    "  # Read files\n",
    "  dt_libd <- fread(file_libd)[, .(CHROM = `#CHROM`, POS, REF, ALT)]\n",
    "  dt_ref <- fread(file_ref)[, .(CHROM = `#CHROM`, POS, ID, REF, ALT)]\n",
    "\n",
    "  # Find non-matching rows\n",
    "  non_matching_libd <- anti_join(dt_libd, dt_ref, by = c(\"CHROM\", \"POS\", \"REF\", \"ALT\"))\n",
    "  non_matching_ref <- anti_join(dt_ref, dt_libd, by = c(\"CHROM\", \"POS\", \"REF\", \"ALT\"))\n",
    "\n",
    "  # Accumulate non-matching entries, including ID from ref\n",
    "  if (nrow(non_matching_ref) > 0) {\n",
    "    all_non_matching[[chr]] <- non_matching_ref[, .(CHROM, POS, ID)]\n",
    "  }\n",
    "\n",
    "  # Report non-matching details\n",
    "  if (nrow(non_matching_libd) > 0 || nrow(non_matching_ref) > 0) {\n",
    "    cat(sprintf(\"%d SNPs on chromosome %d don't match, as shown below:\\n\", nrow(non_matching_libd) + nrow(non_matching_ref), chr))\n",
    "    \n",
    "    if (nrow(non_matching_libd) > 0) {\n",
    "      print(\"Rows in libd not in ref:\")\n",
    "      print(non_matching_libd[, .(CHROM, POS, REF, ALT)])\n",
    "    }\n",
    "    \n",
    "    if (nrow(non_matching_ref) > 0) {\n",
    "      print(\"Rows in ref not in libd:\")\n",
    "      print(non_matching_ref[, .(CHROM, POS, ID, REF, ALT)])\n",
    "    }\n",
    "  } else {\n",
    "    cat(sprintf(\"All SNPs on chromosome %d match perfectly.\\n\\n\", chr))\n",
    "  }\n",
    "    cat(\"\\n\")\n",
    "}\n",
    "\n",
    "# Combine all non-matching entries into a single dataframe\n",
    "final_non_matching <- rbindlist(all_non_matching)\n",
    "final_non_matching <- unique(final_non_matching)\n",
    "# Save the final list to CSV\n",
    "fwrite(final_non_matching, \"30-OUT_multi-allele_exclude_list.csv\", col.names = TRUE)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "24e2f8f9-7988-4d6a-a8c8-ede55ab85800",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "All IDs on chromosome 1 have been verified correctly.\n",
      "\n",
      "All IDs on chromosome 2 have been verified correctly.\n",
      "\n",
      "All IDs on chromosome 3 have been verified correctly.\n",
      "\n",
      "All IDs on chromosome 4 have been verified correctly.\n",
      "\n",
      "All IDs on chromosome 5 have been verified correctly.\n",
      "\n",
      "All IDs on chromosome 6 have been verified correctly.\n",
      "\n",
      "All IDs on chromosome 7 have been verified correctly.\n",
      "\n",
      "All IDs on chromosome 8 have been verified correctly.\n",
      "\n",
      "All IDs on chromosome 9 have been verified correctly.\n",
      "\n",
      "All IDs on chromosome 10 have been verified correctly.\n",
      "\n",
      "All IDs on chromosome 11 have been verified correctly.\n",
      "\n",
      "All IDs on chromosome 12 have been verified correctly.\n",
      "\n",
      "All IDs on chromosome 13 have been verified correctly.\n",
      "\n",
      "All IDs on chromosome 14 have been verified correctly.\n",
      "\n",
      "All IDs on chromosome 15 have been verified correctly.\n",
      "\n",
      "All IDs on chromosome 16 have been verified correctly.\n",
      "\n",
      "All IDs on chromosome 17 have been verified correctly.\n",
      "\n",
      "All IDs on chromosome 18 have been verified correctly.\n",
      "\n",
      "All IDs on chromosome 19 have been verified correctly.\n",
      "\n",
      "All IDs on chromosome 20 have been verified correctly.\n",
      "\n",
      "All IDs on chromosome 21 have been verified correctly.\n",
      "\n",
      "All IDs on chromosome 22 have been verified correctly.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Second loop to verify IDs against CHROM and POS\n",
    "for (chr in chromosomes) {\n",
    "  # Read the reference file again\n",
    "  dt_ref <- fread(paste0(path_ref, \"ref_EUR_flipped_chr\", chr, \".pvar\"))[, .(CHROM = `#CHROM`, POS, ID)]\n",
    "\n",
    "  # Subset final_non_matching for the current chromosome\n",
    "  verify_ids <- final_non_matching[CHROM == chr]\n",
    "\n",
    "  # Check ID correspondence\n",
    "  verification_result <- merge(verify_ids, dt_ref, by = c(\"CHROM\", \"POS\", \"ID\"), all.x = TRUE, all.y = FALSE)\n",
    "  missing_ids <- verification_result[is.na(ID)]\n",
    "\n",
    "  if (nrow(missing_ids) > 0) {\n",
    "    cat(sprintf(\"Verification failed for %d entries on chromosome %d as shown below:\\n\", nrow(missing_ids), chr))\n",
    "    print(missing_ids)\n",
    "  } else {\n",
    "    cat(sprintf(\"All IDs on chromosome %d have been verified correctly.\\n\\n\", chr))\n",
    "  }\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "fd4b6be4-b5b8-4600-a5df-8ea23921e14f",
   "metadata": {},
   "outputs": [],
   "source": [
    "final_non_matching <- unique(final_non_matching)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9332bfa8-c0fd-4753-b428-33a9a70a1ec2",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "fc34e6dc-2bb5-4e5f-adfa-958eb724387e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 21 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>CHROM</th><th scope=col>POS</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td> 1</td><td> 47352558</td></tr>\n",
       "\t<tr><td> 1</td><td>202939940</td></tr>\n",
       "\t<tr><td> 2</td><td> 97349751</td></tr>\n",
       "\t<tr><td> 5</td><td> 29127615</td></tr>\n",
       "\t<tr><td> 6</td><td> 61140626</td></tr>\n",
       "\t<tr><td> 6</td><td> 61150043</td></tr>\n",
       "\t<tr><td> 6</td><td> 61170701</td></tr>\n",
       "\t<tr><td> 6</td><td> 61173330</td></tr>\n",
       "\t<tr><td> 6</td><td> 61185008</td></tr>\n",
       "\t<tr><td> 6</td><td> 61187712</td></tr>\n",
       "\t<tr><td> 6</td><td> 61190696</td></tr>\n",
       "\t<tr><td> 6</td><td> 61197843</td></tr>\n",
       "\t<tr><td> 6</td><td> 61250730</td></tr>\n",
       "\t<tr><td> 6</td><td> 61286007</td></tr>\n",
       "\t<tr><td>10</td><td>108423160</td></tr>\n",
       "\t<tr><td>12</td><td>119689666</td></tr>\n",
       "\t<tr><td>13</td><td>101704483</td></tr>\n",
       "\t<tr><td>13</td><td>106007683</td></tr>\n",
       "\t<tr><td>15</td><td> 93140187</td></tr>\n",
       "\t<tr><td>18</td><td> 10699712</td></tr>\n",
       "\t<tr><td>22</td><td> 47177732</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 21 × 2\n",
       "\\begin{tabular}{ll}\n",
       " CHROM & POS\\\\\n",
       " <int> & <int>\\\\\n",
       "\\hline\n",
       "\t  1 &  47352558\\\\\n",
       "\t  1 & 202939940\\\\\n",
       "\t  2 &  97349751\\\\\n",
       "\t  5 &  29127615\\\\\n",
       "\t  6 &  61140626\\\\\n",
       "\t  6 &  61150043\\\\\n",
       "\t  6 &  61170701\\\\\n",
       "\t  6 &  61173330\\\\\n",
       "\t  6 &  61185008\\\\\n",
       "\t  6 &  61187712\\\\\n",
       "\t  6 &  61190696\\\\\n",
       "\t  6 &  61197843\\\\\n",
       "\t  6 &  61250730\\\\\n",
       "\t  6 &  61286007\\\\\n",
       "\t 10 & 108423160\\\\\n",
       "\t 12 & 119689666\\\\\n",
       "\t 13 & 101704483\\\\\n",
       "\t 13 & 106007683\\\\\n",
       "\t 15 &  93140187\\\\\n",
       "\t 18 &  10699712\\\\\n",
       "\t 22 &  47177732\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 21 × 2\n",
       "\n",
       "| CHROM &lt;int&gt; | POS &lt;int&gt; |\n",
       "|---|---|\n",
       "|  1 |  47352558 |\n",
       "|  1 | 202939940 |\n",
       "|  2 |  97349751 |\n",
       "|  5 |  29127615 |\n",
       "|  6 |  61140626 |\n",
       "|  6 |  61150043 |\n",
       "|  6 |  61170701 |\n",
       "|  6 |  61173330 |\n",
       "|  6 |  61185008 |\n",
       "|  6 |  61187712 |\n",
       "|  6 |  61190696 |\n",
       "|  6 |  61197843 |\n",
       "|  6 |  61250730 |\n",
       "|  6 |  61286007 |\n",
       "| 10 | 108423160 |\n",
       "| 12 | 119689666 |\n",
       "| 13 | 101704483 |\n",
       "| 13 | 106007683 |\n",
       "| 15 |  93140187 |\n",
       "| 18 |  10699712 |\n",
       "| 22 |  47177732 |\n",
       "\n"
      ],
      "text/plain": [
       "   CHROM POS      \n",
       "1   1     47352558\n",
       "2   1    202939940\n",
       "3   2     97349751\n",
       "4   5     29127615\n",
       "5   6     61140626\n",
       "6   6     61150043\n",
       "7   6     61170701\n",
       "8   6     61173330\n",
       "9   6     61185008\n",
       "10  6     61187712\n",
       "11  6     61190696\n",
       "12  6     61197843\n",
       "13  6     61250730\n",
       "14  6     61286007\n",
       "15 10    108423160\n",
       "16 12    119689666\n",
       "17 13    101704483\n",
       "18 13    106007683\n",
       "19 15     93140187\n",
       "20 18     10699712\n",
       "21 22     47177732"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "final_non_matching"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "d71add81-65e8-4d51-a678-b1027e76ac5a",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save the final list to CSV\n",
    "fwrite(final_non_matching, \"30-OUT_multi-allele_exclude_list.csv\", col.names = TRUE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "8c04345b-ff84-4a55-b012-e1d963b4747d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   CHROM       POS          ID\n",
      "   <int>     <int>      <char>\n",
      "1:     1  47352558   rs2622925\n",
      "2:     1 202939940 rs114652390\n",
      "3:     2  97349751   rs9308891\n",
      "4:     5  29127615 rs116564546\n",
      "5:     6  61140626   rs1932643\n",
      "6:     6  61150043  rs62424078\n"
     ]
    }
   ],
   "source": [
    "library(data.table)\n",
    "\n",
    "# Load the exclusion list\n",
    "exclude_list <- fread(\"30-OUT_multi-allele_exclude_list.csv\")\n",
    "\n",
    "# Assuming exclude_list has columns CHROM, POS, and ID\n",
    "# We'll write out only the ID column, which is required by PLINK for exclusion\n",
    "fwrite(exclude_list[, .(ID)], \"30_OUT_variant_exclude_list.txt\", quote = FALSE, sep=\"\\n\", col.names = FALSE)\n",
    "\n",
    "# Print to check structure\n",
    "print(head(exclude_list))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c65bf40b-0e18-46f2-9332-b7ed5ee128cd",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.3.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
