{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "4c14af96-8b5e-4741-bfc9-c4cc800b3686",
   "metadata": {},
   "source": [
    "# Exclude from ref_EUR_flipped files additional alleles that don't match with ref_libd"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8b26c5d0-2359-427a-8482-480e57f96560",
   "metadata": {},
   "source": [
    "Non-matching due to different alternative alleles, or strand difference on Chr. 6 region (imputation artifact?)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b70b948f-9cd1-422b-95ab-a630b0e5d69b",
   "metadata": {},
   "source": [
    "This time we run with the second list of multiallelic sites, from notebook 33. Previously we ran with result from notebook 30."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "c1aee98d-c4f1-4ed5-b9ff-ab1627a88d02",
   "metadata": {},
   "outputs": [],
   "source": [
    "cd /expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "18a966d7-634d-4850-aa2b-40b290aa7c9f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "01_snps_overlaped.pl                  ref_EUR_chr12.log\n",
      "directory_listing.csv                 ref_EUR_chr12.pgen\n",
      "gwas_stat_bp                          ref_EUR_chr12.psam\n",
      "gwas_stat_bp_alleleprocessed.txt      ref_EUR_chr12.pvar\n",
      "gwas_stat_mdd                         ref_EUR_chr13.log\n",
      "gwas_stat_mdd_alleleprocessed.txt     ref_EUR_chr13.pgen\n",
      "gwas_stat_scz                         ref_EUR_chr13.psam\n",
      "gwas_stat_scz_alleleprocessed.txt     ref_EUR_chr13.pvar\n",
      "libd_chr10.log                        ref_EUR_chr14.log\n",
      "libd_chr10.pgen                       ref_EUR_chr14.pgen\n",
      "libd_chr10.psam                       ref_EUR_chr14.psam\n",
      "libd_chr10.pvar                       ref_EUR_chr14.pvar\n",
      "libd_chr11.log                        ref_EUR_chr15.log\n",
      "libd_chr11.pgen                       ref_EUR_chr15.pgen\n",
      "libd_chr11.psam                       ref_EUR_chr15.psam\n",
      "libd_chr11.pvar                       ref_EUR_chr15.pvar\n",
      "libd_chr12.log                        ref_EUR_chr16.log\n",
      "libd_chr12.pgen                       ref_EUR_chr16.pgen\n",
      "libd_chr12.psam                       ref_EUR_chr16.psam\n",
      "libd_chr12.pvar                       ref_EUR_chr16.pvar\n",
      "libd_chr13.log                        ref_EUR_chr17.log\n",
      "libd_chr13.pgen                       ref_EUR_chr17.pgen\n",
      "libd_chr13.psam                       ref_EUR_chr17.psam\n",
      "libd_chr13.pvar                       ref_EUR_chr17.pvar\n",
      "libd_chr14.log                        ref_EUR_chr18.log\n",
      "libd_chr14.pgen                       ref_EUR_chr18.pgen\n",
      "libd_chr14.psam                       ref_EUR_chr18.psam\n",
      "libd_chr14.pvar                       ref_EUR_chr18.pvar\n",
      "libd_chr15.log                        ref_EUR_chr19.log\n",
      "libd_chr15.pgen                       ref_EUR_chr19.pgen\n",
      "libd_chr15.psam                       ref_EUR_chr19.psam\n",
      "libd_chr15.pvar                       ref_EUR_chr19.pvar\n",
      "libd_chr16.log                        ref_EUR_chr1.log\n",
      "libd_chr16.pgen                       ref_EUR_chr1.pgen\n",
      "libd_chr16.psam                       ref_EUR_chr1.psam\n",
      "libd_chr16.pvar                       ref_EUR_chr1.pvar\n",
      "libd_chr17.log                        ref_EUR_chr1_snps.log\n",
      "libd_chr17.pgen                       ref_EUR_chr20.log\n",
      "libd_chr17.psam                       ref_EUR_chr20.pgen\n",
      "libd_chr17.pvar                       ref_EUR_chr20.psam\n",
      "libd_chr18.log                        ref_EUR_chr20.pvar\n",
      "libd_chr18.pgen                       ref_EUR_chr21.log\n",
      "libd_chr18.psam                       ref_EUR_chr21.pgen\n",
      "libd_chr18.pvar                       ref_EUR_chr21.psam\n",
      "libd_chr19.log                        ref_EUR_chr21.pvar\n",
      "libd_chr19.pgen                       ref_EUR_chr22.log\n",
      "libd_chr19.psam                       ref_EUR_chr22.pgen\n",
      "libd_chr19.pvar                       ref_EUR_chr22.psam\n",
      "libd_chr1.log                         ref_EUR_chr22.pvar\n",
      "libd_chr1.pgen                        ref_EUR_chr2.log\n",
      "libd_chr1.psam                        ref_EUR_chr2.pgen\n",
      "libd_chr1.pvar                        ref_EUR_chr2.psam\n",
      "libd_chr1_snps.log                    ref_EUR_chr2.pvar\n",
      "libd_chr20.log                        ref_EUR_chr3.log\n",
      "libd_chr20.pgen                       ref_EUR_chr3.pgen\n",
      "libd_chr20.psam                       ref_EUR_chr3.psam\n",
      "libd_chr20.pvar                       ref_EUR_chr3.pvar\n",
      "libd_chr21.log                        ref_EUR_chr4.log\n",
      "libd_chr21.pgen                       ref_EUR_chr4.pgen\n",
      "libd_chr21.psam                       ref_EUR_chr4.psam\n",
      "libd_chr21.pvar                       ref_EUR_chr4.pvar\n",
      "libd_chr22.log                        ref_EUR_chr5.log\n",
      "libd_chr22.pgen                       ref_EUR_chr5.pgen\n",
      "libd_chr22.psam                       ref_EUR_chr5.psam\n",
      "libd_chr22.pvar                       ref_EUR_chr5.pvar\n",
      "libd_chr2.log                         ref_EUR_chr6.log\n",
      "libd_chr2.pgen                        ref_EUR_chr6.pgen\n",
      "libd_chr2.psam                        ref_EUR_chr6.psam\n",
      "libd_chr2.pvar                        ref_EUR_chr6.pvar\n",
      "libd_chr3.log                         ref_EUR_chr7.log\n",
      "libd_chr3.pgen                        ref_EUR_chr7.pgen\n",
      "libd_chr3.psam                        ref_EUR_chr7.psam\n",
      "libd_chr3.pvar                        ref_EUR_chr7.pvar\n",
      "libd_chr4.log                         ref_EUR_chr8.log\n",
      "libd_chr4.pgen                        ref_EUR_chr8.pgen\n",
      "libd_chr4.psam                        ref_EUR_chr8.psam\n",
      "libd_chr4.pvar                        ref_EUR_chr8.pvar\n",
      "libd_chr5.log                         ref_EUR_chr9.log\n",
      "libd_chr5.pgen                        ref_EUR_chr9.pgen\n",
      "libd_chr5.psam                        ref_EUR_chr9.psam\n",
      "libd_chr5.pvar                        ref_EUR_chr9.pvar\n",
      "libd_chr6.log                         ref_EUR_flipped_chr10.log\n",
      "libd_chr6.pgen                        ref_EUR_flipped_chr10.pgen\n",
      "libd_chr6.psam                        ref_EUR_flipped_chr10.psam\n",
      "libd_chr6.pvar                        ref_EUR_flipped_chr10.pvar\n",
      "libd_chr7.log                         ref_EUR_flipped_chr11.log\n",
      "libd_chr7.pgen                        ref_EUR_flipped_chr11.pgen\n",
      "libd_chr7.psam                        ref_EUR_flipped_chr11.psam\n",
      "libd_chr7.pvar                        ref_EUR_flipped_chr11.pvar\n",
      "libd_chr8.log                         ref_EUR_flipped_chr12.log\n",
      "libd_chr8.pgen                        ref_EUR_flipped_chr12.pgen\n",
      "libd_chr8.psam                        ref_EUR_flipped_chr12.psam\n",
      "libd_chr8.pvar                        ref_EUR_flipped_chr12.pvar\n",
      "libd_chr9.log                         ref_EUR_flipped_chr13.log\n",
      "libd_chr9.pgen                        ref_EUR_flipped_chr13.pgen\n",
      "libd_chr9.psam                        ref_EUR_flipped_chr13.psam\n",
      "libd_chr9.pvar                        ref_EUR_flipped_chr13.pvar\n",
      "\u001b[0m\u001b[38;5;40mplink2\u001b[0m                                ref_EUR_flipped_chr14.log\n",
      "ref_EUR_allele-match-libd_chr10.log   ref_EUR_flipped_chr14.pgen\n",
      "ref_EUR_allele-match-libd_chr10.pgen  ref_EUR_flipped_chr14.psam\n",
      "ref_EUR_allele-match-libd_chr10.psam  ref_EUR_flipped_chr14.pvar\n",
      "ref_EUR_allele-match-libd_chr10.pvar  ref_EUR_flipped_chr15.log\n",
      "ref_EUR_allele-match-libd_chr11.log   ref_EUR_flipped_chr15.pgen\n",
      "ref_EUR_allele-match-libd_chr11.pgen  ref_EUR_flipped_chr15.psam\n",
      "ref_EUR_allele-match-libd_chr11.psam  ref_EUR_flipped_chr15.pvar\n",
      "ref_EUR_allele-match-libd_chr11.pvar  ref_EUR_flipped_chr16.log\n",
      "ref_EUR_allele-match-libd_chr12.log   ref_EUR_flipped_chr16.pgen\n",
      "ref_EUR_allele-match-libd_chr12.pgen  ref_EUR_flipped_chr16.psam\n",
      "ref_EUR_allele-match-libd_chr12.psam  ref_EUR_flipped_chr16.pvar\n",
      "ref_EUR_allele-match-libd_chr12.pvar  ref_EUR_flipped_chr17.log\n",
      "ref_EUR_allele-match-libd_chr13.log   ref_EUR_flipped_chr17.pgen\n",
      "ref_EUR_allele-match-libd_chr13.pgen  ref_EUR_flipped_chr17.psam\n",
      "ref_EUR_allele-match-libd_chr13.psam  ref_EUR_flipped_chr17.pvar\n",
      "ref_EUR_allele-match-libd_chr13.pvar  ref_EUR_flipped_chr18.log\n",
      "ref_EUR_allele-match-libd_chr14.log   ref_EUR_flipped_chr18.pgen\n",
      "ref_EUR_allele-match-libd_chr14.pgen  ref_EUR_flipped_chr18.psam\n",
      "ref_EUR_allele-match-libd_chr14.psam  ref_EUR_flipped_chr18.pvar\n",
      "ref_EUR_allele-match-libd_chr14.pvar  ref_EUR_flipped_chr19.log\n",
      "ref_EUR_allele-match-libd_chr15.log   ref_EUR_flipped_chr19.pgen\n",
      "ref_EUR_allele-match-libd_chr15.pgen  ref_EUR_flipped_chr19.psam\n",
      "ref_EUR_allele-match-libd_chr15.psam  ref_EUR_flipped_chr19.pvar\n",
      "ref_EUR_allele-match-libd_chr15.pvar  ref_EUR_flipped_chr1.log\n",
      "ref_EUR_allele-match-libd_chr16.log   ref_EUR_flipped_chr1.pgen\n",
      "ref_EUR_allele-match-libd_chr16.pgen  ref_EUR_flipped_chr1.psam\n",
      "ref_EUR_allele-match-libd_chr16.psam  ref_EUR_flipped_chr1.pvar\n",
      "ref_EUR_allele-match-libd_chr16.pvar  ref_EUR_flipped_chr20.log\n",
      "ref_EUR_allele-match-libd_chr17.log   ref_EUR_flipped_chr20.pgen\n",
      "ref_EUR_allele-match-libd_chr17.pgen  ref_EUR_flipped_chr20.psam\n",
      "ref_EUR_allele-match-libd_chr17.psam  ref_EUR_flipped_chr20.pvar\n",
      "ref_EUR_allele-match-libd_chr17.pvar  ref_EUR_flipped_chr21.log\n",
      "ref_EUR_allele-match-libd_chr18.log   ref_EUR_flipped_chr21.pgen\n",
      "ref_EUR_allele-match-libd_chr18.pgen  ref_EUR_flipped_chr21.psam\n",
      "ref_EUR_allele-match-libd_chr18.psam  ref_EUR_flipped_chr21.pvar\n",
      "ref_EUR_allele-match-libd_chr18.pvar  ref_EUR_flipped_chr22.log\n",
      "ref_EUR_allele-match-libd_chr19.log   ref_EUR_flipped_chr22.pgen\n",
      "ref_EUR_allele-match-libd_chr19.pgen  ref_EUR_flipped_chr22.psam\n",
      "ref_EUR_allele-match-libd_chr19.psam  ref_EUR_flipped_chr22.pvar\n",
      "ref_EUR_allele-match-libd_chr19.pvar  ref_EUR_flipped_chr2.log\n",
      "ref_EUR_allele-match-libd_chr1.log    ref_EUR_flipped_chr2.pgen\n",
      "ref_EUR_allele-match-libd_chr1.pgen   ref_EUR_flipped_chr2.psam\n",
      "ref_EUR_allele-match-libd_chr1.psam   ref_EUR_flipped_chr2.pvar\n",
      "ref_EUR_allele-match-libd_chr1.pvar   ref_EUR_flipped_chr3.log\n",
      "ref_EUR_allele-match-libd_chr20.log   ref_EUR_flipped_chr3.pgen\n",
      "ref_EUR_allele-match-libd_chr20.pgen  ref_EUR_flipped_chr3.psam\n",
      "ref_EUR_allele-match-libd_chr20.psam  ref_EUR_flipped_chr3.pvar\n",
      "ref_EUR_allele-match-libd_chr20.pvar  ref_EUR_flipped_chr4.log\n",
      "ref_EUR_allele-match-libd_chr21.log   ref_EUR_flipped_chr4.pgen\n",
      "ref_EUR_allele-match-libd_chr21.pgen  ref_EUR_flipped_chr4.psam\n",
      "ref_EUR_allele-match-libd_chr21.psam  ref_EUR_flipped_chr4.pvar\n",
      "ref_EUR_allele-match-libd_chr21.pvar  ref_EUR_flipped_chr5.log\n",
      "ref_EUR_allele-match-libd_chr22.log   ref_EUR_flipped_chr5.pgen\n",
      "ref_EUR_allele-match-libd_chr22.pgen  ref_EUR_flipped_chr5.psam\n",
      "ref_EUR_allele-match-libd_chr22.psam  ref_EUR_flipped_chr5.pvar\n",
      "ref_EUR_allele-match-libd_chr22.pvar  ref_EUR_flipped_chr6.log\n",
      "ref_EUR_allele-match-libd_chr2.log    ref_EUR_flipped_chr6.pgen\n",
      "ref_EUR_allele-match-libd_chr2.pgen   ref_EUR_flipped_chr6.psam\n",
      "ref_EUR_allele-match-libd_chr2.psam   ref_EUR_flipped_chr6.pvar\n",
      "ref_EUR_allele-match-libd_chr2.pvar   ref_EUR_flipped_chr7.log\n",
      "ref_EUR_allele-match-libd_chr3.log    ref_EUR_flipped_chr7.pgen\n",
      "ref_EUR_allele-match-libd_chr3.pgen   ref_EUR_flipped_chr7.psam\n",
      "ref_EUR_allele-match-libd_chr3.psam   ref_EUR_flipped_chr7.pvar\n",
      "ref_EUR_allele-match-libd_chr3.pvar   ref_EUR_flipped_chr8.log\n",
      "ref_EUR_allele-match-libd_chr4.log    ref_EUR_flipped_chr8.pgen\n",
      "ref_EUR_allele-match-libd_chr4.pgen   ref_EUR_flipped_chr8.psam\n",
      "ref_EUR_allele-match-libd_chr4.psam   ref_EUR_flipped_chr8.pvar\n",
      "ref_EUR_allele-match-libd_chr4.pvar   ref_EUR_flipped_chr9.log\n",
      "ref_EUR_allele-match-libd_chr5.log    ref_EUR_flipped_chr9.pgen\n",
      "ref_EUR_allele-match-libd_chr5.pgen   ref_EUR_flipped_chr9.psam\n",
      "ref_EUR_allele-match-libd_chr5.psam   ref_EUR_flipped_chr9.pvar\n",
      "ref_EUR_allele-match-libd_chr5.pvar   snps\n",
      "ref_EUR_allele-match-libd_chr6.log    snps_to_flip_chr10.txt\n",
      "ref_EUR_allele-match-libd_chr6.pgen   snps_to_flip_chr11.txt\n",
      "ref_EUR_allele-match-libd_chr6.psam   snps_to_flip_chr12.txt\n",
      "ref_EUR_allele-match-libd_chr6.pvar   snps_to_flip_chr13.txt\n",
      "ref_EUR_allele-match-libd_chr7.log    snps_to_flip_chr14.txt\n",
      "ref_EUR_allele-match-libd_chr7.pgen   snps_to_flip_chr15.txt\n",
      "ref_EUR_allele-match-libd_chr7.psam   snps_to_flip_chr16.txt\n",
      "ref_EUR_allele-match-libd_chr7.pvar   snps_to_flip_chr17.txt\n",
      "ref_EUR_allele-match-libd_chr8.log    snps_to_flip_chr18.txt\n",
      "ref_EUR_allele-match-libd_chr8.pgen   snps_to_flip_chr19.txt\n",
      "ref_EUR_allele-match-libd_chr8.psam   snps_to_flip_chr1.txt\n",
      "ref_EUR_allele-match-libd_chr8.pvar   snps_to_flip_chr20.txt\n",
      "ref_EUR_allele-match-libd_chr9.log    snps_to_flip_chr21.txt\n",
      "ref_EUR_allele-match-libd_chr9.pgen   snps_to_flip_chr22.txt\n",
      "ref_EUR_allele-match-libd_chr9.psam   snps_to_flip_chr2.txt\n",
      "ref_EUR_allele-match-libd_chr9.pvar   snps_to_flip_chr3.txt\n",
      "ref_EUR_chr10.log                     snps_to_flip_chr4.txt\n",
      "ref_EUR_chr10.pgen                    snps_to_flip_chr5.txt\n",
      "ref_EUR_chr10.psam                    snps_to_flip_chr6.txt\n",
      "ref_EUR_chr10.pvar                    snps_to_flip_chr7.txt\n",
      "ref_EUR_chr11.log                     snps_to_flip_chr8.txt\n",
      "ref_EUR_chr11.pgen                    snps_to_flip_chr9.txt\n",
      "ref_EUR_chr11.psam                    snps_to_flip.txt\n",
      "ref_EUR_chr11.pvar\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "14ac22a9-f0d4-4e05-9487-5893c923d8d3",
   "metadata": {},
   "outputs": [],
   "source": [
    "conda activate mwas"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "9b581b5d-76b3-496e-9258-684b49181ab6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing chromosome 1...\n",
      "PLINK v2.00a5.12LM AVX2 AMD (25 Jun 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ref_EUR_allele-match-a2-libd_chr1.log.\n",
      "Options in effect:\n",
      "  --exclude /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/scripts/30-OUT_multi-allele_exclude_list_pt2-from32a2.txt\n",
      "  --make-pgen\n",
      "  --out ref_EUR_allele-match-a2-libd_chr1\n",
      "  --pfile ref_EUR_allele-match-libd_chr1\n",
      "\n",
      "Start time: Mon Aug 19 20:51:56 2024\n",
      "257485 MiB RAM detected, ~205538 available; reserving 128742 MiB for main\n",
      "workspace.\n",
      "Using up to 12 threads (change this with --threads).\n",
      "489 samples (0 females, 0 males, 489 ambiguous; 489 founders) loaded from\n",
      "ref_EUR_allele-match-libd_chr1.psam.\n",
      "448582 variants loaded from ref_EUR_allele-match-libd_chr1.pvar.\n",
      "Note: No phenotype data present.\n",
      "--exclude: 448580 variants remaining.\n",
      "448580 variants remaining after main filters.\n",
      "Writing ref_EUR_allele-match-a2-libd_chr1.psam ... done.\n",
      "Writing ref_EUR_allele-match-a2-libd_chr1.pvar ... 101011111212131314151516161717181819202021212222232324252526262727282829303031313232333334353536363737383839404041414242434344454546464747484849505051515252535354555556565757585859606061616262636364656566666767686869707071717272737374757576767777787879808081818282838384858586868787888889909091919292939394959596969797989899done.\n",
      "Writing ref_EUR_allele-match-a2-libd_chr1.pgen ... 4387done.\n",
      "End time: Mon Aug 19 20:51:57 2024\n",
      "PLINK has been executed to exclude SNPs for chromosome 1.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Processing chromosome 2...\n",
      "PLINK v2.00a5.12LM AVX2 AMD (25 Jun 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ref_EUR_allele-match-a2-libd_chr2.log.\n",
      "Options in effect:\n",
      "  --exclude /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/scripts/30-OUT_multi-allele_exclude_list_pt2-from32a2.txt\n",
      "  --make-pgen\n",
      "  --out ref_EUR_allele-match-a2-libd_chr2\n",
      "  --pfile ref_EUR_allele-match-libd_chr2\n",
      "\n",
      "Start time: Mon Aug 19 20:51:57 2024\n",
      "257485 MiB RAM detected, ~205532 available; reserving 128742 MiB for main\n",
      "workspace.\n",
      "Using up to 12 threads (change this with --threads).\n",
      "489 samples (0 females, 0 males, 489 ambiguous; 489 founders) loaded from\n",
      "ref_EUR_allele-match-libd_chr2.psam.\n",
      "502799 variants loaded from ref_EUR_allele-match-libd_chr2.pvar.\n",
      "Note: No phenotype data present.\n",
      "--exclude: 502799 variants remaining.\n",
      "502799 variants remaining after main filters.\n",
      "Writing ref_EUR_allele-match-a2-libd_chr2.psam ... done.\n",
      "Writing ref_EUR_allele-match-a2-libd_chr2.pvar ... 1010111112121313141415151616171718181919202021212222232324242525262627272828292930303131323233333434353536363737383839394040414142424343444445454646474748484949505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "Writing ref_EUR_allele-match-a2-libd_chr2.pgen ... 3978done.\n",
      "End time: Mon Aug 19 20:51:57 2024\n",
      "PLINK has been executed to exclude SNPs for chromosome 2.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Processing chromosome 3...\n",
      "PLINK v2.00a5.12LM AVX2 AMD (25 Jun 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ref_EUR_allele-match-a2-libd_chr3.log.\n",
      "Options in effect:\n",
      "  --exclude /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/scripts/30-OUT_multi-allele_exclude_list_pt2-from32a2.txt\n",
      "  --make-pgen\n",
      "  --out ref_EUR_allele-match-a2-libd_chr3\n",
      "  --pfile ref_EUR_allele-match-libd_chr3\n",
      "\n",
      "Start time: Mon Aug 19 20:51:57 2024\n",
      "257485 MiB RAM detected, ~205525 available; reserving 128742 MiB for main\n",
      "workspace.\n",
      "Using up to 12 threads (change this with --threads).\n",
      "489 samples (0 females, 0 males, 489 ambiguous; 489 founders) loaded from\n",
      "ref_EUR_allele-match-libd_chr3.psam.\n",
      "429002 variants loaded from ref_EUR_allele-match-libd_chr3.pvar.\n",
      "Note: No phenotype data present.\n",
      "--exclude: 429002 variants remaining.\n",
      "429002 variants remaining after main filters.\n",
      "Writing ref_EUR_allele-match-a2-libd_chr3.psam ... done.\n",
      "Writing ref_EUR_allele-match-a2-libd_chr3.pvar ... 10101111121213131414151516161717181819192020212122222323242425252626272728282929303031313232333334343535363637373838393940404141424243434444454546464747484849505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "Writing ref_EUR_allele-match-a2-libd_chr3.pgen ... 4591done.\n",
      "End time: Mon Aug 19 20:51:58 2024\n",
      "PLINK has been executed to exclude SNPs for chromosome 3.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Processing chromosome 4...\n",
      "PLINK v2.00a5.12LM AVX2 AMD (25 Jun 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ref_EUR_allele-match-a2-libd_chr4.log.\n",
      "Options in effect:\n",
      "  --exclude /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/scripts/30-OUT_multi-allele_exclude_list_pt2-from32a2.txt\n",
      "  --make-pgen\n",
      "  --out ref_EUR_allele-match-a2-libd_chr4\n",
      "  --pfile ref_EUR_allele-match-libd_chr4\n",
      "\n",
      "Start time: Mon Aug 19 20:51:58 2024\n",
      "257485 MiB RAM detected, ~205520 available; reserving 128742 MiB for main\n",
      "workspace.\n",
      "Using up to 12 threads (change this with --threads).\n",
      "489 samples (0 females, 0 males, 489 ambiguous; 489 founders) loaded from\n",
      "ref_EUR_allele-match-libd_chr4.psam.\n",
      "435606 variants loaded from ref_EUR_allele-match-libd_chr4.pvar.\n",
      "Note: No phenotype data present.\n",
      "--exclude: 435606 variants remaining.\n",
      "435606 variants remaining after main filters.\n",
      "Writing ref_EUR_allele-match-a2-libd_chr4.psam ... done.\n",
      "Writing ref_EUR_allele-match-a2-libd_chr4.pvar ... 10101111121213131414151516161717181819192020212122222323242425252626272728282929303031313232333334343535363637373838393940404141424243434444454546464747484849505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "Writing ref_EUR_allele-match-a2-libd_chr4.pgen ... 4590done.\n",
      "End time: Mon Aug 19 20:51:59 2024\n",
      "PLINK has been executed to exclude SNPs for chromosome 4.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Processing chromosome 5...\n",
      "PLINK v2.00a5.12LM AVX2 AMD (25 Jun 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ref_EUR_allele-match-a2-libd_chr5.log.\n",
      "Options in effect:\n",
      "  --exclude /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/scripts/30-OUT_multi-allele_exclude_list_pt2-from32a2.txt\n",
      "  --make-pgen\n",
      "  --out ref_EUR_allele-match-a2-libd_chr5\n",
      "  --pfile ref_EUR_allele-match-libd_chr5\n",
      "\n",
      "Start time: Mon Aug 19 20:51:59 2024\n",
      "257485 MiB RAM detected, ~205514 available; reserving 128742 MiB for main\n",
      "workspace.\n",
      "Using up to 12 threads (change this with --threads).\n",
      "489 samples (0 females, 0 males, 489 ambiguous; 489 founders) loaded from\n",
      "ref_EUR_allele-match-libd_chr5.psam.\n",
      "380612 variants loaded from ref_EUR_allele-match-libd_chr5.pvar.\n",
      "Note: No phenotype data present.\n",
      "--exclude: 380611 variants remaining.\n",
      "380611 variants remaining after main filters.\n",
      "Writing ref_EUR_allele-match-a2-libd_chr5.psam ... done.\n",
      "Writing ref_EUR_allele-match-a2-libd_chr5.pvar ... 1010111112121313141415151616171718181919202021212222232324242525262627272828292930303131323233333434353536363737383839394040414142424343444445454646474748484949505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "Writing ref_EUR_allele-match-a2-libd_chr5.pgen ... 51done.\n",
      "End time: Mon Aug 19 20:52:00 2024\n",
      "PLINK has been executed to exclude SNPs for chromosome 5.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Processing chromosome 6...\n",
      "PLINK v2.00a5.12LM AVX2 AMD (25 Jun 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ref_EUR_allele-match-a2-libd_chr6.log.\n",
      "Options in effect:\n",
      "  --exclude /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/scripts/30-OUT_multi-allele_exclude_list_pt2-from32a2.txt\n",
      "  --make-pgen\n",
      "  --out ref_EUR_allele-match-a2-libd_chr6\n",
      "  --pfile ref_EUR_allele-match-libd_chr6\n",
      "\n",
      "Start time: Mon Aug 19 20:52:00 2024\n",
      "257485 MiB RAM detected, ~205497 available; reserving 128742 MiB for main\n",
      "workspace.\n",
      "Using up to 12 threads (change this with --threads).\n",
      "489 samples (0 females, 0 males, 489 ambiguous; 489 founders) loaded from\n",
      "ref_EUR_allele-match-libd_chr6.psam.\n",
      "373969 variants loaded from ref_EUR_allele-match-libd_chr6.pvar.\n",
      "Note: No phenotype data present.\n",
      "--exclude: 373969 variants remaining.\n",
      "373969 variants remaining after main filters.\n",
      "Writing ref_EUR_allele-match-a2-libd_chr6.psam ... done.\n",
      "Writing ref_EUR_allele-match-a2-libd_chr6.pvar ... 1010111112121313141415151616171718181919202021212222232324242525262627272828292930303131323233333434353536363737383839394040414142424343444445454646474748484949505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "Writing ref_EUR_allele-match-a2-libd_chr6.pgen ... 52done.\n",
      "End time: Mon Aug 19 20:52:01 2024\n",
      "PLINK has been executed to exclude SNPs for chromosome 6.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Processing chromosome 7...\n",
      "PLINK v2.00a5.12LM AVX2 AMD (25 Jun 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ref_EUR_allele-match-a2-libd_chr7.log.\n",
      "Options in effect:\n",
      "  --exclude /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/scripts/30-OUT_multi-allele_exclude_list_pt2-from32a2.txt\n",
      "  --make-pgen\n",
      "  --out ref_EUR_allele-match-a2-libd_chr7\n",
      "  --pfile ref_EUR_allele-match-libd_chr7\n",
      "\n",
      "Start time: Mon Aug 19 20:52:01 2024\n",
      "257485 MiB RAM detected, ~205497 available; reserving 128742 MiB for main\n",
      "workspace.\n",
      "Using up to 12 threads (change this with --threads).\n",
      "489 samples (0 females, 0 males, 489 ambiguous; 489 founders) loaded from\n",
      "ref_EUR_allele-match-libd_chr7.psam.\n",
      "345874 variants loaded from ref_EUR_allele-match-libd_chr7.pvar.\n",
      "Note: No phenotype data present.\n",
      "--exclude: 345873 variants remaining.\n",
      "345873 variants remaining after main filters.\n",
      "Writing ref_EUR_allele-match-a2-libd_chr7.psam ... done.\n",
      "Writing ref_EUR_allele-match-a2-libd_chr7.pvar ... 1010111112121313141415151616171718181919202021212222232324242525262627272828292930303131323233333434353536363737383839394040414142424343444445454646474748484949505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "Writing ref_EUR_allele-match-a2-libd_chr7.pgen ... 56done.\n",
      "End time: Mon Aug 19 20:52:02 2024\n",
      "PLINK has been executed to exclude SNPs for chromosome 7.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Processing chromosome 8...\n",
      "PLINK v2.00a5.12LM AVX2 AMD (25 Jun 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ref_EUR_allele-match-a2-libd_chr8.log.\n",
      "Options in effect:\n",
      "  --exclude /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/scripts/30-OUT_multi-allele_exclude_list_pt2-from32a2.txt\n",
      "  --make-pgen\n",
      "  --out ref_EUR_allele-match-a2-libd_chr8\n",
      "  --pfile ref_EUR_allele-match-libd_chr8\n",
      "\n",
      "Start time: Mon Aug 19 20:52:02 2024\n",
      "257485 MiB RAM detected, ~205492 available; reserving 128742 MiB for main\n",
      "workspace.\n",
      "Using up to 12 threads (change this with --threads).\n",
      "489 samples (0 females, 0 males, 489 ambiguous; 489 founders) loaded from\n",
      "ref_EUR_allele-match-libd_chr8.psam.\n",
      "336686 variants loaded from ref_EUR_allele-match-libd_chr8.pvar.\n",
      "Note: No phenotype data present.\n",
      "--exclude: 336685 variants remaining.\n",
      "336685 variants remaining after main filters.\n",
      "Writing ref_EUR_allele-match-a2-libd_chr8.psam ... done.\n",
      "Writing ref_EUR_allele-match-a2-libd_chr8.pvar ... 10101111121213131414151516161717181819202021212222232324242525262627272828292930303131323233333434353536363737383839404041414242434344444545464647474848494950505151525253535454555556565757585859606061616262636364646565666667676868696970707171727273737474757576767777787879808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "Writing ref_EUR_allele-match-a2-libd_chr8.pgen ... 58done.\n",
      "End time: Mon Aug 19 20:52:03 2024\n",
      "PLINK has been executed to exclude SNPs for chromosome 8.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Processing chromosome 9...\n",
      "PLINK v2.00a5.12LM AVX2 AMD (25 Jun 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ref_EUR_allele-match-a2-libd_chr9.log.\n",
      "Options in effect:\n",
      "  --exclude /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/scripts/30-OUT_multi-allele_exclude_list_pt2-from32a2.txt\n",
      "  --make-pgen\n",
      "  --out ref_EUR_allele-match-a2-libd_chr9\n",
      "  --pfile ref_EUR_allele-match-libd_chr9\n",
      "\n",
      "Start time: Mon Aug 19 20:52:03 2024\n",
      "257485 MiB RAM detected, ~205487 available; reserving 128742 MiB for main\n",
      "workspace.\n",
      "Using up to 12 threads (change this with --threads).\n",
      "489 samples (0 females, 0 males, 489 ambiguous; 489 founders) loaded from\n",
      "ref_EUR_allele-match-libd_chr9.psam.\n",
      "253860 variants loaded from ref_EUR_allele-match-libd_chr9.pvar.\n",
      "Note: No phenotype data present.\n",
      "--exclude: 253858 variants remaining.\n",
      "253858 variants remaining after main filters.\n",
      "Writing ref_EUR_allele-match-a2-libd_chr9.psam ... done.\n",
      "Writing ref_EUR_allele-match-a2-libd_chr9.pvar ... 10101111121213131414151516161717181819192020212122222323242425252626272728282929303031313232333334343535363637373838393940404141424243434444454546464747484849505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "Writing ref_EUR_allele-match-a2-libd_chr9.pgen ... 77done.\n",
      "End time: Mon Aug 19 20:52:04 2024\n",
      "PLINK has been executed to exclude SNPs for chromosome 9.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Processing chromosome 10...\n",
      "PLINK v2.00a5.12LM AVX2 AMD (25 Jun 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ref_EUR_allele-match-a2-libd_chr10.log.\n",
      "Options in effect:\n",
      "  --exclude /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/scripts/30-OUT_multi-allele_exclude_list_pt2-from32a2.txt\n",
      "  --make-pgen\n",
      "  --out ref_EUR_allele-match-a2-libd_chr10\n",
      "  --pfile ref_EUR_allele-match-libd_chr10\n",
      "\n",
      "Start time: Mon Aug 19 20:52:04 2024\n",
      "257485 MiB RAM detected, ~205483 available; reserving 128742 MiB for main\n",
      "workspace.\n",
      "Using up to 12 threads (change this with --threads).\n",
      "489 samples (0 females, 0 males, 489 ambiguous; 489 founders) loaded from\n",
      "ref_EUR_allele-match-libd_chr10.psam.\n",
      "305527 variants loaded from ref_EUR_allele-match-libd_chr10.pvar.\n",
      "Note: No phenotype data present.\n",
      "--exclude: 305527 variants remaining.\n",
      "305527 variants remaining after main filters.\n",
      "Writing ref_EUR_allele-match-a2-libd_chr10.psam ... done.\n",
      "Writing ref_EUR_allele-match-a2-libd_chr10.pvar ... 1010111112121313141415151616171718181919202021212222232324242525262627272828292930303131323233333434353536363737383839394040414142424343444445454646474748484949505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "Writing ref_EUR_allele-match-a2-libd_chr10.pgen ... 64done.\n",
      "End time: Mon Aug 19 20:52:05 2024\n",
      "PLINK has been executed to exclude SNPs for chromosome 10.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Processing chromosome 11...\n",
      "PLINK v2.00a5.12LM AVX2 AMD (25 Jun 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ref_EUR_allele-match-a2-libd_chr11.log.\n",
      "Options in effect:\n",
      "  --exclude /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/scripts/30-OUT_multi-allele_exclude_list_pt2-from32a2.txt\n",
      "  --make-pgen\n",
      "  --out ref_EUR_allele-match-a2-libd_chr11\n",
      "  --pfile ref_EUR_allele-match-libd_chr11\n",
      "\n",
      "Start time: Mon Aug 19 20:52:05 2024\n",
      "257485 MiB RAM detected, ~205480 available; reserving 128742 MiB for main\n",
      "workspace.\n",
      "Using up to 12 threads (change this with --threads).\n",
      "489 samples (0 females, 0 males, 489 ambiguous; 489 founders) loaded from\n",
      "ref_EUR_allele-match-libd_chr11.psam.\n",
      "293499 variants loaded from ref_EUR_allele-match-libd_chr11.pvar.\n",
      "Note: No phenotype data present.\n",
      "--exclude: 293498 variants remaining.\n",
      "293498 variants remaining after main filters.\n",
      "Writing ref_EUR_allele-match-a2-libd_chr11.psam ... done.\n",
      "Writing ref_EUR_allele-match-a2-libd_chr11.pvar ... 10101111121213131414151516161717181819192020212122222323242425252626272728282929303031313232333334343535363637373838393940404141424243434444454546464747484849505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "Writing ref_EUR_allele-match-a2-libd_chr11.pgen ... 66done.\n",
      "End time: Mon Aug 19 20:52:05 2024\n",
      "PLINK has been executed to exclude SNPs for chromosome 11.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Processing chromosome 12...\n",
      "PLINK v2.00a5.12LM AVX2 AMD (25 Jun 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ref_EUR_allele-match-a2-libd_chr12.log.\n",
      "Options in effect:\n",
      "  --exclude /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/scripts/30-OUT_multi-allele_exclude_list_pt2-from32a2.txt\n",
      "  --make-pgen\n",
      "  --out ref_EUR_allele-match-a2-libd_chr12\n",
      "  --pfile ref_EUR_allele-match-libd_chr12\n",
      "\n",
      "Start time: Mon Aug 19 20:52:05 2024\n",
      "257485 MiB RAM detected, ~205476 available; reserving 128742 MiB for main\n",
      "workspace.\n",
      "Using up to 12 threads (change this with --threads).\n",
      "489 samples (0 females, 0 males, 489 ambiguous; 489 founders) loaded from\n",
      "ref_EUR_allele-match-libd_chr12.psam.\n",
      "278630 variants loaded from ref_EUR_allele-match-libd_chr12.pvar.\n",
      "Note: No phenotype data present.\n",
      "--exclude: 278630 variants remaining.\n",
      "278630 variants remaining after main filters.\n",
      "Writing ref_EUR_allele-match-a2-libd_chr12.psam ... done.\n",
      "Writing ref_EUR_allele-match-a2-libd_chr12.pvar ... 101011111212131314141515161617171818192020212122222323242425252626272728282930303131323233333434353536363737383839404041414242434344444545464647474848495050515152525353545455555656575758585960606161626263636464656566666767686869707071717272737374747575767677777878798080818182828383848485858686878788888990909191929293939494959596969797989899done.\n",
      "Writing ref_EUR_allele-match-a2-libd_chr12.pgen ... 70done.\n",
      "End time: Mon Aug 19 20:52:06 2024\n",
      "PLINK has been executed to exclude SNPs for chromosome 12.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Processing chromosome 13...\n",
      "PLINK v2.00a5.12LM AVX2 AMD (25 Jun 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ref_EUR_allele-match-a2-libd_chr13.log.\n",
      "Options in effect:\n",
      "  --exclude /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/scripts/30-OUT_multi-allele_exclude_list_pt2-from32a2.txt\n",
      "  --make-pgen\n",
      "  --out ref_EUR_allele-match-a2-libd_chr13\n",
      "  --pfile ref_EUR_allele-match-libd_chr13\n",
      "\n",
      "Start time: Mon Aug 19 20:52:06 2024\n",
      "257485 MiB RAM detected, ~205472 available; reserving 128742 MiB for main\n",
      "workspace.\n",
      "Using up to 12 threads (change this with --threads).\n",
      "489 samples (0 females, 0 males, 489 ambiguous; 489 founders) loaded from\n",
      "ref_EUR_allele-match-libd_chr13.psam.\n",
      "225271 variants loaded from ref_EUR_allele-match-libd_chr13.pvar.\n",
      "Note: No phenotype data present.\n",
      "--exclude: 225271 variants remaining.\n",
      "225271 variants remaining after main filters.\n",
      "Writing ref_EUR_allele-match-a2-libd_chr13.psam ... done.\n",
      "Writing ref_EUR_allele-match-a2-libd_chr13.pvar ... 1010111112121313141415151616171718181919202021212222232324242525262627272828292930303131323233333434353536363737383839394040414142424343444445454646474748484949505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "Writing ref_EUR_allele-match-a2-libd_chr13.pgen ... 87done.\n",
      "End time: Mon Aug 19 20:52:08 2024\n",
      "PLINK has been executed to exclude SNPs for chromosome 13.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Processing chromosome 14...\n",
      "PLINK v2.00a5.12LM AVX2 AMD (25 Jun 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ref_EUR_allele-match-a2-libd_chr14.log.\n",
      "Options in effect:\n",
      "  --exclude /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/scripts/30-OUT_multi-allele_exclude_list_pt2-from32a2.txt\n",
      "  --make-pgen\n",
      "  --out ref_EUR_allele-match-a2-libd_chr14\n",
      "  --pfile ref_EUR_allele-match-libd_chr14\n",
      "\n",
      "Start time: Mon Aug 19 20:52:08 2024\n",
      "257485 MiB RAM detected, ~205469 available; reserving 128742 MiB for main\n",
      "workspace.\n",
      "Using up to 12 threads (change this with --threads).\n",
      "489 samples (0 females, 0 males, 489 ambiguous; 489 founders) loaded from\n",
      "ref_EUR_allele-match-libd_chr14.psam.\n",
      "192569 variants loaded from ref_EUR_allele-match-libd_chr14.pvar.\n",
      "Note: No phenotype data present.\n",
      "--exclude: 192568 variants remaining.\n",
      "192568 variants remaining after main filters.\n",
      "Writing ref_EUR_allele-match-a2-libd_chr14.psam ... done.\n",
      "Writing ref_EUR_allele-match-a2-libd_chr14.pvar ... 1010111112121313141415151616171718181919202021212222232324252526262727282829293030313132323333343435353636373738383939404041414242434344444545464647474848495050515152525353545455555656575758585959606061616262636364646565666667676868696970707171727273737475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "Writing ref_EUR_allele-match-a2-libd_chr14.pgen ... done.\n",
      "End time: Mon Aug 19 20:52:08 2024\n",
      "PLINK has been executed to exclude SNPs for chromosome 14.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Processing chromosome 15...\n",
      "PLINK v2.00a5.12LM AVX2 AMD (25 Jun 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ref_EUR_allele-match-a2-libd_chr15.log.\n",
      "Options in effect:\n",
      "  --exclude /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/scripts/30-OUT_multi-allele_exclude_list_pt2-from32a2.txt\n",
      "  --make-pgen\n",
      "  --out ref_EUR_allele-match-a2-libd_chr15\n",
      "  --pfile ref_EUR_allele-match-libd_chr15\n",
      "\n",
      "Start time: Mon Aug 19 20:52:08 2024\n",
      "257485 MiB RAM detected, ~205286 available; reserving 128742 MiB for main\n",
      "workspace.\n",
      "Using up to 12 threads (change this with --threads).\n",
      "489 samples (0 females, 0 males, 489 ambiguous; 489 founders) loaded from\n",
      "ref_EUR_allele-match-libd_chr15.psam.\n",
      "165315 variants loaded from ref_EUR_allele-match-libd_chr15.pvar.\n",
      "Note: No phenotype data present.\n",
      "--exclude: 165315 variants remaining.\n",
      "165315 variants remaining after main filters.\n",
      "Writing ref_EUR_allele-match-a2-libd_chr15.psam ... done.\n",
      "Writing ref_EUR_allele-match-a2-libd_chr15.pvar ... 10101111121213131414151516161717181819202021212222232324242525262627272828292930303131323233333434353536363737383839404041414242434344444545464647474848494950505151525253535454555556565757585859606061616262636364646565666667676868696970707171727273737474757576767777787879808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "Writing ref_EUR_allele-match-a2-libd_chr15.pgen ... done.\n",
      "End time: Mon Aug 19 20:52:09 2024\n",
      "PLINK has been executed to exclude SNPs for chromosome 15.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Processing chromosome 16...\n",
      "PLINK v2.00a5.12LM AVX2 AMD (25 Jun 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ref_EUR_allele-match-a2-libd_chr16.log.\n",
      "Options in effect:\n",
      "  --exclude /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/scripts/30-OUT_multi-allele_exclude_list_pt2-from32a2.txt\n",
      "  --make-pgen\n",
      "  --out ref_EUR_allele-match-a2-libd_chr16\n",
      "  --pfile ref_EUR_allele-match-libd_chr16\n",
      "\n",
      "Start time: Mon Aug 19 20:52:09 2024\n",
      "257485 MiB RAM detected, ~205285 available; reserving 128742 MiB for main\n",
      "workspace.\n",
      "Using up to 12 threads (change this with --threads).\n",
      "489 samples (0 females, 0 males, 489 ambiguous; 489 founders) loaded from\n",
      "ref_EUR_allele-match-libd_chr16.psam.\n",
      "179545 variants loaded from ref_EUR_allele-match-libd_chr16.pvar.\n",
      "Note: No phenotype data present.\n",
      "--exclude: 179545 variants remaining.\n",
      "179545 variants remaining after main filters.\n",
      "Writing ref_EUR_allele-match-a2-libd_chr16.psam ... done.\n",
      "Writing ref_EUR_allele-match-a2-libd_chr16.pvar ... 10101111121213131414151516161717181819202021212222232324242525262627272828292930303131323233333434353536363737383839404041414242434344444545464647474848494950505151525253535454555556565757585859606061616262636364646565666667676868696970707171727273737474757576767777787879808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "Writing ref_EUR_allele-match-a2-libd_chr16.pgen ... done.\n",
      "End time: Mon Aug 19 20:52:12 2024\n",
      "PLINK has been executed to exclude SNPs for chromosome 16.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Processing chromosome 17...\n",
      "PLINK v2.00a5.12LM AVX2 AMD (25 Jun 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ref_EUR_allele-match-a2-libd_chr17.log.\n",
      "Options in effect:\n",
      "  --exclude /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/scripts/30-OUT_multi-allele_exclude_list_pt2-from32a2.txt\n",
      "  --make-pgen\n",
      "  --out ref_EUR_allele-match-a2-libd_chr17\n",
      "  --pfile ref_EUR_allele-match-libd_chr17\n",
      "\n",
      "Start time: Mon Aug 19 20:52:12 2024\n",
      "257485 MiB RAM detected, ~205283 available; reserving 128742 MiB for main\n",
      "workspace.\n",
      "Using up to 12 threads (change this with --threads).\n",
      "489 samples (0 females, 0 males, 489 ambiguous; 489 founders) loaded from\n",
      "ref_EUR_allele-match-libd_chr17.psam.\n",
      "149786 variants loaded from ref_EUR_allele-match-libd_chr17.pvar.\n",
      "Note: No phenotype data present.\n",
      "--exclude: 149786 variants remaining.\n",
      "149786 variants remaining after main filters.\n",
      "Writing ref_EUR_allele-match-a2-libd_chr17.psam ... done.\n",
      "Writing ref_EUR_allele-match-a2-libd_chr17.pvar ... 10101111121213131414151516161717181819192020212122222323242425252626272728282929303031313232333334343535363637373838393940404141424243434444454546464747484849505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "Writing ref_EUR_allele-match-a2-libd_chr17.pgen ... done.\n",
      "End time: Mon Aug 19 20:52:12 2024\n",
      "PLINK has been executed to exclude SNPs for chromosome 17.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Processing chromosome 18...\n",
      "PLINK v2.00a5.12LM AVX2 AMD (25 Jun 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ref_EUR_allele-match-a2-libd_chr18.log.\n",
      "Options in effect:\n",
      "  --exclude /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/scripts/30-OUT_multi-allele_exclude_list_pt2-from32a2.txt\n",
      "  --make-pgen\n",
      "  --out ref_EUR_allele-match-a2-libd_chr18\n",
      "  --pfile ref_EUR_allele-match-libd_chr18\n",
      "\n",
      "Start time: Mon Aug 19 20:52:12 2024\n",
      "257485 MiB RAM detected, ~205280 available; reserving 128742 MiB for main\n",
      "workspace.\n",
      "Using up to 12 threads (change this with --threads).\n",
      "489 samples (0 females, 0 males, 489 ambiguous; 489 founders) loaded from\n",
      "ref_EUR_allele-match-libd_chr18.psam.\n",
      "171245 variants loaded from ref_EUR_allele-match-libd_chr18.pvar.\n",
      "Note: No phenotype data present.\n",
      "--exclude: 171245 variants remaining.\n",
      "171245 variants remaining after main filters.\n",
      "Writing ref_EUR_allele-match-a2-libd_chr18.psam ... done.\n",
      "Writing ref_EUR_allele-match-a2-libd_chr18.pvar ... 10101111121213131414151516161717181819202021212222232324242525262627272828292930303131323233333434353536363737383839404041414242434344444545464647474848494950505151525253535454555556565757585859606061616262636364646565666667676868696970707171727273737474757576767777787879808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "Writing ref_EUR_allele-match-a2-libd_chr18.pgen ... done.\n",
      "End time: Mon Aug 19 20:52:13 2024\n",
      "PLINK has been executed to exclude SNPs for chromosome 18.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Processing chromosome 19...\n",
      "PLINK v2.00a5.12LM AVX2 AMD (25 Jun 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ref_EUR_allele-match-a2-libd_chr19.log.\n",
      "Options in effect:\n",
      "  --exclude /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/scripts/30-OUT_multi-allele_exclude_list_pt2-from32a2.txt\n",
      "  --make-pgen\n",
      "  --out ref_EUR_allele-match-a2-libd_chr19\n",
      "  --pfile ref_EUR_allele-match-libd_chr19\n",
      "\n",
      "Start time: Mon Aug 19 20:52:13 2024\n",
      "257485 MiB RAM detected, ~205279 available; reserving 128742 MiB for main\n",
      "workspace.\n",
      "Using up to 12 threads (change this with --threads).\n",
      "489 samples (0 females, 0 males, 489 ambiguous; 489 founders) loaded from\n",
      "ref_EUR_allele-match-libd_chr19.psam.\n",
      "118848 variants loaded from ref_EUR_allele-match-libd_chr19.pvar.\n",
      "Note: No phenotype data present.\n",
      "--exclude: 118848 variants remaining.\n",
      "118848 variants remaining after main filters.\n",
      "Writing ref_EUR_allele-match-a2-libd_chr19.psam ... done.\n",
      "Writing ref_EUR_allele-match-a2-libd_chr19.pvar ... 1010111112121313141415151616171718181919202021212222232324252526262727282829293030313132323333343435353636373738383939404041414242434344444545464647474848495050515152525353545455555656575758585959606061616262636364646565666667676868696970707171727273737475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "Writing ref_EUR_allele-match-a2-libd_chr19.pgen ... done.\n",
      "End time: Mon Aug 19 20:52:14 2024\n",
      "PLINK has been executed to exclude SNPs for chromosome 19.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Processing chromosome 20...\n",
      "PLINK v2.00a5.12LM AVX2 AMD (25 Jun 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ref_EUR_allele-match-a2-libd_chr20.log.\n",
      "Options in effect:\n",
      "  --exclude /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/scripts/30-OUT_multi-allele_exclude_list_pt2-from32a2.txt\n",
      "  --make-pgen\n",
      "  --out ref_EUR_allele-match-a2-libd_chr20\n",
      "  --pfile ref_EUR_allele-match-libd_chr20\n",
      "\n",
      "Start time: Mon Aug 19 20:52:14 2024\n",
      "257485 MiB RAM detected, ~205278 available; reserving 128742 MiB for main\n",
      "workspace.\n",
      "Using up to 12 threads (change this with --threads).\n",
      "489 samples (0 females, 0 males, 489 ambiguous; 489 founders) loaded from\n",
      "ref_EUR_allele-match-libd_chr20.psam.\n",
      "133791 variants loaded from ref_EUR_allele-match-libd_chr20.pvar.\n",
      "Note: No phenotype data present.\n",
      "--exclude: 133790 variants remaining.\n",
      "133790 variants remaining after main filters.\n",
      "Writing ref_EUR_allele-match-a2-libd_chr20.psam ... done.\n",
      "Writing ref_EUR_allele-match-a2-libd_chr20.pvar ... 101011111212131314141515161617171818192020212122222323242425252626272728282930303131323233333434353536363737383839404041414242434344444545464647474848495050515152525353545455555656575758585960606161626263636464656566666767686869707071717272737374747575767677777878798080818182828383848485858686878788888990909191929293939494959596969797989899done.\n",
      "Writing ref_EUR_allele-match-a2-libd_chr20.pgen ... done.\n",
      "End time: Mon Aug 19 20:52:15 2024\n",
      "PLINK has been executed to exclude SNPs for chromosome 20.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Processing chromosome 21...\n",
      "PLINK v2.00a5.12LM AVX2 AMD (25 Jun 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ref_EUR_allele-match-a2-libd_chr21.log.\n",
      "Options in effect:\n",
      "  --exclude /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/scripts/30-OUT_multi-allele_exclude_list_pt2-from32a2.txt\n",
      "  --make-pgen\n",
      "  --out ref_EUR_allele-match-a2-libd_chr21\n",
      "  --pfile ref_EUR_allele-match-libd_chr21\n",
      "\n",
      "Start time: Mon Aug 19 20:52:15 2024\n",
      "257485 MiB RAM detected, ~205275 available; reserving 128742 MiB for main\n",
      "workspace.\n",
      "Using up to 12 threads (change this with --threads).\n",
      "489 samples (0 females, 0 males, 489 ambiguous; 489 founders) loaded from\n",
      "ref_EUR_allele-match-libd_chr21.psam.\n",
      "82864 variants loaded from ref_EUR_allele-match-libd_chr21.pvar.\n",
      "Note: No phenotype data present.\n",
      "--exclude: 82864 variants remaining.\n",
      "82864 variants remaining after main filters.\n",
      "Writing ref_EUR_allele-match-a2-libd_chr21.psam ... done.\n",
      "Writing ref_EUR_allele-match-a2-libd_chr21.pvar ... 1010111112121313141415151616171718181919202021212222232324252526262727282829293030313132323333343435353636373738383939404041414242434344444545464647474848495050515152525353545455555656575758585959606061616262636364646565666667676868696970707171727273737475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "Writing ref_EUR_allele-match-a2-libd_chr21.pgen ... done.\n",
      "End time: Mon Aug 19 20:52:16 2024\n",
      "PLINK has been executed to exclude SNPs for chromosome 21.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Processing chromosome 22...\n",
      "PLINK v2.00a5.12LM AVX2 AMD (25 Jun 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ref_EUR_allele-match-a2-libd_chr22.log.\n",
      "Options in effect:\n",
      "  --exclude /expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/scripts/30-OUT_multi-allele_exclude_list_pt2-from32a2.txt\n",
      "  --make-pgen\n",
      "  --out ref_EUR_allele-match-a2-libd_chr22\n",
      "  --pfile ref_EUR_allele-match-libd_chr22\n",
      "\n",
      "Start time: Mon Aug 19 20:52:16 2024\n",
      "257485 MiB RAM detected, ~205273 available; reserving 128742 MiB for main\n",
      "workspace.\n",
      "Using up to 12 threads (change this with --threads).\n",
      "489 samples (0 females, 0 males, 489 ambiguous; 489 founders) loaded from\n",
      "ref_EUR_allele-match-libd_chr22.psam.\n",
      "75839 variants loaded from ref_EUR_allele-match-libd_chr22.pvar.\n",
      "Note: No phenotype data present.\n",
      "--exclude: 75839 variants remaining.\n",
      "75839 variants remaining after main filters.\n",
      "Writing ref_EUR_allele-match-a2-libd_chr22.psam ... done.\n",
      "Writing ref_EUR_allele-match-a2-libd_chr22.pvar ... 1010111112121313141415151616171718181919202021212222232324242525262627272828292930303131323233333434353536363737383839394040414142424343444445454646474748484949505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "Writing ref_EUR_allele-match-a2-libd_chr22.pgen ... done.\n",
      "End time: Mon Aug 19 20:52:17 2024\n",
      "PLINK has been executed to exclude SNPs for chromosome 22.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Path to the PLINK binary\n",
    "plink_path=\"./plink2\"\n",
    "\n",
    "# Loop through chromosomes 1 to 22\n",
    "for chr in {1..22}\n",
    "do\n",
    "  echo \"Processing chromosome $chr...\"\n",
    "\n",
    "  # Define file paths for this chromosome\n",
    "  input_pfile=\"ref_EUR_allele-match-libd_chr${chr}\"\n",
    "  output_pfile=\"ref_EUR_allele-match-a2-libd_chr${chr}\"\n",
    "\n",
    "  # Enclose paths in quotes to handle special characters and spaces\n",
    "  exclude_list_path=\"/expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/scripts/30-OUT_multi-allele_exclude_list_pt2-from32a2.txt\"\n",
    "\n",
    "  # Check if the exclusion list file is not empty\n",
    "  if [ -s \"$exclude_list_path\" ]; then\n",
    "    # Run PLINK to exclude the SNPs listed in the exclusion list\n",
    "    $plink_path --pfile \"$input_pfile\" --exclude \"$exclude_list_path\" --make-pgen --out \"$output_pfile\"\n",
    "\n",
    "    echo \"PLINK has been executed to exclude SNPs for chromosome $chr.\"\n",
    "  else\n",
    "    echo \"No SNPs to exclude for chromosome $chr, or an error occurred in finding the exclusion list.\"\n",
    "  fi\n",
    "\n",
    "  # Add multiple blank lines for readability between iterations\n",
    "  echo -e \"\\n\\n\\n\\n\\n\"\n",
    "done\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bea01300-e604-4067-ace9-eacab27dd116",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
