{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "8ab7eb87-1623-48c0-9530-78e76fa64adc",
   "metadata": {},
   "source": [
    "# Again try running stage 2 MWAS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "653dfc4e-c218-4268-b6eb-84ba090ad23d",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: foreach\n",
      "\n",
      "Loading required package: iterators\n",
      "\n",
      "Loading required package: parallel\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(CpGWAS)\n",
    "library(data.table)\n",
    "library(stringr)\n",
    "library(optparse)\n",
    "library(doParallel)\n",
    "library(foreach)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "7a5b9066-82eb-439f-ad20-076be0602d5d",
   "metadata": {},
   "outputs": [],
   "source": [
    "cl <- makeCluster(120)\n",
    "registerDoParallel(cl)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "60cea74c-470e-4c11-98fe-17f2d1aa9c6c",
   "metadata": {},
   "outputs": [],
   "source": [
    "for(chr in 1:22){\n",
    "    opt <- list(genome_file_index = chr,\n",
    "                data_file = \"/expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/scripts/12-OUT_matched_SNP_meth_cov_outputs.csv\",\n",
    "                cores = 120)\n",
    "    \n",
    "    df <- fread(opt$data_file)\n",
    "    \n",
    "    # Load genome files\n",
    "    genome_files <- list.files(\"/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped\",\n",
    "                               pattern = \"EUR\", full.names = TRUE)\n",
    "    genome_files <- genome_files[grepl(\"allele\", genome_files)]\n",
    "    genome_files <- genome_files[grepl(\"pvar\", genome_files)]\n",
    "    genome_files <- data.table(path = genome_files, Chr = NA)\n",
    "    genome_files$Chr <- str_split_fixed(genome_files$path, \"chr\", 2)[, 2]\n",
    "    genome_files$Chr <- gsub(\".pvar\", \"\", genome_files$Chr)\n",
    "    genome_files$Chr <- as.integer(genome_files$Chr)\n",
    "    genome_files <- genome_files[order(genome_files$Chr), ]\n",
    "    \n",
    "    print(\"Starting genome file processing\")\n",
    "    g <- opt$genome_file_index\n",
    "    print(paste(\"Processing genome file index:\", g))\n",
    "    \n",
    "    paths <- list(\n",
    "      pvar_path = genome_files[g]$path,\n",
    "      pgen_path = gsub(\"pvar\", \"pgen\", genome_files[g]$path),\n",
    "      psam_path = gsub(\"pvar\", \"psam\", genome_files[g]$path)\n",
    "    )\n",
    "\n",
    "    df_this_chr <- df[which(df$Chr == genome_files[g]$Chr), ]\n",
    "    \n",
    "    summary_stats_list <- list.files(\"/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped\", pattern = \"alleleprocessed\", full.names = TRUE)\n",
    "    \n",
    "    summary_stats_data <- lapply(summary_stats_list, function(path) {\n",
    "       stats <- suppressWarnings(data.table::fread(path))\n",
    "       setkey(stats, SNP, CHR)\n",
    "       stats <- stats[CHR == genome_files[g]$Chr]\n",
    "       return(stats)\n",
    "    })\n",
    "    \n",
    "    df_parts <- split(df_this_chr, cut(seq_len(nrow(df_this_chr)), opt$cores, labels = FALSE))\n",
    "\n",
    "# Parallel processing\n",
    "    foreach(part_idx = seq_along(df_parts), .packages = c(\"data.table\", \"CpGWAS\")) %dopar% {\n",
    "      df_part <- df_parts[[part_idx]]\n",
    "    \n",
    "      # Load SNPs within each worker\n",
    "      my_SNPs <- CpGWAS::loadSNPData(paths$pvar_path, paths$pgen_path, paths$psam_path)\n",
    "      \n",
    "      for(j in 1:nrow(df_part)) {\n",
    "        if(j %% 10 == 0) print(j)\n",
    "        print(paste0(\"File number: \", j))\n",
    "        if (grepl(\"empty\", df_part$path[j])) {\n",
    "          message(paste0(\"no model for \", df_part$path[j]))\n",
    "          next\n",
    "        }\n",
    "    \n",
    "        all_files_exist <- TRUE\n",
    "        outnames <- vector(\"character\", length(summary_stats_list))\n",
    "    \n",
    "        for (k in 1:length(summary_stats_list)) {\n",
    "          outnames[k] <- gsub(\"\\\\.rds$\", paste0(\"_\", basename(tools::file_path_sans_ext(summary_stats_list[[k]])), \"_results-par.rds\"), df_part$path[j])\n",
    "          if (!file.exists(outnames[k])) {\n",
    "            all_files_exist <- FALSE\n",
    "            break\n",
    "          }\n",
    "        }\n",
    "    \n",
    "        if (all_files_exist) {\n",
    "          print(paste(\"All output files already exist for\", df_part$path[j], \"- Skipping\"))\n",
    "          next\n",
    "        }\n",
    "\n",
    "        # Load my_rds within each worker\n",
    "        my_rds <- tryCatch({\n",
    "          readRDS(df_part$path[j])\n",
    "        }, error = function(e) {\n",
    "          message(\"ALERT!!! Error reading RDS file: \", e$message)\n",
    "          return(NULL)\n",
    "        })\n",
    "    \n",
    "        if (is.null(my_rds)) {\n",
    "          next\n",
    "        }\n",
    "    \n",
    "        print(paste(\"Loaded RDS file:\", df_part$path[j]))\n",
    "    \n",
    "        for (k in 1:length(summary_stats_list)) {\n",
    "          outname <- gsub(\"\\\\.rds$\", paste0(\"_\", basename(tools::file_path_sans_ext(summary_stats_list[[k]])), \"_results-par.rds\"), df_part$path[j])\n",
    "          if(file.exists(outname)) next\n",
    "          summary_stats <- summary_stats_data[[k]]\n",
    "    \n",
    "          MWASmodels <- vector(\"list\", length(my_rds@models))\n",
    "    \n",
    "          for (i in seq_along(my_rds@models)) {\n",
    "            this_MethylationBase <- my_rds@models[[i]]\n",
    "            if (length(this_MethylationBase@snpWeights) == 0) {\n",
    "              next\n",
    "            }\n",
    "            MWASmodels[[i]] <- process_model(this_MethylationBase, my_SNPs, summary_stats)\n",
    "            if (is.null(MWASmodels[[i]])) {\n",
    "              next\n",
    "            }\n",
    "            MWASmodels[[i]][\"bp\"] <- this_MethylationBase@methylationPosition\n",
    "          }\n",
    "\n",
    "          MWASmodels <- Filter(Negate(is.null), MWASmodels)\n",
    "          results <- MWASresults(MWASmodels, paths$pvar_path, paths$pgen_path, paths$psam_path, summary_stats_list[[k]], df_this_chr$path[j])\n",
    "          saveRDS(results, outname)\n",
    "    \n",
    "          df <- do.call(rbind, results@MWASmodels)\n",
    "          df <- as.data.frame(df)\n",
    "    \n",
    "          outname_csv <- gsub(\"rds\", \"csv\", outname)\n",
    "          fwrite(df, outname_csv)\n",
    "    \n",
    "          print(paste(\"Saved results to:\", outname))\n",
    "        }\n",
    "      }\n",
    "    }\n",
    "    \n",
    "\n",
    "    }"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ed7bec3e-1469-47fd-9ea7-1283066a8cae",
   "metadata": {},
   "outputs": [],
   "source": [
    "stopCluster(cl)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "776bd0de-35b5-4b45-8286-34de3c2e91d6",
   "metadata": {},
   "outputs": [],
   "source": [
    "Sys.time() #started 1:23pm with 6 cores"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.3.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
