{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "8ab7eb87-1623-48c0-9530-78e76fa64adc",
   "metadata": {},
   "source": [
    "# Again try running stage 2 MWAS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ab8169e3-b8ce-4c28-8fc2-2424864b8826",
   "metadata": {},
   "outputs": [],
   "source": [
    "library(CpGWAS)\n",
    "library(data.table)\n",
    "library(stringr)\n",
    "library(optparse)\n",
    "library(doParallel)\n",
    "library(foreach)\n",
    "\n",
    "opt <- list(genome_file_index = 21,\n",
    "            data_file = \"/expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/scripts/12-OUT_matched_SNP_meth_cov_outputs.csv\",\n",
    "            cores = 6)\n",
    "\n",
    "if(opt$cores > 1) registerDoParallel(opt$cores)\n",
    "\n",
    "df <- fread(opt$data_file)\n",
    "\n",
    "# Load genome files\n",
    "genome_files <- list.files(\"/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped\",\n",
    "                           pattern = \"EUR\", full.names = TRUE)\n",
    "genome_files <- genome_files[grepl(\"allele\", genome_files)]\n",
    "\n",
    "genome_files <- genome_files[grepl(\"pvar\", genome_files)]\n",
    "\n",
    "genome_files <- data.table(path = genome_files, Chr = NA)\n",
    "\n",
    "genome_files$Chr <- str_split_fixed(genome_files$path, \"chr\", 2)[, 2]\n",
    "genome_files$Chr <- gsub(\".pvar\", \"\", genome_files$Chr)\n",
    "\n",
    "genome_files$Chr <- as.integer(genome_files$Chr)\n",
    "genome_files <- genome_files[order(genome_files$Chr), ]\n",
    "\n",
    "\n",
    "\n",
    "print(\"Starting genome file processing\")\n",
    "# Process the specified genome file\n",
    "g <- opt$genome_file_index\n",
    "print(paste(\"Processing genome file index:\", g))\n",
    "\n",
    "paths <- list(\n",
    "  pvar_path = genome_files[g]$path,\n",
    "  pgen_path = gsub(\"pvar\", \"pgen\", genome_files[g]$path),\n",
    "  psam_path = gsub(\"pvar\", \"psam\", genome_files[g]$path)\n",
    ")\n",
    "\n",
    "chr <- as.numeric(gsub(\"\\\\.pvar\", \"\",\n",
    "            stringr::str_split_fixed(paths$pvar_path, \"chr\", 2)[, 2]))\n",
    "\n",
    "my_SNPs <- CpGWAS::loadSNPData(paths$pvar_path, paths$pgen_path, paths$psam_path)\n",
    "setkey(my_SNPs$pvar_dt, `#CHROM`, POS)\n",
    "df_this_chr <- df[which(df$Chr == genome_files[g]$Chr), ]\n",
    "\n",
    "summary_stats_list <- list.files(\"/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped\", pattern = \"alleleprocessed\", full.names = TRUE)\n",
    "\n",
    "# Pre-load all summary stats files into a list and clean/standardize column names\n",
    "summary_stats_data <- lapply(summary_stats_list, function(path) {\n",
    "   stats <- suppressWarnings(data.table::fread(path))\n",
    "   \n",
    "   # Rename column if needed (uncomment if column name needs changing from \"#CHROM\" to \"CHR\")\n",
    "   # setnames(stats, \"#CHROM\", \"CHR\")\n",
    "   \n",
    "   # Set key for efficient subsetting\n",
    "   setkey(stats, SNP, CHR)\n",
    "   \n",
    "   # Subset to only include rows where 'CHR' matches the 'chr' variable\n",
    "   stats <- stats[CHR == chr]\n",
    "\n",
    "   # Clean and standardize column names if needed\n",
    "   # clean_and_standardize_colnames(stats)\n",
    "   \n",
    "   return(stats)\n",
    "})\n",
    "\n",
    "summary_stats_data <- lapply(summary_stats_data, function(stats) stats[`CHR` == genome_files[g]$Chr])\n",
    "\n",
    "print(\"Loaded SNP data\")\n",
    "print(\"Files for this Chr:\")\n",
    "print(nrow(df_this_chr))\n",
    "                             \n",
    "\n",
    "for(j in 1:nrow(df_this_chr)){\n",
    "  if(j %% 10 == 0) print(j)\n",
    "  print(paste0(\"File number: \", j))\n",
    "  if (grepl(\"empty\", df_this_chr$path[j])) {\n",
    "    message(paste0(\"no model for \", df_this_chr$path[j]))\n",
    "    next\n",
    "  }\n",
    "\n",
    "  all_files_exist <- TRUE\n",
    "  outnames <- vector(\"character\", length(summary_stats_list))\n",
    "\n",
    "  for (k in 1:length(summary_stats_list)) {\n",
    "    outnames[k] <- gsub(\"\\\\.rds$\", paste0(\"_\", basename(tools::file_path_sans_ext(summary_stats_list[[k]])), \"_results.rds\"), df_this_chr$path[j])\n",
    "    if (!file.exists(outnames[k])) {\n",
    "      all_files_exist <- FALSE\n",
    "      break  # Exit the loop early if any file does not exist\n",
    "    }\n",
    "  }\n",
    "\n",
    "  if (all_files_exist) {\n",
    "    print(paste(\"All output files already exist for\", df_this_chr$path[j], \"- Skipping\"))\n",
    "    next  # Skip to the next file if all outputs already exist\n",
    "  }\n",
    "  #\n",
    "  my_rds <- tryCatch({\n",
    "    readRDS(df_this_chr$path[j])\n",
    "  }, error = function(e) {\n",
    "    # Print an error message and skip this iteration\n",
    "    message(\"ALERT!!! Error reading RDS file: \", e$message)\n",
    "    return(NULL)  # Return NULL to signal failure\n",
    "  })\n",
    "    \n",
    "  # Check if the readRDS call returned NULL (which indicates an error)\n",
    "  if (is.null(my_rds)) {\n",
    "    next  # Skip the rest of this loop iteration\n",
    "  }\n",
    "\n",
    "  print(paste(\"Loaded RDS file:\", df_this_chr$path[j]))\n",
    "\n",
    "  for (k in 1:length(summary_stats_list)) {\n",
    "    outname <- gsub(\"\\\\.rds$\", paste0(\"_\", basename(tools::file_path_sans_ext(summary_stats_list[[k]])), \"_results.rds\"), df_this_chr$path[j])\n",
    "    if(file.exists(outname)) next\n",
    "    summary_stats <- summary_stats_data[[k]]\n",
    "\n",
    "    MWASmodels <- foreach(this_MethylationBase = my_rds@models, .packages = c(\"data.table\", \"CpGWAS\"), .combine = 'c', .export = c(\"my_SNPs\", \"summary_stats\")) %dopar% {\n",
    "      \n",
    "      if(length(this_MethylationBase@snpWeights) == 0) {\n",
    "        return(list(z=NA, p=NA, n=NA, bp=this_MethylationBase@methylationPosition))\n",
    "      }\n",
    "      \n",
    "      model <- process_model(this_MethylationBase, my_SNPs, summary_stats)\n",
    "      \n",
    "      if(is.null(model)) {\n",
    "        return(list(z=NA, p=NA, n=NA, bp=this_MethylationBase@methylationPosition))\n",
    "      }\n",
    "      \n",
    "      model['bp'] <- this_MethylationBase@methylationPosition\n",
    "      return(list(model))  # Return as a single list\n",
    "    }\n",
    "      \n",
    "    # MWASmodels <- future_lapply(seq_along(my_rds@models), function(i) {\n",
    "    #   library(CpGWAS)\n",
    "    #   library(data.table)\n",
    "    #   this_MethylationBase <- my_rds@models[[i]]\n",
    "      \n",
    "    #   # Skip if no SNP weights\n",
    "    #   if (length(this_MethylationBase@snpWeights) == 0) {\n",
    "    #     return(NULL)\n",
    "    #   }\n",
    "      \n",
    "    #   # Process the model\n",
    "    #   model_result <- process_model(this_MethylationBase, my_SNPs, summary_stats)\n",
    "      \n",
    "    #   # Skip if processing returns NULL\n",
    "    #   if (is.null(model_result)) {\n",
    "    #     return(NULL)\n",
    "    #   }\n",
    "      \n",
    "    #   # Add bp information\n",
    "    #   model_result['bp'] <- this_MethylationBase@methylationPosition\n",
    "    #   return(model_result)\n",
    "    # }, future.seed = TRUE)  # Set future.seed to TRUE if random number generation is involved\n",
    "    \n",
    "    # # Filter out NULL results\n",
    "    # MWASmodels <- Filter(Negate(is.null), MWASmodels)\n",
    "\n",
    "\n",
    "    #MWASmodels <- vector(\"list\", length(my_rds@models))\n",
    "    #    if (is.null(summary_stats)) {\n",
    "    #      summary_stats <- suppressWarnings(fread(summary_stats_list[[k]]))\n",
    "    #      summary_stats <- clean_and_standardize_colnames(summary_stats)\n",
    "    #    }\n",
    "\n",
    "    # Version 1 - not working\n",
    "    # Using foreach to parallelize the loop\n",
    "    # results <- foreach(i = seq_along(my_rds@models), .combine='c', .packages=c(\"data.table\", \"CpGWAS\")) %dopar% {\n",
    "    #   this_MethylationBase <- my_rds@models[[i]]\n",
    "    #   if(length(this_MethylationBase@snpWeights) == 0){\n",
    "    #     return(NULL)  # Skip this iteration entirely if no SNP weights\n",
    "    #   }\n",
    "      \n",
    "    #   model_result <- process_model(this_MethylationBase, my_SNPs, summary_stats)\n",
    "    #   if(is.null(model_result)){\n",
    "    #     return(NULL)  # Skip this iteration if model processing returns NULL\n",
    "    #   }\n",
    "      \n",
    "    #   model_result['bp'] <- this_MethylationBase@methylationPosition\n",
    "    #   return(model_result)\n",
    "    # }\n",
    "    \n",
    "    # # Filter out NULL entries and reassign to MWASmodels\n",
    "    # MWASmodels <- Filter(Negate(is.null), results)\n",
    "\n",
    "\n",
    "    # Version 2 not parallel\n",
    "    # for (i in seq_along(my_rds@models)) {\n",
    "    #   this_MethylationBase <- my_rds@models[[i]]\n",
    "    #   if(length(this_MethylationBase@snpWeights) == 0){\n",
    "    #     next\n",
    "    #   }\n",
    "    #   MWASmodels[[i]] <- process_model(this_MethylationBase, my_SNPs, summary_stats)\n",
    "    #   MWASmodels[[i]]['bp'] <- this_MethylationBase@methylationPosition\n",
    "    #   #pb$tick()\n",
    "    # }\n",
    "\n",
    "    for (i in seq_along(my_rds@models)) {\n",
    "      this_MethylationBase <- my_rds@models[[i]]\n",
    "      if(length(this_MethylationBase@snpWeights) == 0){\n",
    "        next\n",
    "      }\n",
    "      MWASmodels[[i]] <- process_model(this_MethylationBase, my_SNPs, summary_stats)\n",
    "\n",
    "      if(is.null(MWASmodels[[i]])){\n",
    "        next\n",
    "      }\n",
    "      \n",
    "      MWASmodels[[i]]['bp'] <- this_MethylationBase@methylationPosition\n",
    "      #pb$tick()\n",
    "    }\n",
    "\n",
    "    # results <- MWASresults(MWASmodels, paths$pvar_path, paths$pgen_path, paths$psam_path, summary_stats_list[[k]], df_this_chr$path[j])\n",
    "\n",
    "      \n",
    "    # version 3 not working  \n",
    "    #results <- foreach(i = seq_along(my_rds@models), .packages = c(\"CpGWAS\", \"data.table\"), .combine='c') %dopar% {\n",
    "    # this_MethylationBase <- my_rds@models[[i]]\n",
    "    # if (length(this_MethylationBase@snpWeights) == 0) {\n",
    "    #   return(NULL)  # Return NULL for models that should be skipped\n",
    "    # }\n",
    "    # MWASmodel <- process_model(this_MethylationBase, my_SNPs, summary_stats)\n",
    "    # MWASmodel['bp'] <- this_MethylationBase@methylationPosition\n",
    "    # return(MWASmodel)\n",
    "    # }\n",
    "    results@MWASmodels <- Filter(Negate(is.null), results@MWASmodels)  # Remove NULL entries\n",
    "    \n",
    "    saveRDS(results, outname)\n",
    "\n",
    "    #filtered_models <- Filter(Negate(is.null), results@MWASmodels)\n",
    "    df <- do.call(rbind, results@MWASmodels)\n",
    "    df <- as.data.frame(df)\n",
    "\n",
    "    outname_csv <- gsub(\"rds\", \"csv\", outname)\n",
    "    fwrite(df, outname_csv)\n",
    "    \n",
    "    print(paste(\"Saved results to:\", outname))\n",
    "  }\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8786b7bd-9d65-42a3-99eb-64f253f47337",
   "metadata": {},
   "outputs": [],
   "source": [
    "New approach"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4743114c-a158-4d0d-9f0d-6cb9edd6707f",
   "metadata": {},
   "outputs": [],
   "source": [
    "for(j in 1:nrow(df_this_chr)){\n",
    "  if(j %% 10 == 0) print(j)\n",
    "  print(paste0(\"File number: \", j))\n",
    "  if (grepl(\"empty\", df_this_chr$path[j])) {\n",
    "    message(paste0(\"no model for \", df_this_chr$path[j]))\n",
    "    next\n",
    "  }\n",
    "\n",
    "  all_files_exist <- TRUE\n",
    "  outnames <- vector(\"character\", length(summary_stats_list))\n",
    "\n",
    "  for (k in 1:length(summary_stats_list)) {\n",
    "    outnames[k] <- gsub(\"\\\\.rds$\", paste0(\"_\", basename(tools::file_path_sans_ext(summary_stats_list[[k]])), \"_results.rds\"), df_this_chr$path[j])\n",
    "    if (!file.exists(outnames[k])) {\n",
    "      all_files_exist <- FALSE\n",
    "      break  # Exit the loop early if any file does not exist\n",
    "    }\n",
    "  }\n",
    "\n",
    "  if (all_files_exist) {\n",
    "    print(paste(\"All output files already exist for\", df_this_chr$path[j], \"- Skipping\"))\n",
    "    next  # Skip to the next file if all outputs already exist\n",
    "  }\n",
    "  #\n",
    "  my_rds <- tryCatch({\n",
    "    readRDS(df_this_chr$path[j])\n",
    "  }, error = function(e) {\n",
    "    # Print an error message and skip this iteration\n",
    "    message(\"ALERT!!! Error reading RDS file: \", e$message)\n",
    "    return(NULL)  # Return NULL to signal failure\n",
    "  })\n",
    "    \n",
    "  # Check if the readRDS call returned NULL (which indicates an error)\n",
    "  if (is.null(my_rds)) {\n",
    "    next  # Skip the rest of this loop iteration\n",
    "  }\n",
    "\n",
    "  print(paste(\"Loaded RDS file:\", df_this_chr$path[j]))\n",
    "\n",
    "  for (k in 1:length(summary_stats_list)) {\n",
    "    outname <- gsub(\"\\\\.rds$\", paste0(\"_\", basename(tools::file_path_sans_ext(summary_stats_list[[k]])), \"_results.rds\"), df_this_chr$path[j])\n",
    "    if(file.exists(outname)) next\n",
    "    summary_stats <- summary_stats_data[[k]]\n",
    "\n",
    "    MWASmodels <- foreach(this_MethylationBase = my_rds@models, .packages = c(\"data.table\", \"CpGWAS\"), .combine = 'c', .export = c(\"my_SNPs\", \"summary_stats\")) %dopar% {\n",
    "      \n",
    "      if(length(this_MethylationBase@snpWeights) == 0) {\n",
    "        return(list(z=NA, p=NA, n=NA, bp=this_MethylationBase@methylationPosition))\n",
    "      }\n",
    "      \n",
    "      model <- process_model(this_MethylationBase, my_SNPs, summary_stats)\n",
    "      \n",
    "      if(is.null(model)) {\n",
    "        return(list(z=NA, p=NA, n=NA, bp=this_MethylationBase@methylationPosition))\n",
    "      }\n",
    "      \n",
    "      model['bp'] <- this_MethylationBase@methylationPosition\n",
    "      return(list(model))  # Return as a single list\n",
    "    }\n",
    "\n",
    "    results@MWASmodels <- Filter(Negate(is.null), results@MWASmodels)  # Remove NULL entries\n",
    "    \n",
    "    saveRDS(results, outname)\n",
    "\n",
    "    #filtered_models <- Filter(Negate(is.null), results@MWASmodels)\n",
    "    df <- do.call(rbind, results@MWASmodels)\n",
    "    df <- as.data.frame(df)\n",
    "\n",
    "    outname_csv <- gsub(\"rds\", \"csv\", outname)\n",
    "    fwrite(df, outname_csv)\n",
    "    \n",
    "    print(paste(\"Saved results to:\", outname))\n",
    "  }\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "92ba70b8-9a3f-48f3-b3d4-f8d47135041a",
   "metadata": {},
   "outputs": [],
   "source": [
    "Sys.time() # Started at 4L17"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0598577d-62f4-413a-b6fa-9660e9447ed3",
   "metadata": {},
   "outputs": [],
   "source": [
    "traceback()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1f378a86-ef8f-47ee-911b-92f0bd13abb2",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Estimate memory usage for key objects\n",
    "mem_my_SNPs <- object.size(my_SNPs)\n",
    "mem_summary_stats <- object.size(summary_stats)\n",
    "mem_this_MethylationBase <- mean(sapply(my_rds@models, object.size))\n",
    "\n",
    "# Calculate total memory required per core\n",
    "mem_per_core <- mem_my_SNPs + mem_summary_stats + mem_this_MethylationBase\n",
    "\n",
    "# Convert to MB\n",
    "mem_per_core_MB <- mem_per_core / (1024^2)\n",
    "\n",
    "# Print memory usage per core in MB\n",
    "print(mem_per_core_MB)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ef339598-a21e-416e-b9e6-a7c03759bee8",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "46ca3d99-d3a4-4bbf-97f1-5e220dea1de3",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ffc3188b-646e-4316-ae47-421b674d2bcc",
   "metadata": {},
   "outputs": [],
   "source": [
    "length(results@MWASmodels)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6e9d9545-7b2a-4edc-a98f-b40e0a7b1cdc",
   "metadata": {},
   "outputs": [],
   "source": [
    "names(attributes(results))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "379ac061-7914-4abf-9293-0574548a0a21",
   "metadata": {},
   "outputs": [],
   "source": [
    "head(results)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2ea536dd-3bb7-4d44-82c4-e85d508698a3",
   "metadata": {},
   "outputs": [],
   "source": [
    "head(summary_stats)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "776bd0de-35b5-4b45-8286-34de3c2e91d6",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.3.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
