{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "f16633e6-7964-4c69-84a1-ab6c50da3435",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: foreach\n",
      "\n",
      "Loading required package: iterators\n",
      "\n",
      "Loading required package: parallel\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Version to run in console\n",
    "\n",
    "files_skipped <- 0\n",
    "files_not_skipped <- 0\n",
    "\n",
    "library(CpGWAS)\n",
    "library(data.table)\n",
    "setDTthreads(6)\n",
    "\n",
    "library(stringr)\n",
    "library(optparse)\n",
    "library(doParallel)\n",
    "library(foreach)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "1994fdf6-5674-4c97-a85d-edbce3f6cc10",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: future\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# library(future.apply)\n",
    "# plan(multicore)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "f8c5a357-a4a0-40fc-be06-b777a510dddd",
   "metadata": {},
   "outputs": [],
   "source": [
    "cl <- makeCluster(6)  # Leave one core free for other tasks\n",
    "registerDoParallel(cl)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "2f90851b-b03c-4f4b-91b7-65da2a34c554",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"chr: 9\"\n",
      "[1] \"Starting genome file processing\"\n",
      "[1] \"Processing genome file index: 9\"\n",
      "[1] \"Loaded SNP data\"\n",
      "[1] \"Files for this Chr:\"\n",
      "[1] 486\n"
     ]
    }
   ],
   "source": [
    "chr_list <- 1:22\n",
    "\n",
    "this_chr <- 9\n",
    "\n",
    "machine <- \"notlocal\"\n",
    "\n",
    "print(paste0(\"chr: \", this_chr))\n",
    "opt <- list(genome_file_index = this_chr,\n",
    "            data_file = \"/expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/scripts/12-OUT_matched_SNP_meth_cov_outputs.csv\",\n",
    "            cores = 1)\n",
    "\n",
    "#if(opt$cores > 1) registerDoParallel(opt$cores)\n",
    "\n",
    "# Load genome files\n",
    "genome_files <- list.files(\"/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped\",\n",
    "                           pattern = \"EUR\", full.names = TRUE)\n",
    "genome_files <- genome_files[grepl(\"allele\", genome_files)]\n",
    "genome_files <- genome_files[grepl(\"a2\", genome_files)]\n",
    "\n",
    "genome_files <- genome_files[grepl(\"pvar\", genome_files)]\n",
    "\n",
    "genome_files <- data.table(path = genome_files, Chr = NA)\n",
    "\n",
    "genome_files$Chr <- str_split_fixed(genome_files$path, \"chr\", 2)[, 2]\n",
    "genome_files$Chr <- gsub(\"\\\\.pvar\", \"\", genome_files$Chr)\n",
    "\n",
    "genome_files$Chr <- as.integer(genome_files$Chr)\n",
    "genome_files <- genome_files[order(genome_files$Chr), ]\n",
    "\n",
    "df <- fread(opt$data_file)\n",
    "\n",
    "if(machine == \"local\"){\n",
    "  df$path <- gsub(\"..//output_EXPANSE_a2\", \"~/data/stage1_output_a3/output_EXPANSE_dt\", df$path)\n",
    "}\n",
    "\n",
    "\n",
    "df$path <- gsub(\"_a2_\", \"_dt_\", df$path)\n",
    "df$path <- gsub(\"\\\\.rds\", \"_dt_combined.csv\", df$path)\n",
    "\n",
    "print(\"Starting genome file processing\")\n",
    "# Process the specified genome file\n",
    "g <- opt$genome_file_index\n",
    "print(paste(\"Processing genome file index:\", g))\n",
    "\n",
    "paths <- list(\n",
    "  pvar_path = genome_files[g]$path,\n",
    "  pgen_path = gsub(\"pvar\", \"pgen\", genome_files[g]$path),\n",
    "  psam_path = gsub(\"pvar\", \"psam\", genome_files[g]$path)\n",
    ")\n",
    "\n",
    "chr <- as.numeric(gsub(\"\\\\.pvar\", \"\",\n",
    "                       stringr::str_split_fixed(paths$pvar_path, \"chr\", 2)[, 2]))\n",
    "\n",
    "my_SNPs <- CpGWAS::loadSNPData(paths$pvar_path, paths$pgen_path, paths$psam_path)\n",
    "setkey(my_SNPs$pvar_dt, POS)\n",
    "df_this_chr <- df[which(df$Chr == genome_files[g]$Chr), ]\n",
    "\n",
    "\n",
    "summary_stats_list <- list.files(\"/expanse/lustre/projects/jhu152/naglemi/mwas/gwas\", pattern = \"alleleprocessed\", full.names = TRUE)\n",
    "summary_stats_list <- summary_stats_list[grepl(\"a3\", summary_stats_list)]\n",
    "\n",
    "# Pre-load all summary stats files into a list and clean/standardize column names\n",
    "summary_stats_data <- lapply(summary_stats_list, function(path) {\n",
    "  cols_to_load <- c(\"CHR\", \"BP\", \"SNP\", \"A1\", \"A2\", \"Z\", \"SE\", \"PVAL\")\n",
    "  stats <- suppressWarnings(data.table::fread(path, select = cols_to_load))\n",
    "  \n",
    "  # Rename column if needed (uncomment if column name needs changing from \"#CHROM\" to \"CHR\")\n",
    "  # setnames(stats, \"#CHROM\", \"CHR\")\n",
    "  \n",
    "  # Set key for efficient subsetting\n",
    "  setkey(stats, BP)\n",
    "  \n",
    "  # Subset to only include rows where 'CHR' matches the 'chr' variable\n",
    "  stats <- stats[CHR == chr]\n",
    "  \n",
    "  # Clean and standardize column names if needed\n",
    "  # clean_and_standardize_colnames(stats)\n",
    "  \n",
    "  return(stats)\n",
    "})\n",
    "\n",
    "summary_stats_data <- lapply(summary_stats_data, function(stats) stats[`CHR` == genome_files[g]$Chr])\n",
    "\n",
    "print(\"Loaded SNP data\")\n",
    "print(\"Files for this Chr:\")\n",
    "print(nrow(df_this_chr))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "f3ad1694-3338-4512-829f-508cd051b8e0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"File number: 1\"\n"
     ]
    }
   ],
   "source": [
    "j <- 1\n",
    "\n",
    "print(paste0(\"File number: \", j))\n",
    "\n",
    "start_time <- Sys.time()  # Start timing the loop iteration\n",
    "\n",
    "if (grepl(\"empty\", df_this_chr$path[j])) {\n",
    "message(paste0(\"No model for \", df_this_chr$path[j]))\n",
    "next\n",
    "}\n",
    "\n",
    "all_files_exist <- TRUE\n",
    "outnames <- vector(\"character\", length(summary_stats_list))\n",
    "\n",
    "# Check if output files already exist\n",
    "for (k in 1:length(summary_stats_list)) {\n",
    "outnames[k] <- gsub(\"\\\\.rds$\", paste0(\"_\", basename(tools::file_path_sans_ext(summary_stats_list[[k]])), \"_results.rds\"), df_this_chr$path[j])\n",
    "if (!file.exists(outnames[k])) {\n",
    "  all_files_exist <- FALSE\n",
    "  break  # Exit the loop early if any file does not exist\n",
    "}\n",
    "}\n",
    "\n",
    "# Load the CSV file\n",
    "load_csv_start <- Sys.time()\n",
    "my_csv <- fread(df_this_chr$path[j])\n",
    "cgs <- levels(factor(my_csv$cg))\n",
    "cgs <- as.integer(cgs)\n",
    "\n",
    "setkey(my_csv,\n",
    "     #cg,\n",
    "     features)\n",
    "\n",
    "my_csv_list <- split(my_csv, by = \"cg\")\n",
    "\n",
    "# Ensure that 'cgs' is ordered and unique if it's not already\n",
    "cgs <- unique(cgs[order(cgs)])\n",
    "\n",
    "#print(paste(\"Loaded csv file:\", df_this_chr$path[j]))\n",
    "load_csv_end <- Sys.time()\n",
    "#print(paste(\"Time to load CSV:\", load_csv_end - load_csv_start))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "029d296d-09be-495c-918e-eefab608b657",
   "metadata": {},
   "outputs": [],
   "source": [
    "k <- 1\n",
    "process_start <- Sys.time()\n",
    "\n",
    "outname <- gsub(\"_dt_combined.csv\", paste0(\"_\", basename(tools::file_path_sans_ext(summary_stats_list[[k]])), \"_results.rds\"), df_this_chr$path[j])\n",
    "\n",
    "summary_stats <- summary_stats_data[[k]]\n",
    "\n",
    "matching_bps <- unique(my_csv$features)\n",
    "\n",
    "# Subset summary_stats to only include rows with BPs in the matching_bps vector\n",
    "summary_stats_sub <- summary_stats[BP %in% matching_bps]\n",
    "\n",
    "MWASmodels <- vector(\"list\", length(cgs))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "eb720d55-fcfd-42ab-89c2-1a564b665cb7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[1] \"2024-08-29 06:37:49 PDT\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "Sys.time()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "46b87820-9228-4ef5-9457-3562b844bd0e",
   "metadata": {},
   "outputs": [],
   "source": [
    "library(parallel)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "2784d6d0-75e4-4591-8652-70be53ec97b2",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in mclapply(function(this_cg_data) {:\n",
      "“all scheduled cores encountered errors in user code”\n"
     ]
    }
   ],
   "source": [
    "# Using lapply with indices passed properly using a wrapper list\n",
    "MWASmodels <- mclapply(function(this_cg_data) {\n",
    "  if (!is.null(this_cg_data)) {\n",
    "    model_result <- process_model_csv(this_cg_data, my_SNPs, summary_stats_sub)\n",
    "    return(model_result)\n",
    "  } else {\n",
    "    return(NULL)  # Return NULL for null data to maintain index alignment\n",
    "  }\n",
    "}, my_csv_list, SIMPLIFY = FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "b53f4eef-1545-4794-b63f-71e5d58fc20e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[1] \"2024-08-29 06:37:50 PDT\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "Sys.time()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "4045b727-2183-4d7b-a411-98e0353fc981",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol>\n",
       "\t<li><style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'CpGWAS'</li><li>'data.table'</li><li>'stats'</li><li>'graphics'</li><li>'grDevices'</li><li>'utils'</li><li>'datasets'</li><li>'methods'</li><li>'base'</li></ol>\n",
       "</li>\n",
       "\t<li><style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'CpGWAS'</li><li>'data.table'</li><li>'stats'</li><li>'graphics'</li><li>'grDevices'</li><li>'utils'</li><li>'datasets'</li><li>'methods'</li><li>'base'</li></ol>\n",
       "</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate}\n",
       "\\item \\begin{enumerate*}\n",
       "\\item 'CpGWAS'\n",
       "\\item 'data.table'\n",
       "\\item 'stats'\n",
       "\\item 'graphics'\n",
       "\\item 'grDevices'\n",
       "\\item 'utils'\n",
       "\\item 'datasets'\n",
       "\\item 'methods'\n",
       "\\item 'base'\n",
       "\\end{enumerate*}\n",
       "\n",
       "\\item \\begin{enumerate*}\n",
       "\\item 'CpGWAS'\n",
       "\\item 'data.table'\n",
       "\\item 'stats'\n",
       "\\item 'graphics'\n",
       "\\item 'grDevices'\n",
       "\\item 'utils'\n",
       "\\item 'datasets'\n",
       "\\item 'methods'\n",
       "\\item 'base'\n",
       "\\end{enumerate*}\n",
       "\n",
       "\\end{enumerate}\n"
      ],
      "text/markdown": [
       "1. 1. 'CpGWAS'\n",
       "2. 'data.table'\n",
       "3. 'stats'\n",
       "4. 'graphics'\n",
       "5. 'grDevices'\n",
       "6. 'utils'\n",
       "7. 'datasets'\n",
       "8. 'methods'\n",
       "9. 'base'\n",
       "\n",
       "\n",
       "\n",
       "2. 1. 'CpGWAS'\n",
       "2. 'data.table'\n",
       "3. 'stats'\n",
       "4. 'graphics'\n",
       "5. 'grDevices'\n",
       "6. 'utils'\n",
       "7. 'datasets'\n",
       "8. 'methods'\n",
       "9. 'base'\n",
       "\n",
       "\n",
       "\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[[1]]\n",
       "[1] \"CpGWAS\"     \"data.table\" \"stats\"      \"graphics\"   \"grDevices\" \n",
       "[6] \"utils\"      \"datasets\"   \"methods\"    \"base\"      \n",
       "\n",
       "[[2]]\n",
       "[1] \"CpGWAS\"     \"data.table\" \"stats\"      \"graphics\"   \"grDevices\" \n",
       "[6] \"utils\"      \"datasets\"   \"methods\"    \"base\"      \n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cl <- parallel::makeCluster(2)\n",
    "parallel::clusterExport(cl, c(\"paths\", \"summary_stats_sub\", \"process_model_csv\"))\n",
    "parallel::clusterEvalQ(cl, {\n",
    "  library(data.table)\n",
    "  library(CpGWAS)\n",
    "})\n",
    "\n",
    "MWASmodels <- parallel::parLapply(cl, my_csv_list, function(this_cg_data) {\n",
    "  my_SNPs <- CpGWAS::loadSNPData(paths$pvar_path, paths$pgen_path, paths$psam_path)\n",
    "  setkey(my_SNPs$pvar_dt, POS)\n",
    "\n",
    "  if (!is.null(this_cg_data)) {\n",
    "    return(process_model_csv(this_cg_data, my_SNPs, summary_stats_sub))\n",
    "  } else {\n",
    "    return(NULL)\n",
    "  }\n",
    "})\n",
    "\n",
    "parallel::stopCluster(cl)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3b1bfad8-e981-4cad-9e03-acffa5f640b2",
   "metadata": {},
   "outputs": [],
   "source": [
    "Sys.time()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "1f3e6798-9adc-4c13-8136-18e546a38d5c",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in parallel::mclapply(my_csv_list, function(this_cg_data) {:\n",
      "“scheduled core 2 encountered error in user code, all values of the job will be affected”\n"
     ]
    }
   ],
   "source": [
    "MWASmodels <- parallel::mclapply(my_csv_list, function(this_cg_data) {\n",
    "  if (!is.null(this_cg_data)) {\n",
    "    return(process_model_csv(this_cg_data, my_SNPs, summary_stats_sub))\n",
    "  } else {\n",
    "    return(NULL)\n",
    "  }\n",
    "}, mc.cores = 2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "55983fca-5439-403c-8182-c256cba6c313",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[1] \"2024-08-29 06:29:14 PDT\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "Sys.time()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "6e774909-e4a2-4125-ac1c-cb63f841f5d0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<dl>\n",
       "\t<dt>$`172730`</dt>\n",
       "\t\t<dd><style>\n",
       ".dl-inline {width: auto; margin:0; padding: 0}\n",
       ".dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}\n",
       ".dl-inline>dt::after {content: \":\\0020\"; padding-right: .5ex}\n",
       ".dl-inline>dt:not(:first-of-type) {padding-left: .5ex}\n",
       "</style><dl class=dl-inline><dt>z</dt><dd>0.976280701725673</dd><dt>p</dt><dd>0.328925380581425</dd><dt>n</dt><dd>1</dd></dl>\n",
       "</dd>\n",
       "\t<dt>$`172895`</dt>\n",
       "\t\t<dd><style>\n",
       ".dl-inline {width: auto; margin:0; padding: 0}\n",
       ".dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}\n",
       ".dl-inline>dt::after {content: \":\\0020\"; padding-right: .5ex}\n",
       ".dl-inline>dt:not(:first-of-type) {padding-left: .5ex}\n",
       "</style><dl class=dl-inline><dt>z</dt><dd>-0.939398852109057</dd><dt>p</dt><dd>0.347526002152833</dd><dt>n</dt><dd>2</dd></dl>\n",
       "</dd>\n",
       "\t<dt>$`173292`</dt>\n",
       "\t\t<dd><style>\n",
       ".dl-inline {width: auto; margin:0; padding: 0}\n",
       ".dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}\n",
       ".dl-inline>dt::after {content: \":\\0020\"; padding-right: .5ex}\n",
       ".dl-inline>dt:not(:first-of-type) {padding-left: .5ex}\n",
       "</style><dl class=dl-inline><dt>z</dt><dd>0.976280701725673</dd><dt>p</dt><dd>0.328925380581425</dd><dt>n</dt><dd>1</dd></dl>\n",
       "</dd>\n",
       "\t<dt>$`173359`</dt>\n",
       "\t\t<dd><style>\n",
       ".dl-inline {width: auto; margin:0; padding: 0}\n",
       ".dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}\n",
       ".dl-inline>dt::after {content: \":\\0020\"; padding-right: .5ex}\n",
       ".dl-inline>dt:not(:first-of-type) {padding-left: .5ex}\n",
       "</style><dl class=dl-inline><dt>z</dt><dd>-0.937103405812875</dd><dt>p</dt><dd>0.348705369702289</dd><dt>n</dt><dd>2</dd></dl>\n",
       "</dd>\n",
       "\t<dt>$`173509`</dt>\n",
       "\t\t<dd><style>\n",
       ".dl-inline {width: auto; margin:0; padding: 0}\n",
       ".dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}\n",
       ".dl-inline>dt::after {content: \":\\0020\"; padding-right: .5ex}\n",
       ".dl-inline>dt:not(:first-of-type) {padding-left: .5ex}\n",
       "</style><dl class=dl-inline><dt>z</dt><dd>0.976280701725673</dd><dt>p</dt><dd>0.328925380581425</dd><dt>n</dt><dd>1</dd></dl>\n",
       "</dd>\n",
       "\t<dt>$`173918`</dt>\n",
       "\t\t<dd><style>\n",
       ".dl-inline {width: auto; margin:0; padding: 0}\n",
       ".dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}\n",
       ".dl-inline>dt::after {content: \":\\0020\"; padding-right: .5ex}\n",
       ".dl-inline>dt:not(:first-of-type) {padding-left: .5ex}\n",
       "</style><dl class=dl-inline><dt>z</dt><dd>0.976280701725673</dd><dt>p</dt><dd>0.328925380581425</dd><dt>n</dt><dd>1</dd></dl>\n",
       "</dd>\n",
       "</dl>\n"
      ],
      "text/latex": [
       "\\begin{description}\n",
       "\\item[\\$`172730`] \\begin{description*}\n",
       "\\item[z] 0.976280701725673\n",
       "\\item[p] 0.328925380581425\n",
       "\\item[n] 1\n",
       "\\end{description*}\n",
       "\n",
       "\\item[\\$`172895`] \\begin{description*}\n",
       "\\item[z] -0.939398852109057\n",
       "\\item[p] 0.347526002152833\n",
       "\\item[n] 2\n",
       "\\end{description*}\n",
       "\n",
       "\\item[\\$`173292`] \\begin{description*}\n",
       "\\item[z] 0.976280701725673\n",
       "\\item[p] 0.328925380581425\n",
       "\\item[n] 1\n",
       "\\end{description*}\n",
       "\n",
       "\\item[\\$`173359`] \\begin{description*}\n",
       "\\item[z] -0.937103405812875\n",
       "\\item[p] 0.348705369702289\n",
       "\\item[n] 2\n",
       "\\end{description*}\n",
       "\n",
       "\\item[\\$`173509`] \\begin{description*}\n",
       "\\item[z] 0.976280701725673\n",
       "\\item[p] 0.328925380581425\n",
       "\\item[n] 1\n",
       "\\end{description*}\n",
       "\n",
       "\\item[\\$`173918`] \\begin{description*}\n",
       "\\item[z] 0.976280701725673\n",
       "\\item[p] 0.328925380581425\n",
       "\\item[n] 1\n",
       "\\end{description*}\n",
       "\n",
       "\\end{description}\n"
      ],
      "text/markdown": [
       "$`172730`\n",
       ":   z\n",
       ":   0.976280701725673p\n",
       ":   0.328925380581425n\n",
       ":   1\n",
       "\n",
       "\n",
       "$`172895`\n",
       ":   z\n",
       ":   -0.939398852109057p\n",
       ":   0.347526002152833n\n",
       ":   2\n",
       "\n",
       "\n",
       "$`173292`\n",
       ":   z\n",
       ":   0.976280701725673p\n",
       ":   0.328925380581425n\n",
       ":   1\n",
       "\n",
       "\n",
       "$`173359`\n",
       ":   z\n",
       ":   -0.937103405812875p\n",
       ":   0.348705369702289n\n",
       ":   2\n",
       "\n",
       "\n",
       "$`173509`\n",
       ":   z\n",
       ":   0.976280701725673p\n",
       ":   0.328925380581425n\n",
       ":   1\n",
       "\n",
       "\n",
       "$`173918`\n",
       ":   z\n",
       ":   0.976280701725673p\n",
       ":   0.328925380581425n\n",
       ":   1\n",
       "\n",
       "\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "$`172730`\n",
       "        z         p         n \n",
       "0.9762807 0.3289254 1.0000000 \n",
       "\n",
       "$`172895`\n",
       "         z          p          n \n",
       "-0.9393989  0.3475260  2.0000000 \n",
       "\n",
       "$`173292`\n",
       "        z         p         n \n",
       "0.9762807 0.3289254 1.0000000 \n",
       "\n",
       "$`173359`\n",
       "         z          p          n \n",
       "-0.9371034  0.3487054  2.0000000 \n",
       "\n",
       "$`173509`\n",
       "        z         p         n \n",
       "0.9762807 0.3289254 1.0000000 \n",
       "\n",
       "$`173918`\n",
       "        z         p         n \n",
       "0.9762807 0.3289254 1.0000000 \n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(MWASmodels)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "faa8bd56-7acd-4ae7-a5f4-7dc44f4128b3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[1] \"2024-08-29 06:25:15 PDT\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "Sys.time()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "93ce8ede-748f-48b5-b832-498b577dea07",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in mclapply(seq_along(my_csv_list), function(idx) {:\n",
      "“scheduled core 2 encountered error in user code, all values of the job will be affected”\n"
     ]
    }
   ],
   "source": [
    "library(parallel)\n",
    "\n",
    "MWASmodels <- mclapply(seq_along(my_csv_list), function(idx) {\n",
    "  this_cg_data <- my_csv_list[[idx]]\n",
    "  if (!is.null(this_cg_data)) {\n",
    "    model_result <- process_model_csv(this_cg_data, my_SNPs, summary_stats_sub)\n",
    "    model_result['bp'] <- cgs[idx]\n",
    "    return(model_result)\n",
    "  } else {\n",
    "    return(NULL)\n",
    "  }\n",
    "}, mc.cores = 2)\n",
    "\n",
    "MWASmodels <- Filter(Negate(is.null), MWASmodels)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "1a509ad9-4da6-48f7-a902-93b41981fc1b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[1] \"2024-08-29 06:25:24 PDT\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "Sys.time()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "cc3e22e2-95fb-45d4-8f9a-349ca73613cf",
   "metadata": {},
   "outputs": [],
   "source": [
    "# # Using lapply with indices passed properly using a wrapper list\n",
    "# MWASmodels <- mapply(function(this_cg_data, idx) {\n",
    "#   if (!is.null(this_cg_data)) {\n",
    "#     model_result <- process_model_csv(this_cg_data, my_SNPs, summary_stats_sub)\n",
    "#     model_result['bp'] <- cgs[idx]  # Assign the correct CG identifier\n",
    "#     return(model_result)\n",
    "#   } else {\n",
    "#     return(NULL)  # Return NULL for null data to maintain index alignment\n",
    "#   }\n",
    "# }, my_csv_list, seq_along(my_csv_list), SIMPLIFY = FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "120b5dbb-941c-44cb-a440-e94ff5db0fd0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[1] \"2024-08-29 06:23:22 PDT\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "Sys.time()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "c45f748a-557b-4242-b0e7-c525458b3ea2",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# # Now, use this pre-structured data in your loop to eliminate the bottleneck\n",
    "# for (i in seq_len(length(cgs))) {\n",
    "#   print(i)\n",
    "#   print(Sys.time())\n",
    "#   this_cg <- cgs[i]\n",
    "#   this_cg_data <- my_csv_list[[as.character(this_cg)]]\n",
    "#   if (!is.null(this_cg_data)) {\n",
    "#     MWASmodels[[i]] <- process_model_csv(this_cg_data, my_SNPs, summary_stats_sub)\n",
    "#     MWASmodels[[i]]['bp'] <- MWASmodels[[i]]['bp'] <- this_cg\n",
    "#   }\n",
    "# }"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "7045f3c2-6d58-4832-a5b3-c6864b9f6706",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[1] \"2024-08-29 06:14:39 PDT\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "Sys.time()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "09450128-2bee-4fc4-aee6-45964b771496",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<dl>\n",
       "\t<dt>$`172730`</dt>\n",
       "\t\t<dd><style>\n",
       ".dl-inline {width: auto; margin:0; padding: 0}\n",
       ".dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}\n",
       ".dl-inline>dt::after {content: \":\\0020\"; padding-right: .5ex}\n",
       ".dl-inline>dt:not(:first-of-type) {padding-left: .5ex}\n",
       "</style><dl class=dl-inline><dt>z</dt><dd>0.976280701725673</dd><dt>p</dt><dd>0.328925380581425</dd><dt>n</dt><dd>1</dd><dt>bp</dt><dd>172730</dd></dl>\n",
       "</dd>\n",
       "\t<dt>$`172895`</dt>\n",
       "\t\t<dd><style>\n",
       ".dl-inline {width: auto; margin:0; padding: 0}\n",
       ".dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}\n",
       ".dl-inline>dt::after {content: \":\\0020\"; padding-right: .5ex}\n",
       ".dl-inline>dt:not(:first-of-type) {padding-left: .5ex}\n",
       "</style><dl class=dl-inline><dt>z</dt><dd>-0.939398852109057</dd><dt>p</dt><dd>0.347526002152833</dd><dt>n</dt><dd>2</dd><dt>bp</dt><dd>172895</dd></dl>\n",
       "</dd>\n",
       "\t<dt>$`173292`</dt>\n",
       "\t\t<dd><style>\n",
       ".dl-inline {width: auto; margin:0; padding: 0}\n",
       ".dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}\n",
       ".dl-inline>dt::after {content: \":\\0020\"; padding-right: .5ex}\n",
       ".dl-inline>dt:not(:first-of-type) {padding-left: .5ex}\n",
       "</style><dl class=dl-inline><dt>z</dt><dd>0.976280701725673</dd><dt>p</dt><dd>0.328925380581425</dd><dt>n</dt><dd>1</dd><dt>bp</dt><dd>173292</dd></dl>\n",
       "</dd>\n",
       "\t<dt>$`173359`</dt>\n",
       "\t\t<dd><style>\n",
       ".dl-inline {width: auto; margin:0; padding: 0}\n",
       ".dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}\n",
       ".dl-inline>dt::after {content: \":\\0020\"; padding-right: .5ex}\n",
       ".dl-inline>dt:not(:first-of-type) {padding-left: .5ex}\n",
       "</style><dl class=dl-inline><dt>z</dt><dd>-0.937103405812875</dd><dt>p</dt><dd>0.348705369702289</dd><dt>n</dt><dd>2</dd><dt>bp</dt><dd>173359</dd></dl>\n",
       "</dd>\n",
       "\t<dt>$`173509`</dt>\n",
       "\t\t<dd><style>\n",
       ".dl-inline {width: auto; margin:0; padding: 0}\n",
       ".dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}\n",
       ".dl-inline>dt::after {content: \":\\0020\"; padding-right: .5ex}\n",
       ".dl-inline>dt:not(:first-of-type) {padding-left: .5ex}\n",
       "</style><dl class=dl-inline><dt>z</dt><dd>0.976280701725673</dd><dt>p</dt><dd>0.328925380581425</dd><dt>n</dt><dd>1</dd><dt>bp</dt><dd>173509</dd></dl>\n",
       "</dd>\n",
       "\t<dt>$`173918`</dt>\n",
       "\t\t<dd><style>\n",
       ".dl-inline {width: auto; margin:0; padding: 0}\n",
       ".dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}\n",
       ".dl-inline>dt::after {content: \":\\0020\"; padding-right: .5ex}\n",
       ".dl-inline>dt:not(:first-of-type) {padding-left: .5ex}\n",
       "</style><dl class=dl-inline><dt>z</dt><dd>0.976280701725673</dd><dt>p</dt><dd>0.328925380581425</dd><dt>n</dt><dd>1</dd><dt>bp</dt><dd>173918</dd></dl>\n",
       "</dd>\n",
       "</dl>\n"
      ],
      "text/latex": [
       "\\begin{description}\n",
       "\\item[\\$`172730`] \\begin{description*}\n",
       "\\item[z] 0.976280701725673\n",
       "\\item[p] 0.328925380581425\n",
       "\\item[n] 1\n",
       "\\item[bp] 172730\n",
       "\\end{description*}\n",
       "\n",
       "\\item[\\$`172895`] \\begin{description*}\n",
       "\\item[z] -0.939398852109057\n",
       "\\item[p] 0.347526002152833\n",
       "\\item[n] 2\n",
       "\\item[bp] 172895\n",
       "\\end{description*}\n",
       "\n",
       "\\item[\\$`173292`] \\begin{description*}\n",
       "\\item[z] 0.976280701725673\n",
       "\\item[p] 0.328925380581425\n",
       "\\item[n] 1\n",
       "\\item[bp] 173292\n",
       "\\end{description*}\n",
       "\n",
       "\\item[\\$`173359`] \\begin{description*}\n",
       "\\item[z] -0.937103405812875\n",
       "\\item[p] 0.348705369702289\n",
       "\\item[n] 2\n",
       "\\item[bp] 173359\n",
       "\\end{description*}\n",
       "\n",
       "\\item[\\$`173509`] \\begin{description*}\n",
       "\\item[z] 0.976280701725673\n",
       "\\item[p] 0.328925380581425\n",
       "\\item[n] 1\n",
       "\\item[bp] 173509\n",
       "\\end{description*}\n",
       "\n",
       "\\item[\\$`173918`] \\begin{description*}\n",
       "\\item[z] 0.976280701725673\n",
       "\\item[p] 0.328925380581425\n",
       "\\item[n] 1\n",
       "\\item[bp] 173918\n",
       "\\end{description*}\n",
       "\n",
       "\\end{description}\n"
      ],
      "text/markdown": [
       "$`172730`\n",
       ":   z\n",
       ":   0.976280701725673p\n",
       ":   0.328925380581425n\n",
       ":   1bp\n",
       ":   172730\n",
       "\n",
       "\n",
       "$`172895`\n",
       ":   z\n",
       ":   -0.939398852109057p\n",
       ":   0.347526002152833n\n",
       ":   2bp\n",
       ":   172895\n",
       "\n",
       "\n",
       "$`173292`\n",
       ":   z\n",
       ":   0.976280701725673p\n",
       ":   0.328925380581425n\n",
       ":   1bp\n",
       ":   173292\n",
       "\n",
       "\n",
       "$`173359`\n",
       ":   z\n",
       ":   -0.937103405812875p\n",
       ":   0.348705369702289n\n",
       ":   2bp\n",
       ":   173359\n",
       "\n",
       "\n",
       "$`173509`\n",
       ":   z\n",
       ":   0.976280701725673p\n",
       ":   0.328925380581425n\n",
       ":   1bp\n",
       ":   173509\n",
       "\n",
       "\n",
       "$`173918`\n",
       ":   z\n",
       ":   0.976280701725673p\n",
       ":   0.328925380581425n\n",
       ":   1bp\n",
       ":   173918\n",
       "\n",
       "\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "$`172730`\n",
       "           z            p            n           bp \n",
       "9.762807e-01 3.289254e-01 1.000000e+00 1.727300e+05 \n",
       "\n",
       "$`172895`\n",
       "            z             p             n            bp \n",
       "-9.393989e-01  3.475260e-01  2.000000e+00  1.728950e+05 \n",
       "\n",
       "$`173292`\n",
       "           z            p            n           bp \n",
       "9.762807e-01 3.289254e-01 1.000000e+00 1.732920e+05 \n",
       "\n",
       "$`173359`\n",
       "            z             p             n            bp \n",
       "-9.371034e-01  3.487054e-01  2.000000e+00  1.733590e+05 \n",
       "\n",
       "$`173509`\n",
       "           z            p            n           bp \n",
       "9.762807e-01 3.289254e-01 1.000000e+00 1.735090e+05 \n",
       "\n",
       "$`173918`\n",
       "           z            p            n           bp \n",
       "9.762807e-01 3.289254e-01 1.000000e+00 1.739180e+05 \n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(MWASmodels)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "7c8ae75e-9826-4622-b46d-1893fdf788fc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[1] \"2024-08-29 06:14:47 PDT\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "Sys.time()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "38560a72-c1e5-44ee-83d0-8505ff3cab63",
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error: 'future_options' is not an exported object from 'namespace:future'\n",
     "output_type": "error",
     "traceback": [
      "Error: 'future_options' is not an exported object from 'namespace:future'\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "library(future.apply)\n",
    "plan(multisession)  # or multicore if on Unix-like systems for testing\n",
    "\n",
    "# Use options to capture and debug future errors\n",
    "options(future.globals.maxSize = Inf)  # Adjust if global size is an issue\n",
    "future::future_options(future.globals.onReference = \"error\")\n",
    "\n",
    "MWASmodels2 <- future_mapply(function(this_cg_data, idx) {\n",
    "    tryCatch({\n",
    "        if (!is.null(this_cg_data)) {\n",
    "            model_result <- process_model_csv(this_cg_data, my_SNPs, summary_stats_sub)\n",
    "            model_result['bp'] <- cgs[idx]  # Assign the correct CG identifier\n",
    "            return(model_result)\n",
    "        } else {\n",
    "            return(NULL)  # Handle NULL data\n",
    "        }\n",
    "    }, error = function(e) {\n",
    "        message(\"Error in processing: \", e$message)  # Print error message\n",
    "        return(NULL)  # Return NULL on error to maintain list structure\n",
    "    })\n",
    "}, my_csv_list, seq_along(my_csv_list), SIMPLIFY = FALSE, future.seed = TRUE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "cb47201d-b505-45df-9118-0f6d3f5fb953",
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): PgrGetD() error 6\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): PgrGetD() error 6\nTraceback:\n",
      "1. future_mapply(function(this_cg_data, idx) {\n .     if (!is.null(this_cg_data)) {\n .         model_result <- process_model_csv(this_cg_data, my_SNPs, \n .             summary_stats_sub)\n .         model_result[\"bp\"] <- cgs[idx]\n .         return(model_result)\n .     }\n .     else {\n .         return(NULL)\n .     }\n . }, my_csv_list, seq_along(my_csv_list), SIMPLIFY = FALSE, future.seed = TRUE)",
      "2. future_xapply(FUN = FUN, nX = nX, chunk_args = dots, MoreArgs = MoreArgs, \n .     get_chunk = function(X, chunk) lapply(X, FUN = `chunkWith[[`, \n .         chunk), expr = expr, envir = envir, future.envir = future.envir, \n .     future.globals = future.globals, future.packages = future.packages, \n .     future.scheduling = future.scheduling, future.chunk.size = future.chunk.size, \n .     future.stdout = future.stdout, future.conditions = future.conditions, \n .     future.seed = future.seed, future.label = future.label, fcn_name = fcn_name, \n .     args_name = args_name, debug = debug)",
      "3. value(fs)",
      "4. value.list(fs)",
      "5. resolve(y, result = TRUE, stdout = stdout, signal = signal, force = TRUE)",
      "6. resolve.list(y, result = TRUE, stdout = stdout, signal = signal, \n .     force = TRUE)",
      "7. signalConditionsASAP(obj, resignal = FALSE, pos = ii)",
      "8. signalConditions(obj, exclude = getOption(\"future.relay.immediate\", \n .     \"immediateCondition\"), resignal = resignal, ...)"
     ]
    }
   ],
   "source": [
    "MWASmodels2 <- future_mapply(function(this_cg_data, idx) {\n",
    "  if (!is.null(this_cg_data)) {\n",
    "    model_result <- process_model_csv(this_cg_data, my_SNPs, summary_stats_sub)\n",
    "    model_result['bp'] <- cgs[idx]  # Assign the correct CG identifier\n",
    "    return(model_result)\n",
    "  } else {\n",
    "    return(NULL)  # Return NULL for null data to maintain index alignment\n",
    "  }\n",
    "}, my_csv_list, seq_along(my_csv_list), SIMPLIFY = FALSE, future.seed = TRUE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "947ec4c9-cb04-45f9-bb31-c2f99d5cfb09",
   "metadata": {},
   "outputs": [],
   "source": [
    "Sys.time()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "807fd698-1e28-40bd-9ac2-1edaec09cfd3",
   "metadata": {},
   "outputs": [],
   "source": [
    "MWASmodels <- Filter(Negate(is.null), MWASmodels)  # Remove NULL entries\n",
    "results_end <- Sys.time()\n",
    "# print(paste(\"Time to create MWAS results:\", results_end - results_start))\n",
    "\n",
    "# Save the results\n",
    "save_start <- Sys.time()\n",
    "#saveRDS(results, outname)\n",
    "\n",
    "df <- do.call(rbind, MWASmodels)\n",
    "df <- as.data.frame(df)\n",
    "\n",
    "outname_csv <- gsub(\"rds\", \"csv\", outname)\n",
    "fwrite(df, outname_csv)\n",
    "save_end <- Sys.time()\n",
    "#print(paste(\"Saved results to:\", outname))\n",
    "#print(paste(\"Time to save results:\", save_end - save_start))\n",
    "\n",
    "process_end <- Sys.time()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "491e12f3-7101-417c-af84-e802fb6d6544",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>z</th><th scope=col>p</th><th scope=col>n</th><th scope=col>bp</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>172730</th><td> 0.9762807</td><td>0.3289254</td><td>1</td><td>172730</td></tr>\n",
       "\t<tr><th scope=row>172895</th><td>-0.9393989</td><td>0.3475260</td><td>2</td><td>172895</td></tr>\n",
       "\t<tr><th scope=row>173292</th><td> 0.9762807</td><td>0.3289254</td><td>1</td><td>173292</td></tr>\n",
       "\t<tr><th scope=row>173359</th><td>-0.9371034</td><td>0.3487054</td><td>2</td><td>173359</td></tr>\n",
       "\t<tr><th scope=row>173509</th><td> 0.9762807</td><td>0.3289254</td><td>1</td><td>173509</td></tr>\n",
       "\t<tr><th scope=row>173918</th><td> 0.9762807</td><td>0.3289254</td><td>1</td><td>173918</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 4\n",
       "\\begin{tabular}{r|llll}\n",
       "  & z & p & n & bp\\\\\n",
       "  & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t172730 &  0.9762807 & 0.3289254 & 1 & 172730\\\\\n",
       "\t172895 & -0.9393989 & 0.3475260 & 2 & 172895\\\\\n",
       "\t173292 &  0.9762807 & 0.3289254 & 1 & 173292\\\\\n",
       "\t173359 & -0.9371034 & 0.3487054 & 2 & 173359\\\\\n",
       "\t173509 &  0.9762807 & 0.3289254 & 1 & 173509\\\\\n",
       "\t173918 &  0.9762807 & 0.3289254 & 1 & 173918\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 4\n",
       "\n",
       "| <!--/--> | z &lt;dbl&gt; | p &lt;dbl&gt; | n &lt;dbl&gt; | bp &lt;dbl&gt; |\n",
       "|---|---|---|---|---|\n",
       "| 172730 |  0.9762807 | 0.3289254 | 1 | 172730 |\n",
       "| 172895 | -0.9393989 | 0.3475260 | 2 | 172895 |\n",
       "| 173292 |  0.9762807 | 0.3289254 | 1 | 173292 |\n",
       "| 173359 | -0.9371034 | 0.3487054 | 2 | 173359 |\n",
       "| 173509 |  0.9762807 | 0.3289254 | 1 | 173509 |\n",
       "| 173918 |  0.9762807 | 0.3289254 | 1 | 173918 |\n",
       "\n"
      ],
      "text/plain": [
       "       z          p         n bp    \n",
       "172730  0.9762807 0.3289254 1 172730\n",
       "172895 -0.9393989 0.3475260 2 172895\n",
       "173292  0.9762807 0.3289254 1 173292\n",
       "173359 -0.9371034 0.3487054 2 173359\n",
       "173509  0.9762807 0.3289254 1 173509\n",
       "173918  0.9762807 0.3289254 1 173918"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "94cc044b-9b89-4412-9a1c-8e025fccf9ec",
   "metadata": {},
   "outputs": [],
   "source": [
    "for(j in 1:nrow(df_this_chr)){\n",
    "#for(j in 1:3){\n",
    "  print(paste0(\"File number: \", j))\n",
    "  \n",
    "  start_time <- Sys.time()  # Start timing the loop iteration\n",
    "  \n",
    "  if (grepl(\"empty\", df_this_chr$path[j])) {\n",
    "    message(paste0(\"No model for \", df_this_chr$path[j]))\n",
    "    next\n",
    "  }\n",
    "  \n",
    "  all_files_exist <- TRUE\n",
    "  outnames <- vector(\"character\", length(summary_stats_list))\n",
    "  \n",
    "  # Check if output files already exist\n",
    "  for (k in 1:length(summary_stats_list)) {\n",
    "    outnames[k] <- gsub(\"\\\\.rds$\", paste0(\"_\", basename(tools::file_path_sans_ext(summary_stats_list[[k]])), \"_results.rds\"), df_this_chr$path[j])\n",
    "    if (!file.exists(outnames[k])) {\n",
    "      all_files_exist <- FALSE\n",
    "      break  # Exit the loop early if any file does not exist\n",
    "    }\n",
    "  }\n",
    "  \n",
    "  # Load the CSV file\n",
    "  load_csv_start <- Sys.time()\n",
    "  my_csv <- fread(df_this_chr$path[j])\n",
    "  cgs <- levels(factor(my_csv$cg))\n",
    "  cgs <- as.integer(cgs)\n",
    "  \n",
    "  setkey(my_csv,\n",
    "         #cg,\n",
    "         features)\n",
    "  \n",
    "  my_csv_list <- split(my_csv, by = \"cg\")\n",
    "  \n",
    "  # Ensure that 'cgs' is ordered and unique if it's not already\n",
    "  cgs <- unique(cgs[order(cgs)])\n",
    "  \n",
    "  #print(paste(\"Loaded csv file:\", df_this_chr$path[j]))\n",
    "  load_csv_end <- Sys.time()\n",
    "  #print(paste(\"Time to load CSV:\", load_csv_end - load_csv_start))\n",
    "  \n",
    "  # Process each summary statistics file\n",
    "  for (k in 1:length(summary_stats_list)) {\n",
    "    process_start <- Sys.time()\n",
    "    \n",
    "    outname <- gsub(\"_dt_combined.csv\", paste0(\"_\", basename(tools::file_path_sans_ext(summary_stats_list[[k]])), \"_results.rds\"), df_this_chr$path[j])\n",
    "\n",
    "    summary_stats <- summary_stats_data[[k]]\n",
    "    \n",
    "    matching_bps <- unique(my_csv$features)\n",
    "\n",
    "    # Subset summary_stats to only include rows with BPs in the matching_bps vector\n",
    "    summary_stats_sub <- summary_stats[BP %in% matching_bps]\n",
    "    \n",
    "    MWASmodels <- vector(\"list\", length(cgs))\n",
    "    \n",
    "    \n",
    "    # Now, use this pre-structured data in your loop to eliminate the bottleneck\n",
    "    for (i in seq_len(length(cgs))) {\n",
    "      print(i)\n",
    "      print(Sys.time())\n",
    "      this_cg <- cgs[i]\n",
    "      this_cg_data <- my_csv_list[[as.character(this_cg)]]\n",
    "      if (!is.null(this_cg_data)) {\n",
    "        MWASmodels[[i]] <- process_model_csv(this_cg_data, my_SNPs, summary_stats_sub)\n",
    "        MWASmodels[[i]]['bp'] <- MWASmodels[[i]]['bp'] <- this_cg\n",
    "      }\n",
    "    }\n",
    "  \n",
    "    \n",
    "    MWASmodels <- Filter(Negate(is.null), MWASmodels)  # Remove NULL entries\n",
    "    results_end <- Sys.time()\n",
    "    # print(paste(\"Time to create MWAS results:\", results_end - results_start))\n",
    "    \n",
    "    # Save the results\n",
    "    save_start <- Sys.time()\n",
    "    #saveRDS(results, outname)\n",
    "    \n",
    "    df <- do.call(rbind, MWASmodels)\n",
    "    df <- as.data.frame(df)\n",
    "    \n",
    "    outname_csv <- gsub(\"rds\", \"csv\", outname)\n",
    "    fwrite(df, outname_csv)\n",
    "    save_end <- Sys.time()\n",
    "    #print(paste(\"Saved results to:\", outname))\n",
    "    #print(paste(\"Time to save results:\", save_end - save_start))\n",
    "    \n",
    "    process_end <- Sys.time()\n",
    "    #print(paste(\"Total time to process summary stats file\", k, \":\", process_end - process_start))\n",
    "  }\n",
    "  \n",
    "  loop_end_time <- Sys.time()  # End timing the loop iteration\n",
    "  print(paste(\"Total time for file number\", j, \":\", loop_end_time - start_time))\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "acd3b0b7-e6ad-4e15-865e-e8ef1228463a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [],
      "text/latex": [],
      "text/markdown": [],
      "text/plain": [
       "character(0)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "summary_stats_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f553c680-dab2-4941-bca6-e31b3c56f1b6",
   "metadata": {},
   "outputs": [],
   "source": [
    "files_skipped <- 0\n",
    "files_not_skipped <- 0\n",
    "\n",
    "library(CpGWAS)\n",
    "library(data.table)\n",
    "library(stringr)\n",
    "library(optparse)\n",
    "library(doParallel)\n",
    "library(foreach)\n",
    "\n",
    "chr_list <- 1:22\n",
    "\n",
    "for(this_chr in chr_list){\n",
    "print(paste0(\"chr: \", this_chr))\n",
    "opt <- list(genome_file_index = this_chr,\n",
    "            data_file = \"/expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/scripts/12-OUT_matched_SNP_meth_cov_outputs.csv\",\n",
    "            cores = 1)\n",
    "\n",
    "#if(opt$cores > 1) registerDoParallel(opt$cores)\n",
    "\n",
    "# Load genome files\n",
    "genome_files <- list.files(\"/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped\",\n",
    "                           pattern = \"EUR\", full.names = TRUE)\n",
    "genome_files <- genome_files[grepl(\"allele\", genome_files)]\n",
    "genome_files <- genome_files[grepl(\"a2\", genome_files)]\n",
    "\n",
    "genome_files <- genome_files[grepl(\"pvar\", genome_files)]\n",
    "\n",
    "genome_files <- data.table(path = genome_files, Chr = NA)\n",
    "\n",
    "genome_files$Chr <- str_split_fixed(genome_files$path, \"chr\", 2)[, 2]\n",
    "genome_files$Chr <- gsub(\"\\\\.pvar\", \"\", genome_files$Chr)\n",
    "\n",
    "genome_files$Chr <- as.integer(genome_files$Chr)\n",
    "genome_files <- genome_files[order(genome_files$Chr), ]\n",
    "\n",
    "df <- fread(opt$data_file)\n",
    "df$path <- gsub(\"_a2_\", \"_dt_\", df$path)\n",
    "df$path <- gsub(\"\\\\.rds\", \"_dt_combined.csv\", df$path)\n",
    "\n",
    "print(\"Starting genome file processing\")\n",
    "# Process the specified genome file\n",
    "g <- opt$genome_file_index\n",
    "print(paste(\"Processing genome file index:\", g))\n",
    "\n",
    "paths <- list(\n",
    "  pvar_path = genome_files[g]$path,\n",
    "  pgen_path = gsub(\"pvar\", \"pgen\", genome_files[g]$path),\n",
    "  psam_path = gsub(\"pvar\", \"psam\", genome_files[g]$path)\n",
    ")\n",
    "\n",
    "chr <- as.numeric(gsub(\"\\\\.pvar\", \"\",\n",
    "            stringr::str_split_fixed(paths$pvar_path, \"chr\", 2)[, 2]))\n",
    "\n",
    "my_SNPs <- CpGWAS::loadSNPData(paths$pvar_path, paths$pgen_path, paths$psam_path)\n",
    "setkey(my_SNPs$pvar_dt, `#CHROM`, POS)\n",
    "df_this_chr <- df[which(df$Chr == genome_files[g]$Chr), ]\n",
    "\n",
    "\n",
    "summary_stats_list <- list.files(\"/expanse/lustre/projects/jhu152/naglemi/mwas/gwas\", pattern = \"alleleprocessed\", full.names = TRUE)\n",
    "summary_stats_list <- summary_stats_list[grepl(\"a3\", summary_stats_list)]\n",
    "\n",
    "# Pre-load all summary stats files into a list and clean/standardize column names\n",
    "summary_stats_data <- lapply(summary_stats_list, function(path) {\n",
    "   stats <- suppressWarnings(data.table::fread(path))\n",
    "   \n",
    "   # Rename column if needed (uncomment if column name needs changing from \"#CHROM\" to \"CHR\")\n",
    "   # setnames(stats, \"#CHROM\", \"CHR\")\n",
    "   \n",
    "   # Set key for efficient subsetting\n",
    "   setkey(stats, SNP, CHR)\n",
    "   \n",
    "   # Subset to only include rows where 'CHR' matches the 'chr' variable\n",
    "   stats <- stats[CHR == chr]\n",
    "\n",
    "   # Clean and standardize column names if needed\n",
    "   # clean_and_standardize_colnames(stats)\n",
    "   \n",
    "   return(stats)\n",
    "})\n",
    "\n",
    "summary_stats_data <- lapply(summary_stats_data, function(stats) stats[`CHR` == genome_files[g]$Chr])\n",
    "\n",
    "print(\"Loaded SNP data\")\n",
    "print(\"Files for this Chr:\")\n",
    "print(nrow(df_this_chr))\n",
    "\n",
    "for(j in 1:nrow(df_this_chr)){\n",
    "#for(j in 1:10){\n",
    "  print(paste0(\"File number: \", j))\n",
    "  \n",
    "  start_time <- Sys.time()  # Start timing the loop iteration\n",
    "\n",
    "  if (grepl(\"empty\", df_this_chr$path[j])) {\n",
    "    message(paste0(\"No model for \", df_this_chr$path[j]))\n",
    "    next\n",
    "  }\n",
    "\n",
    "  all_files_exist <- TRUE\n",
    "  outnames <- vector(\"character\", length(summary_stats_list))\n",
    "\n",
    "  # Check if output files already exist\n",
    "  for (k in 1:length(summary_stats_list)) {\n",
    "    outnames[k] <- gsub(\"\\\\.rds$\", paste0(\"_\", basename(tools::file_path_sans_ext(summary_stats_list[[k]])), \"_results.rds\"), df_this_chr$path[j])\n",
    "    if (!file.exists(outnames[k])) {\n",
    "      all_files_exist <- FALSE\n",
    "      break  # Exit the loop early if any file does not exist\n",
    "    }\n",
    "  }\n",
    "\n",
    "  #if (all_files_exist) {\n",
    "  #  print(paste(\"All output files already exist for\", df_this_chr$path[j], \"- Skipping\"))\n",
    "  #  next  # Skip to the next file if all outputs already exist\n",
    "  #}\n",
    "\n",
    "  # Load the CSV file\n",
    "  load_csv_start <- Sys.time()\n",
    "  my_csv <- fread(df_this_chr$path[j])\n",
    "  setkey(my_csv, cg)\n",
    "  cgs <- levels(factor(my_csv$cg))\n",
    "  #print(paste(\"Loaded csv file:\", df_this_chr$path[j]))\n",
    "  load_csv_end <- Sys.time()\n",
    "  #print(paste(\"Time to load CSV:\", load_csv_end - load_csv_start))\n",
    "\n",
    "  # Process each summary statistics file\n",
    "  for (k in 1:length(summary_stats_list)) {\n",
    "    process_start <- Sys.time()\n",
    "    \n",
    "    outname <- gsub(\"_dt_combined.csv\", paste0(\"_\", basename(tools::file_path_sans_ext(summary_stats_list[[k]])), \"_results.rds\"), df_this_chr$path[j])\n",
    "    if(file.exists(outname)){\n",
    "        files_skipped <- files_skipped + 1\n",
    "        next\n",
    "    } else {\n",
    "        files_not_skipped <- files_not_skipped + 1\n",
    "    }\n",
    "    summary_stats <- summary_stats_data[[k]]\n",
    "\n",
    "    MWASmodels <- vector(\"list\", length(cgs))\n",
    "\n",
    "    # Process each CpG site\n",
    "    for (i in seq_along(cgs)) {\n",
    "    #for(i in 1:10){\n",
    "      this_cg <- cgs[i]\n",
    "      \n",
    "      process_cg_start <- Sys.time()\n",
    "\n",
    "      this_cg_data <- my_csv[cg == this_cg]\n",
    "      MWASmodels[[i]] <- process_model_csv(this_cg_data, my_SNPs, summary_stats)\n",
    "\n",
    "      if(is.null(MWASmodels[[i]])){\n",
    "        next\n",
    "      }\n",
    "      \n",
    "      MWASmodels[[i]]['bp'] <- this_cg\n",
    "\n",
    "      process_cg_end <- Sys.time()\n",
    "  #    print(paste(\"Time to process CpG:\", this_cg, \":\", process_cg_end - process_cg_start))\n",
    "    }\n",
    "\n",
    "    # Create MWAS results\n",
    "    results_start <- Sys.time()\n",
    "    results <- MWASresults(MWASmodels, paths$pvar_path, paths$pgen_path, paths$psam_path, summary_stats_list[[k]], df_this_chr$path[j])\n",
    "    results@MWASmodels <- Filter(Negate(is.null), results@MWASmodels)  # Remove NULL entries\n",
    "    results_end <- Sys.time()\n",
    "   # print(paste(\"Time to create MWAS results:\", results_end - results_start))\n",
    "    \n",
    "    # Save the results\n",
    "    save_start <- Sys.time()\n",
    "    #saveRDS(results, outname)\n",
    "    \n",
    "    df <- do.call(rbind, results@MWASmodels)\n",
    "    df <- as.data.frame(df)\n",
    "\n",
    "    outname_csv <- gsub(\"rds\", \"csv\", outname)\n",
    "    fwrite(df, outname_csv)\n",
    "    save_end <- Sys.time()\n",
    "    #print(paste(\"Saved results to:\", outname))\n",
    "    #print(paste(\"Time to save results:\", save_end - save_start))\n",
    "\n",
    "    process_end <- Sys.time()\n",
    "    #print(paste(\"Total time to process summary stats file\", k, \":\", process_end - process_start))\n",
    "  }\n",
    "  \n",
    "  loop_end_time <- Sys.time()  # End timing the loop iteration\n",
    "  print(paste(\"Total time for file number\", j, \":\", loop_end_time - start_time))\n",
    "}\n",
    "}\n",
    "\n",
    "opt <- list(genome_file_index = this_chr,\n",
    "            data_file = \"/expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/scripts/12-OUT_matched_SNP_meth_cov_outputs.csv\",\n",
    "            cores = 1)\n",
    "\n",
    "#if(opt$cores > 1) registerDoParallel(opt$cores)\n",
    "\n",
    "# Load genome files\n",
    "genome_files <- list.files(\"/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped\",\n",
    "                           pattern = \"EUR\", full.names = TRUE)\n",
    "genome_files <- genome_files[grepl(\"allele\", genome_files)]\n",
    "genome_files <- genome_files[grepl(\"a2\", genome_files)]\n",
    "\n",
    "genome_files <- genome_files[grepl(\"pvar\", genome_files)]\n",
    "\n",
    "genome_files <- data.table(path = genome_files, Chr = NA)\n",
    "\n",
    "genome_files$Chr <- str_split_fixed(genome_files$path, \"chr\", 2)[, 2]\n",
    "genome_files$Chr <- gsub(\"\\\\.pvar\", \"\", genome_files$Chr)\n",
    "\n",
    "genome_files$Chr <- as.integer(genome_files$Chr)\n",
    "genome_files <- genome_files[order(genome_files$Chr), ]\n",
    "\n",
    "df <- fread(opt$data_file)\n",
    "df$path <- gsub(\"_a2_\", \"_dt_\", df$path)\n",
    "df$path <- gsub(\"\\\\.rds\", \"_dt_combined.csv\", df$path)\n",
    "\n",
    "print(\"Starting genome file processing\")\n",
    "# Process the specified genome file\n",
    "g <- opt$genome_file_index\n",
    "print(paste(\"Processing genome file index:\", g))\n",
    "\n",
    "paths <- list(\n",
    "  pvar_path = genome_files[g]$path,\n",
    "  pgen_path = gsub(\"pvar\", \"pgen\", genome_files[g]$path),\n",
    "  psam_path = gsub(\"pvar\", \"psam\", genome_files[g]$path)\n",
    ")\n",
    "\n",
    "chr <- as.numeric(gsub(\"\\\\.pvar\", \"\",\n",
    "            stringr::str_split_fixed(paths$pvar_path, \"chr\", 2)[, 2]))\n",
    "\n",
    "my_SNPs <- CpGWAS::loadSNPData(paths$pvar_path, paths$pgen_path, paths$psam_path)\n",
    "setkey(my_SNPs$pvar_dt, `#CHROM`, POS)\n",
    "df_this_chr <- df[which(df$Chr == genome_files[g]$Chr), ]\n",
    "\n",
    "\n",
    "summary_stats_list <- list.files(\"/expanse/lustre/projects/jhu152/naglemi/mwas/gwas\", pattern = \"alleleprocessed\", full.names = TRUE)\n",
    "summary_stats_list <- summary_stats_list[grepl(\"a3\", summary_stats_list)]\n",
    "\n",
    "# Pre-load all summary stats files into a list and clean/standardize column names\n",
    "summary_stats_data <- lapply(summary_stats_list, function(path) {\n",
    "   stats <- suppressWarnings(data.table::fread(path))\n",
    "   \n",
    "   # Rename column if needed (uncomment if column name needs changing from \"#CHROM\" to \"CHR\")\n",
    "   # setnames(stats, \"#CHROM\", \"CHR\")\n",
    "   \n",
    "   # Set key for efficient subsetting\n",
    "   setkey(stats, SNP, CHR)\n",
    "   \n",
    "   # Subset to only include rows where 'CHR' matches the 'chr' variable\n",
    "   stats <- stats[CHR == chr]\n",
    "\n",
    "   # Clean and standardize column names if needed\n",
    "   # clean_and_standardize_colnames(stats)\n",
    "   \n",
    "   return(stats)\n",
    "})\n",
    "\n",
    "summary_stats_data <- lapply(summary_stats_data, function(stats) stats[`CHR` == genome_files[g]$Chr])\n",
    "\n",
    "print(\"Loaded SNP data\")\n",
    "print(\"Files for this Chr:\")\n",
    "print(nrow(df_this_chr))\n",
    "\n",
    "for(j in 1:nrow(df_this_chr)){\n",
    "#for(j in 1:10){\n",
    "  print(paste0(\"File number: \", j))\n",
    "  \n",
    "  start_time <- Sys.time()  # Start timing the loop iteration\n",
    "\n",
    "  if (grepl(\"empty\", df_this_chr$path[j])) {\n",
    "    message(paste0(\"No model for \", df_this_chr$path[j]))\n",
    "    next\n",
    "  }\n",
    "\n",
    "  all_files_exist <- TRUE\n",
    "  outnames <- vector(\"character\", length(summary_stats_list))\n",
    "\n",
    "  # Check if output files already exist\n",
    "  for (k in 1:length(summary_stats_list)) {\n",
    "    outnames[k] <- gsub(\"\\\\.rds$\", paste0(\"_\", basename(tools::file_path_sans_ext(summary_stats_list[[k]])), \"_results.rds\"), df_this_chr$path[j])\n",
    "    if (!file.exists(outnames[k])) {\n",
    "      all_files_exist <- FALSE\n",
    "      break  # Exit the loop early if any file does not exist\n",
    "    }\n",
    "  }\n",
    "\n",
    "  #if (all_files_exist) {\n",
    "  #  print(paste(\"All output files already exist for\", df_this_chr$path[j], \"- Skipping\"))\n",
    "  #  next  # Skip to the next file if all outputs already exist\n",
    "  #}\n",
    "\n",
    "  # Load the CSV file\n",
    "  load_csv_start <- Sys.time()\n",
    "  my_csv <- fread(df_this_chr$path[j])\n",
    "  setkey(my_csv, cg)\n",
    "  cgs <- levels(factor(my_csv$cg))\n",
    "  #print(paste(\"Loaded csv file:\", df_this_chr$path[j]))\n",
    "  load_csv_end <- Sys.time()\n",
    "  #print(paste(\"Time to load CSV:\", load_csv_end - load_csv_start))\n",
    "\n",
    "  # Process each summary statistics file\n",
    "  for (k in 1:length(summary_stats_list)) {\n",
    "    process_start <- Sys.time()\n",
    "    \n",
    "    outname <- gsub(\"_dt_combined.csv\", paste0(\"_\", basename(tools::file_path_sans_ext(summary_stats_list[[k]])), \"_results.rds\"), df_this_chr$path[j])\n",
    "    #if(file.exists(outname)) next\n",
    "    summary_stats <- summary_stats_data[[k]]\n",
    "\n",
    "    MWASmodels <- vector(\"list\", length(cgs))\n",
    "\n",
    "    # Process each CpG site\n",
    "    for (i in seq_along(cgs)) {\n",
    "    #for(i in 1:10){\n",
    "      this_cg <- cgs[i]\n",
    "      \n",
    "      process_cg_start <- Sys.time()\n",
    "\n",
    "      this_cg_data <- my_csv[cg == this_cg]\n",
    "      MWASmodels[[i]] <- process_model_csv(this_cg_data, my_SNPs, summary_stats)\n",
    "\n",
    "      if(is.null(MWASmodels[[i]])){\n",
    "        next\n",
    "      }\n",
    "      \n",
    "      MWASmodels[[i]]['bp'] <- this_cg\n",
    "\n",
    "      process_cg_end <- Sys.time()\n",
    "  #    print(paste(\"Time to process CpG:\", this_cg, \":\", process_cg_end - process_cg_start))\n",
    "    }\n",
    "\n",
    "    # Create MWAS results\n",
    "    results_start <- Sys.time()\n",
    "    results <- MWASresults(MWASmodels, paths$pvar_path, paths$pgen_path, paths$psam_path, summary_stats_list[[k]], df_this_chr$path[j])\n",
    "    results@MWASmodels <- Filter(Negate(is.null), results@MWASmodels)  # Remove NULL entries\n",
    "    results_end <- Sys.time()\n",
    "   # print(paste(\"Time to create MWAS results:\", results_end - results_start))\n",
    "    \n",
    "    # Save the results\n",
    "    save_start <- Sys.time()\n",
    "    #saveRDS(results, outname)\n",
    "    \n",
    "    df <- do.call(rbind, results@MWASmodels)\n",
    "    df <- as.data.frame(df)\n",
    "\n",
    "    outname_csv <- gsub(\"rds\", \"csv\", outname)\n",
    "    fwrite(df, outname_csv)\n",
    "    save_end <- Sys.time()\n",
    "    #print(paste(\"Saved results to:\", outname))\n",
    "    #print(paste(\"Time to save results:\", save_end - save_start))\n",
    "\n",
    "    process_end <- Sys.time()\n",
    "    #print(paste(\"Total time to process summary stats file\", k, \":\", process_end - process_start))\n",
    "  }\n",
    "  \n",
    "  loop_end_time <- Sys.time()  # End timing the loop iteration\n",
    "  print(paste(\"Total time for file number\", j, \":\", loop_end_time - start_time))\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c9d21986-6221-4b48-a3fc-a9127205a41a",
   "metadata": {},
   "source": [
    "Version without timing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c136fd9e-8d48-496b-b820-9a3ab85cfe94",
   "metadata": {},
   "outputs": [],
   "source": [
    "                             \n",
    "# for(j in 1:nrow(df_this_chr)){\n",
    "#   print(paste0(\"File number: \", j))\n",
    "#   if (grepl(\"empty\", df_this_chr$path[j])) {\n",
    "#     message(paste0(\"no model for \", df_this_chr$path[j]))\n",
    "#     next\n",
    "#   }\n",
    "\n",
    "#   all_files_exist <- TRUE\n",
    "#   outnames <- vector(\"character\", length(summary_stats_list))\n",
    "\n",
    "#   for (k in 1:length(summary_stats_list)) {\n",
    "#     outnames[k] <- gsub(\"\\\\.rds$\", paste0(\"_\", basename(tools::file_path_sans_ext(summary_stats_list[[k]])), \"_results.rds\"), df_this_chr$path[j])\n",
    "#     if (!file.exists(outnames[k])) {\n",
    "#       all_files_exist <- FALSE\n",
    "#       break  # Exit the loop early if any file does not exist\n",
    "#     }\n",
    "#   }\n",
    "\n",
    "#   #if (all_files_exist) {\n",
    "#    # print(paste(\"All output files already exist for\", df_this_chr$path[j], \"- Skipping\"))\n",
    "#    # next  # Skip to the next file if all outputs already exist\n",
    "#   #}\n",
    "#   #\n",
    "#   my_csv <- fread(df_this_chr$path[j])\n",
    "#   setkey(my_csv, cg)\n",
    "#   cgs <- levels(factor(my_csv$cg))\n",
    "#   print(paste(\"Loaded csv file:\", df_this_chr$path[j]))\n",
    "\n",
    "#   for (k in 1:length(summary_stats_list)) {\n",
    "#     outname <- gsub(\"_dt_combined.csv\", paste0(\"_\", basename(tools::file_path_sans_ext(summary_stats_list[[k]])), \"_results.rds\"), df_this_chr$path[j])\n",
    "#     #if(file.exists(outname)) next\n",
    "#     summary_stats <- summary_stats_data[[k]]\n",
    "\n",
    "#     MWASmodels <- vector(\"list\", length(cgs))\n",
    "# #    if (is.null(summary_stats)) {\n",
    "# #      summary_stats <- suppressWarnings(fread(summary_stats_list[[k]]))\n",
    "# #      summary_stats <- clean_and_standardize_colnames(summary_stats)\n",
    "# #    }\n",
    "\n",
    "#     # for (i in seq_along(my_rds@models)) {\n",
    "#     #   this_MethylationBase <- my_rds@models[[i]]\n",
    "#     #   if(length(this_MethylationBase@snpWeights) == 0){\n",
    "#     #     next\n",
    "#     #   }\n",
    "#     #   MWASmodels[[i]] <- process_model(this_MethylationBase, my_SNPs, summary_stats)\n",
    "#     #   MWASmodels[[i]]['bp'] <- this_MethylationBase@methylationPosition\n",
    "#     #   #pb$tick()\n",
    "#     # }\n",
    "\n",
    "#     for (i in seq_along(cgs)) {\n",
    "#       this_cg <- cgs[i]\n",
    "      \n",
    "#       this_cg_data <- my_csv[cg == this_cg]\n",
    "#       MWASmodels[[i]] <- process_model_csv(this_cg_data, my_SNPs, summary_stats)\n",
    "\n",
    "#       if(is.null(MWASmodels[[i]])){\n",
    "#         next\n",
    "#       }\n",
    "      \n",
    "#       MWASmodels[[i]]['bp'] <- MWASmodels[[i]]['bp'] <- this_cg\n",
    "#       #pb$tick()\n",
    "#     }\n",
    "\n",
    "#     results <- MWASresults(MWASmodels, paths$pvar_path, paths$pgen_path, paths$psam_path, summary_stats_list[[k]], df_this_chr$path[j])\n",
    "\n",
    "    \n",
    "#     # results <- foreach(i = seq_along(my_rds@models), .packages = c(\"CpGWAS\", \"data.table\"), .combine='c') %dopar% {\n",
    "#     # this_MethylationBase <- my_rds@models[[i]]\n",
    "#     # if (length(this_MethylationBase@snpWeights) == 0) {\n",
    "#     #   return(NULL)  # Return NULL for models that should be skipped\n",
    "#     # }\n",
    "#     # MWASmodel <- process_model(this_MethylationBase, my_SNPs, summary_stats)\n",
    "#     # MWASmodel['bp'] <- this_MethylationBase@methylationPosition\n",
    "#     # return(MWASmodel)\n",
    "#     # }\n",
    "    \n",
    "#     results@MWASmodels <- Filter(Negate(is.null), results@MWASmodels)  # Remove NULL entries\n",
    "    \n",
    "#     saveRDS(results, outname)\n",
    "\n",
    "#     #filtered_models <- Filter(Negate(is.null), results@MWASmodels)\n",
    "#     df <- do.call(rbind, results@MWASmodels)\n",
    "#     df <- as.data.frame(df)\n",
    "\n",
    "#     outname_csv <- gsub(\"rds\", \"csv\", outname)\n",
    "#     fwrite(df, outname_csv)\n",
    "    \n",
    "#     print(paste(\"Saved results to:\", outname))\n",
    "#   }\n",
    "# }"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.3.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
