{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "f16633e6-7964-4c69-84a1-ab6c50da3435",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: foreach\n",
      "\n",
      "Loading required package: iterators\n",
      "\n",
      "Loading required package: parallel\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Version to run in console\n",
    "\n",
    "files_skipped <- 0\n",
    "files_not_skipped <- 0\n",
    "\n",
    "library(CpGWAS)\n",
    "library(data.table)\n",
    "setDTthreads(6)\n",
    "\n",
    "library(stringr)\n",
    "library(optparse)\n",
    "library(doParallel)\n",
    "library(foreach)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "1994fdf6-5674-4c97-a85d-edbce3f6cc10",
   "metadata": {},
   "outputs": [],
   "source": [
    "# library(future.apply)\n",
    "# plan(multicore)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "f8c5a357-a4a0-40fc-be06-b777a510dddd",
   "metadata": {},
   "outputs": [],
   "source": [
    "cl <- makeCluster(6)  # Leave one core free for other tasks\n",
    "registerDoParallel(cl)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "2f90851b-b03c-4f4b-91b7-65da2a34c554",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"chr: 9\"\n",
      "[1] \"Starting genome file processing\"\n",
      "[1] \"Processing genome file index: 9\"\n",
      "[1] \"Loaded SNP data\"\n",
      "[1] \"Files for this Chr:\"\n",
      "[1] 486\n"
     ]
    }
   ],
   "source": [
    "chr_list <- 1:22\n",
    "\n",
    "this_chr <- 9\n",
    "\n",
    "machine <- \"notlocal\"\n",
    "\n",
    "print(paste0(\"chr: \", this_chr))\n",
    "opt <- list(genome_file_index = this_chr,\n",
    "            data_file = \"/expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/scripts/12-OUT_matched_SNP_meth_cov_outputs.csv\",\n",
    "            cores = 1)\n",
    "\n",
    "#if(opt$cores > 1) registerDoParallel(opt$cores)\n",
    "\n",
    "# Load genome files\n",
    "genome_files <- list.files(\"/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped\",\n",
    "                           pattern = \"EUR\", full.names = TRUE)\n",
    "genome_files <- genome_files[grepl(\"allele\", genome_files)]\n",
    "genome_files <- genome_files[grepl(\"a2\", genome_files)]\n",
    "\n",
    "genome_files <- genome_files[grepl(\"pvar\", genome_files)]\n",
    "\n",
    "genome_files <- data.table(path = genome_files, Chr = NA)\n",
    "\n",
    "genome_files$Chr <- str_split_fixed(genome_files$path, \"chr\", 2)[, 2]\n",
    "genome_files$Chr <- gsub(\"\\\\.pvar\", \"\", genome_files$Chr)\n",
    "\n",
    "genome_files$Chr <- as.integer(genome_files$Chr)\n",
    "genome_files <- genome_files[order(genome_files$Chr), ]\n",
    "\n",
    "df <- fread(opt$data_file)\n",
    "\n",
    "if(machine == \"local\"){\n",
    "  df$path <- gsub(\"..//output_EXPANSE_a2\", \"~/data/stage1_output_a3/output_EXPANSE_dt\", df$path)\n",
    "}\n",
    "\n",
    "\n",
    "df$path <- gsub(\"_a2_\", \"_dt_\", df$path)\n",
    "df$path <- gsub(\"\\\\.rds\", \"_dt_combined.csv\", df$path)\n",
    "\n",
    "print(\"Starting genome file processing\")\n",
    "# Process the specified genome file\n",
    "g <- opt$genome_file_index\n",
    "print(paste(\"Processing genome file index:\", g))\n",
    "\n",
    "paths <- list(\n",
    "  pvar_path = genome_files[g]$path,\n",
    "  pgen_path = gsub(\"pvar\", \"pgen\", genome_files[g]$path),\n",
    "  psam_path = gsub(\"pvar\", \"psam\", genome_files[g]$path)\n",
    ")\n",
    "\n",
    "chr <- as.numeric(gsub(\"\\\\.pvar\", \"\",\n",
    "                       stringr::str_split_fixed(paths$pvar_path, \"chr\", 2)[, 2]))\n",
    "\n",
    "my_SNPs <- CpGWAS::loadSNPData(paths$pvar_path, paths$pgen_path, paths$psam_path)\n",
    "setkey(my_SNPs$pvar_dt, POS)\n",
    "df_this_chr <- df[which(df$Chr == genome_files[g]$Chr), ]\n",
    "\n",
    "\n",
    "summary_stats_list <- list.files(\"/expanse/lustre/projects/jhu152/naglemi/mwas/gwas\", pattern = \"alleleprocessed\", full.names = TRUE)\n",
    "summary_stats_list <- summary_stats_list[grepl(\"a3\", summary_stats_list)]\n",
    "\n",
    "# Pre-load all summary stats files into a list and clean/standardize column names\n",
    "summary_stats_data <- lapply(summary_stats_list, function(path) {\n",
    "  cols_to_load <- c(\"CHR\", \"BP\", \"SNP\", \"A1\", \"A2\", \"Z\", \"SE\", \"PVAL\")\n",
    "  stats <- suppressWarnings(data.table::fread(path, select = cols_to_load))\n",
    "  \n",
    "  # Rename column if needed (uncomment if column name needs changing from \"#CHROM\" to \"CHR\")\n",
    "  # setnames(stats, \"#CHROM\", \"CHR\")\n",
    "  \n",
    "  # Set key for efficient subsetting\n",
    "  setkey(stats, BP)\n",
    "  \n",
    "  # Subset to only include rows where 'CHR' matches the 'chr' variable\n",
    "  stats <- stats[CHR == chr]\n",
    "  \n",
    "  # Clean and standardize column names if needed\n",
    "  # clean_and_standardize_colnames(stats)\n",
    "  \n",
    "  return(stats)\n",
    "})\n",
    "\n",
    "summary_stats_data <- lapply(summary_stats_data, function(stats) stats[`CHR` == genome_files[g]$Chr])\n",
    "\n",
    "print(\"Loaded SNP data\")\n",
    "print(\"Files for this Chr:\")\n",
    "print(nrow(df_this_chr))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "94cc044b-9b89-4412-9a1c-8e025fccf9ec",
   "metadata": {},
   "outputs": [],
   "source": [
    "# for(j in 1:nrow(df_this_chr)){\n",
    "# #for(j in 1:3){\n",
    "#   print(paste0(\"File number: \", j))\n",
    "  \n",
    "#   start_time <- Sys.time()  # Start timing the loop iteration\n",
    "  \n",
    "#   if (grepl(\"empty\", df_this_chr$path[j])) {\n",
    "#     message(paste0(\"No model for \", df_this_chr$path[j]))\n",
    "#     next\n",
    "#   }\n",
    "  \n",
    "#   all_files_exist <- TRUE\n",
    "#   outnames <- vector(\"character\", length(summary_stats_list))\n",
    "  \n",
    "#   # Check if output files already exist\n",
    "#   for (k in 1:length(summary_stats_list)) {\n",
    "#     outnames[k] <- gsub(\"\\\\.rds$\", paste0(\"_\", basename(tools::file_path_sans_ext(summary_stats_list[[k]])), \"_results.rds\"), df_this_chr$path[j])\n",
    "#     if (!file.exists(outnames[k])) {\n",
    "#       all_files_exist <- FALSE\n",
    "#       break  # Exit the loop early if any file does not exist\n",
    "#     }\n",
    "#   }\n",
    "  \n",
    "#   # Load the CSV file\n",
    "#   load_csv_start <- Sys.time()\n",
    "#   my_csv <- fread(df_this_chr$path[j])\n",
    "#   cgs <- levels(factor(my_csv$cg))\n",
    "#   cgs <- as.integer(cgs)\n",
    "  \n",
    "#   setkey(my_csv,\n",
    "#          #cg,\n",
    "#          features)\n",
    "  \n",
    "#   my_csv_list <- split(my_csv, by = \"cg\")\n",
    "  \n",
    "#   # Ensure that 'cgs' is ordered and unique if it's not already\n",
    "#   cgs <- unique(cgs[order(cgs)])\n",
    "  \n",
    "#   #print(paste(\"Loaded csv file:\", df_this_chr$path[j]))\n",
    "#   load_csv_end <- Sys.time()\n",
    "#   #print(paste(\"Time to load CSV:\", load_csv_end - load_csv_start))\n",
    "  \n",
    "#   # Process each summary statistics file\n",
    "#   for (k in 1:length(summary_stats_list)) {\n",
    "#     process_start <- Sys.time()\n",
    "    \n",
    "#     outname <- gsub(\"_dt_combined.csv\", paste0(\"_\", basename(tools::file_path_sans_ext(summary_stats_list[[k]])), \"_results.rds\"), df_this_chr$path[j])\n",
    "\n",
    "#     summary_stats <- summary_stats_data[[k]]\n",
    "    \n",
    "#     matching_bps <- unique(my_csv$features)\n",
    "\n",
    "#     # Subset summary_stats to only include rows with BPs in the matching_bps vector\n",
    "#     summary_stats_sub <- summary_stats[BP %in% matching_bps]\n",
    "    \n",
    "#     MWASmodels <- vector(\"list\", length(cgs))\n",
    "    \n",
    "    \n",
    "#     # Now, use this pre-structured data in your loop to eliminate the bottleneck\n",
    "#     for (i in seq_len(length(cgs))) {\n",
    "#       print(i)\n",
    "#       print(Sys.time())\n",
    "#       this_cg <- cgs[i]\n",
    "#       this_cg_data <- my_csv_list[[as.character(this_cg)]]\n",
    "#       if (!is.null(this_cg_data)) {\n",
    "#         MWASmodels[[i]] <- process_model_csv(this_cg_data, my_SNPs, summary_stats_sub)\n",
    "#         MWASmodels[[i]]['bp'] <- MWASmodels[[i]]['bp'] <- this_cg\n",
    "#       }\n",
    "#     }\n",
    "  \n",
    "    \n",
    "#     MWASmodels <- Filter(Negate(is.null), MWASmodels)  # Remove NULL entries\n",
    "#     results_end <- Sys.time()\n",
    "#     # print(paste(\"Time to create MWAS results:\", results_end - results_start))\n",
    "    \n",
    "#     # Save the results\n",
    "#     save_start <- Sys.time()\n",
    "#     #saveRDS(results, outname)\n",
    "    \n",
    "#     df <- do.call(rbind, MWASmodels)\n",
    "#     df <- as.data.frame(df)\n",
    "    \n",
    "#     outname_csv <- gsub(\"rds\", \"csv\", outname)\n",
    "#     fwrite(df, outname_csv)\n",
    "#     save_end <- Sys.time()\n",
    "#     #print(paste(\"Saved results to:\", outname))\n",
    "#     #print(paste(\"Time to save results:\", save_end - save_start))\n",
    "    \n",
    "#     process_end <- Sys.time()\n",
    "#     #print(paste(\"Total time to process summary stats file\", k, \":\", process_end - process_start))\n",
    "#   }\n",
    "  \n",
    "#   loop_end_time <- Sys.time()  # End timing the loop iteration\n",
    "#   print(paste(\"Total time for file number\", j, \":\", loop_end_time - start_time))\n",
    "# }"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "847e8999-9256-4a01-b6c2-98c591dfc4f9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[1] \"2024-08-29 06:46:31 PDT\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "Sys.time()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "7b40e88f-0b3d-4981-8fcd-5faf44ecee3a",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in e$fun(obj, substitute(ex), parent.frame(), e$data):\n",
      "“already exporting variable(s): df, df_this_chr, my_SNPs, paths, summary_stats_data, summary_stats_list”\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<ol>\n",
       "\t<li>'Total time for file number 1 : 1.14866422812144'</li>\n",
       "\t<li>'Total time for file number 2 : 1.26822197039922'</li>\n",
       "\t<li>'Total time for file number 3 : 1.07017720540365'</li>\n",
       "\t<li>'Total time for file number 4 : 1.21630582809448'</li>\n",
       "\t<li>'Total time for file number 5 : 1.24973820447922'</li>\n",
       "\t<li>'Total time for file number 6 : 1.13368099530538'</li>\n",
       "\t<li>'Total time for file number 7 : 1.30762934684753'</li>\n",
       "\t<li>'Total time for file number 8 : 1.19594549338023'</li>\n",
       "\t<li>'Total time for file number 9 : 1.13294479846954'</li>\n",
       "\t<li>'Total time for file number 10 : 1.18814138571421'</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate}\n",
       "\\item 'Total time for file number 1 : 1.14866422812144'\n",
       "\\item 'Total time for file number 2 : 1.26822197039922'\n",
       "\\item 'Total time for file number 3 : 1.07017720540365'\n",
       "\\item 'Total time for file number 4 : 1.21630582809448'\n",
       "\\item 'Total time for file number 5 : 1.24973820447922'\n",
       "\\item 'Total time for file number 6 : 1.13368099530538'\n",
       "\\item 'Total time for file number 7 : 1.30762934684753'\n",
       "\\item 'Total time for file number 8 : 1.19594549338023'\n",
       "\\item 'Total time for file number 9 : 1.13294479846954'\n",
       "\\item 'Total time for file number 10 : 1.18814138571421'\n",
       "\\end{enumerate}\n"
      ],
      "text/markdown": [
       "1. 'Total time for file number 1 : 1.14866422812144'\n",
       "2. 'Total time for file number 2 : 1.26822197039922'\n",
       "3. 'Total time for file number 3 : 1.07017720540365'\n",
       "4. 'Total time for file number 4 : 1.21630582809448'\n",
       "5. 'Total time for file number 5 : 1.24973820447922'\n",
       "6. 'Total time for file number 6 : 1.13368099530538'\n",
       "7. 'Total time for file number 7 : 1.30762934684753'\n",
       "8. 'Total time for file number 8 : 1.19594549338023'\n",
       "9. 'Total time for file number 9 : 1.13294479846954'\n",
       "10. 'Total time for file number 10 : 1.18814138571421'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[[1]]\n",
       "[1] \"Total time for file number 1 : 1.14866422812144\"\n",
       "\n",
       "[[2]]\n",
       "[1] \"Total time for file number 2 : 1.26822197039922\"\n",
       "\n",
       "[[3]]\n",
       "[1] \"Total time for file number 3 : 1.07017720540365\"\n",
       "\n",
       "[[4]]\n",
       "[1] \"Total time for file number 4 : 1.21630582809448\"\n",
       "\n",
       "[[5]]\n",
       "[1] \"Total time for file number 5 : 1.24973820447922\"\n",
       "\n",
       "[[6]]\n",
       "[1] \"Total time for file number 6 : 1.13368099530538\"\n",
       "\n",
       "[[7]]\n",
       "[1] \"Total time for file number 7 : 1.30762934684753\"\n",
       "\n",
       "[[8]]\n",
       "[1] \"Total time for file number 8 : 1.19594549338023\"\n",
       "\n",
       "[[9]]\n",
       "[1] \"Total time for file number 9 : 1.13294479846954\"\n",
       "\n",
       "[[10]]\n",
       "[1] \"Total time for file number 10 : 1.18814138571421\"\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "library(foreach)\n",
    "library(doParallel)\n",
    "library(data.table)\n",
    "\n",
    "# Set up parallel backend\n",
    "n_cores <- 4\n",
    "cl <- makeCluster(n_cores)\n",
    "registerDoParallel(cl)\n",
    "\n",
    "# Export entire environment\n",
    "foreach(j = 1:10, .packages = c(\"data.table\", \"CpGWAS\"), .export = ls()) %dopar% {\n",
    "    \n",
    "  my_SNPs <- CpGWAS::loadSNPData(paths$pvar_path, paths$pgen_path, paths$psam_path)\n",
    "  setkey(my_SNPs$pvar_dt, POS)\n",
    "  print(paste0(\"File number: \", j))\n",
    "  \n",
    "  start_time <- Sys.time()\n",
    "\n",
    "  if (grepl(\"empty\", df_this_chr$path[j])) {\n",
    "    message(paste0(\"No model for \", df_this_chr$path[j]))\n",
    "    next\n",
    "  }\n",
    "  \n",
    "  all_files_exist <- TRUE\n",
    "  outnames <- vector(\"character\", length(summary_stats_list))\n",
    "\n",
    "  for (k in 1:length(summary_stats_list)) {\n",
    "    outnames[k] <- gsub(\"\\\\.rds$\", paste0(\"_\", basename(tools::file_path_sans_ext(summary_stats_list[[k]])), \"_results.rds\"), df_this_chr$path[j])\n",
    "    if (!file.exists(outnames[k])) {\n",
    "      all_files_exist <- FALSE\n",
    "      break\n",
    "    }\n",
    "  }\n",
    "\n",
    "  load_csv_start <- Sys.time()\n",
    "  my_csv <- fread(df_this_chr$path[j])\n",
    "  cgs <- unique(as.integer(my_csv$cg))\n",
    "  setkey(my_csv, features)\n",
    "  my_csv_list <- split(my_csv, by = \"cg\")\n",
    "  load_csv_end <- Sys.time()\n",
    "\n",
    "  for (k in 1:length(summary_stats_list)) {\n",
    "    process_start <- Sys.time()\n",
    "\n",
    "    outname <- gsub(\"_dt_combined.csv\", paste0(\"_\", basename(tools::file_path_sans_ext(summary_stats_list[[k]])), \"_results.rds\"), df_this_chr$path[j])\n",
    "    summary_stats <- summary_stats_data[[k]]\n",
    "    matching_bps <- unique(my_csv$features)\n",
    "    summary_stats_sub <- summary_stats[BP %in% matching_bps]\n",
    "\n",
    "    MWASmodels <- vector(\"list\", length(cgs))\n",
    "\n",
    "    for (i in seq_len(length(cgs))) {\n",
    "      this_cg_data <- my_csv_list[[as.character(cgs[i])]]\n",
    "      if (!is.null(this_cg_data)) {\n",
    "        MWASmodels[[i]] <- process_model_csv(this_cg_data, my_SNPs, summary_stats_sub)\n",
    "        MWASmodels[[i]]['bp'] <- cgs[i]\n",
    "      }\n",
    "    }\n",
    "\n",
    "    MWASmodels <- Filter(Negate(is.null), MWASmodels)\n",
    "    \n",
    "    df <- do.call(rbind, MWASmodels)\n",
    "    df <- as.data.frame(df)\n",
    "    outname_csv <- gsub(\"rds\", \"csv\", outname)\n",
    "    fwrite(df, outname_csv)\n",
    "\n",
    "    process_end <- Sys.time()\n",
    "  }\n",
    "  \n",
    "  loop_end_time <- Sys.time()\n",
    "  print(paste(\"Total time for file number\", j, \":\", loop_end_time - start_time))\n",
    "}\n",
    "\n",
    "# Stop the cluster\n",
    "stopCluster(cl)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "ba132edd-bb7a-48e0-adb8-ee4edfdf6ddb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[1] \"2024-08-29 06:52:52 PDT\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "Sys.time() # started 9:42am"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "73bee966-c729-4e0c-8429-e70a5e45bdb9",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "aa3b51be-a37c-48f5-99c8-25e6f7a59110",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[1] \"2024-08-29 06:56:42 PDT\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "Sys.time()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "b7cc1b0e-063e-4c06-a4db-599d9450278b",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in e$fun(obj, substitute(ex), parent.frame(), e$data):\n",
      "“already exporting variable(s): df, df_this_chr, my_SNPs, paths, summary_stats_data, summary_stats_list”\n"
     ]
    },
    {
     "ename": "ERROR",
     "evalue": "Error in unserialize(socklist[[n]]): error reading from connection\n",
     "output_type": "error",
     "traceback": [
      "Error in unserialize(socklist[[n]]): error reading from connection\nTraceback:\n",
      "1. foreach(j = 1:10, .packages = c(\"data.table\", \"CpGWAS\"), .export = ls()) %dopar% \n .     {\n .         my_SNPs <- CpGWAS::loadSNPData(paths$pvar_path, paths$pgen_path, \n .             paths$psam_path)\n .         setkey(my_SNPs$pvar_dt, POS)\n .         print(paste0(\"File number: \", j))\n .         start_time <- Sys.time()\n .         if (grepl(\"empty\", df_this_chr$path[j])) {\n .             message(paste0(\"No model for \", df_this_chr$path[j]))\n .             next\n .         }\n .         all_files_exist <- TRUE\n .         outnames <- vector(\"character\", length(summary_stats_list))\n .         for (k in 1:length(summary_stats_list)) {\n .             outnames[k] <- gsub(\"\\\\.rds$\", paste0(\"_\", basename(tools::file_path_sans_ext(summary_stats_list[[k]])), \n .                 \"_results.rds\"), df_this_chr$path[j])\n .             if (!file.exists(outnames[k])) {\n .                 all_files_exist <- FALSE\n .                 break\n .             }\n .         }\n .         load_csv_start <- Sys.time()\n .         my_csv <- fread(df_this_chr$path[j])\n .         cgs <- unique(as.integer(my_csv$cg))\n .         setkey(my_csv, features)\n .         my_csv_list <- split(my_csv, by = \"cg\")\n .         load_csv_end <- Sys.time()\n .         for (k in 1:length(summary_stats_list)) {\n .             process_start <- Sys.time()\n .             outname <- gsub(\"_dt_combined.csv\", paste0(\"_\", basename(tools::file_path_sans_ext(summary_stats_list[[k]])), \n .                 \"_results.rds\"), df_this_chr$path[j])\n .             summary_stats <- summary_stats_data[[k]]\n .             matching_bps <- unique(my_csv$features)\n .             summary_stats_sub <- summary_stats[BP %in% matching_bps]\n .             MWASmodels <- vector(\"list\", length(cgs))\n .             for (i in seq_len(length(cgs))) {\n .                 this_cg_data <- my_csv_list[[as.character(cgs[i])]]\n .                 if (!is.null(this_cg_data)) {\n .                   MWASmodels[[i]] <- process_model_csv(this_cg_data, \n .                     my_SNPs, summary_stats_sub)\n .                   MWASmodels[[i]][\"bp\"] <- cgs[i]\n .                 }\n .             }\n .             MWASmodels <- Filter(Negate(is.null), MWASmodels)\n .             df <- do.call(rbind, MWASmodels)\n .             df <- as.data.frame(df)\n .             outname_csv <- gsub(\"rds\", \"csv\", outname)\n .             fwrite(df, outname_csv)\n .             process_end <- Sys.time()\n .         }\n .         loop_end_time <- Sys.time()\n .         print(paste(\"Total time for file number\", j, \":\", loop_end_time - \n .             start_time))\n .     }",
      "2. e$fun(obj, substitute(ex), parent.frame(), e$data)",
      "3. clusterApplyLB(cl, argsList, evalWrapper)",
      "4. dynamicClusterApply(cl, fun, length(x), argfun)",
      "5. recvOneResult(cl)",
      "6. recvOneData(cl)",
      "7. recvOneData.SOCKcluster(cl)",
      "8. unserialize(socklist[[n]])"
     ]
    }
   ],
   "source": [
    "library(foreach)\n",
    "library(doParallel)\n",
    "library(data.table)\n",
    "\n",
    "# Set up parallel backend\n",
    "n_cores <- 4\n",
    "cl <- makeCluster(n_cores)\n",
    "registerDoParallel(cl)\n",
    "\n",
    "# Export entire environment\n",
    "foreach(j = 1:10, .packages = c(\"data.table\", \"CpGWAS\"), .export = ls()) %dopar% {\n",
    "    \n",
    "  my_SNPs <- CpGWAS::loadSNPData(paths$pvar_path, paths$pgen_path, paths$psam_path)\n",
    "  setkey(my_SNPs$pvar_dt, POS)\n",
    "  print(paste0(\"File number: \", j))\n",
    "  \n",
    "  start_time <- Sys.time()\n",
    "\n",
    "  if (grepl(\"empty\", df_this_chr$path[j])) {\n",
    "    message(paste0(\"No model for \", df_this_chr$path[j]))\n",
    "    next\n",
    "  }\n",
    "  \n",
    "  all_files_exist <- TRUE\n",
    "  outnames <- vector(\"character\", length(summary_stats_list))\n",
    "\n",
    "  for (k in 1:length(summary_stats_list)) {\n",
    "    outnames[k] <- gsub(\"\\\\.rds$\", paste0(\"_\", basename(tools::file_path_sans_ext(summary_stats_list[[k]])), \"_results.rds\"), df_this_chr$path[j])\n",
    "    if (!file.exists(outnames[k])) {\n",
    "      all_files_exist <- FALSE\n",
    "      break\n",
    "    }\n",
    "  }\n",
    "\n",
    "  load_csv_start <- Sys.time()\n",
    "  my_csv <- fread(df_this_chr$path[j])\n",
    "  cgs <- unique(as.integer(my_csv$cg))\n",
    "  setkey(my_csv, features)\n",
    "  my_csv_list <- split(my_csv, by = \"cg\")\n",
    "  load_csv_end <- Sys.time()\n",
    "\n",
    "  for (k in 1:length(summary_stats_list)) {\n",
    "    process_start <- Sys.time()\n",
    "\n",
    "    outname <- gsub(\"_dt_combined.csv\", paste0(\"_\", basename(tools::file_path_sans_ext(summary_stats_list[[k]])), \"_results.rds\"), df_this_chr$path[j])\n",
    "    summary_stats <- summary_stats_data[[k]]\n",
    "    matching_bps <- unique(my_csv$features)\n",
    "    summary_stats_sub <- summary_stats[BP %in% matching_bps]\n",
    "\n",
    "    MWASmodels <- vector(\"list\", length(cgs))\n",
    "\n",
    "    for (i in seq_len(length(cgs))) {\n",
    "      this_cg_data <- my_csv_list[[as.character(cgs[i])]]\n",
    "      if (!is.null(this_cg_data)) {\n",
    "        MWASmodels[[i]] <- process_model_csv(this_cg_data, my_SNPs, summary_stats_sub)\n",
    "        MWASmodels[[i]]['bp'] <- cgs[i]\n",
    "      }\n",
    "    }\n",
    "\n",
    "    MWASmodels <- Filter(Negate(is.null), MWASmodels)\n",
    "    \n",
    "    df <- do.call(rbind, MWASmodels)\n",
    "    df <- as.data.frame(df)\n",
    "    outname_csv <- gsub(\"rds\", \"csv\", outname)\n",
    "    fwrite(df, outname_csv)\n",
    "\n",
    "    process_end <- Sys.time()\n",
    "  }\n",
    "  \n",
    "  loop_end_time <- Sys.time()\n",
    "  print(paste(\"Total time for file number\", j, \":\", loop_end_time - start_time))\n",
    "}\n",
    "\n",
    "# Stop the cluster\n",
    "stopCluster(cl)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cbd881a9-0a26-4f91-be6c-017054963793",
   "metadata": {},
   "outputs": [],
   "source": [
    "Sys.time() # started 9:42am"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "60f6be20-54e1-4dc1-acc5-9a24b0a8c166",
   "metadata": {},
   "outputs": [],
   "source": [
    "my_SNPs <- CpGWAS::loadSNPData(paths$pvar_path, paths$pgen_path, paths$psam_path)\n",
    "setkey(my_SNPs$pvar_dt, POS)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "debe751d-e894-4c24-8127-aa2c6dc9a9ff",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "25af5d7e-7469-4c63-b167-26e4741d583b",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "acd3b0b7-e6ad-4e15-865e-e8ef1228463a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [],
      "text/latex": [],
      "text/markdown": [],
      "text/plain": [
       "character(0)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "summary_stats_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f553c680-dab2-4941-bca6-e31b3c56f1b6",
   "metadata": {},
   "outputs": [],
   "source": [
    "files_skipped <- 0\n",
    "files_not_skipped <- 0\n",
    "\n",
    "library(CpGWAS)\n",
    "library(data.table)\n",
    "library(stringr)\n",
    "library(optparse)\n",
    "library(doParallel)\n",
    "library(foreach)\n",
    "\n",
    "chr_list <- 1:22\n",
    "\n",
    "for(this_chr in chr_list){\n",
    "print(paste0(\"chr: \", this_chr))\n",
    "opt <- list(genome_file_index = this_chr,\n",
    "            data_file = \"/expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/scripts/12-OUT_matched_SNP_meth_cov_outputs.csv\",\n",
    "            cores = 1)\n",
    "\n",
    "#if(opt$cores > 1) registerDoParallel(opt$cores)\n",
    "\n",
    "# Load genome files\n",
    "genome_files <- list.files(\"/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped\",\n",
    "                           pattern = \"EUR\", full.names = TRUE)\n",
    "genome_files <- genome_files[grepl(\"allele\", genome_files)]\n",
    "genome_files <- genome_files[grepl(\"a2\", genome_files)]\n",
    "\n",
    "genome_files <- genome_files[grepl(\"pvar\", genome_files)]\n",
    "\n",
    "genome_files <- data.table(path = genome_files, Chr = NA)\n",
    "\n",
    "genome_files$Chr <- str_split_fixed(genome_files$path, \"chr\", 2)[, 2]\n",
    "genome_files$Chr <- gsub(\"\\\\.pvar\", \"\", genome_files$Chr)\n",
    "\n",
    "genome_files$Chr <- as.integer(genome_files$Chr)\n",
    "genome_files <- genome_files[order(genome_files$Chr), ]\n",
    "\n",
    "df <- fread(opt$data_file)\n",
    "df$path <- gsub(\"_a2_\", \"_dt_\", df$path)\n",
    "df$path <- gsub(\"\\\\.rds\", \"_dt_combined.csv\", df$path)\n",
    "\n",
    "print(\"Starting genome file processing\")\n",
    "# Process the specified genome file\n",
    "g <- opt$genome_file_index\n",
    "print(paste(\"Processing genome file index:\", g))\n",
    "\n",
    "paths <- list(\n",
    "  pvar_path = genome_files[g]$path,\n",
    "  pgen_path = gsub(\"pvar\", \"pgen\", genome_files[g]$path),\n",
    "  psam_path = gsub(\"pvar\", \"psam\", genome_files[g]$path)\n",
    ")\n",
    "\n",
    "chr <- as.numeric(gsub(\"\\\\.pvar\", \"\",\n",
    "            stringr::str_split_fixed(paths$pvar_path, \"chr\", 2)[, 2]))\n",
    "\n",
    "my_SNPs <- CpGWAS::loadSNPData(paths$pvar_path, paths$pgen_path, paths$psam_path)\n",
    "setkey(my_SNPs$pvar_dt, `#CHROM`, POS)\n",
    "df_this_chr <- df[which(df$Chr == genome_files[g]$Chr), ]\n",
    "\n",
    "\n",
    "summary_stats_list <- list.files(\"/expanse/lustre/projects/jhu152/naglemi/mwas/gwas\", pattern = \"alleleprocessed\", full.names = TRUE)\n",
    "summary_stats_list <- summary_stats_list[grepl(\"a3\", summary_stats_list)]\n",
    "\n",
    "# Pre-load all summary stats files into a list and clean/standardize column names\n",
    "summary_stats_data <- lapply(summary_stats_list, function(path) {\n",
    "   stats <- suppressWarnings(data.table::fread(path))\n",
    "   \n",
    "   # Rename column if needed (uncomment if column name needs changing from \"#CHROM\" to \"CHR\")\n",
    "   # setnames(stats, \"#CHROM\", \"CHR\")\n",
    "   \n",
    "   # Set key for efficient subsetting\n",
    "   setkey(stats, SNP, CHR)\n",
    "   \n",
    "   # Subset to only include rows where 'CHR' matches the 'chr' variable\n",
    "   stats <- stats[CHR == chr]\n",
    "\n",
    "   # Clean and standardize column names if needed\n",
    "   # clean_and_standardize_colnames(stats)\n",
    "   \n",
    "   return(stats)\n",
    "})\n",
    "\n",
    "summary_stats_data <- lapply(summary_stats_data, function(stats) stats[`CHR` == genome_files[g]$Chr])\n",
    "\n",
    "print(\"Loaded SNP data\")\n",
    "print(\"Files for this Chr:\")\n",
    "print(nrow(df_this_chr))\n",
    "\n",
    "for(j in 1:nrow(df_this_chr)){\n",
    "#for(j in 1:10){\n",
    "  print(paste0(\"File number: \", j))\n",
    "  \n",
    "  start_time <- Sys.time()  # Start timing the loop iteration\n",
    "\n",
    "  if (grepl(\"empty\", df_this_chr$path[j])) {\n",
    "    message(paste0(\"No model for \", df_this_chr$path[j]))\n",
    "    next\n",
    "  }\n",
    "\n",
    "  all_files_exist <- TRUE\n",
    "  outnames <- vector(\"character\", length(summary_stats_list))\n",
    "\n",
    "  # Check if output files already exist\n",
    "  for (k in 1:length(summary_stats_list)) {\n",
    "    outnames[k] <- gsub(\"\\\\.rds$\", paste0(\"_\", basename(tools::file_path_sans_ext(summary_stats_list[[k]])), \"_results.rds\"), df_this_chr$path[j])\n",
    "    if (!file.exists(outnames[k])) {\n",
    "      all_files_exist <- FALSE\n",
    "      break  # Exit the loop early if any file does not exist\n",
    "    }\n",
    "  }\n",
    "\n",
    "  #if (all_files_exist) {\n",
    "  #  print(paste(\"All output files already exist for\", df_this_chr$path[j], \"- Skipping\"))\n",
    "  #  next  # Skip to the next file if all outputs already exist\n",
    "  #}\n",
    "\n",
    "  # Load the CSV file\n",
    "  load_csv_start <- Sys.time()\n",
    "  my_csv <- fread(df_this_chr$path[j])\n",
    "  setkey(my_csv, cg)\n",
    "  cgs <- levels(factor(my_csv$cg))\n",
    "  #print(paste(\"Loaded csv file:\", df_this_chr$path[j]))\n",
    "  load_csv_end <- Sys.time()\n",
    "  #print(paste(\"Time to load CSV:\", load_csv_end - load_csv_start))\n",
    "\n",
    "  # Process each summary statistics file\n",
    "  for (k in 1:length(summary_stats_list)) {\n",
    "    process_start <- Sys.time()\n",
    "    \n",
    "    outname <- gsub(\"_dt_combined.csv\", paste0(\"_\", basename(tools::file_path_sans_ext(summary_stats_list[[k]])), \"_results.rds\"), df_this_chr$path[j])\n",
    "    if(file.exists(outname)){\n",
    "        files_skipped <- files_skipped + 1\n",
    "        next\n",
    "    } else {\n",
    "        files_not_skipped <- files_not_skipped + 1\n",
    "    }\n",
    "    summary_stats <- summary_stats_data[[k]]\n",
    "\n",
    "    MWASmodels <- vector(\"list\", length(cgs))\n",
    "\n",
    "    # Process each CpG site\n",
    "    for (i in seq_along(cgs)) {\n",
    "    #for(i in 1:10){\n",
    "      this_cg <- cgs[i]\n",
    "      \n",
    "      process_cg_start <- Sys.time()\n",
    "\n",
    "      this_cg_data <- my_csv[cg == this_cg]\n",
    "      MWASmodels[[i]] <- process_model_csv(this_cg_data, my_SNPs, summary_stats)\n",
    "\n",
    "      if(is.null(MWASmodels[[i]])){\n",
    "        next\n",
    "      }\n",
    "      \n",
    "      MWASmodels[[i]]['bp'] <- this_cg\n",
    "\n",
    "      process_cg_end <- Sys.time()\n",
    "  #    print(paste(\"Time to process CpG:\", this_cg, \":\", process_cg_end - process_cg_start))\n",
    "    }\n",
    "\n",
    "    # Create MWAS results\n",
    "    results_start <- Sys.time()\n",
    "    results <- MWASresults(MWASmodels, paths$pvar_path, paths$pgen_path, paths$psam_path, summary_stats_list[[k]], df_this_chr$path[j])\n",
    "    results@MWASmodels <- Filter(Negate(is.null), results@MWASmodels)  # Remove NULL entries\n",
    "    results_end <- Sys.time()\n",
    "   # print(paste(\"Time to create MWAS results:\", results_end - results_start))\n",
    "    \n",
    "    # Save the results\n",
    "    save_start <- Sys.time()\n",
    "    #saveRDS(results, outname)\n",
    "    \n",
    "    df <- do.call(rbind, results@MWASmodels)\n",
    "    df <- as.data.frame(df)\n",
    "\n",
    "    outname_csv <- gsub(\"rds\", \"csv\", outname)\n",
    "    fwrite(df, outname_csv)\n",
    "    save_end <- Sys.time()\n",
    "    #print(paste(\"Saved results to:\", outname))\n",
    "    #print(paste(\"Time to save results:\", save_end - save_start))\n",
    "\n",
    "    process_end <- Sys.time()\n",
    "    #print(paste(\"Total time to process summary stats file\", k, \":\", process_end - process_start))\n",
    "  }\n",
    "  \n",
    "  loop_end_time <- Sys.time()  # End timing the loop iteration\n",
    "  print(paste(\"Total time for file number\", j, \":\", loop_end_time - start_time))\n",
    "}\n",
    "}\n",
    "\n",
    "opt <- list(genome_file_index = this_chr,\n",
    "            data_file = \"/expanse/lustre/projects/jhu152/naglemi/mwas/CpGWAS/scripts/12-OUT_matched_SNP_meth_cov_outputs.csv\",\n",
    "            cores = 1)\n",
    "\n",
    "#if(opt$cores > 1) registerDoParallel(opt$cores)\n",
    "\n",
    "# Load genome files\n",
    "genome_files <- list.files(\"/expanse/lustre/projects/jhu152/naglemi/mwas/gwas_flipped\",\n",
    "                           pattern = \"EUR\", full.names = TRUE)\n",
    "genome_files <- genome_files[grepl(\"allele\", genome_files)]\n",
    "genome_files <- genome_files[grepl(\"a2\", genome_files)]\n",
    "\n",
    "genome_files <- genome_files[grepl(\"pvar\", genome_files)]\n",
    "\n",
    "genome_files <- data.table(path = genome_files, Chr = NA)\n",
    "\n",
    "genome_files$Chr <- str_split_fixed(genome_files$path, \"chr\", 2)[, 2]\n",
    "genome_files$Chr <- gsub(\"\\\\.pvar\", \"\", genome_files$Chr)\n",
    "\n",
    "genome_files$Chr <- as.integer(genome_files$Chr)\n",
    "genome_files <- genome_files[order(genome_files$Chr), ]\n",
    "\n",
    "df <- fread(opt$data_file)\n",
    "df$path <- gsub(\"_a2_\", \"_dt_\", df$path)\n",
    "df$path <- gsub(\"\\\\.rds\", \"_dt_combined.csv\", df$path)\n",
    "\n",
    "print(\"Starting genome file processing\")\n",
    "# Process the specified genome file\n",
    "g <- opt$genome_file_index\n",
    "print(paste(\"Processing genome file index:\", g))\n",
    "\n",
    "paths <- list(\n",
    "  pvar_path = genome_files[g]$path,\n",
    "  pgen_path = gsub(\"pvar\", \"pgen\", genome_files[g]$path),\n",
    "  psam_path = gsub(\"pvar\", \"psam\", genome_files[g]$path)\n",
    ")\n",
    "\n",
    "chr <- as.numeric(gsub(\"\\\\.pvar\", \"\",\n",
    "            stringr::str_split_fixed(paths$pvar_path, \"chr\", 2)[, 2]))\n",
    "\n",
    "my_SNPs <- CpGWAS::loadSNPData(paths$pvar_path, paths$pgen_path, paths$psam_path)\n",
    "setkey(my_SNPs$pvar_dt, `#CHROM`, POS)\n",
    "df_this_chr <- df[which(df$Chr == genome_files[g]$Chr), ]\n",
    "\n",
    "\n",
    "summary_stats_list <- list.files(\"/expanse/lustre/projects/jhu152/naglemi/mwas/gwas\", pattern = \"alleleprocessed\", full.names = TRUE)\n",
    "summary_stats_list <- summary_stats_list[grepl(\"a3\", summary_stats_list)]\n",
    "\n",
    "# Pre-load all summary stats files into a list and clean/standardize column names\n",
    "summary_stats_data <- lapply(summary_stats_list, function(path) {\n",
    "   stats <- suppressWarnings(data.table::fread(path))\n",
    "   \n",
    "   # Rename column if needed (uncomment if column name needs changing from \"#CHROM\" to \"CHR\")\n",
    "   # setnames(stats, \"#CHROM\", \"CHR\")\n",
    "   \n",
    "   # Set key for efficient subsetting\n",
    "   setkey(stats, SNP, CHR)\n",
    "   \n",
    "   # Subset to only include rows where 'CHR' matches the 'chr' variable\n",
    "   stats <- stats[CHR == chr]\n",
    "\n",
    "   # Clean and standardize column names if needed\n",
    "   # clean_and_standardize_colnames(stats)\n",
    "   \n",
    "   return(stats)\n",
    "})\n",
    "\n",
    "summary_stats_data <- lapply(summary_stats_data, function(stats) stats[`CHR` == genome_files[g]$Chr])\n",
    "\n",
    "print(\"Loaded SNP data\")\n",
    "print(\"Files for this Chr:\")\n",
    "print(nrow(df_this_chr))\n",
    "\n",
    "for(j in 1:nrow(df_this_chr)){\n",
    "#for(j in 1:10){\n",
    "  print(paste0(\"File number: \", j))\n",
    "  \n",
    "  start_time <- Sys.time()  # Start timing the loop iteration\n",
    "\n",
    "  if (grepl(\"empty\", df_this_chr$path[j])) {\n",
    "    message(paste0(\"No model for \", df_this_chr$path[j]))\n",
    "    next\n",
    "  }\n",
    "\n",
    "  all_files_exist <- TRUE\n",
    "  outnames <- vector(\"character\", length(summary_stats_list))\n",
    "\n",
    "  # Check if output files already exist\n",
    "  for (k in 1:length(summary_stats_list)) {\n",
    "    outnames[k] <- gsub(\"\\\\.rds$\", paste0(\"_\", basename(tools::file_path_sans_ext(summary_stats_list[[k]])), \"_results.rds\"), df_this_chr$path[j])\n",
    "    if (!file.exists(outnames[k])) {\n",
    "      all_files_exist <- FALSE\n",
    "      break  # Exit the loop early if any file does not exist\n",
    "    }\n",
    "  }\n",
    "\n",
    "  #if (all_files_exist) {\n",
    "  #  print(paste(\"All output files already exist for\", df_this_chr$path[j], \"- Skipping\"))\n",
    "  #  next  # Skip to the next file if all outputs already exist\n",
    "  #}\n",
    "\n",
    "  # Load the CSV file\n",
    "  load_csv_start <- Sys.time()\n",
    "  my_csv <- fread(df_this_chr$path[j])\n",
    "  setkey(my_csv, cg)\n",
    "  cgs <- levels(factor(my_csv$cg))\n",
    "  #print(paste(\"Loaded csv file:\", df_this_chr$path[j]))\n",
    "  load_csv_end <- Sys.time()\n",
    "  #print(paste(\"Time to load CSV:\", load_csv_end - load_csv_start))\n",
    "\n",
    "  # Process each summary statistics file\n",
    "  for (k in 1:length(summary_stats_list)) {\n",
    "    process_start <- Sys.time()\n",
    "    \n",
    "    outname <- gsub(\"_dt_combined.csv\", paste0(\"_\", basename(tools::file_path_sans_ext(summary_stats_list[[k]])), \"_results.rds\"), df_this_chr$path[j])\n",
    "    #if(file.exists(outname)) next\n",
    "    summary_stats <- summary_stats_data[[k]]\n",
    "\n",
    "    MWASmodels <- vector(\"list\", length(cgs))\n",
    "\n",
    "    # Process each CpG site\n",
    "    for (i in seq_along(cgs)) {\n",
    "    #for(i in 1:10){\n",
    "      this_cg <- cgs[i]\n",
    "      \n",
    "      process_cg_start <- Sys.time()\n",
    "\n",
    "      this_cg_data <- my_csv[cg == this_cg]\n",
    "      MWASmodels[[i]] <- process_model_csv(this_cg_data, my_SNPs, summary_stats)\n",
    "\n",
    "      if(is.null(MWASmodels[[i]])){\n",
    "        next\n",
    "      }\n",
    "      \n",
    "      MWASmodels[[i]]['bp'] <- this_cg\n",
    "\n",
    "      process_cg_end <- Sys.time()\n",
    "  #    print(paste(\"Time to process CpG:\", this_cg, \":\", process_cg_end - process_cg_start))\n",
    "    }\n",
    "\n",
    "    # Create MWAS results\n",
    "    results_start <- Sys.time()\n",
    "    results <- MWASresults(MWASmodels, paths$pvar_path, paths$pgen_path, paths$psam_path, summary_stats_list[[k]], df_this_chr$path[j])\n",
    "    results@MWASmodels <- Filter(Negate(is.null), results@MWASmodels)  # Remove NULL entries\n",
    "    results_end <- Sys.time()\n",
    "   # print(paste(\"Time to create MWAS results:\", results_end - results_start))\n",
    "    \n",
    "    # Save the results\n",
    "    save_start <- Sys.time()\n",
    "    #saveRDS(results, outname)\n",
    "    \n",
    "    df <- do.call(rbind, results@MWASmodels)\n",
    "    df <- as.data.frame(df)\n",
    "\n",
    "    outname_csv <- gsub(\"rds\", \"csv\", outname)\n",
    "    fwrite(df, outname_csv)\n",
    "    save_end <- Sys.time()\n",
    "    #print(paste(\"Saved results to:\", outname))\n",
    "    #print(paste(\"Time to save results:\", save_end - save_start))\n",
    "\n",
    "    process_end <- Sys.time()\n",
    "    #print(paste(\"Total time to process summary stats file\", k, \":\", process_end - process_start))\n",
    "  }\n",
    "  \n",
    "  loop_end_time <- Sys.time()  # End timing the loop iteration\n",
    "  print(paste(\"Total time for file number\", j, \":\", loop_end_time - start_time))\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c9d21986-6221-4b48-a3fc-a9127205a41a",
   "metadata": {},
   "source": [
    "Version without timing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c136fd9e-8d48-496b-b820-9a3ab85cfe94",
   "metadata": {},
   "outputs": [],
   "source": [
    "                             \n",
    "# for(j in 1:nrow(df_this_chr)){\n",
    "#   print(paste0(\"File number: \", j))\n",
    "#   if (grepl(\"empty\", df_this_chr$path[j])) {\n",
    "#     message(paste0(\"no model for \", df_this_chr$path[j]))\n",
    "#     next\n",
    "#   }\n",
    "\n",
    "#   all_files_exist <- TRUE\n",
    "#   outnames <- vector(\"character\", length(summary_stats_list))\n",
    "\n",
    "#   for (k in 1:length(summary_stats_list)) {\n",
    "#     outnames[k] <- gsub(\"\\\\.rds$\", paste0(\"_\", basename(tools::file_path_sans_ext(summary_stats_list[[k]])), \"_results.rds\"), df_this_chr$path[j])\n",
    "#     if (!file.exists(outnames[k])) {\n",
    "#       all_files_exist <- FALSE\n",
    "#       break  # Exit the loop early if any file does not exist\n",
    "#     }\n",
    "#   }\n",
    "\n",
    "#   #if (all_files_exist) {\n",
    "#    # print(paste(\"All output files already exist for\", df_this_chr$path[j], \"- Skipping\"))\n",
    "#    # next  # Skip to the next file if all outputs already exist\n",
    "#   #}\n",
    "#   #\n",
    "#   my_csv <- fread(df_this_chr$path[j])\n",
    "#   setkey(my_csv, cg)\n",
    "#   cgs <- levels(factor(my_csv$cg))\n",
    "#   print(paste(\"Loaded csv file:\", df_this_chr$path[j]))\n",
    "\n",
    "#   for (k in 1:length(summary_stats_list)) {\n",
    "#     outname <- gsub(\"_dt_combined.csv\", paste0(\"_\", basename(tools::file_path_sans_ext(summary_stats_list[[k]])), \"_results.rds\"), df_this_chr$path[j])\n",
    "#     #if(file.exists(outname)) next\n",
    "#     summary_stats <- summary_stats_data[[k]]\n",
    "\n",
    "#     MWASmodels <- vector(\"list\", length(cgs))\n",
    "# #    if (is.null(summary_stats)) {\n",
    "# #      summary_stats <- suppressWarnings(fread(summary_stats_list[[k]]))\n",
    "# #      summary_stats <- clean_and_standardize_colnames(summary_stats)\n",
    "# #    }\n",
    "\n",
    "#     # for (i in seq_along(my_rds@models)) {\n",
    "#     #   this_MethylationBase <- my_rds@models[[i]]\n",
    "#     #   if(length(this_MethylationBase@snpWeights) == 0){\n",
    "#     #     next\n",
    "#     #   }\n",
    "#     #   MWASmodels[[i]] <- process_model(this_MethylationBase, my_SNPs, summary_stats)\n",
    "#     #   MWASmodels[[i]]['bp'] <- this_MethylationBase@methylationPosition\n",
    "#     #   #pb$tick()\n",
    "#     # }\n",
    "\n",
    "#     for (i in seq_along(cgs)) {\n",
    "#       this_cg <- cgs[i]\n",
    "      \n",
    "#       this_cg_data <- my_csv[cg == this_cg]\n",
    "#       MWASmodels[[i]] <- process_model_csv(this_cg_data, my_SNPs, summary_stats)\n",
    "\n",
    "#       if(is.null(MWASmodels[[i]])){\n",
    "#         next\n",
    "#       }\n",
    "      \n",
    "#       MWASmodels[[i]]['bp'] <- MWASmodels[[i]]['bp'] <- this_cg\n",
    "#       #pb$tick()\n",
    "#     }\n",
    "\n",
    "#     results <- MWASresults(MWASmodels, paths$pvar_path, paths$pgen_path, paths$psam_path, summary_stats_list[[k]], df_this_chr$path[j])\n",
    "\n",
    "    \n",
    "#     # results <- foreach(i = seq_along(my_rds@models), .packages = c(\"CpGWAS\", \"data.table\"), .combine='c') %dopar% {\n",
    "#     # this_MethylationBase <- my_rds@models[[i]]\n",
    "#     # if (length(this_MethylationBase@snpWeights) == 0) {\n",
    "#     #   return(NULL)  # Return NULL for models that should be skipped\n",
    "#     # }\n",
    "#     # MWASmodel <- process_model(this_MethylationBase, my_SNPs, summary_stats)\n",
    "#     # MWASmodel['bp'] <- this_MethylationBase@methylationPosition\n",
    "#     # return(MWASmodel)\n",
    "#     # }\n",
    "    \n",
    "#     results@MWASmodels <- Filter(Negate(is.null), results@MWASmodels)  # Remove NULL entries\n",
    "    \n",
    "#     saveRDS(results, outname)\n",
    "\n",
    "#     #filtered_models <- Filter(Negate(is.null), results@MWASmodels)\n",
    "#     df <- do.call(rbind, results@MWASmodels)\n",
    "#     df <- as.data.frame(df)\n",
    "\n",
    "#     outname_csv <- gsub(\"rds\", \"csv\", outname)\n",
    "#     fwrite(df, outname_csv)\n",
    "    \n",
    "#     print(paste(\"Saved results to:\", outname))\n",
    "#   }\n",
    "# }"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.3.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
